---
layout: topic
title: Reference databases
date: 2019-02-13 00:00:01
bibliography:
- '../tail/bibliography.bib'
tags: genomics
subject: Precision medicine
---
{{ page.title }}
================
<p class="meta">13 Feb 2019</p>

* TOC
{:toc}

## General genomics:
[Ensembl](http://www.ensembl.org/info/website/index.html)\
[RefSeq](http://www.ncbi.nlm.nih.gov/refseq/)\
[HGNC](http://www.genenames.org/)\
[The Human Protein Atlas](https://www.proteinatlas.org/about)\
[GTEx](https://gtexportal.org/home/)\
[GHR Genes](https://ghr.nlm.nih.gov/gene)\
[UniProt](http://www.uniprot.org/)\
[BioCarta](https://maayanlab.cloud/Harmonizome/dataset/Biocarta+Pathways)\
[GNF/Atlas](https://www.ebi.ac.uk/gxa/home)\
[Gene Ontology](http://geneontology.org/)\
[KEGG](https://www.genome.jp/kegg/)\
[egenetics](BioMart)\
[BioCarta](https://maayanlab.cloud/Harmonizome/dataset/Biocarta+Pathways)\
[GNF/Atlas](https://www.ebi.ac.uk/gxa/home)\
[Gene Ontology](http://geneontology.org/)\
[BioMart](https://grch37.ensembl.org/info/data/biomart/index.html)\
[KGGSeq](http://pmglab.top/kggseq/)\
[HaploReg](https://pubs.broadinstitute.org/mammals/haploreg/haploreg.php)


## Annotation resources:
[dbNSFP](https://sites.google.com/site/jpopgen/dbNSFP)\
[ANNOVAR](https://annovar.openbioinformatics.org/en/latest/)\
[UCSC hgVai](https://genome.ucsc.edu/cgi-bin/)\
[Ensembl VEP](https://www.ensembl.org/info/docs/tools/vep/index.html)\
[SnpEff, SnpSift](http://pcingola.github.io/SnpEff/)

## Genotype analysis:
[PLINK](http://zzz.bwh.harvard.edu/plink/)\
[KING](https://www.kingrelatedness.com/manual.shtml)\
[SHAPEIT2](https://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.html)\
[IMPUTE2](https://mathgen.stats.ox.ac.uk/impute/impute_v2.html)\
[GCTA](https://cnsgenomics.com/software/gcta/)\
[LocusZoom](http://locuszoom.org)\
[LDLink](https://ldlink.nci.nih.gov)\
[FUMA](https://fuma.ctglab.nl)

## Imputation: 
[Sanger](https://www.sanger.ac.uk/tool/sanger-imputation-service/)\
[MACH](http://csg.sph.umich.edu/abecasis/MACH/tour/imputation.html)\
[TOPMed](https://imputation.biodatacatalyst.nhlbi.nih.gov)

## Population/conservation data:
[Exac](http://exac.broadinstitute.org/)\
[gnomAD](http://gnomad.broadinstitute.org/)\
[PHAST](http://compgen.cshl.edu/phast/)\
[Mitomap](https://www.mitomap.org/foswiki/bin/view/MITOMAP/WebHome)\
[fitCons](http://compgen.bscb.cornell.edu/fitCons/)\
[phastCons](http://hgdownload.soe.ucsc.edu/goldenPath/hg38/)\
[kaviar3](http://db.systemsbiology.net/kaviar/Kaviar.downloads.html)\
[dbSNP](http://www.ncbi.nlm.nih.gov/snp)\
[dbVar](https://ftp.ncbi.nlm.nih.gov/pub/dbVar/data/Homo_sapiens/by_assembly/)\
[EVS, NHLBI ESP6500](https://evs.gs.washington.edu/EVS/)\
[DGV](http://dgv.tcag.ca/dgv/app/home)\
[Bravo](https://bravo.sph.umich.edu/freeze5/hg38/)\
[FinnGen](https://r4.finngen.fi)\
[1000 Genomes Project (1KG](https://www.internationalgenome.org/data)\
[UKBB](https://www.ukbiobank.ac.uk)

## Drug-gene:
[CPIC Genes-Drugs](https://cpicpgx.org/genes-drugs/)\
[AACT](https://clinicaltrials.gov)\
[EMA Approved Drugs](https://www.ema.europa.eu/en)\
[FDA Approved Drugs](https://www.fda.gov/drugs/informationondrugs/ucm079750.htm)\
[FDA Pharmacogenomics](https://www.fda.gov/drugs/science-and-research-drugs/table-pharmacogenomic-biomarkers-drug-labeling)\
[PharmGKB](https://www.pharmgkb.org/)\
[DGI](http://dgidb.genome.wustl.edu/)

## Cancer:
[TP53](https://tp53.isb-cgc.org)\
[ICGC somatic](https://icgc.org/)\
[CancerHotspots](https://www.cancerhotspots.org)\
[OncoTree](http://oncotree.mskcc.org/##/home)\
[cBioPortal](https://www.cbioportal.org/)\
[GDC](https://gdc.cancer.gov/)\
[Cancer Gene Census](https://cancer.sanger.ac.uk/census)\
[COSMIC](http://cancer.sanger.ac.uk/cosmic)\
[FusionGDB](https://ccsm.uth.edu/FusionGDB/)\
[CIViC](https://civic.genome.wustl.edu/##/home) 

## Association:
[ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/)\
[ClinGen](https://ftp.clinicalgenome.org/)\
[GWAS Catalog](http://www.ebi.ac.uk/)\
[gene2phenotype](https://www.ebi.ac.uk/gene2phenotype/about)\
[GenCC](https://thegencc.org/)\
[Mastermind](https://mastermind.genomenon.com)\
[GE PanelApp](https://panelapp.genomicsengland.co.uk/)\
[CKB](https://ckb.jax.org/)\
[Mondo](https://monarchinitiative.org/)\
[CGD](http://research.nhgri.nih.gov/CGD/)\
[HPO](https://hpo.jax.org/app/)\
[DailyMed](https://dailymed.nlm.nih.gov)\
[DECIPHER](https://decipher.sanger.ac.uk/)\
[PMKB](https://pmkb.weill.cornell.edu/)\
[Consensus](http://cpdb.molgen.mpg.de/)\
[HGMD](http://www.hgmd.cf.ac.uk/ac/index.php)

## Viral genetics:
[asn2fsa](https://www.huge-man-linux.net/man1/asn2fsa.html)\
[clc novo assemble](qiagenbioinformatics.com)\
[ClustalO](https://www.ebi.ac.uk/Tools/msa/clustalo/)\
[GenBank](https://www.ncbi.nlm.nih.gov/genbank/)\
[IQ-Tree](https://www.iqtree.org/)\
[MAFFT](https://mafft.cbrc.jp/alignment/software)\
[Tbl2asn](https://www.ncbi.nlm.nih.gov/genbank/tbl2asn2/)\
[VIGOR](https://sourceforge.net/projects/jcvi-vigor/files/)


## Protein structure:
[RCSB PDB](https://www.rcsb.org)\
[alphafold](https://alphafold.com)\
[RoseTTA](https://new.rosettacommons.org/docs/latest/Home)\
[FoldX](http://foldxsuite.crg.eu)\
[DynaMut](https://pubmed.ncbi.nlm.nih.gov/32881105/) 

## Prediction:
[GERP](http://mendel.stanford.edu/SidowLab/downloads/gerp/)\
[DANN SNVs](https://cbcl.ics.uci.edu/doku.php/data##dann_data)\
[Domino](https://wwwfbm.unil.ch/domino/index.php)\
[CADD](https://cadd.gs.washington.edu/)\
[dbscSNV](http://www.liulab.science/dbscsnv.html)\
[Essential Genes](https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003484)\
[GDI](https://www.pnas.org/content/112/44/13615)\
[GHIS](https://academic.oup.com/nar/article/43/15/e101/2414292)\
[HIPred](https://academic.oup.com/bioinformatics/article/33/12/1751/2964486)\
[LoFTool](https://academic.oup.com/bioinformatics/article/33/4/471/2525582)\
[PHI Score](https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1001154)\
[Prec Score](https://science.sciencemag.org/content/335/6070/823)\
[RVIS](http://genic-intolerance.org/)\
[ALoFT](http://aloft.gersteinlab.org/)\
[BayesDel](http://fengbj-laboratory.org/BayesDel/BayesDel.html)\
[DEOGEN2](https://deogen2.mutaframe.com/)\
[Eigen](http://www.columbia.edu/~ii2135/eigen.html)\
[FATHMM](http://fathmm.biocompute.org.uk)\
[FATHMM](http://fathmm.biocompute.org.uk/fathmmMKL.htm)\
[HIPred](https://academic.oup.com/bioinformatics/article/33/12/1751/2964486)\
[KEGG](https://www.genome.jp/kegg/)\
[LIST-S2](https://precomputed.list-s2.msl.ubc.ca/)\
[LRT](http://www.genetics.wustl.edu/jflab/lrt_query.html)\
[LoFTool](https://academic.oup.com/bioinformatics/article/33/4/471/2525582)\
[M-CAP](http://bejerano.stanford.edu/MCAP/)\
[MPC](https://www.biorxiv.org/content/10.1101/148353v1)\
[MVP](https://github.com/ShenLab/missense)\
[MetaLR,SVM,RNN](http://www.liulab.science)\
[MutPred](http://mutpred.mutdb.org/)\
[MutationAssessor](http://mutationassessor.org/)\
[MutationTaster](http://www.mutationtaster.org/)\
[PROVEAN](http://provean.jcvi.org/index.php)\
[PrimateAI](https://github.com/Illumina/PrimateAI)\
[REVEL](https://sites.google.com/site/revelgenomics/)\
[SIFT](https://sift.bii.a-star.edu.sg)\
[SiPhy](https://www.broadinstitute.org/mammals-models/29-mammals-project-supplementary-info)\
[bStatistic](http://cadd.gs.washington.edu/)\
[Polyphen-2](http://genetics.bwh.harvard.edu/pph2/)\
[EVE](https://evemodel.org/)

## Population statistics
[GWAS Catalog](https://www.ebi.ac.uk/gwas/home)\
[CFTR2 Variant List](https://cftr2.org/mutations_history)\
[Cystic Fibrosis Mutation Database](http://genet.sickkids.on.ca/LinkPage.html)\
[Explore UK Biobank data](https://cran.rstudio.com/web/packages/ukbtools/vignettes/explore-ukb-data.html)\
[UKB : Browse by Category](http://biobank.ctsu.ox.ac.uk/crystal/label.cgi)\
[St. Jude Cloud](https://stjude.cloud/)\
[Simons Genome Diversity Project](https://www.simonsfoundation.org/simons-genome-diversity-project/)\
[openSNP](https://opensnp.org/)\
[Catalog of Published GWAS - NHGRI](https://www.genome.gov/gwastudies/)\
[PopHuman](https://pophuman.uab.cat/?loc=chr6%3A137998001..138396200&amp;tracks=DNA%2Cgene_annotations%2CPi_CEU_10kb%2Ctheta_CEU_10kb%2CTajima_D_CEU_10kb%2Chap_diversity_within_CEU_10kb%2CKelly_ZnS_CEU_10kb%2CDoS_CEU_10kb%2CK_CEU_10kb&amp;highlight=)\
[Geographic genomic variant browser](https://popgen.uchicago.edu/ggv/?data=%221000genomes%22&amp;chr=1&amp;pos=222087833)\
[SweGen](https://www.nature.com/articles/ejhg2017130)\
[FinnGen](https://www.finngen.fi/en)\
[LOVD - Open Source DNA variation database system](http://www.lovd.nl/3.0/home)\
[dbGaP - NCBI](https://www.ncbi.nlm.nih.gov/gap)\
[dbNSFP - Jpopgen](https://sites.google.com/site/jpopgen/dbNSFP)\
[gnomad](https://gnomad.broadinstitute.org/)

## Disease database
[Primary immunodeficiency disorders](https://panelapp.genomicsengland.co.uk/panels/398/)\
[Panel app GE](https://panelapp.genomicsengland.co.uk/panels/398/)\
[Matchmaker Exchange](http://www.matchmakerexchange.org/)\
[Cafe Variome](https://www.cafevariome.org/home)\
[SEP MS cohort grant and data](https://www.multiplesklerose.ch/fr/propos-de-la-sep/recherche/demande-de-soutien-de-votre-projet-de-recherche/)\
[guidelines](https://www.multiplesklerose.ch/de/ueber-ms/aus-der-forschung/forschungsantraege/##c8514)\
[Consortium on Asthma among African-ancestry Populations in the Americas CAAPA](https://www.caapa-project.org)

## Functionl databases
[VariantValidator](https://variantvalidator.org/##variantValidator)\
[DrugBank](https://www.drugbank.ca/releases/latest)\
[PhylomeDB - large scale phylogenetic data](http://phylomedb.org/)\
[InnateDB: Systems Biology of the Innate Immune Response](https://www.innatedb.com/annotatedGenes.do?type=innatedb)\
[Nextprot](https://www.nextprot.org/about/nextprot)\
[Human Phenotype Ontology](https://hpo.jax.org/app/download/annotation)\
[SIFTS PDBe EMBL-EBI](http://www.ebi.ac.uk/pdbe/docs/sifts/quick.html)\
[List of gene lists for genomic analyses.](https://github.com/macarthur-lab/gene_lists)\
[Genome Mastermind - published variants](https://mastermind.genomenon.com/detail?disease=all%20diseases&amp;gene=RAG1&amp;mutation=RAG1:S225R)\
[Broad GDAC Firehose](https://gdac.broadinstitute.org/)\
[Monarch gentype to phenotype](https://monarchinitiative.org/)\
[HumanNet Search](https://www.inetbio.org/humannet/)\
[Varsome](https://varsome.com/variant/hg19/RAG1%3AR737H)\
[DiscovEHR genomics and EHR](http://www.discovehrshare.com/)

## Methods
[Genome-Wide Association Studies Fact Sheet - National Human Genome Research Institute NHGRI](https://www.genome.gov/20019523/)\
[Genomics England uses MongoDB to power the 100,000 Genomes Project Genomics England](https://www.genomicsengland.co.uk/genomics-england-uses-mongodb-to-power-the-data-science-behind-the-100000-genomes-project/)\
[Genomics England uses MongoDB to Power the 100,000 Genomes Project MongoDB](https://www.mongodb.com/press/genomics-england-uses-mongodb-to-power-the-data-science-behind-the-100000-genomes-project)\
[Confluence Mobile - OpenCB](http://docs.opencb.org/plugins/servlet/mobile?contentId=327756##content/view/327756)\
[neXtProt platform The human protein knowledge base and annotation database](https://www.nextprot.org/)\
[Integrative Genomics Viewer - CCRs](https://s3.us-east-2.amazonaws.com/ccrs/ccr.html)\
[VDJdb](https://vdjdb.cdr3.net/)\
[VDJviz: home](https://vdjviz.cdr3.net/)\
[Personal Genome Project: United Kingdom](https://www.personalgenomes.org.uk/data/)\
[MUSCLE multiple sequence align](http://www.drive5.com/muscle/)\
[Clinical Data Interchange Standards Consortium](https://www.cdisc.org/about)\
[European i2b2 tranSMART Academic Users Group](http://transmartfoundation.org/europe-fall-meeting-2018/)\
[RD-Connect](https://rd-connect.eu/)
[Tools and Software Wellcome Sanger Institute](https://www.sanger.ac.uk/science/tools)

## Pathogens
[JGI Integrated Microbial Viral Genome Home](https://img.jgi.doe.gov/cgi-bin/vr/main.cgi)\
[HIV sequence database main page](https://www.hiv.lanl.gov/content/sequence/HIV/mainpage.html)\
[Ecoli Ab resistance example](https://academic.oup.com/bioinformatics/article/34/13/i89/5045729)

## Misc
[Advanced Bash-Scripting Guide](http://tldp.org/LDP/abs/html/)\
[Hadley Readings in applied data science](https://github.com/hadley/stats337##readings)\
[Dataset Search](https://toolbox.google.com/datasetsearch)\
[Registry of Open Data on AWS](https://registry.opendata.aws/)\
[LibGen literature](http://libgen.io/)

## Commercial
[Saphetor](http://saphetor.com/technology_platform/)\
[Dante labs genome sequencing](https://www.dantelabs.com)
