---
layout: topic
title: Genomic analysis tools
date: 2018-08-08 00:00:01
tags: genomics
subject: Precision medicine
---
{{ page.title }}
================
<p class="meta">26 Apr 2020</p>

* TOC
{:toc}

# command line tool  
[AmpliconArchitect](https://github.com/virajbdeshpande/AmpliconArchitect) AmpliconArchitect is used to identify circular DNA fragments in genomic data.  
[Anaconda2](https://anaconda.org)   
[Anaconda3](https://anaconda.org)   
[annovar](http://annovar.openbioinformatics.org/) Annotate functional consequences of genetic variation from sequencing data  
[ASCAT](https://github.com/cancerit/ascatNgs )   
[ASCAT](https://github.com/Crick-CancerGenomics/ascat)   
[Autoconf](https://www.gnu.org/software/autoconf/autoconf.html)   
[Automake](https://www.gnu.org/software/automake/)   
[Autotools](https://www.gnu.org/software/automake/manual/html_node/Autotools-Introduction.html)   
[Battenberg](https://github.com/cancerit/cgpBattenberg)   
[Battenberg](https://github.com/Wedge-Oxford/battenberg)   
[bayesianDeNovoFilter](https://github.com/andyrimmer/Platypus/blob/master/extensions/DeNovo/bayesianDeNovoFilter.py)   
[bcftools](http://www.htslib.org/doc/bcftools.html) Utilities for variant calling and manipulating VCFs and BCFs  
[bedops](http://bedops.readthedocs.io/) Toolkit that performs highly efficient and scalable Boolean and other set operations, statistical calculations, archiving, conversion and other management of genomic data.  
[bedtools](http://bedtools.readthedocs.io/) Set of tools for a wide-range of genomics analysis tasks.  
[bertha](NA)   
[binutils](https://www.gnu.org/software/binutils/)   
[Bison](https://www.gnu.org/software/bison/)   
[Blast](ftp://ftp.ncbi.nlm.nih.gov/blast/db/) It searches through non-human sequence looking for bacteria/viruses.  
[bzip2](http://www.bzip.org)   
[canvas](http://github.com/Illumina/canvas) A tool for calling copy number variants [CNVs) from human DNA sequencing data  
[cellbase](https://github.com/opencb/cellbase/wiki) A comprehensive collection of RESTful web services for retrieving relevant biological information from heterogeneous sources  
[Circular RNA analysis](https://github.com/lindaszabo/KNIFE)   
[cluster](http://bonsai.hgc.jp/%7Emdehoon/software/cluster/software.htm)   
[CNVnator](https://github.com/abyzovlab/CNVnator) A tool for CNV discovery and genotyping from depth-of-coverage by mapped reads  
[CUDA](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=2&cad=rja&uact=8&ved=2ahUKEwia5f_lu93cAhUI3KQKHTZNC1kQFjABegQIAhAB&url=https%3A%2F%2Fdeveloper.nvidia.com%2Fcuda-zone&usg=AOvVaw2J1e7C-ir5D_lz8OgoiUKF)   
[cuDNN](https://developer.nvidia.com/cudnn)   
[curl](https://curl.haxx.se)   
[dndscv](https://github.com/im3sanger/dndscv) Ability to calculate the non-synonymous to synonymous ratio [dN/dS) in cancer to find genes under negative or positive selection.  
[dotnet](https://github.com/dotnet)   
[Eagle](NA)   
[EasyBuild](https://github.com/easybuilders/easybuild)   
[ERDS](NA)   
[Eric-script](https://sites.google.com/site/bioericscript/home)   
[expat](https://libexpat.github.io)   
[FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) All the necessary tools for RNA-Seq analysis from raw-sequence read quality assessment, to alignment, transcript quantification, and differential expression. Also it includes analysis for gene-fusion analysis and circular RNA analysis.  
[FFTW](http://www.fftw.org)   
[flashPCA](https://github.com/gabraham/flashpca) flashPCA performs fast principal component analysis [PCA) of single nucleotide polymorphism [SNP) data  
[fle](xhttps://github.com/westes/flex)   
[fontconfig](https://www.freedesktop.org/wiki/Software/fontconfig/)   
[freetype](https://www.freetype.org)   
[GATK](https://software.broadinstitute.org/gatk/) Toolkit with a primary focus on variant discovery and genotyping  
[GCC](http://gcc.gnu.org)   
[GCCcore](https://github.com/easybuilders/easybuild-easyconfigs/tree/master/easybuild/easyconfigs/g/GCCcore)   
[gettext](https://www.gnu.org/software/gettext/)   
[GMP](https://gmplib.org)   
[gompi](NA)   
[hapcompass](http://www.brown.edu/Research/Istrail_Lab/resources/hapcompass_manual.html) A software package for haplotype assembly of diploid, polyploid, and tumor genomes  
[hapcut](https://github.com/vibansal/hapcut/blob/master/README.md) A max-cut based algorithm for haplotype assembly that uses the mix of sequenced fragments from the two chromosomes of an individual  
[help2man](https://www.gnu.org/software/help2man/)   
[HiSAT](https://ccb.jhu.edu/software/hisat2/index.shtml)   
[htslib](http://www.htslib.org/doc/#manual-pages) A C library for reading/writing high-throughput sequencing data  
[hwloc](https://www.open-mpi.org/projects/hwloc/)   
[impute2](http://mathgen.stats.ox.ac.uk/impute/impute_v2.html ) IMPUTE version 2 [also known as IMPUTE2) is a genotype imputation and haplotype phasing program based on ideas from Howie et al. 2009  
[install velvet](https://www.ebi.ac.uk/~zerbino/velvet/) It Assembles unmapped reads into longer contigs, to help identify their source.  
[java](https://java.com/en/download/)   
[Junckey](https://github.com/comprna/Junckey)   
[Kallisto](https://pachterlab.github.io/kallisto/download.htm)   
[king](http://people.virginia.edu/~wc9c/KING/manual.html) A toolset to explore genotype data from a genome-wide association study and a sequencing project  
[LeafCutter](https://github.com/davidaknowles/leafcutter)   
[libpng](http://www.libpng.org/pub/png/libpng.html)   
[libreadline](https://tiswww.case.edu/php/chet/readline/rltop.html)   
[libtool](https://www.gnu.org/software/libtool/)   
[libxml2](http://xmlsoft.org)   
[LiftOver](https://genome.ucsc.edu/util.html)   
[lumpy](https://github.com/arq5x/lumpy-sv/blob/master/README.md) A probabilistic framework for structural variant discovery  
[M4](https://www.gnu.org/software/m4/m4.html)   
[MAJIQ](https://majiq.biociphers.org/)   
[manta](https://getmanta.com)   
[maven](https://maven.apache.org)   
[msings](https://bitbucket.org/uwlabmed/msings)   
[MultiQC](http://multiqc.info/docs/#manual-installation) MultiQC is a program that generates reports from the log files of common bioinformatics tools, and would be very good to have in the environment. It provides report creation, information aggregation functionalities.  
[ncurses](https://www.gnu.org/software/ncurses/)   
[numactl](https://linux.die.net/man/8/numactl)   
[OpenBLAS](https://www.openblas.net)   
[OpenMPI](https://www.open-mpi.org)   
[Pandoc](https://pandoc.org/) Pandoc is a Haskell library for converting from one markup format to another, and a command-line tool that uses this library.  
[parallel](https://www.gnu.org/software/parallel/)   
[perl](https://www.perl.org)   
[phylip](http://evolution.genetics.washington.edu/phylip.html)   
[picard](https://broadinstitute.github.io/picard/)   
[pixman](http://www.pixman.org)   
[pkg-config](https://www.freedesktop.org/wiki/Software/pkg-config/)   
[Platypus](http://www.well.ox.ac.uk/platypus-doc) A Haplotype-Based Variant Caller For Next Generation Sequence Data  
[PLINK](http://zzz.bwh.harvard.edu/plink/) A whole genome association analysis toolset, designed to perform a range of basic, large-scale analyses in a computationally efficient manner  
[python](https://www.python.org)   
[R](https://www.r-project.org)   
[rMATS](http://rnaseq-mats.sourceforge.net/)   
[RMySQL R package](https://bioconductor.org/biocLite.R)   
[rtg-tools](https://github.com/RealTimeGenomics/rtg-tools) Utilities for accurate VCF comparison and manipulation  
[ruby](https://www.ruby-lang.org/en/)   
[Salmon](https://github.com/COMBINE-lab/salmon) For direct transcript quantification from FastQ  
[Sambamba](http://lomereiter.github.io/sambamba/index.html) Utilities for viewing, manipulating and merging bam files  
[samblaster](https://github.com/GregoryFaust/samblaster) A tool to mark duplicates and extract discordant and split reads from sam files  
[samtools](http://www.htslib.org/doc/samtools.html) Utilities for accessing and manipulating sam files  
[ScaLAPACK](https://www.osc.edu/resources/available_software/software_list/scalapack)   
[SHAPEIT2](http://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.html ) SHAPEIT is a fast and accurate method for estimation of haplotypes [aka phasing) from genotype or sequencing data.  
[STAR](https://github.com/alexdobin/STAR)   
[STAR-fusion](https://github.com/STAR-Fusion/STAR-Fusion/wiki)   
[SUPPA](https://github.com/comprna/SUPPA)   
[texlive](https://tug.org/texlive/)   
[vcflib](https://github.com/vcflib/vcflib/blob/master/README.md) A C++ library for parsing and manipulating VCF files  
[vcftools](https://vcftools.github.io/index.html) A program package designed for working with VCF files  
[vep](https://github.com/Ensembl/ensembl-vep) Determines the effect of variants [SNPs, insertions, deletions, CNVs or structural variants) on genes, transcripts, and protein sequence, as well as regulatory regions  
[verifybamid](https://github.com/statgen/verifyBamID/releases)   
[vt](http://www.lstmed.ac.uk/vtbuilder) A tool for the inference of non-chimeric contigs from read data that has been sequenced from complex multi-isoformic transcriptomes  
[weCall](NA)   
[xz](https://tukaani.org/xz/)   
[zlib](https://www.zlib.net)   
# desktop_applications  
[atom-ide-ui](https://atom.io/packages/atom-ide-ui )   
[Firefo](xhttps://www.mozilla.org/en-US/firefox/new/)   
[ide-python](https://atom.io/packages/ide-python ) requires server https://github.com/palantir/python-language-server  
[LibreOffice](https://www.libreoffice.org)   
[RStudio](https://www.rstudio.com)   
# Websites  
[ClinVar](www.ncbi.nlm.nih.gov/clinvar/*) www.ncbi.nlm.nih.gov/clinvar/  
[Compbio HPO Browser](compbio.charite.de/hpoweb/*) http://compbio.charite.de/hpoweb/  
[Consensus CDS Project](www.ncbi.nlm.nih.gov/CCDS/*) www.ncbi.nlm.nih.gov/CCDS/  
[COSMIC](cancer.sanger.ac.uk/wgs/*) http://cancer.sanger.ac.uk/cosmic/  
[DBsnp](www.ncbi.nlm.nih.gov/snp/*) www.ncbi.nlm.nih.gov/snp/  
[DECIPHER](decipher.sanger.ac.uk/*) http://decipher.sanger.ac.uk/  
[EMBL EBI](www.ebi.ac.uk/*) www.ebi.ac.uk/  
[EMBL EBI](www.ebi.ac.uk/interpro/*)   
[EMBL EBI](www.ebi.ac.uk/wen_guidelines/*)   
[Ensembl](static.ensembl.org/*)   
[Ensembl](static.ensembl.org/minified/*)   
[Ensembl](www.ensembl.org/id/*)   
[Ensembl Human](ensembl.org/*) http://ensembl.org/Homo_sapiens/  
[Ensembl Human](ensembl.org/Homo_sapiens/Gene/*)   
[Ensembl Human](ensembl.org/Homo_sapiens/*)   
[exAC Browser beta](http://exac.broadinstitute.org) http://exac.broadinstitute.org  
[Genetics Home Reference Gene Browser](ghr.nlm.nih.gov/*) http://ghr.nlm.nih.gov/gene  
[Genetics Home Reference Gene Browser](ghr.nlm.nih.gov/gene/*)   
[Genetics Home Reference Gene Browser](ghr.nlm.nih.gov/images/*)   
[Genomics Plc Misc.](.incoming01.genomicsplc.com)   
[gnomAD Browser beta](http://gnomad.broadinstitute.org) http://gnomad.broadinstitute.org  
[Google Misc.](www.gstatic.com/charts/*)   
[HUGO Gene Nomenclature Committee](www.genenames.org/*) http://www.genenames.org/  
[HUGO Gene Nomenclature Committee](www.genenames.org/data/*)   
[HUGO Gene Nomenclature Committee](www.genenames.org/sites/*)   
[HUGO Gene Nomenclature Committee](www.genenames.org/css/*)   
[HUGO Gene Nomenclature Committee](www.genenames.org/js/*)   
[HUGO Gene Nomenclature Committee](rest.genenames.org/*)   
[Human Phenotype Ontology](www.human-phenotype-ontology.org/hpoweb/*) http://compbio.charite.de/hpoweb/showterm  
[Human Phenotype Ontology](www.human-phenotype-ontology.org/*)   
[Human Phenotype Ontology](human-phenotype-ontology.github.io/*)   
[IGV Browser](igv.broadinstitute.org/*)   
[IGV Browser](igvdata.broadinstitute.org/*)   
[NCBI](www.ncbi.nlm.nih.gov/portal*) www.ncbi.nlm.nih.gov/portal  
[NCBI](www.nlm.nih.gov/core/*)   
[NCBI](.nlm.nih.gov/*)   
[NCBI](www.ncbi.nlm.nih.gov/gtr/tests/*)   
[NCBI](www.ncbi.nlm.nih.gov/core/*)   
[NCBI](www.ncbi.nlm.nih.gov/project/*)   
[NCBI Gene](www.ncbi.nlm.nih.gov/gene*) www.ncbi.nlm.nih.gov/gene  
[NCBI Nucleotide](www.ncbi.nlm.nih.gov/nuccore/*) www.ncbi.nlm.nih.gov/nuccore/  
[OMIM](omim.org/*) http://omim.org  
[OMIM](omim.org/entry/*)   
[OMIM](omim.org/static/*)   
[OMIM](omim.org/search/*)   
[OMIM](api.europe.omim.org/api/*)   
[Panelapp](https://panelapp.genomicsengland.co.uk/) https://panelapp.genomicsengland.co.uk/  
[Pubmed](static.pubmed.gov/*) www.ncbi.nlm.nih.gov/pubmed/  
[Pubmed](www.ncbi.nlm.nih.gov/pubmed/*)   
[Pubmed](dev-static.pubmed.gov/*)   
[Sanger](cancer.sanger.ac.uk/_asset/*)   
[Sanger](cancer.sanger.ac.uk/cancergenome/*)   
[Sanger](cancer.sanger.ac.uk/javascripts/*)   
[Sanger](cancer.sanger.ac.uk/cosmic/*)   
[SMART [a Simple Modular Architecture Research Tool)](smart.embl-heidelberg.de/*) http://smart.embl-heidelberg.de  
[Swiss Institute of Bioinformatics](https://www.sib.swiss/*) https://www.sib.swiss  
[The Human Gene Mutation Database](www.hgmd.cf.ac.uk/ac/*) www.hgmd.cf.ac.uk  
[The Human Gene Mutation Database](www.hgmd.cf.ac.uk/*)   
[The Sequence Ontology  database](www.sequenceontology.org/miso/current_release/term/*) www.sequenceontology.org/  
[The Sequence Ontology  database](www.sequenceontology.org/css/*)   
[The Sequence Ontology  database](www.sequenceontology.org/js/*)   
[The Sequence Ontology  database](www.sequenceontology.org/img/*)   
[The Sequence Ontology  database](www.sequenceontology.org/browser/*)   
[The Sequence Ontology  database](www.sequenceontology.org/*)   
[UCSC Genome Browser](www.genome.ucsc.edu/*)   
[Xfam [Pfam, Rfam, Dfam, Treefam, iPfam, Antifam)](pfam.sanger.ac.uk/family/*) http://xfam.org  
[Xfam [Pfam, Rfam, Dfam, Treefam, iPfam, Antifam)](pfam.sanger.ac.uk/*)   
[Xfam [Pfam, Rfam, Dfam, Treefam, iPfam, Antifam)](xfam.org/*)   
