<!DOCTYPE html>
<html xmlns="https://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
	<!-- link to MathJax for LaTeX stlye eqn -->
	 <script type="text/javascript" async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-MML-AM_CHTML">
</script>

	 
	 <!-- https://realfavicongenerator.net -->
	 <link rel="apple-touch-icon" sizes="180x180" href="/images/logos/favicon_package_v0.16_bw/apple-touch-icon.png">
	 <link rel="icon" type="image/png" sizes="32x32" href="/images/logos/favicon_package_v0.16_bw/favicon-32x32.png">
	 <link rel="icon" type="image/png" sizes="16x16" href="/images/logos/favicon_package_v0.16_bw/favicon-16x16.png">
	 <link rel="manifest" href="/images/logos/favicon_package_v0.16_bw/site.webmanifest">
	 <meta name="msapplication-TileColor" content="#da532c">
	 <meta name="theme-color" content="#ffffff">
	
	 <meta name="viewport" content="width=device-width, initial-scale=1">
	
	<meta http-equiv="content-type" content="text/html; charset=utf-8">
	<meta name="description" content="A home for topics in human precision medicine, genomic analysis, and data visualisation.">
	<title>Unnecessary complexity in precision medicine</title>
	<meta name="author" content="LawlessGenomics">
	<link href="https://lawlessgenomics.com/feed.xml" rel="alternate" title="LawlessGenomics" type="application/atom+xml">
	<!-- <meta name="readability-verification" content="QCzSs992GxmRYRKVpPeZ6LE2tS8aYKxsSSQKV8YM"/> -->
	
	<!-- include a google font -->
	<link href="https://fonts.googleapis.com/css?family=Roboto+Mono%7CRoboto+Slab:300%7CRoboto:500" rel="stylesheet">
	<link href="https://fonts.googleapis.com/css?family=Lora" rel="stylesheet">
	
	<!-- syntax highlighting CSS -->
	<link rel="stylesheet" href="/css/syntax.css" type="text/css">
	
	<!-- Homepage CSS -->
	<link rel="stylesheet" href="/css/screen.css" type="text/css" media="screen, projection">
	
	<!-- Typekit -->
	<script type="text/javascript" src="https://use.typekit.com/jpd0pfm.js"></script>
	<script type="text/javascript">try{Typekit.load();}catch(e){}</script>
</head>
<body>

<!-- ClickTale Top part -->
<script type="text/javascript">
	var WRInitTime=(new Date()).getTime();
</script>
<!-- ClickTale end of Top part -->

<!-- Logo and navigation bar -->
<div class="site">
	<div class="title">
		<div class="nav-container">
			<nav>

			<div class="logoimage">
				<a href="/">
				<img style="margin-bottom:-16px" src="/images/logos/logo4.1_150ppi.png" alt="Logo image" width="35.7" height="55"></a>
			</div>

			<div class="logo">
				<a href="/">LawlessGenomics</a>
			</div>

				<input type="checkbox" id="nav-toggle">
					<label for="nav-toggle" class="gg-menu"></label>
					<div class="right-menu">
						<a href="/resume">About</a>
						<a href="/">Home</a>
					</div>
			</nav>
		</div>
	</div>

	<div id="post">
<h1 id="unnecessary-complexity-in-precision-medicine">Unnecessary complexity in precision medicine</h1>
<p class="meta">26 Apr 2020</p>
<p><br></p>
<p><br>
Singh_and_Gupta.2020.GenoMed;<br>
Singh, R.S., Gupta, B.P. Genes and genomes and unnecessary complexity in precision medicine. npj Genom. Med. 5, 21 (2020). https://doi.org/10.1038/s41525-020-0128-1 Open Access
<a href="https://doi.org/10.1038/s41525-020-0128-1">https://doi.org/10.1038/s41525-020-0128-1</a></p>

<h1 id="abstract">Abstract</h1>

<p>Risk factors, gene to phenotype, are more complex that anticipated in early genomics. 
Genotype - phenotype
Molecular and evolutionary complexity.
Molecular contigency - chance driven mutation, redundancy, molectular pathway cross-over (shared phenotypes).
Necessary complexity versus unnecessary - evolutionary baggage due to molecular constraint and blind evolution.</p>

<h1 id="introduction">Introduction</h1>

<p>“Two individuals bearing the same set of risk factors may not present (or exhibit the symptoms of) the same disease. This is the kind of problem that precision medicine is expected to overcome.”
From familial studies, we often see that even clear monogenic, dominant disease does not always manifest equally among family members. “The lack of genetic determinism in disease is the result of molecular contingency.”</p>

<p>An interesting point is made that I haven’t though much about; “Whether evolution is deterministic, i.e., predictable and repeatable, or contingent and unpredictable is an interesting topic in evolutionary biology (ref 1) (for a recent review see ref. 2)”
Although, this is just a smaller version of a question in physics that I do think about frequently: if every irreducibly small particle/force in the universe could be detected with a mapped trajectory, then we could “see” both forward and reverse in time; causal determinism. Since the particle physics answer will not come anytime soon, a discussion on the evolutionary version is reasonable.</p>

<p>”..phenotypic evolutionary ‘repeatability’ is common when the founding populations are closely related, perhaps resulting from shared genes and developmental pathways, whereas different outcomes become more likely as historical divergences increase.”</p>

<p>I believe that in some cases we can produces actionable levels of confidence in mutation prediction to make this a worthwhile endeavour. Knowing that a mutation is likely to occur (and be evolutionarily selected against) and result in disease means that we can prepare a treatment / cure for when it does occur naturally. 
For example, each nucleotide within each individual cell will have a particular probability for the energy required to mutate. 
Many factors affect this value; nucleotide type, methylation, DNA/RNA folding, bound proteins, chemical/UV insult, etc. 
Some of these factors can be estimated based on large studies, and some may never be quantifiable in the real world.
However, we can at least quantify some values to produce mutation predictor value X and and response Y and therefore quantify estimate coefficients, and therefore predict that mutation 1 is more likely to occur than mutation 2.</p>

<p>Reference is made to “‘Evolution and Tinkering,’ Jacob (ref 4)” who compares natural selection to a tinkerer as opposed to an engineer.
Singh and Gupta propose “a theory of molecular complexity, consisting of necessary and unnecessary complexity in living systems”, that “exceedingly high level of unnecessary complexity” is more due to blind evolution rather than randomness of mutation. 
“Unecessary and unnecessary complexity is relevant to current discussions on genomics and precision medicine.”</p>

<h1 id="complexity-in-living-systems">Complexity in living systems</h1>
<h1 id="is-complexity-necessary">Is complexity necessary?</h1>
<h1 id="evolution-of-complexity-genic-genomic-and-developmental">Evolution of complexity: genic, genomic, and developmental</h1>
<h1 id="unnecessary-complexity-in-precision-medicine-1">Unnecessary complexity in precision medicine</h1>
<h1 id="unnecessary-complexity-and-the-etiology-of-cancer">Unnecessary complexity and the etiology of cancer</h1>
<h1 id="unnecessary-complexity-beyond-precision-medicine">Unnecessary complexity beyond precision medicine</h1>
<h1 id="conclusions">Conclusions</h1>


<h1 id="unnecessary-complexity-in-precision-medicine">Unnecessary complexity in precision medicine</h1>
<p class="meta">26 Apr 2020</p>
<p><br></p>
<p><br>
Singh_and_Gupta.2020.GenoMed;<br>
Singh, R.S., Gupta, B.P. Genes and genomes and unnecessary complexity in precision medicine. npj Genom. Med. 5, 21 (2020). https://doi.org/10.1038/s41525-020-0128-1 Open Access
<a href="https://doi.org/10.1038/s41525-020-0128-1">https://doi.org/10.1038/s41525-020-0128-1</a></p>

<h1 id="abstract">Abstract</h1>

<p>Risk factors, gene to phenotype, are more complex that anticipated in early genomics. 
Genotype - phenotype
Molecular and evolutionary complexity.
Molecular contigency - chance driven mutation, redundancy, molectular pathway cross-over (shared phenotypes).
Necessary complexity versus unnecessary - evolutionary baggage due to molecular constraint and blind evolution.</p>

<h1 id="introduction">Introduction</h1>

<p>“Two individuals bearing the same set of risk factors may not present (or exhibit the symptoms of) the same disease. This is the kind of problem that precision medicine is expected to overcome.”
From familial studies, we often see that even clear monogenic, dominant disease does not always manifest equally among family members. “The lack of genetic determinism in disease is the result of molecular contingency.”</p>

<p>An interesting point is made that I haven’t though much about; “Whether evolution is deterministic, i.e., predictable and repeatable, or contingent and unpredictable is an interesting topic in evolutionary biology (ref 1) (for a recent review see ref. 2)”
Although, this is just a smaller version of a question in physics that I do think about frequently: if every irreducibly small particle/force in the universe could be detected with a mapped trajectory, then we could “see” both forward and reverse in time; causal determinism. Since the particle physics answer will not come anytime soon, a discussion on the evolutionary version is reasonable.</p>

<p>”..phenotypic evolutionary ‘repeatability’ is common when the founding populations are closely related, perhaps resulting from shared genes and developmental pathways, whereas different outcomes become more likely as historical divergences increase.”</p>

<p>I believe that in some cases we can produces actionable levels of confidence in mutation prediction to make this a worthwhile endeavour. Knowing that a mutation is likely to occur (and be evolutionarily selected against) and result in disease means that we can prepare a treatment / cure for when it does occur naturally. 
For example, each nucleotide within each individual cell will have a particular probability for the energy required to mutate. 
Many factors affect this value; nucleotide type, methylation, DNA/RNA folding, bound proteins, chemical/UV insult, etc. 
Some of these factors can be estimated based on large studies, and some may never be quantifiable in the real world.
However, we can at least quantify some values to produce mutation predictor value X and and response Y and therefore quantify estimate coefficients, and therefore predict that mutation 1 is more likely to occur than mutation 2.</p>

<p>Reference is made to “‘Evolution and Tinkering,’ Jacob (ref 4)” who compares natural selection to a tinkerer as opposed to an engineer.
Singh and Gupta propose “a theory of molecular complexity, consisting of necessary and unnecessary complexity in living systems”, that “exceedingly high level of unnecessary complexity” is more due to blind evolution rather than randomness of mutation. 
“Unecessary and unnecessary complexity is relevant to current discussions on genomics and precision medicine.”</p>

<h1 id="complexity-in-living-systems">Complexity in living systems</h1>
<h1 id="is-complexity-necessary">Is complexity necessary?</h1>
<h1 id="evolution-of-complexity-genic-genomic-and-developmental">Evolution of complexity: genic, genomic, and developmental</h1>
<h1 id="unnecessary-complexity-in-precision-medicine-1">Unnecessary complexity in precision medicine</h1>
<h1 id="unnecessary-complexity-and-the-etiology-of-cancer">Unnecessary complexity and the etiology of cancer</h1>
<h1 id="unnecessary-complexity-beyond-precision-medicine">Unnecessary complexity beyond precision medicine</h1>
<h1 id="conclusions">Conclusions</h1>


</div>

<div id="related">
  <h2>Related Posts</h2>
  <ul class="posts">
    
      <li>
<span>21 Sep 2021</span> » <a href="/2021/09/21/whos_afraid.html">Who's afraid of the big bad geneticist?</a>
</li>
    
      <li>
<span>13 Sep 2021</span> » <a href="/2021/09/13/weeknotes_16.html">Weeknotes</a>
</li>
    
      <li>
<span>06 Sep 2021</span> » <a href="/2021/09/06/weeknotes_15.html">Weeknotes</a>
</li>
    
  </ul>
</div>



	<div class="footer">
		<div class="foot-container">
				<a href="/" class="logo-foot">
				<img style="margin-bottom:-16px" src="/images/logos/logo4.2_150ppi.png" alt="Logo image" width="35.7" height="55"></a>
					<div class="right-root">
						<p> Maintained by Dylan Lawless. Postdoc at <a href="https://people.epfl.ch/dylan.lawless">EPFL</a>.
						<a href="/resume">Resume</a>
						and <a href="https://scholar.google.com/citations?user=RPBxP1wAAAAJ&hl=en&oi=ao">Google scholar</a>.
						<a href="https://lawlessgenomics.com/feed.xml">RSS feed</a>
						</p>
					</div>
		</div>
	</div>

</div>


<!-- Banner on top right corner linking to github
<a href="https://github.com/mojombo"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_red_aa0000.png" alt="Fork me on GitHub" /></a>
-->

<!-- ClickTale Bottom part -->
<div id="ClickTaleDiv" style="display: none;"></div>
<script type="text/javascript">
if(document.location.protocol!='https:')
  document.write(unescape("%3Cscript%20src='https://s.clicktale.net/WRb.js'%20type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
if(typeof ClickTale=='function') ClickTale(206,0.3,"www03");
</script>
<!-- ClickTale end of Bottom part -->

<!-- Google Analytics -->
<!--
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "https://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
var pageTracker = _gat._getTracker("UA-6016902-1");
pageTracker._trackPageview();
</script>
-->
<!-- Google Analytics end -->

</body>
</html>
