<!DOCTYPE html>
<html xmlns="https://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
	<!-- link to MathJax for LaTeX stlye eqn -->
	 <script type="text/javascript" async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-MML-AM_CHTML">
</script>

	 
	 <!-- https://realfavicongenerator.net -->
	 <link rel="apple-touch-icon" sizes="180x180" href="/images/logos/favicon_package_v0.16_bw/apple-touch-icon.png">
	 <link rel="icon" type="image/png" sizes="32x32" href="/images/logos/favicon_package_v0.16_bw/favicon-32x32.png">
	 <link rel="icon" type="image/png" sizes="16x16" href="/images/logos/favicon_package_v0.16_bw/favicon-16x16.png">
	 <link rel="manifest" href="/images/logos/favicon_package_v0.16_bw/site.webmanifest">
	 <meta name="msapplication-TileColor" content="#da532c">
	 <meta name="theme-color" content="#ffffff">
	
	 <meta name="viewport" content="width=device-width, initial-scale=1">
	
	<meta http-equiv="content-type" content="text/html; charset=utf-8">
	<meta name="description" content="A home for topics in human precision medicine, genomic analysis, and data visualisation.">
	<title>Integrating databases</title>
	<meta name="author" content="LawlessGenomics">
	<link href="https://lawlessgenomics.com/feed.xml" rel="alternate" title="LawlessGenomics" type="application/atom+xml">
	<!-- <meta name="readability-verification" content="QCzSs992GxmRYRKVpPeZ6LE2tS8aYKxsSSQKV8YM"/> -->
	
	<!-- include a google font -->
	<link href="https://fonts.googleapis.com/css?family=Roboto+Mono%7CRoboto+Slab:300%7CRoboto:500" rel="stylesheet">
	<link href="https://fonts.googleapis.com/css?family=Lora" rel="stylesheet">
	
	<!-- syntax highlighting CSS -->
	<link rel="stylesheet" href="/css/syntax.css" type="text/css">
	
	<!-- Homepage CSS -->
	<link rel="stylesheet" href="/css/screen.css" type="text/css" media="screen, projection">
	
	<!-- Google font -->
	<link rel="preconnect" href="https://fonts.googleapis.com"> 
	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin> 
	<link href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,400;0,600;0,700;1,400;1,600;1,700&amp;display=swap" rel="stylesheet">
	
	<!-- Typekit -->
	<script type="text/javascript" src="https://use.typekit.com/jpd0pfm.js"></script>
	<script type="text/javascript">try{Typekit.load();}catch(e){}</script>
</head>
<body>

<!-- ClickTale Top part -->
<script type="text/javascript">
	var WRInitTime=(new Date()).getTime();
</script>
<!-- ClickTale end of Top part -->

<!-- Logo and navigation bar -->



<!-- SITE START --------------------------------------------------------- -->
<div class="site">

<!-- NAV START --------------------------------------------------------- -->
<div class="sitewidthnav">
	<div class="title">
		<div class="nav-container">
			<nav>

			<div class="logoimage">
				<a href="/">
				<img style="margin-bottom:-16px" src="/images/logos/logo4.2_150ppi.png" alt="Logo image" width="25" height="38.5"></a>
				<!-- alt="Logo image" width="35.7" height="55" /></a> -->
			</div>

			<div class="logo">
				<a href="/">LawlessGenomics</a>
			</div>

				<input type="checkbox" id="nav-toggle">
					<label for="nav-toggle" class="gg-menu"></label>
					<div class="right-menu">
						<a href="/resume">About</a>
						<a href="/">Home</a>
					</div>
			</nav>
		</div>
	</div>



</div>
<!-- NAV END ----------------------------------------------------------- -->

<!-- MAIN START -------------------------------------------------------- -->
<div class="sitewidth">
	<div id="topic">
Reading time  10 minutes
<h1 id="integrating-databases">Integrating databases</h1>

<ul id="markdown-toc">
  <li><a href="#subsec:gnomad" id="markdown-toc-subsec:gnomad">Population genetics</a></li>
  <li><a href="#phenotype-genotype-and-function" id="markdown-toc-phenotype-genotype-and-function">Phenotype, genotype, and function</a></li>
  <li><a href="#references" id="markdown-toc-references">References</a></li>
</ul>

<h2 id="subsec:gnomad">Population genetics</h2>
<p>GnomAD (version r2.0.2)  <a class="citation" href="#lek2016analysis">(Lek et al., 2016)</a> was used in these studies as
the best source of population genetics data. The reference genome is
GRCh37. Offline local database mirrors were used in most cases. Input
sets used GnomAD variant allele frequencies and reference sequences
processed as VCF and CSV files. outlines a specific data transformation
using the gnomAD database, but in general, gnomAD was used as a
filtering threshold for determining the expected population frequency of
each variant. A strict threshold for rare variants could be set to
ignore and candidate variants that are more frequent than 0.001.
However, in most cases a more lenient level is used and any remaining
benign or common variants are removed by “technical control” (filter on
cohort to remove common variants between individuals that do not share a
phenotype). A more modest cut-off threshold allows us to sometimes
identify variant that are present in the general population, which are
responsible for a recessive disease with no predictable heterozygous
loss-of-function intolerance.</p>

<p>Other sources of population genetics data comes from resources such as
CliVar and dbSNP, which as they grow in size become an annotated and
curated for of population data. These resources allow us to calculate
the expected frequencies for disease-causing variants. However, since
these are manually curated database and predominantly European based,
they are inherently biased and not reliable for statistical
applications.</p>

<h2 id="phenotype-genotype-and-function">Phenotype, genotype, and function</h2>
<p>Population genetics database gnomAD has been individually addressed in
section [subsec:gnomad], as this is the most important type of
annotation and filtering criteria for genetic determinants of rare
disease. Additionally, in these studies many phenotype and genotype
databases have been used for annotation and interpretation.
Specifically, the most frequently used data came from MGI Phenotype,
MorbidMap, VOC MammalianPhenotype, Gencode symbol, UniProtKB, Enterez
ID, ENSGene ID, GO ID, Description, OMIM, BIOGRID interactions, HGMD
human phenotype, ClinVar, and dbSNP. In most cases, every candidate
variant was annotated with the main information per gene from a local
database containing the information from each of the listed resources.</p>

<p>These are the “basic” information databases that we used to annotate
variants. In a cohort study, data mining can find correlations and was
therefore included for posterity as it does not significantly increase
the data storage. Even if an obvious cause of disease was found we may
later return to the data to find other cofactors or genetic modifiers.
Or for example, in a single case study, a variant of unknown
significance may have no statistical basis to be selected or ignored. We
use this information to decide if that mutation is worth consideration:
Is it in a protein domain of known function? Are there other cases
reported with the same phenotype? What is the gene function, ontology,
etc.?</p>

<p>We have also used some gene lists that are specific to disease,
druggability, etc. A major contributor for collecting these gene lists
has been the Mac Arthur et al.  <a class="citation" href="#macarthurgit">(Lab, 2018)</a>. These gene lists can be
used is special cases. For example, a study looking at (1) dominant
pathogenic mutations, and (2) in known immune genes might filter to
included only those known observables. We could decide to only study
SNPs in FDA-approved drug targets.</p>

<div class="table-wrapper">

  <table>
    <thead>
      <tr>
        <th>Mechanism</th>
        <th>Gene Count</th>
        <th>Name</th>
        <th>Reference</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td> </td>
        <td>19,194</td>
        <td>HUGO 2018</td>
        <td><a class="citation" href="#HUGO2018">(at the European Bioinformatics Institute, 2018)</a></td>
      </tr>
      <tr>
        <td>FDA-approved drug targets</td>
        <td>385</td>
        <td>Wishart 2018</td>
        <td><a class="citation" href="#Wishart2018">(Wishart et al., 2017)</a></td>
      </tr>
      <tr>
        <td>Drug targets</td>
        <td>201</td>
        <td>Nelson 2012</td>
        <td><a class="citation" href="#Nelson2012">(Nelson et al., 2012)</a></td>
      </tr>
      <tr>
        <td>Autosomal dominant genes</td>
        <td>307</td>
        <td>Blekhman 2008</td>
        <td><a class="citation" href="#Blekhman2008">(Blekhman et al., 2008)</a></td>
      </tr>
      <tr>
        <td>Autosomal dominant genes</td>
        <td>631</td>
        <td>Berg 2013</td>
        <td><a class="citation" href="#Berg2013">(Berg et al., 2013)</a></td>
      </tr>
      <tr>
        <td>Autosomal recessive genes</td>
        <td>527</td>
        <td>Blekhman 2008</td>
        <td><a class="citation" href="#Blekhman2008">(Blekhman et al., 2008)</a></td>
      </tr>
      <tr>
        <td>Autosomal recessive genes</td>
        <td>1073</td>
        <td>Berg 2013</td>
        <td><a class="citation" href="#Berg2013">(Berg et al., 2013)</a></td>
      </tr>
      <tr>
        <td>X-linked genes</td>
        <td>66</td>
        <td>Blekhman 2008</td>
        <td><a class="citation" href="#Blekhman2008">(Blekhman et al., 2008)</a></td>
      </tr>
      <tr>
        <td>X-linked recessive genes</td>
        <td>102</td>
        <td>Berg 2013</td>
        <td><a class="citation" href="#Berg2013">(Berg et al., 2013)</a></td>
      </tr>
      <tr>
        <td>X-linked dominant genes</td>
        <td>34</td>
        <td>Berg 2013</td>
        <td><a class="citation" href="#Berg2013">(Berg et al., 2013)</a></td>
      </tr>
      <tr>
        <td>X-linked ClinVar genes</td>
        <td>61</td>
        <td>Landrum 2014</td>
        <td><a class="citation" href="#Landrum2014">(Landrum et al., 2013)</a></td>
      </tr>
      <tr>
        <td>All dominant genes</td>
        <td>709</td>
        <td>Blekhman 2008, Berg 2013</td>
        <td>(missing reference)</td>
      </tr>
      <tr>
        <td>All recessive genes</td>
        <td>1183</td>
        <td>Blekhman 2008, Berg 2013</td>
        <td>(missing reference)</td>
      </tr>
      <tr>
        <td>Homozygous LoF tolerant</td>
        <td>330</td>
        <td>Lek 2016</td>
        <td><a class="citation" href="#lek2016analysis">(Lek et al., 2016)</a></td>
      </tr>
      <tr>
        <td>Essential in culture</td>
        <td>283</td>
        <td>Hart 2014</td>
        <td><a class="citation" href="#Hart2014">(Hart et al., 2014)</a></td>
      </tr>
      <tr>
        <td>Essential in culture</td>
        <td>683</td>
        <td>Hart 2017</td>
        <td><a class="citation" href="#Hart2017">(Hart et al., 2017)</a></td>
      </tr>
      <tr>
        <td>Non-essential in culture</td>
        <td>913</td>
        <td>Hart 2017</td>
        <td><a class="citation" href="#Hart2017">(Hart et al., 2017)</a></td>
      </tr>
      <tr>
        <td>Essential in mice</td>
        <td>2,454</td>
        <td>Blake ‘11, Georgi ‘13, Liu ‘13</td>
        <td>(missing reference)</td>
      </tr>
      <tr>
        <td>Genes nearest to GWAS peaks</td>
        <td>6,336</td>
        <td>MacArthur 2017</td>
        <td><a class="citation" href="#MacArthur2017">(MacArthur et al., 2016)</a></td>
      </tr>
      <tr>
        <td>DNA Repair Genes</td>
        <td>178</td>
        <td>Wood 2005</td>
        <td><a class="citation" href="#Wood2005">(Wood et al., 2005)</a></td>
      </tr>
      <tr>
        <td>DNA Repair Genes</td>
        <td>151</td>
        <td>Kang 2012</td>
        <td><a class="citation" href="#Kang2012">(Kang et al., 2012)</a></td>
      </tr>
      <tr>
        <td>ClinGen haploinsufficient genes</td>
        <td>294</td>
        <td>Rehm 2015</td>
        <td><a class="citation" href="#Rehm2015">(Rehm et al., 2015)</a></td>
      </tr>
      <tr>
        <td>Olfactory receptors</td>
        <td>371</td>
        <td>Mainland 2015</td>
        <td><a class="citation" href="#Mainland2015">(Mainland et al., 2015)</a></td>
      </tr>
      <tr>
        <td>Reported in ClinVar</td>
        <td>3078</td>
        <td>Landrum 2014</td>
        <td><a class="citation" href="#Landrum2014">(Landrum et al., 2013)</a></td>
      </tr>
      <tr>
        <td>Kinases</td>
        <td>347</td>
        <td>UniProt 2016</td>
        <td><a class="citation" href="#UniProt2016">(Consortium, 2016)</a></td>
      </tr>
      <tr>
        <td>GPCRs from guide to pharmacology</td>
        <td>391</td>
        <td>Alexander 2017, Harding 2018.</td>
        <td>(missing reference)</td>
      </tr>
      <tr>
        <td>GPCRs from Uniprot</td>
        <td>756</td>
        <td>UniProt 2016</td>
        <td><a class="citation" href="#UniProt2016">(Consortium, 2016)</a></td>
      </tr>
      <tr>
        <td>Natural product targets</td>
        <td>37</td>
        <td>Dancik 2010</td>
        <td><a class="citation" href="#Dancik2010">(Dančı́k Vlado et al., 2010)</a></td>
      </tr>
      <tr>
        <td>BROCA - Cancer Risk Panel</td>
        <td>66</td>
        <td>BROCA Cancer Risk Panel</td>
        <td><a class="citation" href="#BROCACancerRiskPanel">(Deptartment of Laboratory Medicine, n.d.)</a></td>
      </tr>
      <tr>
        <td>ACMG V2.0</td>
        <td>59</td>
        <td>Kalia 2017</td>
        <td><a class="citation" href="#Kalia2016">(Kalia et al., 2016)</a></td>
      </tr>
      <tr>
        <td>GPI-anchored proteins</td>
        <td>135</td>
        <td>UniProt 2016</td>
        <td><a class="citation" href="#UniProt2016">(Consortium, 2016)</a></td>
      </tr>
    </tbody>
  </table>

</div>

<p><a class="citation" href="#verma2018rare">(Verma et al., 2018)</a> take an interesting approach to comparing druggable
targets with population genetics data. DrugBank is a database for over
800 genes with over 950 unique drugs. Genetic data can be filtered for
these genes and targeted for LoF variants. Association analysis consists
of logistic regression using the ICD-9 codes, and linear regression
using quantitative variables. This gene binding and regression analysis
steps are done using BioBin.</p>

<p>The International Statistical Classification of Diseases and Related
Health Problems (commonly known as the ICD) provides alpha-numeric codes
to classify diseases and a wide variety of signs, symptoms, abnormal
findings, complaints, social circumstances and external causes of injury
or disease. Nearly every health condition can be assigned to a unique
category and given a code, up to six characters long. Such categories
usually include a set of similar diseases</p>

<p>BioBin relies on the Library of Knowledge Integration (LOKI), which
integrates multiple databases providing a comprehensive biological
knowledge platform for variant binning  <a class="citation" href="#pendergrass2013genomic">(Pendergrass et al., 2013)</a>. The
LOKI database consolidates biological information from several sources,
most notably the National Center for Biotechnology (NCBI) dbSNP and
Entrez Gene, Kyoto Encyclopedia of Genes and Genomes (KEGG), Reactome,
Gene Ontology (GO), Protein families database (Pfam), NetPath-signal
transduction pathways, amongst others
 (missing reference).</p>

<h2 id="references">References</h2>

<ol class="bibliography">
<li><span id="lek2016analysis">Lek, M., Karczewski, K. J., Minikel, E. V., Samocha, K. E., Banks, E., Fennell, T., O’Donnell-Luria, A. H., Ware, J. S., Hill, A. J., Cummings, B. B., &amp; others. (2016). Analysis of protein-coding genetic variation in 60,706 humans. <i>Nature</i>, <i>536</i>(7616), 285.</span></li>
<li><span id="macarthurgit">Lab, M. A. (2018). List of gene lists for genomic analysis. <i>GitHub</i>. https://github.com/macarthur-lab/gene_lists</span></li>
<li><span id="HUGO2018">at the European Bioinformatics Institute, H. U. G. O. G. N. C. (2018). <i>Genenames.org</i>.</span></li>
<li><span id="Wishart2018">Wishart, D. S., Feunang, Y. D., Guo, A. C., Lo, E. J., Marcu, A., Grant, J. R., Sajed, T., Johnson, D., Li, C., Sayeeda, Z., Assempour, N., Iynkkaran, I., Liu, Y., Maciejewski, A., Gale, N., Wilson, A., Chin, L., Cummings, R., Le, D., … Wilson, M. (2017). DrugBank 5.0: a major update to the DrugBank database for 2018
. <i>Nucleic Acids Research</i>, <i>46</i>(D1), D1074–D1082. https://doi.org/10.1093/nar/gkx1037</span></li>
<li><span id="Nelson2012">Nelson, M. R., Wegmann, D., Ehm, M. G., Kessner, D., St. Jean, P., Verzilli, C., Shen, J., Tang, Z., Bacanu, S.-A., Fraser, D., Warren, L., Aponte, J., Zawistowski, M., Liu, X., Zhang, H., Zhang, Y., Li, J., Li, Y., Li, L., … Mooser, V. (2012). An Abundance of Rare Functional Variants in 202 Drug Target Genes Sequenced in 14,002 People. <i>Science</i>, <i>337</i>(6090), 100–104. https://doi.org/10.1126/science.1217876</span></li>
<li><span id="Blekhman2008">Blekhman, R., Man, O., Herrmann, L., Boyko, A. R., Indap, A., Kosiol, C., Bustamante, C. D., Teshima, K. M., &amp; Przeworski, M. (2008). Natural Selection on Genes that Underlie Human Disease Susceptibility. <i>Current Biology</i>, <i>18</i>(12), 883–889. https://doi.org/https://doi.org/10.1016/j.cub.2008.04.074</span></li>
<li><span id="Berg2013">Berg, J. S., Adams, M., Nassar, N., Bizon, C., Lee, K., Schmitt, C. P., Wilhelmsen, K. C., &amp; Evans, J. P. (2013). An informatics approach to analyzing the incidentalome. <i>Genetics in Medicine</i>, <i>15</i>(1), 36.</span></li>
<li><span id="Landrum2014">Landrum, M. J., Lee, J. M., Riley, G. R., Jang, W., Rubinstein, W. S., Church, D. M., &amp; Maglott, D. R. (2013). ClinVar: public archive of relationships among sequence variation and human phenotype
. <i>Nucleic Acids Research</i>, <i>42</i>(D1), D980–D985. https://doi.org/10.1093/nar/gkt1113</span></li>
<li><span id="Hart2014">Hart, T., Brown, K. R., Sircoulomb, F., Rottapel, R., &amp; Moffat, J. (2014). Measuring error rates in genomic perturbation screens: gold standards for human functional genomics. <i>Molecular Systems Biology</i>, <i>10</i>(7).</span></li>
<li><span id="Hart2017">Hart, T., Tong, A. H. Y., Chan, K., Van Leeuwen, J., Seetharaman, A., Aregger, M., Chandrashekhar, M., Hustedt, N., Seth, S., Noonan, A., &amp; others. (2017). Evaluation and design of genome-wide CRISPR/SpCas9 knockout screens. <i>G3: Genes, Genomes, Genetics</i>, <i>7</i>(8), 2719–2727.</span></li>
<li><span id="MacArthur2017">MacArthur, J., Bowler, E., Cerezo, M., Gil, L., Hall, P., Hastings, E., Junkins, H., McMahon, A., Milano, A., Morales, J., Pendlington, Z. M., Welter, D., Burdett, T., Hindorff, L., Flicek, P., Cunningham, F., &amp; Parkinson, H. (2016). The new NHGRI-EBI Catalog of published genome-wide association studies (GWAS Catalog)
. <i>Nucleic Acids Research</i>, <i>45</i>(D1), D896–D901. https://doi.org/10.1093/nar/gkw1133</span></li>
<li><span id="Wood2005">Wood, R. D., Mitchell, M., &amp; Lindahl, T. (2005). Human DNA repair genes, 2005. <i>Mutation Research/Fundamental and Molecular Mechanisms of Mutagenesis</i>, <i>577</i>(1), 275–283. https://doi.org/https://doi.org/10.1016/j.mrfmmm.2005.03.007</span></li>
<li><span id="Kang2012">Kang, J., Dâ€™Andrea, A. D., &amp; Kozono, D. (2012). A DNA Repair Pathwayâ€“Focused Score for Prediction of Outcomes in Ovarian Cancer Treated With Platinum-Based Chemotherapy
. <i>JNCI: Journal of the National Cancer Institute</i>, <i>104</i>(9), 670–681. https://doi.org/10.1093/jnci/djs177</span></li>
<li><span id="Rehm2015">Rehm, H. L., Berg, J. S., Brooks, L. D., Bustamante, C. D., Evans, J. P., Landrum, M. J., Ledbetter, D. H., Maglott, D. R., Martin, C. L., Nussbaum, R. L., Plon, S. E., Ramos, E. M., Sherry, S. T., &amp; Watson, M. S. (2015). ClinGen — The Clinical Genome Resource. <i>New England Journal of Medicine</i>, <i>372</i>(23), 2235–2242. https://doi.org/10.1056/NEJMsr1406261</span></li>
<li><span id="Mainland2015">Mainland, J. D., Li, Y. R., Zhou, T., Liu, W. L. L., &amp; Matsunami, H. (2015). Human olfactory receptor responses to odorants. <i>Scientific Data</i>, <i>2</i>, 150002 EP  -. https://doi.org/10.1038/sdata.2015.2</span></li>
<li><span id="UniProt2016">Consortium, T. U. P. (2016). UniProt: the universal protein knowledgebase. <i>Nucleic Acids Research</i>, <i>45</i>(D1), D158–D169. https://doi.org/10.1093/nar/gkw1099</span></li>
<li><span id="Dancik2010">Dančı́k Vlado, Seiler, K. P., Young, D. W., Schreiber, S. L., &amp; Clemons, P. A. (2010). Distinct Biological Network Properties between the Targets of Natural Products and Disease Genes. <i>Journal of the American Chemical Society</i>, <i>132</i>(27), 9259–9261. https://doi.org/10.1021/ja102798t</span></li>
<li><span id="BROCACancerRiskPanel">Deptartment of Laboratory Medicine, U. of W. <i>BROCA-CancerRiskPanel</i>. http://depts.washington.edu/labweb/Divisions/MolDiag/MolDiagGen/index.htm</span></li>
<li><span id="Kalia2016">Kalia, S. S., Adelman, K., Bale, S. J., Chung, W. K., Eng, C., Evans, J. P., Herman, G. E., Hufnagel, S. B., Klein, T. E., Korf, B. R., McKelvey, K. D., Ormond, K. E., Richards, C. S., Vlangos, C. N., Watson, M., Martin, C. L., &amp; Miller, D. T. (2016). Recommendations for reporting of secondary findings in clinical exome and genome sequencing, 2016 update (ACMG SF v2.0): a policy statement of the American College of Medical Genetics and Genomics. <i>Genetics In Medicine</i>, <i>19</i>, 249 EP  -. https://doi.org/10.1038/gim.2016.190</span></li>
<li><span id="verma2018rare">Verma, S. S., Josyula, N., Verma, A., Zhang, X., Veturi, Y., Dewey, F. E., Hartzel, D. N., Leader, J., Ritchie, M. D., &amp; Pendergrass, S. A. (2018). Rare variants in drug target genes contributing to complex diseases, phenome-wide. <i>Scientific Reports</i>, <i>8</i>(1), 4624.</span></li>
<li><span id="pendergrass2013genomic">Pendergrass, S. A., Frase, A., Wallace, J., Wolfe, D., Katiyar, N., Moore, C., &amp; Ritchie, M. D. (2013). Genomic analyses with biofilter 2.0: knowledge driven filtering, annotation, and model development. <i>BioData Mining</i>, <i>6</i>(1), 25.</span></li>
</ol>

</div>

<!--

<div id="related">
  <h2>Related Topics</h2>
  <ul class="topics">
    
  </ul>
</div>

-->

















</div>
<!-- # MAIN END ---------------------------------------------------------- -->

<!-- FOOT START -------------------------------------------------------- -->
<div class="sitewidthnav">
	<div class="footer">
		<div class="foot-container">
				<a href="/" class="logo-foot">
				<img style="margin-bottom:-16px" src="/images/logos/logo4.2_150ppi.png" alt="Logo image" width="25" height="38.5"></a>
					<div class="right-root">
						<p><a href="https://github.com/DylanLawless/DylanLawless.github.io/tree/source">Maintained</a> by Dylan Lawless. Scientist at <a href="https://people.epfl.ch/dylan.lawless">EPFL</a>.
						<a href="/resume">Resume</a>
						and <a href="https://scholar.google.com/citations?user=RPBxP1wAAAAJ&hl=en&oi=ao">Google scholar</a>.
						<a href="https://lawlessgenomics.com/feed.xml">RSS feed</a>
						</p>
					</div>
		</div>
	</div>
</div>
<!-- FOOT END ---------------------------------------------------------- -->

</div>
<!-- SITE END ----------------------------------------------------------- -->


<!-- Banner on top right corner linking to github
<a href="https://github.com/mojombo"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_red_aa0000.png" alt="Fork me on GitHub" /></a>
-->

<!-- ClickTale Bottom part -->
<div id="ClickTaleDiv" style="display: none;"></div>
<script type="text/javascript">
if(document.location.protocol!='https:')
  document.write(unescape("%3Cscript%20src='https://s.clicktale.net/WRb.js'%20type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
if(typeof ClickTale=='function') ClickTale(206,0.3,"www03");
</script>
<!-- ClickTale end of Bottom part -->

<!-- Google Analytics -->
<!--
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "https://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
var pageTracker = _gat._getTracker("UA-6016902-1");
pageTracker._trackPageview();
</script>
-->
<!-- Google Analytics end -->

</body>
</html>
