<!DOCTYPE html>
<html xmlns="https://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
	<!-- link to MathJax for LaTeX stlye eqn -->
	 <script type="text/javascript" async
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-MML-AM_CHTML">
</script>

	 
	 <!-- https://realfavicongenerator.net -->
	 <link rel="apple-touch-icon" sizes="180x180" href="/images/logos/favicon_package_v0.16_bw/apple-touch-icon.png">
	 <link rel="icon" type="image/png" sizes="32x32" href="/images/logos/favicon_package_v0.16_bw/favicon-32x32.png">
	 <link rel="icon" type="image/png" sizes="16x16" href="/images/logos/favicon_package_v0.16_bw/favicon-16x16.png">
	 <link rel="manifest" href="/images/logos/favicon_package_v0.16_bw/site.webmanifest">
	 <meta name="msapplication-TileColor" content="#da532c">
	 <meta name="theme-color" content="#ffffff">
	
	 <meta name="viewport" content="width=device-width, initial-scale=1">
	
	<meta http-equiv="content-type" content="text/html; charset=utf-8" />
	<meta name="description"
content="A home for topics in human precision medicine, genomic analysis, and data visualisation.">
	<title>Rare disease cohort analysis</title>
	<meta name="author" content="LawlessGenomics" />
	<!-- <link href="https://feeds.feedburner.com/tom-preston-werner" rel="alternate" title="Dyaln feed title" type="application/atom+xml" /> -->
	<!-- <meta name="readability-verification" content="QCzSs992GxmRYRKVpPeZ6LE2tS8aYKxsSSQKV8YM"/> -->
	
	<!-- include a google font -->
	<link href="https://fonts.googleapis.com/css?family=Roboto+Mono|Roboto+Slab:300|Roboto:500" rel="stylesheet">
	<link href="https://fonts.googleapis.com/css?family=Lora" rel="stylesheet">
	
	<!-- syntax highlighting CSS -->
	<link rel="stylesheet" href="/css/syntax.css" type="text/css" />
	
	<!-- Homepage CSS -->
	<link rel="stylesheet" href="/css/screen.css" type="text/css" media="screen, projection" />
	
	<!-- Typekit -->
	<script type="text/javascript" src="https://use.typekit.com/jpd0pfm.js"></script>
	<script type="text/javascript">try{Typekit.load();}catch(e){}</script>
</head>
<body>

<!-- ClickTale Top part -->
<script type="text/javascript">
	var WRInitTime=(new Date()).getTime();
</script>
<!-- ClickTale end of Top part -->

<!-- Logo and navigation bar -->
<div class="site">
	<div class="title">
		<div class="nav-container">
			<nav>

			<div class="logoimage">
				<a href="/">
				<img style="margin-bottom:-16px" 
				src="/images/logos/logo4.1_150ppi.png" 
				alt="Logo image" width="35.7" height="55" /></a>
			</div>

			<div class="logo">
				<a href="/">LawlessGenomics</a>
			</div>

				<input type="checkbox" id="nav-toggle">
					<label for="nav-toggle" class="gg-menu"></label>
					<div class="right-menu">
						<a href="/resume">About</a>
						<a href="/">Home</a>
					</div>
			</nav>
		</div>
	</div>

	<div id="topic">
<h1 id="rare-disease-cohort-analysis">Rare disease cohort analysis</h1>

<ul id="markdown-toc">
  <li><a href="#rare-disease-cohort-analysis" id="markdown-toc-rare-disease-cohort-analysis">Rare disease cohort analysis</a></li>
  <li><a href="#sec:cohort_network_analysis" id="markdown-toc-sec:cohort_network_analysis">Rare disease cohort network analysis</a>    <ul>
      <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
      <li><a href="#exome-analysis" id="markdown-toc-exome-analysis">Exome analysis</a></li>
      <li><a href="#cluster-list-preparation" id="markdown-toc-cluster-list-preparation">Cluster list preparation</a></li>
      <li><a href="#sec:net_construction" id="markdown-toc-sec:net_construction">Network construction</a></li>
      <li><a href="#sec:random_sample" id="markdown-toc-sec:random_sample">Random sampling</a></li>
      <li><a href="#expanding-damaged-gene-mcl-clusters" id="markdown-toc-expanding-damaged-gene-mcl-clusters">Expanding damaged gene MCL clusters</a></li>
      <li><a href="#sec:burden_rank" id="markdown-toc-sec:burden_rank">Burden rank</a></li>
      <li><a href="#sec:number_test" id="markdown-toc-sec:number_test">Determining the number of tests <em>m</em></a></li>
      <li><a href="#significance-testing" id="markdown-toc-significance-testing">Significance testing</a></li>
      <li><a href="#sec:enrichment_test" id="markdown-toc-sec:enrichment_test">Enrichment testing</a></li>
    </ul>
  </li>
</ul>

<h1 id="sec:cohort_network_analysis">Rare disease cohort network analysis</h1>
<h2 id="introduction">Introduction</h2>
<p>The exome sequencing is most commonly used for genetic diagnosis in
single use cases. Over the next decade exome and genome sequencing will
become very commonplace for the average person at least in high-GDP
countries. A massive expanse in population genetics data will provide
the information that GWAS studies have always sought. We will still be
left with large genomic black holes; that is conserved coding and
non-coding regions that are obviously important as the determinants of
human health within Mendelian genetics. To uncover the function of these
genetic loci we will still be at the mercy of cohort size in rare
disorder studies. For true rare genetic disorders, a disease frequency
of 0.01% equates to approximately 800,000 cases worldwide for diseases
that are not embryonic lethal and where lifespan is about normal. If we
consider high income populations where genomic sequencing would be
common, then we may have 100,000 cases. A very well organised rare
disease study would do well to recruit 1 per 1000 in a clearly defined
disease; this is currently the situation although most studies cannot
finance full genomic investigation. Therefore, now and well into the
future, rare disease studies will generally be limited to a maximum
number of living participants on the scale of hundreds.</p>

<p>Current best practices in genomic analysis will first identify “low
hanging fruit”; single cases in a cohort with a clear genetic
determinant (e.g. haploinsufficiency of a well-defined dominant gene).
The second order will identify commonly mutated genes or loci based on
burden testing comparing cases to controls or background population
genetics. Many disorders have a phenotype that can be derived from
mutations in several different genes. The encoded genes generally are a
part of the same protein pathway, even directly upstream and downstream
of each other in some cases. For example, covers this topic with
individual cases of RAG1 and RAG2 deficiency.</p>

<p>Proposed here is a statistically robust and unbiased method to find
variants in protein-coding genes that share a common functional protein
pathway for a disease cohort. <strong>Figure [fig:method]</strong> provides a
high-level graphical summary of the concept. <strong>Figure [fig:abstract]</strong>
conveys the theory of the procedures for this method in more detail with
the major datasets explicitly shown.</p>

<p><strong>Deleterious rare variants in damaged protein pathways in rare
disease.</strong> A. GATK best practices were used for whole exome analysis
with joint genotyping for cases and controls; 200 in total. Custom
filtering [@vcfhacks2015Parry] extracted variants of high impact
consequence (ostensibly loss-of-function (LoF)), present only in cohort
cases. B. Genes harboring rare predicted LoF variants were grouped based
on protein-protein interactions [@String2017Szklarczyk] using a Markov
cluster algorithm [@Enright2002efficienct]. C. Case-control testing was
performed on each protein pathway cluster.</p>

<p><strong>FigureLabel</strong>
method
<img src="/images/net_analysis/method.png" width="40%" /></p>

<p><strong>Rare variant analysis and protein pathway significant enrichment.</strong>
A. DNA is collected and sequenced. B. Routine genomic analysis is
carried out according to best practices, for both (i) control and (ii)
case groups of patients. First, all rare variants are output, followed
by a smaller subset of loss-of-function (LoF) variants. C.
Variant-carrying genes are cluster of protein-protein interactions based
on function and ontology. D. A clustering method is applied to break a
large highly connected network into smaller individual ones. E. The
number of tests can be reduced by, for example, testing only networks
that carry a threshold level of LoFs and are therefore biologically
relevant to disease. F. Deleterious variant load per network was tested
for enrichment in cases, controls, or random sampling. G. Multiple
testing correction is applied to identify the critical significant
threshold.
<strong>FigureLabel</strong>
abstract
<img src="/images/net_analysis/abstract.pdf" width="40%" /></p>

<h2 id="exome-analysis">Exome analysis</h2>
<p>Exome sequencing analyses has been discussed in detail. The rare disease
cohort network analysis requires less tailored analysis steps than
traditional variant interpretation. Therefore, the data preparation is
briefly outlined here.</p>

<p>Sequences were trimmed and quality controlled using FastQC via
Trim-galore. Reads were aligned to GrCh37 using BWA-MEM. GATK “best
practices” were used for marking duplicate reads, recalibration, and
whole cohort variant quality score recalibration before generating
genomic VCFs with HaplotypeCaller and joint genotyping. Filtering and
prediction of functional consequences was performed using Variant Effect
Predictor<br />
(http://www.ensembl.org/info/docs/tools/vep/index.html), Exome Variant
Server<br />
(http://evs.gs.washington.edu/EVS/), The Single Nucleotide Polymorphism
database<br />
(https://www.ncbi.nlm.nih.gov/projects/SNP/) and ClinVar
(https://www.ncbi.nlm.nih.gov/ clinvar/), The Exome Aggregation
Consortium and The Genome Aggregation Database<br />
(http://gnomad.broadinstitute.org). Filtering of common variations and
annotation was performed using VCFhacks<br />
(https://github.com/gantzgraf/vcfhack). Candidate variants were required
to pass the following filtering conditions: frequency (count/coverage)
between 20-100%, according to VEP-annotation at least one canonical
transcript is affected with one of the following consequence: variants
of the coding sequence, frameshift, missense, protein altering, splice
acceptor, splice donor, or splice region; an inframe insertion or
deletion; a start lost, stop gained, or stop retained, or according to
VEP an GnomAD frequency unknown, &lt;=0.01, or with clinical
significance ‘path’. VCFhacks [@vcfhacks2015Parry] was used for
cohort-specific filtering retained functional variants that were present
in at least one case but absent in controls (for case-driven PPI
clustering). The same criteria were used to also collect functional
variants that were present in at least one control but absent in
controls (for control-driven PPI clustering).</p>

<h2 id="cluster-list-preparation">Cluster list preparation</h2>
<p>Group-specific variant data was extracted from the joint cohort.
Specifically, the datasets came from the routine analysis pipeline show
in <strong>Figure [fig:analysis_flow]</strong> as the output of the process
“filter on Sample” and converted from VCF to tsv format using the
process “annovcftoSimple” using the tool VCFhacks [@vcfhacks2015Parry].
Four gene lists were prepared consisting of the following groups; (1)
variants present in controls and (2) variants present in cases and
further divided for genes that harboured either (i) all rare variants or
(ii) only potential loss-of-function variants. Specifically, the
datasets for (i) all rare variants came from the output of the process
“filter on sample” via “get functional variants”. The datasets for (ii)
potential loss-of-function variants is a subset of (i), processed by the
R script at the step where “damaging variants” are written out to file.</p>

<p>Analysis workflow structure. Tools used are shown in square boxes.
Reference data used secondary to inputs are shown as light boxes with
curved sides. Key output files are shown by light slanted boxes. Storage
structure is divided between long-term and short-term
storage.
<strong>FigureLabel</strong>
analysis_flow
<img src="/images/net_analysis/analysis_flow.pdf" width="80%" /></p>

<h2 id="sec:net_construction">Network construction</h2>
<p>Group-specific gene lists [1 (i-ii) and 2 (i-ii)] were assessed for
PPI using the STRING database [@String2017Szklarczyk] via Cytoscape
[@Shannon2003cytoscape]. An initial PPI network was generated for each
of the 4 dataset groups. The STRINGdb default confidence score cut-off
(0.4) was used for these tests. This score is the measure of evidence
required to create an interaction between two nodes. A stricter value
can be set if networks are too large. Query genes were defined as nodes,
PPI were defined as edges, and networks of proteins linked through PPI
were defined as clusters. Clusters or networks can also be generally
considered as making up a part of a protein pathway.</p>

<p>[*[5]{}[Z]{}]{}</p>

<p><br />
&amp; [Network cluster]{} &amp; [Number of nodes]{} &amp; [Number of edges]{} &amp;
[Number of clusters]{}<br />
&amp; [Total ]{} &amp; [1956]{} &amp; [9559]{} &amp; [114]{}<br />
&amp; [No edges]{} &amp; [1]{} &amp; [0]{} &amp; [107]{}<br />
&amp; [One edge]{} &amp; [2]{} &amp; [1]{} &amp; [6]{}<br />
&amp; [Large multi-edge]{} &amp; [1837]{} &amp; [9553]{} &amp; [1]{}<br />
&amp; [Total]{} &amp; [2305]{} &amp; [14139]{} &amp; [102]{}<br />
&amp; [No edges]{} &amp; [1]{} &amp; [0]{} &amp; [77]{}<br />
&amp; [One edge]{} &amp; [2]{} &amp; [1]{} &amp; [3]{}<br />
&amp; [Two edges]{} &amp; [3]{} &amp; [2]{} &amp; [1]{}<br />
&amp; [Large multi-edge]{} &amp; [2219]{} &amp; [14134]{} &amp; [1]{}\</p>

<p><strong>Table [table:node_summary]</strong> lists the characteristics of PPIs for
genes found to harbour functional, potential LoF rare variants in cases
and controls (i.e. gene lists 2 [i-ii]). Most query proteins were seen
to cluster into one large multi-edge node which contained many weak
interactions. The data used in this table is represented again visually
in <strong>Figure [fig:damage_list_case]</strong>. Each dot, or node represents a
protein-coding gene that has at least one potentially damaging variant.
The edges, or lines connecting nodes, represent known PPI data that link
proteins. This visual information clearly illustrates the body of
functional protein data that can be included in variant analysis.
However, since nearly every protein has some potential evidence of
effect on many other proteins, then no clear definable protein pathway
can be seen.</p>

<p><strong>Genes harbouring potentially damaging variants in a disease
cohort.</strong> A visual representation of PPI occurring in all genes that
harbour potentially damaging functional variants in a typical disease
cohort.
<strong>FigureLabel</strong>
damage_list_case
<img src="/images/net_analysis/damage_list_case.png" width="50%" /></p>

<p>To segregate protein pathways and refine the number of genes (nodes) in
each cluster, the Markov cluster algorithm (MCL) was used
[@van2000graph; @Enright2002efficienct]. The principal data-specific
adjustment required for using MCL is the inflation operator, which
regulates cluster granularity or tightness. The optimum inflation
parameter for separating protein pathways was found to be 2.5, using a
measure of uniform distribution across datasets. <strong>Figure
[fig:inflation]</strong> illustrates an optimal inflation of a large PPI
network into smaller, clearly defined protein pathway clusters.</p>

<p>As a reference example, <strong>table [table:node_edge]</strong> lists three
inflation parameters tested for most consistent separation (2.5, 3, 4)
and shows the effect of adjustment on the total number of edges (protein
interactions). The median number of nodes (query proteins) are shown for
cases and controls (also shown as total number of nodes in <strong>table
[table:node_summary]</strong>).</p>

<p>[@l@l@]{} [0pt][0pt]</p>

<p>[Inflation separates protein pathways][<strong>=0 Inflation separates
protein pathways.</strong> A visual representation the ideal inflation
parameter used on a PPI cluster. Weak bonds are broken and strong bonds
draw nodes closer together. No bonds are retained between clusters. With
this type of inflation each protein network cluster can be investigated
without considering overlaps.]{} [fig:inflation]</p>

<p>[*[4]{}[Z]{}]{}</p>

<p><br />
&amp; &amp; [Total count median]{} &amp; [Node/Edge ratio ]{}<br />
&amp; &amp; [Case/control $\pm$ S.D.]{} &amp; [Case/control $\pm$ S.D.]{}<br />
&amp; &amp; [2130.5 $\pm$ 246.78]{} &amp;<br />
&amp; [PPI only *]{} &amp; [11849 $\pm$ 3238.55]{} &amp; [0.18 $\pm$ 0.03]{}<br />
&amp; [Inflation 2.5]{} &amp; [2787.5 $\pm$ 740.34]{} &amp; [0.78 $\pm$ 0.12]{}<br />
&amp; [Inflation 3]{} &amp; [4229.5 $\pm$ 3669.18]{} &amp; [0.77 $\pm$ 0.61]{}<br />
&amp; [Inflation 4]{} &amp; [1199.5 $\pm$ 146.37]{} &amp; [1.78 $\pm$ 0.01]{}\</p>

<p><strong>Figure [fig:network_size_inflation]</strong> illustrates the effect of
adjusting the inflation parameter for MCL clustering on protein
networks. After MCL clustering, cases and controls were found to group
into 928 and 1034 networks clusters respectively. Of these, 494 and 568
were single-node (single-protein) “clusters” which shared no interaction
with another protein while 434 and 466 clusters had at least one
interaction between proteins. The cumulative probability plot (figure
[fig:cumulative_sum_net_rank]) shows the cumulative sum of proteins
per network against network rank size. <strong>Figure [fig:qqplot_2]</strong>
shows qqplot for the same data for distribution compared between groups
after inflation at 2.5.</p>

<p><strong>Effect of inflation on network size distribution.</strong> The outcome on
network size is demonstrated to compare effect of two inflation
parameters. An ideal separation of networks should provide an geometric
decrease in the number of proteins per network regardless of the sample
group. Inflation parameter 2.5 produced the ideal distribution while
inflation parameter 3 produced one large, poorly separated network and a
large increase in single-protein nodes on one group. Binwidth of
10.
<strong>FigureLabel</strong>
network_size_inflation
<img src="/images/net_analysis/network_size_inflation_free.pdf" width="40%" /></p>

<p><strong>Cumulative sum of network rank by size</strong>. The effect of inflation on
network size distribution could be potentially measured automatically by
quantifying the cumulative sum of network rank by size and determining
the best inflation parameter to use. This process would reduce user
bias.
<strong>FigureLabel</strong>
cumulative_sum_net_rank
<img src="/images/net_analysis/cumulative_sum_net_rank.pdf" width="40%" /></p>

<p><strong>QQ plot illustrating uniform inflation.</strong> The data presented in
figure [fig:cumulative_sum_net_rank] is used to produce the
quantile-quantile plot for the most uniform distribution between the
case and control groups after all inflation parameters were
tested.
<strong>FigureLabel</strong>
qqplot_2
<img src="/images/net_analysis/qqplot_2.pdf" width="40%" /></p>

<p><strong>Figure [fig:network_size_nodes]</strong> shows the number of proteins per
network. For example, 235 clusters (470 protein nodes) were seen for
cases where only one interaction was shared between two proteins. A
median of 0.78 nodes-per-edge (proteins-per-interaction) was found in
the cases group; naturally the majority of edges appear in large network
clusters and therefore the frequency of nodes-per-edge increases as
network sizes decrease.</p>

<p>Number of proteins per network for case-driven clustering. The size of
protein networks has a geometric distribution that decreases until
protein (nodes) with no interactions remain; in this cases approximately
200 out of 400 proteins did not play a major role in a single
pathway.
<strong>FigureLabel</strong>
network_size_nodes
<img src="/images/net_analysis/network_size_nodes.pdf" width="40%" /></p>

<h2 id="sec:random_sample">Random sampling</h2>
<p>With our group-specific gene lists [1 (i-ii) and 2 (i-ii)], prepared
in section [sec:net_construction], we found the distribution of genes
per networks and output the list of genes in each network for all 4
datasets. The mean number of genes per network rank was found between
cases and controls, again for (i) all rare variants and (ii) only
potential loss-of-function variants. A third gene cluster list was
produced by random sampling gene symbols in artificial networks equal to
the same median size as case-driven and control-driven clusters in from
datasets (i) and (ii). The resulting dataset [3 (i-ii)] mirrors those
of cases and controls but instead of true PPI networks, the networks
contained randomly assigned genes.</p>

<h2 id="expanding-damaged-gene-mcl-clusters">Expanding damaged gene MCL clusters</h2>
<p>For each of the 4 MCL-clustered datasets, cases and controls [1 (i-ii)
and 2 (i-ii)], the cluster ID and list of gene symbols was extracted.
The gene lists of network clusters made from datasets (ii) (potential
LoF) were used to find the network clusters in (i), all-variant gene
clusters, the contained the same overlapping genes. This occurs where
list (ii) is a subset of list (i). The clusters that contained gene
overlaps were extracted since they contained at least one potential LoF
per network. Using this output, the total variant load in “damaged
pathways” could be compared. For clarity, this procedure is summarised
again in Box [box:expanding]; items <strong>A-B</strong>. Item <strong>C</strong> outlines the
remaining steps. <strong>Figure [fig:damage_list_case]</strong> illustrates the
effect of inflation with an ideal inflation parameter. The large network
of PPI were separated into individually contained protein networks.</p>

<p>[box:expanding]</p>

<h2 id="sec:burden_rank">Burden rank</h2>
<p>Our downstream case-control testing compares the mean total variant load
per network. To prevent dilution of our significance threshold due to
multiple testing an arbitrarily high number of networks we assumed that
protein networks harbouring loss-of-function variants at a consistent
frequency in all groups were unlikely to contain genes of interest. To
remove these networks, we firstly found (<em>p</em>) the ratio of LoF to all
variants within the group per network, and secondly found (<em>q</em>); the
ratio of <em>p</em> between groups per network. Networks were ranked by value
<em>q</em>. Values passing a threshold of 0.7 were included for total variant
load means testing (i.e. 70% of ostensibly damaging variants occurred in
cases regardless of the proportion of total variants). This also has the
effect that even if there is no significant difference in a case/control
total-variant means test downstream, potential false negatives may be
rescued by checking for LoF enrichment. This method is applied to real
data in section [sec:enrichment_test] and <strong>table
[table:bh_real]</strong>.</p>

<h2 id="sec:number_test">Determining the number of tests <em>m</em></h2>
<p>The number of tests should be determined by the predefine LoF ratio per
network, <em>q</em>. This value is arbitrarily set and has the problem that an
investigator can decide to use a higher threshold to nude the critical
significant threshold in a desired direction. However, testing roughly
the top 20-30% of networks is suggested. In our experiments we set our
test number as the top 25% of burden-ranked networks. This will be
approximately 10 networks to test (the asymptote of network numbers peak
when the study size increases over approx. 400 samples as all of the
possible PPIs are saturated once the maximum queryable genes are
included). Study sizes that are much larger than this will likely only
(1) be for disease that are not very rare and (2) be large enough to
start expecting single gene significance levels without requiring
network analysis. However, some very strict filtering rules could allow
larger studies with this method.</p>

<h2 id="significance-testing">Significance testing</h2>
<p>We hypothesised firstly that no variant enrichment would be seen in
random sampling or control-driven gene clusters, and secondly enrichment
would only be seen in case-driven clusters for protein-pathways that
provide susceptibility to viral infection. For measuring a significant
enrichment of functional variants in a protein network, there are three
factors to consider.</p>

<ol>
  <li>
    <p>Our aim is to do a comparison of means between case and control, for
total variant load per network.</p>
  </li>
  <li>
    <p>This is done in three iterations; [1] control-driven, [2]
case-driven, and [3] random sample-driven.</p>
  </li>
  <li>
    <p>We correct our significance threshold to account for multiple
testing using the Benjamini-Hochberg procedure.</p>
  </li>
</ol>

<p>With our group-specific gene lists [1 (i-ii), 2 (i-ii), and [3
(i-ii)], prepared in sections [sec:net_construction] and
[sec:random_sample], we found the distribution of genes per networks
and output the list of genes in each network for all 6 datasets. In each
of the 3 “all variant” datasets we simply do a comparison of means for
total variant load per network comparing case to control, or random.</p>

<p>While the test is not complicated, the significance threshold deserves
an in-depth explanation; this is a novel method and most people
replicating this study will not have experience with the statistical
procedures required. The statistical significance also only allows a
narrow margin for successful discovery. When a large number of tests are
performed, one is likely to produce P-values that are “statistically
significant” by chance (P &lt; 0.05), even if the null hypothesis is
true. The null hypothesis would state that “random controls and people
with disease have the same average frequency of potentially pathogenic
variants in some protein pathway”. The alternate hypothesis would state
that “people with disease have an increased frequency of potentially
pathogenic variants in some protein pathway than random controls”.</p>

<p>Traditionally, Bonferroni correction has been used in cases like this.
For each “family” (network means test) being tested one must correct the
critical P-value. For example, for one test a significant P-value might
be 0.05 and below this we consider the result to be significant. The
chance of getting this result if the null hypothesis was true would be
5%. That does not mean that there is 5% chance that it is true. The
following examples are reiterated summary of the topic found in the
Handbook of biological statistics [@mcdonald2009handbook].</p>

<p>For multiple tests of “families” then we need to adjust the P-value
since we are more likely to get false positives by chance. In a
published example, @garcia2014calorie tested 25 associations with
mammographic density, which is an important risk factor for breast
cancer. The 25 “families” tested were dietary variables including “Total
calories”, “Olive oil”, “whole milk”, “white meat”, etc. For each
variable a P-value was given for its association with mammographic
density, i.e. total calories P &lt; 0.001, olive oil P = 0.008, whole
milk P = 0.039.</p>

<p>To perform a Bonferroni correction, the critical P-value (or significant
threshold) should be divided by the number of tests, 0.05/25 = 0.002.
Therefore, only “total calories” would be significantly associated with
the risk factor. If 75 more variables were measured (100 total) then the
critical P-value would have to be 0.05/100 = 0.0005. However, it may not
be reasonable to invalidate the significance of the original findings.
Using Bonferroni correction for family-wise error rate can mean
extremely small P-values. So instead we use a more powerful method for
controlling the false discovery rate; the Benjamini-Hochberg procedure
[@simes1986improved; @benjamini1995controlling].</p>

<p>In this procedure, we compare each individual P-value to its
Benjamini-Hochberg critical value, $(i/m)Q$, where $i$ is the rank, $m$
is the total number of tests, and $Q$ is the chosen false discovery
rate. The largest P-value that has $P&lt;(i/m)Q$ (i.e. P less than
BH-critical value) is significant, and all of the P-values smaller than
it are also significant, even the ones that aren’t less than their own
Benjamini-Hochberg critical value.</p>

<p>So in the same example, with 25 tests and Benjamini-Hochberg critical
value for a false discovery rate set to 0.25, table
[table:bh_example] shows the outcome. The largest P-value that is
less than its $(i/m)Q$ values is 0.042 for protein. Therefore, the first
5 variables are significantly associated, including whole milk and white
meat despite the fact that their BH-critical value is higher than their
P-value. If we were to never have measured protein in this example, $m$
the number of tests would be 24, slightly increasing the BH-critical
value, and again identify a significant association for the first 4
tests. Someone interested can recalculate this table to see this effect.</p>

<p>[*[2]{}[Z]{}]{}</p>

<p><br />
&amp; [P value]{}<br />
&amp; [&lt;0.001]{}<br />
[Olive oil]{} &amp; [0.008]{}<br />
[Whole milk]{} &amp; [0.039]{}<br />
[White meat]{} &amp; [0.041]{}<br />
[Proteins]{} &amp; [0.042]{}<br />
[Nuts]{} &amp; [0.06]{}<br />
[Cereals and pasta]{} &amp; [0.074]{}<br />
[White fish]{} &amp; [0.205]{}<br />
[Butter]{} &amp; [0.212]{}<br />
[Vegetables]{} &amp; [0.216]{}<br />
[Skimmed milk]{} &amp; [0.222]{}<br />
[Red meat]{} &amp; [0.251]{}<br />
[Fruit]{} &amp; [0.269]{}<br />
[Eggs]{} &amp; [0.275]{}<br />
[Blue fish]{} &amp; [0.34]{}<br />
[Legumes]{} &amp; [0.341]{}<br />
[Carbohydrates]{} &amp; [0.384]{}<br />
[Potatoes]{} &amp; [0.569]{}<br />
[Bread]{} &amp; [0.594]{}<br />
[Fats]{} &amp; [0.696]{}<br />
[Sweets]{} &amp; [0.762]{}<br />
[Dairy products]{} &amp; [0.94]{}<br />
[Semi-skimmed milk]{} &amp; [0.942]{}<br />
[Total meat]{} &amp; [0.975]{}<br />
[Processed meat]{} &amp; [0.986]{}\</p>

<p>The choice of a false discover rate depends on the application. False
positives can waste time, resources, and pollute future work. Minimising
false negatives could result in missing a very important finding, that
is, when there is a real effect but it is not deemed statistically
significant. Allowing a pre-determined level false negatives is often
reasonable. As in our application, finding enriched protein networks is
the main goal, and downstream work will also be done such as clinical
interpretation or functional studies which will catch false negatives.
Therefore, the false discovery rate does not have to be very small;
consider that our input dataset is already filtered down to ostensibly
damaging rare variants. Furthermore, the input dataset is essentially
the result of traditional best practices in exome or genome sequencing
analysis.</p>

<h2 id="sec:enrichment_test">Enrichment testing</h2>
<p>For all networks, the top 30 networks in size (largest to smallest;
1-30) were ordered using the burden rank (sec [sec:burden_rank]).
From these, the number of tests was set (according to the rules defined
in sec [sec:number_test], so that only the top 8 burden-ranked
networks were means tested for their total variant load. <strong>Figure
[fig:means_test]</strong> shows the test of means for the top 8 protein
pathway networks. Table [table:bh_real] lists the P-values assessed
for significance using the BH-procedure. We found that only one of the
networks was significantly associated with a pathogen-specific
immunodeficiency. The variant load was significantly higher than for
controls. The total potential LoF variants only accounted for 30.5% of
total variants in the network but was ranked high during the burden rank
(see sec [sec:burden_rank]) because no controls harboured potential
LoF variants in this network and therefore 100% occurred in cases. This
protein network contained genes responsible for pathogen detection; some
genes <em>might have been</em> identified as candidates using the routine exome
analysis pipeline such as the antiviral receptors and antiviral
interferon regulatory factors. However, most of the other genes that are
integral to this pathway would not have been identified by standard best
practices. The protein network is shown in <strong>Figure
[fig:immune_cluster]</strong> where potential LoF variants-harbouring genes
are coloured in red. Gene candidates with variants of unknown
significance are coloured in red and, anecdotally, the colouring
thereafter becomes lighter (orange to yellow) based on the likelihood of
candidates being identified by manual interpretation of unknown
candidates.</p>

<p><strong>Case and control means test.</strong> The total rare variants per network
are shown, comparing groups. A test of means was conducted in this test
dataset and P-values are
shown.
<strong>FigureLabel</strong>
means_test
<img src="/images/net_analysis/means_test.pdf" width="30%" /></p>

<p>[*[6]{}[Z]{}]{}</p>

<p><br />
&amp; [<strong>LoF freq in cases</strong>]{} &amp; [<strong>LoF freq due to cases per network</strong>]{}
&amp; [<strong>P-value</strong>]{} &amp; [<strong>rank</strong>]{} &amp; [<strong>$(i/m)Q$</strong>]{}<br />
&amp; [0.306]{} &amp; [1]{} &amp; [0.023]{} &amp; [1]{} &amp; [0.025]{}<br />
[27]{} &amp; [0.429]{} &amp; [1]{} &amp; [0.12]{} &amp; [2]{} &amp; [0.05]{}<br />
[16]{} &amp; [0.6]{} &amp; [0.919]{} &amp; [0.13]{} &amp; [3]{} &amp; [0.075]{}<br />
[19]{} &amp; [0.281]{} &amp; [0.835]{} &amp; [0.14]{} &amp; [4]{} &amp; [0.1]{}<br />
[25]{} &amp; [0.25]{} &amp; [1]{} &amp; [0.28]{} &amp; [5]{} &amp; [0.125]{}<br />
[11]{} &amp; [0.357]{} &amp; [0.838]{} &amp; [0.33]{} &amp; [6]{} &amp; [0.15]{}<br />
[10]{} &amp; [0.516]{} &amp; [0.856]{} &amp; [0.34]{} &amp; [7]{} &amp; [0.175]{}<br />
[18]{} &amp; [0.474]{} &amp; [0.85]{} &amp; [0.47]{} &amp; [8]{} &amp; [0.2]{}\</p>

<p>[@l@l@]{} [0pt][0pt]</p>

<p>[Protein network with significantly enriched variant load.][<strong>=0
Protein network with significantly enriched variant load.</strong> From the
example data, network 22 was significantly enriched for rare variants.
The same clustering method was again used on all variants with a less
stringent variant frequency (&lt;1% in general population and present in
any cohort sample). With the resulting, more common variants, the full
protein network can be seen (about double in size compared to only very
rare variants). Gene candidates with variants of unknown significance
are coloured in red and, anecdotally, the colouring thereafter becomes
lighter (orange to yellow) based on the likelihood of candidates being
identified by manual interpretation of unknown candidates. ]{}
[fig:immune_cluster]</p>

</div>

<!--

<div id="related">
  <h2>Related Topics</h2>
  <ul class="topics">
    
  </ul>
</div>

-->


	<div class="footer">
		<div class="foot-container">
				<a href="/" class="logo-foot">
				<img style="margin-bottom:-16px" 
				src="/images/logos/logo4.2_150ppi.png" 
				alt="Logo image" width="35.7" height="55"/></a>
					<div class="right-root">
						<p> Maintained by Dylan Lawless. Postdoc at <a href="https://people.epfl.ch/dylan.lawless">EPFL</a>.
						<a href="/resume">Resume</a>
						and <a href="https://scholar.google.com/citations?user=RPBxP1wAAAAJ&hl=en&oi=ao">Google scholar</a>.
						</p>
					</div>
		</div>
	</div>

</div>


<!-- Banner on top right corner linking to github
<a href="https://github.com/mojombo"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_red_aa0000.png" alt="Fork me on GitHub" /></a>
-->

<!-- ClickTale Bottom part -->
<div id="ClickTaleDiv" style="display: none;"></div>
<script type="text/javascript">
if(document.location.protocol!='https:')
  document.write(unescape("%3Cscript%20src='https://s.clicktale.net/WRb.js'%20type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
if(typeof ClickTale=='function') ClickTale(206,0.3,"www03");
</script>
<!-- ClickTale end of Bottom part -->

<!-- Google Analytics -->
<!--
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "https://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
var pageTracker = _gat._getTracker("UA-6016902-1");
pageTracker._trackPageview();
</script>
-->
<!-- Google Analytics end -->

</body>
</html>
