<!DOCTYPE html>
<html>
<head>
<style>
body {
	background-color: white;
	font-family: Baskerville;
	}

h1, h2, h3, h4, h5, h6 { 
	width:70%;
	margin: 0 auto;
	align:left;
	color: black;
}

p {
	width:70%;
	margin: 0 auto;
	align:center;
	color: black;
}

@media only screen and (max-width:320px)  { /* smartphones, iPhone, portrait below 320 phones */ 
	.subjects { margin-top: 2.5rem ; }
	.logo { font-size: 1rem; }
	body {
		font-family: 'Garamond';
		font-size: 0.8rem;
		line-height: 1.4rem;
		}
	h1 { 
		font: 1.2rem 'Didot';
		color: black;
	}
	h2, h3, h4, h5, h6 {
		font: 1.0rem 'Didot';
		font-style: italic;
		color: black;
		}
    }

@media only screen and (min-width:320px)  { /* smartphones, iPhone, portrait 480x320 phones */ 
	.subjects { margin-top: 2.5rem ; }
	.logo { font-size: 1.4rem; }
	body {
		font-family: 'Garamond';
		font-size: 1.1rem;
		line-height: 1.7rem;
		}
	h1 { 
		font: 1.2rem 'Didot';
		color: black;
		}
	h2, h3, h4, h5, h6 {
		font: 1.1rem 'Didot';
		font-style: italic;
		color: black;
		}
    }

@media only screen and (min-width:481px)  { /* portrait e-readers (Nook/Kindle), smaller tablets @ 600 or @ 640 wide. */ 
	.subjects { margin-top: 2.5rem ; }
	.logo { font-size: 1.5rem; }
	body {
		font-family: 'Garamond';
		font-size: 1.1rem;
		line-height: 1.7rem;
		}
	h1 { 
		font: 1.2rem 'Didot';
		color: black;
		}
	h2, h3, h4, h5, h6 {
		font: 1.1rem 'Didot';
		font-style: italic;
		color: black;
		}
    }

@media only screen and (min-width:641px)  { /* portrait tablets, portrait iPad, landscape e-readers, landscape 800x480 or 854x480 phones */ 
	.subjects { margin-top: 3rem ; }
	.logo { font-size: 1.6rem; }
	body {
		font-size: 1.2rem;
		line-height: 2rem;
		}
	h1 { 
		font: 1.6rem 'Didot';
		color: black;
		}
	h2, h3, h4, h5, h6 {
		font: 1.3rem 'Didot';
		font-style: italic;
		color: black;
		}
    }

@media only screen and (min-width:961px)  { /* tablet, landscape iPad, lo-res laptops ands desktops */
	.subjects { margin-top: 4rem ; }
	.logo { font-size: 1.7rem; }
	body {
		font-size: 1.3rem;
		line-height: 2rem;
		}
	h1 { 
		font: 1.7rem 'Didot';
		color: black;
		}
	h2 ,h3, h4, h5, h6 {
		font: 1.3rem 'Didot';
		font-style: italic;
		color: black;
		}
    }

@media only screen and (min-width:1025px) { /* big landscape tablets, laptops, and desktops */
	.subjects { margin-top: 4rem ; }
	.logo { font-size: 1.7rem; }
	body {
		font-size: 1.3rem;
		line-height: 2rem;
		}
	h1 { 
		font: 1.7rem 'Didot';
		color: black;
		}
	h2, h3, h4, h5, h6 {
		font: 1.3rem 'Didot';
		font-style: italic;
		color: black;
		}
    }

@media only screen and (min-width:1281px) { /* hi-res laptops and desktops */ 
	.subjects { margin-top: 4rem ; }
	body {
		font-size: 1.3rem;
		line-height: 2rem;
		}
	h1 { 
		font: 1.7rem 'Didot';
		color: black;
		}
	h2, h3, h4, h5, h6 {
		font: 1.3rem 'Didot';
		color: black;
	}
    }

.graph-container {
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  text-align:center; 
  float:left;
}

.graph {
  width: 100%;
}

.graph img {
  width: 100%;
}

</style>
</head>
<body>

<h1>Evidence builder</h1>
<p>
This page is a work in progress. 
I publish this intermediate version to demonstrate the purpose and to recruit both data curators and front-end devs.


Shown here: two randomly chosen coding variants in DDX58.
Top - protein-coding functional data with coding positions, bottom - transcript sequence data with genomic coordinates, GRCh37.
Blue dotted lines indicate the variants of interest.
The canonical transcript ENST00000379883.2 is indicated by the green diamond.
Protein annotations match that of Uniprot:ENST00000379883; ENSP00000369213; ENSG00000107201 [O95786-1].
The variant tables lists gnomAD data for the variants of interest. 


<div class="graph-container">
  <div class="graph left" margin-bottom="3vh">

<iframe id="igraph" scrolling="yes" style="border:none;" seamless="seamless" src="
	https://lawlessgenomics.com/topic/gnomad_table.html
	"
	height="200" 
	width="80%"></iframe>

<iframe id="igraph" scrolling="no" style="border:none;" seamless="seamless" src="
	https://lawlessgenomics.com/topic/gene_illustrate.html
	"
	height="800" 
	width="80%"></iframe>

<!-- https://lawlessgenomics.com/pages/portfolio_live/minard_napoleon_1812_main.html -->
 </div>
</div>

<p>
Under dev and coming soon: <br>
	- hover tip annotations for all positions, <br>
	- browsable variant positions, <br>
	- links to <br>
	- <a href="https://lawlessgenomics.com/topic/evidence_finder_gene_all.html">gene evidence finder</a><br>
	- <a href="https://lawlessgenomics.com/topic/evidence-finder-variant">variant evidence finder</a><br>
	- with external links to evidence sources that have been used for analysis; e.g. <br>
	- <a href="https://gnomad.broadinstitute.org">gnomAD</a><br>
	- <a href="http://database.liulab.science/dbNSFP">dbNSFP</a><br>
	- <a href="https://grch37.ensembl.org/Tools/VEP">VEP</a><br>
	- <a href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a><br>
	- <a href="https://www.clinicalgenome.org">ClinGen</a><br>
	- <a href="http://www.hgmd.cf.ac.uk/ac/index.php">HGMD</a><br>
	- and many more including the top pathogenicity prediction databases<br>
	- evidence scores based on the standards and guidelines for the interpretation of sequence variants from ACMG (doi:<a href="https://www.nature.com/articles/gim201530">10.1038/gim.2015.30</a>).<br>
</p>



</body>
</html>
