<!DOCTYPE html>
<html xmlns="https://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
   <!-- link to MathJax for LaTeX stlye eqn -->
    <script type="text/javascript" async
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-MML-AM_CHTML">
</script>

	
	<!-- favicon -->
	<!--
	<link rel="icon" type="image/png" sizes="32x32" href="/images/logos/logo_2.8.png">
	-->
	<!-- https://realfavicongenerator.net -->
	<link rel="apple-touch-icon" sizes="180x180" href="/images/logos/favicon_package_v0.16_color/apple-touch-icon.png">
	<link rel="icon" type="image/png" sizes="32x32" href="/images/logos/favicon_package_v0.16_color/favicon-32x32.png">
	<link rel="icon" type="image/png" sizes="16x16" href="/images/logos/favicon_package_v0.16_color/favicon-16x16.png">
	<link rel="manifest" href="/images/logos/favicon_package_v0.16_color/site.webmanifest">
	<meta name="msapplication-TileColor" content="#da532c">
	<meta name="theme-color" content="#ffffff">

	<meta name="viewport" content="width=device-width, initial-scale=1">

   <!--  -->
   <meta http-equiv="content-type" content="text/html; charset=utf-8" />
   <title>Pharmacogenomics for personal medicine</title>
   <meta name="author" content="LawlessGenomics" />
   <link href="https://feeds.feedburner.com/tom-preston-werner" rel="alternate" title="Dyaln feed title" type="application/atom+xml" />
   <meta name="readability-verification" content="QCzSs992GxmRYRKVpPeZ6LE2tS8aYKxsSSQKV8YM"/>

   <!-- include a google font -->
   <link href="https://fonts.googleapis.com/css?family=Roboto+Mono|Roboto+Slab:300|Roboto:500" rel="stylesheet">
   <link href="https://fonts.googleapis.com/css?family=Lora" rel="stylesheet">

   <!-- syntax highlighting CSS -->
   <link rel="stylesheet" href="/css/syntax.css" type="text/css" />

   <!-- Homepage CSS -->
   <link rel="stylesheet" href="/css/screen.css" type="text/css" media="screen, projection" />

   <!-- Typekit -->
   <script type="text/javascript" src="https://use.typekit.com/jpd0pfm.js"></script>
   <script type="text/javascript">try{Typekit.load();}catch(e){}</script>
</head>
<body>

<!-- ClickTale Top part -->
<script type="text/javascript">
var WRInitTime=(new Date()).getTime();
</script>
<!-- ClickTale end of Top part -->

<div class="site">
  <div class="title">
    <img style="margin-bottom:-8px" 
	src="/images/logos/logo_2.8.png" 
	alt="Logo image" width="55" height="37.3"/>
    <a href="/">LawlessGenomics</a>
    <a class="extra" href="/">Home</a>
  </div>

  <div id="topic">
<h1 id="pharmacogenomics-for-personal-medicine">Pharmacogenomics for personal medicine</h1>
<p class="meta">26 Apr 2020</p>

<ul id="markdown-toc">
  <li><a href="#pharmacogenomics-for-personal-medicine" id="markdown-toc-pharmacogenomics-for-personal-medicine">Pharmacogenomics for personal medicine</a></li>
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#preparing-an-example-vcf" id="markdown-toc-preparing-an-example-vcf">Preparing an example VCF</a></li>
  <li><a href="#comparing-annotated-genetic-data-to-drug-lists" id="markdown-toc-comparing-annotated-genetic-data-to-drug-lists">Comparing annotated genetic data to drug lists</a>    <ul>
      <li><a href="#small-example-check-of-gene-list-versus-drug-gene-list" id="markdown-toc-small-example-check-of-gene-list-versus-drug-gene-list">Small example check of gene list versus drug-gene list</a></li>
    </ul>
  </li>
  <li><a href="#a-real-example-of-merging-genetic-and-pharmacogenomic-data" id="markdown-toc-a-real-example-of-merging-genetic-and-pharmacogenomic-data">A real example of merging genetic and pharmacogenomic data</a></li>
  <li><a href="#drug-indication" id="markdown-toc-drug-indication">Drug indication</a></li>
  <li><a href="#annotation" id="markdown-toc-annotation">Annotation</a>    <ul>
      <li><a href="#optimising-vcf-annotation" id="markdown-toc-optimising-vcf-annotation">Optimising VCF annotation</a></li>
      <li><a href="#how-to-get-coordinates-for-a-gene-list" id="markdown-toc-how-to-get-coordinates-for-a-gene-list">How to get coordinates for a gene list</a></li>
      <li><a href="#extracting-regions-from-a-vcf-using-a-bed-file" id="markdown-toc-extracting-regions-from-a-vcf-using-a-bed-file">Extracting regions from a VCF using a bed file</a></li>
    </ul>
  </li>
  <li><a href="#unknown-variants" id="markdown-toc-unknown-variants">Unknown variants</a></li>
  <li><a href="#gene-dosage" id="markdown-toc-gene-dosage">Gene dosage</a></li>
  <li><a href="#drug-indication-1" id="markdown-toc-drug-indication-1">Drug indication</a></li>
  <li><a href="#a-large-scale-example-summary" id="markdown-toc-a-large-scale-example-summary">A large scale example summary</a></li>
  <li><a href="#funding-strategy" id="markdown-toc-funding-strategy">Funding strategy</a></li>
  <li><a href="#legal-requirements" id="markdown-toc-legal-requirements">Legal requirements</a></li>
  <li><a href="#more-questions" id="markdown-toc-more-questions">More questions</a></li>
  <li><a href="#references" id="markdown-toc-references">References</a></li>
</ul>

<h1 id="introduction">Introduction</h1>
<p>With the popularisation of commercial genetics services, more and more people are now able to “decode” their genetic data.
Questions that might arise from this information include “do I have potentially disease-causing variants that can be treated with a drug?”, or “am I taking a drug that will be affected by my genetics?”.
To tackle such questions with an example, we use public data in combination with pharmacogenomics.
Outside of genotype data (offered by <a href="https://www.23andme.com">23andMe</a> for example), the most common file type will be VCF:
<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531692-VCF-Variant-Call-Format">What is a vcf and how should I interpret it?</a>.</p>

<p>As an example input, try download Craig Venter’s genome VCF:
<a href="ftp://ftp.ensembl.org/pub/release-75/variation/vcf/homo_sapiens/Venter.vcf.gz">ftp://ftp.ensembl.org/pub/release-75/variation/vcf/homo_sapiens/Venter.vcf.gz</a>.</p>

<p>When I wrote this post, I used a data source with different genetic data files;
<a href="https://my.pgp-hms.org/public_genetic_data">https://my.pgp-hms.org/public_genetic_data</a>.
To check that it works OK, I tried a quick version of this challenge.
I picked the first whole genome VCF file that I saw (hu24385B 2019-04-07.vcf.g_z)
<a href="https://my.pgp-hms.org/profile/hu24385B">https://my.pgp-hms.org/profile/hu24385B</a>. 
The VCF has 3,461,639 variants.
VCF files can contain a large range of information for each variant, however only the first 7 column are strictly neccessary; Chromosome, position, ID, Reference, Alternate, Qulaity, Filter, info. 
<a href="https://gatkforums.broadinstitute.org/gatk/discussion/1268/what-is-a-vcf-and-how-should-i-interpret-it">The details are explained on this GATK forum post</a>.
Annotation information about the gene name (or related diseases) is often not present when the VCF is generated and only added later.
Therefore the most common input source may be lacking gene symbols.
To get the gene names of a single file, the simplest way was is to upload a VCF (or a part of it) to <a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/">Variant Effect Predictor</a> to get the gene symbol (and any other information that you wish).
<img src="https://dylanlawlessblog.files.wordpress.com/2019/05/screenshot-2019-05-07-at-17.01.45.png" width="80%" /><br />
<img src="https://dylanlawlessblog.files.wordpress.com/2019/05/screenshot-2019-05-07-at-17.01.58.png" width="80%" /></p>

<p>To reduce the time and output you can limit the options.
Split the file and run in batches to save time.
Here I tried the first ~1800 variants.</p>

<p><br />
<code class="language-plaintext highlighter-rouge">head -2000 56001801068861_WGZ.snp.vcf &gt; test.vcf</code></p>

<p><br />
And then annotated that extract with <a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/">Variant Effect Predictor.</a>
The results would be retained by a URL address such as this, for a few days, but will be deleted by the time your read this.<br />
<br />
<a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?db=core;tl=jNYYW5ONeVFYnaMM-5265700">http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?db=core;tl=jNYYW5ONeVFYnaMM-5265700</a> <br />
<br />
<img src="https://dylanlawlessblog.files.wordpress.com/2019/05/screenshot-2019-05-07-at-17.01.10.png" width="80%" /><br />
<br />
Make certain that you use the same reference genome as used on the input data.
The VCF file was made using reference genome GRCh37.
Therefore the Ensembl/VEP website URL should be for that genome build (grch37, not the default GRCh38).</p>

<h1 id="preparing-an-example-vcf">Preparing an example VCF</h1>
<p>If you would like to try this using a whole genome using the Ensembl web 
interface you will need to split your VCF into smaller block first.
For routine usage the command-line version of VEP and it’s databases should be 
installed on run locally.
There are several bioinformatics tools that are commonly used for manipulating 
genetic file formats such as VCFtools. 
To get a real understanding of the data type, I inlcude here a method using 
command line bash to split a VCF file into smaller blocks. 
A bash script is printed below where I use very mainstream traditional 
command-line tools to wrangle data, including 
<a href="https://en.wikipedia.org/wiki/Gzip">gunzip</a>
to unzip compressed files, 
<a href="https://en.wikipedia.org/wiki/wc_(Unix)">wc</a>
to count lines, 
<a href="https://en.wikipedia.org/wiki/Cat_(Unix)">cat</a>
to print a file, 
<a href="https://en.wikipedia.org/wiki/Head_(Unix)">head</a>
to read the top of a file, 
<a href="https://en.wikipedia.org/wiki/Sed">sed</a>
to edit lines, 
<a href="https://en.wikipedia.org/wiki/AWK">awk</a>
for data extraction, and
<a href="https://en.wikipedia.org/wiki/Grep">grep</a>
which is not used here but it fits well with these other tools - for text search.
Creating a file containing the code below and ending with the filename extension with “.sh” will allow it to be run by your terminal, in this case using the bash language.
I encourage you to read each line and figure out what should happen. If it makes sense then it is reasonable to swap such a manual method with a more efficient specialised tool.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c">#!/bin/bash</span>
<span class="c"># VEP accept files of &lt;50MB size.</span>
<span class="c"># We will split our large VCF into smaller files.</span>
<span class="c"># Each file requires the same original </span>
<span class="c"># headers and file extension ".vcf"</span>

<span class="c"># Unzip the VCF.gz</span>
<span class="nb">gunzip </span>56001801068861_WGZ.snp.vcf.gz

<span class="c"># Count the number of lines in vcf</span>
<span class="nb">wc</span> <span class="nt">-l</span> 56001801068861_WGZ.snp.vcf

<span class="c"># How should a vcf file look?</span>
<span class="c"># See the links posted in this tutorial above</span>

<span class="c"># Take a look at the header</span>
<span class="c"># This VCF has 140 lines of header metadata (beginning with "#")</span>
<span class="c"># Line 141 shows the column headers: CHROM	POS	ID	REF	ALT...</span>
<span class="c"># Line 142 starts with the first variant</span>
<span class="nb">head</span> <span class="nt">-142</span> 56001801068861_WGZ.snp.vcf

<span class="c"># Print the header to a new file for later</span>
<span class="nb">head</span> <span class="nt">-141</span> 56001801068861_WGZ.snp.vcf <span class="o">&gt;</span> header
<span class="c"># Print everything else (the body) to </span>
<span class="c"># a new file that we will then split.</span>
<span class="nb">sed</span> <span class="s1">'1,141d'</span> 56001801068861_WGZ.snp.vcf <span class="o">&gt;</span> body.vcf

<span class="c"># Make a new directory for the next step</span>
<span class="nb">mkdir </span>split_files
<span class="c"># Move the large file inside</span>
<span class="nb">mv </span>body.vcf split_files/
<span class="nb">cd </span>split_files
<span class="c"># Now split the body.vcf into smaller </span>
<span class="c"># files of 200,000 lines each</span>
<span class="nb">split</span> <span class="nt">-l</span> 150000 body.vcf

<span class="c"># You will now how ~10 files "xaa, xab, etc."</span>
<span class="c"># Add the header back onto all of these files to make them VCFs again.</span>
<span class="c"># This "for loop" will do the following for each file:</span>
<span class="c"># Print the header and the vcf body to </span>
<span class="c"># a file with the same name, </span>
<span class="c"># adding a file extension ".vcf".</span>
<span class="c"># Then remove the vcf body file that </span>
<span class="c"># does not have the ".vcf" extension</span>
<span class="c"># leaving you with the original whole genome VCF split</span>
<span class="c"># into smaller files, each with the same headers.</span>

<span class="k">for </span>file <span class="k">in</span> ./x<span class="k">*</span> <span class="p">;</span> 
    <span class="k">do </span><span class="nb">cat</span> ../header <span class="nv">$file</span> <span class="o">&gt;&gt;</span> <span class="nv">$file</span>.vcf <span class="o">&amp;&amp;</span> <span class="nb">rm</span> <span class="nv">$file</span> <span class="p">;</span>
<span class="k">done</span>

<span class="c"># These should be small enough to run on VEP online.</span>
<span class="c"># You could edit the split command to make a </span>
<span class="c"># reasonable number of files, </span>
<span class="c"># uploading &gt;10 is not efficient.</span>

</code></pre></div></div>

<h1 id="comparing-annotated-genetic-data-to-drug-lists">Comparing annotated genetic data to drug lists</h1>
<p>Uploading either a small example or your split-whole-genome example, VEP will process the data and annotate it will whatever features you require, including gene names, variant consequence, pahtnogenicy prediction, etc.
The next aim will be to see if any of your output genes are also present in your drug-gene database.
The method will require merging both dataset (gene and drug datasets) based on shared features. 
A simple sanity test would be useful first:</p>

<h2 id="small-example-check-of-gene-list-versus-drug-gene-list">Small example check of gene list versus drug-gene list</h2>
<p>From the VEP output I extracted the gene symbols column and compared against a list of druggable target genes from 
<a href="https://www.drugbank.ca">DrugBank</a> 
(because I happen to have their data on hand, FDA may be more reliable).
I used another command-line method to do this efficienctly:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cut</span> <span class="nt">-f1</span> <span class="nt">-d</span> <span class="s2">","</span> vep_output_file.csv | <span class="nb">uniq</span> <span class="o">&gt;</span> unique.genes.txt
</code></pre></div></div>

<ul>
  <li>Cut column 1 (f1)</li>
  <li>with a delimiter comma (,)</li>
  <li>from the vep output csv file (or tsv, or text file)</li>
  <li>then pipe (|) that result into another program (sort) to sort the result in alphabetic order</li>
  <li>pipe (|) again this result into(uniq)</li>
  <li>so that only one unique gene name is output</li>
  <li>then (&gt;) write the output into the new file “unique.genes.txt”.<br />
<br />
Repeat the same type of method on the DrugBank dataset to get a list of the gene names contained in DrugBank into a file called “unique.druggable.txt”
The next command will then compare both lists.
<br /></li>
</ul>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">sort </span>unique.genes.txt unique.druggable.txt | <span class="nb">uniq</span> <span class="nt">-c</span> <span class="nt">-i</span> | <span class="nb">grep</span> <span class="nt">-v</span> <span class="s1">'1 '</span>
</code></pre></div></div>

<p>This command also used “uniq -c” to count how many times each gen name occurs and then “grep -v ‘1 ‘” meaning ignore genes that are only present 1 time. 
We want the genes that are present twice, once in each list.
The genes which were in present in both the variant list and DrugBank list are:
From a 2,000 line VCF file:
<a href="https://www.drugbank.ca/bio_entities/BE0003599">GABRD</a>,
<a href="https://www.drugbank.ca/bio_entities/BE0004895">PRKCZ</a>,
<a href="https://www.drugbank.ca/bio_entities/BE0000495">SCNN1D</a><br />
From a 10,000 VCF line file = 
<a href="https://www.drugbank.ca/bio_entities/BE0003599">GABRD</a>,
<a href="https://www.drugbank.ca/bio_entities/BE0004895">PRKCZ</a>,
<a href="https://www.drugbank.ca/bio_entities/BE0000495">SCNN1D</a>,
<a href="https://www.drugbank.ca/bio_entities/BE0008994">TP73</a>.</p>

<h1 id="a-real-example-of-merging-genetic-and-pharmacogenomic-data">A real example of merging genetic and pharmacogenomic data</h1>
<p>Now that a small example has shown us the logic of the process, we can try a more complex real-world example. 
The following R language script is used to merge the VEP annotated VCF file with a DrugBank database based on the gene names that are common to both datasets.
Read each line and try to understand the process. 
The are many alternative ways to do the same thing in different programming languages. 
This example is done using R. 
I recommending installing R and then installing R studio to edit and run your commands.</p>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Comment lines are ignored because of "#" symbol.</span><span class="w">
</span><span class="c1"># Command lines are run by clicking "Run" or "command+enter" on Mac</span><span class="w">

</span><span class="c1"># csv = comma sep file</span><span class="w">
</span><span class="c1"># tsv = tab spaced</span><span class="w">
</span><span class="c1"># txt = white space</span><span class="w">

</span><span class="c1">#///////////////////////////////</span><span class="w">
</span><span class="c1"># To do</span><span class="w">
</span><span class="c1">#///////////////////////////////</span><span class="w">
</span><span class="c1"># Files: vepfile, drugs</span><span class="w">
</span><span class="c1"># merge based on Gene symbol</span><span class="w">

</span><span class="c1">#///////////////////////////////</span><span class="w">
</span><span class="c1"># import the VEP file</span><span class="w">
</span><span class="c1">#///////////////////////////////</span><span class="w">

</span><span class="c1"># Important note:</span><span class="w">
</span><span class="c1"># The VEP file will start with a header line </span><span class="w">
</span><span class="c1"># that begins with "#" symbol. But this is being ignored by R.</span><span class="w">
</span><span class="c1"># Open the file with text edit and remove that symbol.</span><span class="w">
</span><span class="c1"># Maybe there is an R command to do this on import, whatever is faster.</span><span class="w">

</span><span class="c1"># Split vcf into ~ 10 files. </span><span class="w">
</span><span class="c1"># file 1 "xaa" was anylised on VEP, </span><span class="w">
</span><span class="c1"># download the TXT version, </span><span class="w">
</span><span class="c1"># or unzip my provided example one - cfKJCLRm0eKXsaEG.txt.zip</span><span class="w">
</span><span class="c1"># https://grch37.ensembl.org/Homo_sapiens/Tools/VEP/</span><span class="w">

</span><span class="c1"># Import the VEP output</span><span class="w">
</span><span class="n">vepfile</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="w">
  </span><span class="n">file</span><span class="o">=</span><span class="s2">"cfKJCLRm0eKXsaEG.txt"</span><span class="p">,</span><span class="w">
  </span><span class="n">na.strings</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s2">""</span><span class="p">,</span><span class="w"> </span><span class="s2">"NA"</span><span class="p">),</span><span class="w">
  </span><span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">,</span><span class="w">
  </span><span class="n">header</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w">

</span><span class="c1"># Import drugbank table</span><span class="w">
</span><span class="c1"># the "fill=TRUE" is needed because not all </span><span class="w">
</span><span class="c1"># file lines have the same number of elements.</span><span class="w">
</span><span class="n">drugs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="w">
  </span><span class="n">file</span><span class="o">=</span><span class="s2">"all.txt"</span><span class="p">,</span><span class="w">
  </span><span class="n">na.strings</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s2">""</span><span class="p">,</span><span class="w"> </span><span class="s2">"NA"</span><span class="p">),</span><span class="w">
  </span><span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">,</span><span class="w">
  </span><span class="n">header</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w">
  </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="w"> </span><span class="p">)</span><span class="w">


</span><span class="c1"># We can merge these two files based on 1 common column.</span><span class="w">
</span><span class="c1"># However, the gene name column does not have the same name.</span><span class="w">
</span><span class="c1"># One of them needs to be renamed:</span><span class="w">
</span><span class="c1"># vepfile="SYMBOL"</span><span class="w">
</span><span class="c1"># drugs="Gene.Name"</span><span class="w">

</span><span class="c1"># colnames(df)[colnames(df) == 'oldName'] &lt;- 'newName'</span><span class="w">
</span><span class="n">colnames</span><span class="p">(</span><span class="n">vepfile</span><span class="p">)[</span><span class="n">colnames</span><span class="p">(</span><span class="n">vepfile</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"SYMBOL"</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"Gene.Name"</span><span class="w">

</span><span class="c1"># Merge keeping only matches</span><span class="w">
</span><span class="n">merged</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">merge</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">vepfile</span><span class="p">,</span><span class="w">
                </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">drugs</span><span class="p">,</span><span class="w">
                </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Gene.Name"</span><span class="p">,</span><span class="w"> </span><span class="n">all</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">


</span><span class="c1"># Remove empty data "NA"</span><span class="w">
</span><span class="c1"># Install packeges once (comment out then)</span><span class="w">
</span><span class="c1"># Load library each time to use "%&gt;%" (command join) and filtering</span><span class="w">
</span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"tidyr"</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">tidyr</span><span class="p">)</span><span class="w">
</span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"dplyr"</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span><span class="w">
</span><span class="n">df1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">merged</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">drop_na</span><span class="p">(</span><span class="n">Drug.IDs</span><span class="p">)</span><span class="w">

</span><span class="c1"># Make a list of benign variant types that should be removed</span><span class="w">
</span><span class="n">filter_out</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> 
    </span><span class="s1">'synonymous_variant|UTR|NMD_transcript\
	|non_coding|downstream|upstream\
	|intron|mature_miRNA_variant'</span><span class="w">

</span><span class="c1"># Then filter out anything matching these terms.</span><span class="w">
</span><span class="n">df2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">df1</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">filter_all</span><span class="p">(</span><span class="n">all_vars</span><span class="p">(</span><span class="o">!</span><span class="n">grepl</span><span class="p">(</span><span class="n">filter_out</span><span class="p">,</span><span class="n">.</span><span class="p">)))</span><span class="w">

</span><span class="c1"># Save an output tsv file for Excel, etc.</span><span class="w">
</span><span class="n">write.table</span><span class="p">((</span><span class="n">df2</span><span class="p">),</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="s1">'./output.tsv'</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">,</span><span class="w"> </span><span class="err">\</span><span class="w">
</span><span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>If you complete this process the output will contain a perfectly merged dataset.
So in this simple example it takes just 5 minutes to get from a real genome VCF to possibly druggable target genes (see further note on <em>drug indication</em> below).
The difficulty lies downstream in interpreting which variants can have an effect that would justify the use of the drug.
Anyone implementing a usable version of this method will incur several obstacles;
e.g. are non-coding or synonymous variants worth reporting?, genes have multiple transcripts which means one variant can be both coding and non-coding depending on transcript splicing, etc.
Other sources of sequence data, including the sequences of Watson and Venter;<br />
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/">http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/</a><br />
23andMe open snp data; <a href="https://opensnp.org/genotypes">https://opensnp.org/genotypes</a>.
There are many layers to a this problem to create a usable product.
For example, how to integrate pharmacodynamics, covariats to drug response, contraindications, variant pathogenicity, etc.
However, this is a good start as a learning experience.</p>

<h1 id="drug-indication">Drug indication</h1>
<p>My example used <a href="https://www.drugbank.ca">DrugBank</a> for pharmacogenomic information.
However, it may be safest to use the <a href="https://www.fda.gov/drugs/science-research-drugs/table-pharmacogenomic-biomarkers-drug-labeling">FDA information</a> as the primary source, but including <a href="https://www.drugbank.ca">DrugBank</a> info is no problem.
Drugs might be either a treatment for a genetic determinant, or a warning for drug usage in someone who also has a genetic variation that might effect their treatment.
The “Labelling Section” listed by FDA might offer the best information.</p>

<p><a href="https://www.fda.gov/drugs/science-research-drugs/table-pharmacogenomic-biomarkers-drug-labeling">https://www.fda.gov/drugs/</a>
For example, if we go and check the Prescribing Information PDF to compare two drugs we see that</p>

<p><br />
<strong>(1)</strong> One is used to directly block a gene product,<br />
<strong>(2)</strong> The second warns about use with certain genetic complications.</p>

<p><br />
<strong>Drug 1</strong>: <a href="https://www.accessdata.fda.gov/scripts/cder/daf/index.cfm?event=overview.process&amp;varApplNo=761034">Atezolizumab</a> (1),<br />
<strong>Gene</strong>: <a href="https://www.fda.gov/drugs/science-research-drugs/table-pharmacogenomic-biomarkers-drug-labeling"><em>CD274</em></a> <a href="https://www.fda.gov/drugs/science-research-drugs/table-pharmacogenomic-biomarkers-drug-labeling">(PD-L1)</a><br />
<strong>Labeling</strong>: Indications and Usage<br />
<strong>PRESCRIBING</strong> <strong>INFORMATION</strong>: TECENTRIQ (Atezolizumab) is a programmed death-ligand 1 (PD-L1) blocking antibody indicated for the treatment of patients with…
<a href="https://www.accessdata.fda.gov/drugsatfda_docs/label/2016/761034Orig1s000lbl.pdf">linked PDF</a>.<br />
<strong>Explained</strong>: Genetic disorder and the drug to treat it, exactly what you want.</p>

<p><br />
<strong>Drug 2</strong>: Avatrombopag (3)<br />
<strong>Gene</strong>: <a href="https://www.fda.gov/drugs/science-research-drugs/table-pharmacogenomic-biomarkers-drug-labeling"><em>PROC</em></a><br />
<strong>Labeling</strong>: Warnings and Precautions<br />
<strong>PRESCRIBING INFORMATION</strong>: Thrombotic/Thromboembolic Complications: DOPTELET is a thrombopoietin (TPO) receptor agonist… Monitor platelet counts and for thromboembolic events
<a href="https://www.accessdata.fda.gov/drugsatfda_docs/label/2019/210238s002lbl.pdf">linked PDF</a>.<br />
<strong>Explained</strong>: Atezolizumab is used to treat thrombocytopenia (low levels of thrombocytes).<br />
You <em>do not want to give</em> this to someone who has <a href="https://omim.org/entry/176860?search=proc&amp;highlight=proc"><em>PROC</em></a><a href="https://omim.org/entry/176860?search=proc&amp;highlight=proc"> deficiency</a>;
their disease is <a href="https://en.wikipedia.org/wiki/Thrombophilia">Thrombophilia</a> (hypercoagulability, or <a href="https://en.wikipedia.org/wiki/Thrombosis">thrombosis</a>).
With this in mind, perhaps an application doing this job could work two ways.
(1) If someone has a genetic disorder, the drug, gene, and Indicated usage appears.
(2) If someone is prescribed a drug a suggestion appears to check their genetics, with a link to the gene and Warnings and Precautions.</p>

<h1 id="annotation">Annotation</h1>
<p><a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/">Variant Effect Predictor (VEP)</a> is very useful.
During variant annotation, VEP supplies a “consequence” column.
Consequences are general and based on translation of genetic code in humans. 
The Loss-of-function (LoF) consequence is the simplest example (splice, stop mutations).
The variant consequence may be one of the defining criteria by which variants can 
be included in analysis since they are <em>interpretable</em> or of ostensibly <em>known significance</em>.
<em>Note: Using this alone could introduce spurious results so it is  best to have a solid criteria 
for selecting consequences of interest</em>.
The consequences provided by VEP are too long to discuss in detail here.
The table from the ensembl website is worth reading; the HIGH impact variants 
might be a simple method for selecting candidates:
<a href="https://grch37.ensembl.org/info/genome/variation/prediction/predicted_data.html#consequences">Ensembl Variation - Calculated variant consequences</a>.<br />
<img src="/images/VEP_consequences.jpg" width="100%" /></p>

<p><em>Note: For a real product, the code should be run offline (a perl program with a few local library dependencies). The databases/cache that it uses are a bit too large to include on in a user software. In the real world you would have to send anonymised packets from the user via an API for accessing the genomic databases hosted on your servers. Make sure to check their license to see if you can use oftware and databases in a commercial product</em>.
<a href="http://www.ensembl.org/info/about/legal/code_licence.html">http://www.ensembl.org/info/about/legal/code_licence.html</a></p>

<p>Running the software:</p>

<ul>
  <li>Using VEP is a vital part of converting the DNA variant information (genome position and nucleotide change) into annotated variant effects (protein coding change, gene name, predicted pathogenicity).</li>
  <li>It requires the VEP code to run and requires a copy of the database files (reference genome, gene information, etc.).</li>
  <li>You can upload a small number of variants to the online VEP web server to do this, or you can download the database and code to run on your own computer/server.</li>
</ul>

<p>So to process your customer/patient data, you have to choose one of these methods:</p>

<ol>
  <li>Customer must upload their entire file to your server that runs VEP (1GB - 30GB per individual genome data).</li>
  <li>Customer must download the database and VEP code to run on their own computer (complex and large download for them, not recommended).</li>
</ol>

<p>Number 1 is better. But sending one large file often has problems.
If they have a VCF file, you could:</p>

<ul>
  <li>Break it into small equal sized blocks to upload the data to you. Anonymisation is normal for all internet connections now, but you could just mention that some method of anonymising these blocks is important since someone “hacker” might try to steal information if most of the network data being sent to you contains small VCF format files.</li>
  <li>You could also run a very small piece of code on the cusotmers software application that could extract just the main parts of the VCF file that you need, instead of sending everything. This is explained in sections 7-9. You could say this, but don’t need to actually have working. i.e. the drugbank information only includes a certain number of human genes so perhaps you could just extract these using a list of genome coordinates before processing with VEP.</li>
</ul>

<p>For the license:</p>

<ul>
  <li>Anyone is free to download and use VEP code.</li>
  <li>However, if you modify or reuse the code commercially it might affect the possibility of getting a patent for your product.</li>
  <li>Your product uses VEP as an intermediate step, so you probably only need to include credit, or other legal info to say you have used it.</li>
  <li>If there were a reason to prevent you using the software commercially, you might be able to make a simple replacement that could give the minimum outputs that you need - gene name and mutation type. If the topic happens to interest you, you can read about <a href="https://en.wikipedia.org/wiki/Reverse_engineering#Software">reverse engineering software</a>.</li>
  <li>As an aside, you could also decide that you don’t want to commercialise and offer this tool for free which would prevent bigger companies (like Google) from offering this service in return for harvesting the public’s genetic data.</li>
</ul>

<h2 id="optimising-vcf-annotation">Optimising VCF annotation</h2>
<p>The slowest part of the method is VCF annotation.
You can significantly increase the speed by first reducing the input to contain only regions of interst.
That is, prepare a list of coordinates for each gene, and select for those regions in your input VCF or genotype data before annotation (VEP).</p>

<h2 id="how-to-get-coordinates-for-a-gene-list">How to get coordinates for a gene list</h2>
<p>Use Biomart.
Their main server was down when I tried, so I went via Ensembl, data access section:<br />
<a href="http://www.ensembl.org/info/data/biomart/index.html">http://www.ensembl.org/info/data/biomart/index.html</a><br />
Then to use the BioMart data mining tool<br />
<a href="http://www.ensembl.org/biomart/martview/28fdaf82da6c02dc5892f99b757e2c44">http://www.ensembl.org/biomart/martview/</a><br />
I actually needed the positions using GRCh37 (rather than 38), so I switched to the old Ensembl using<br />
<a href="http://www.ensembl.org/info/website/tutorials/grch37.html">http://www.ensembl.org/info/website/tutorials/grch37.html</a><br />
to get to <a href="http://grch37.ensembl.org/index.html">http://grch37.ensembl.org/index.html</a> 
then the Biomart section<br />
<a href="http://grch37.ensembl.org/biomart/martview/04f009257dadbafbe595155ba910eb5e">http://grch37.ensembl.org/biomart/martview/</a></p>

<p>Choose DataBase: Genes 93 Dataset: Human Filter -&gt; Gene -&gt; Input external ref ID list -&gt; (change dropdown) Gene
Name paste your list.
e.g. VPS45 PSMB8 BLNK NEFL NLRP7 SMAD4 PSMB9<br />
To set the output type: Attributes -&gt; Gene -&gt; select “gene start”, “gene stop”, “gene name”, or anything extra.
Select the “Results” button at the top and export.
The results can be tsv or csv.
You would have to figure out how to extract the regions from the vcf (sed, grep, awk, R code, etc.).
When I needed this, I used my own tools which required converting to format like this “X:1-2000”, and ordered by number and alphabetic (some positions in the reference genome were patches added later and have an alphanumeric instead of the normal chromosome).
If you use this list to extract regions from a VCF, remember to include all the original VCF header information.</p>

<h2 id="extracting-regions-from-a-vcf-using-a-bed-file">Extracting regions from a VCF using a bed file</h2>
<p>The early part of this tutorial shows how old-school command line tools can be used to extract data. 
Indeed, this may be computationally most efficient but there are some specialised tools that make the process easier in general.
You can use VCFtools to extract specified regions.<br />
<a href="https://vcftools.github.io/man_latest.html">https://vcftools.github.io/man_latest.html</a><br />
You could use a list of defined genome position to reduce the size of your dataset. 
The defined genomic coordinates are generally supplied in a file format called the (BED file](https://en.wikipedia.org/wiki/BED_(file_format))
Note that sometimes the bed file “chrom” ID - the name of the chromosome (e.g. chr3) does not match if the VCF file uses “3” instead of “chr3”.
You might need to edit the bed.
My bed file was like this:<br />
(tab spaced), ref.bed<br />
<br />
chrom    chromStart    chromEnd<br />
1    3549    13555<br />
<br />
And this command ran OK for me to give “output_prefix.recode.vcf”<br /></p>
<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$:</span>~/tools/vcftools_0.1.13/bin <span class="se">\</span>
./vcftools <span class="se">\</span>
<span class="nt">--vcf</span> ~/input.vcf <span class="se">\</span>
<span class="nt">--bed</span> ~/ref.bed <span class="se">\</span>
<span class="nt">--out</span> output_prefix <span class="se">\</span>
<span class="nt">--recode</span> <span class="nt">--keep-INFO-all</span>
</code></pre></div></div>

<p>This new VCF will now only contain gene regions that are potentially “druggable”, or at least included on the FDA list.
VCF annotation will be <em>much faster</em> than annotation of the whole genome.</p>

<h1 id="unknown-variants">Unknown variants</h1>
<p>In the majority of situations you will be stuck with <em>variants of unknown significance</em>.
In the absence of tailored analysis and interpretation of each invidual variant, one must often rank unknown variants based on a predicted pathogenicity.
Carefully consider that predictions can be completed wrong and address how such an annotation should be presented. 
One can rank unknown variants based on PHRED-scaled CADD score, highest being more predicted pathogenic.
https://cadd.gs.washington.edu/info<br />
<a href="http://genetics.bwh.harvard.edu/pph2/">Polyphen</a> gives a predicted outcome label and a probability score 0-1 from benign to probably damaging.
See what other pathogenicity prediction tools you can find and estimate how widespread/accepted their usage is.</p>

<h1 id="gene-dosage">Gene dosage</h1>
<p>An important cosideration of variant effect depends on gene dosage.
A <a href="https://en.wikipedia.org/wiki/Dominance_(genetics)">dominant gene</a> may be affected by a single heterozgous variant while a recessive gene may be able to compensate against the negative effect of a heterozyous variant due the presence of a second functional gene copy.
Therefore, the presence of heterozygous or homozygous allele is an important consideration.
Some genes may be sensitive to a <a href="https://en.wikipedia.org/wiki/Zygosity">hemizygous</a> effect, low frequency <a href="https://en.wikipedia.org/wiki/Somatic_(biology)">somatic variants</a>,
<a href="https://en.wikipedia.org/wiki/Mosaic_(genetics)">mosaisism</a>, etc.
<a href="https://en.wikipedia.org/wiki/SNV_calling_from_NGS_data">SNV calling in NGS</a> is a broad topic, but it is safe to say that at least the allele dosage (generally heterozygous or homozygous) should be included in result summary.
If possible, when a gene is linked to a specific disease then the <a href="https://en.wikipedia.org/wiki/Heredity">inheritance type</a> associated with that gene-disease should also be included. 
For example, <a href="https://www.omim.org">https://www.omim.org</a> is a good place to see examles.
The genetic disease <a href="https://www.omim.org/entry/219700?search=cystic%20fibrosis&amp;highlight=cystic%20fibrosi">cystic fibrosis</a> is shown with an inheritance type AR (autosomal recessive) meaning that damaging variants on both gene alleles are required to cause disease. 
The gene <em>cftr</em>, which is the genetic determinant of cystic fibrosis, also has an <a href="https://www.omim.org/entry/602421?search=cftr&amp;highlight=cftr">OMIM page <em>cftr</em></a> that also lists AR inheritance.
An excellent resource for matching gene to disease is the <a href="https://panelapp.genomicsengland.co.uk">https://panelapp.genomicsengland.co.uk</a>.
Individual genes can be explored, or “panels” of disease-specific gene lists can be explored. For example, here is the “<a href="https://panelapp.genomicsengland.co.uk/panels/545/">Bleeding and platelet disorders</a>” panel. 
This shows the “Mode of inheritance” and colour-coded confidence in the disease-gene relationship.
Integrating this type of expert-curated open datasets can be extremely useful.</p>

<h1 id="drug-indication-1">Drug indication</h1>
<p>The indication or warning can be difficult to automate.
For the example drug<br />
<a href="https://www.drugbank.ca/drugs/DB11595">https://www.drugbank.ca/drugs/DB11595</a><br />
the section “Pharmacology” “Indication” has the Indication info.<br />
The FDA label is contained as a PDF attachment in the section “REFERENCES” FDA label Download (245 KB).
If I had to automate the process I would add a URL link for each drug:<br />
for gene name CD274<br />
the drugbank column Drug IDs has these:<br />
DB11595; DB11714; DB11945<br />
and for each ID you could append the ID onto the drugbank URL to link to the webpage
<a href="https://www.drugbank.ca/drugs/">https://www.drugbank.ca/drugs/</a>.
You can do this in R with some technical how-to reading, or do it manually for a quick example like this and removing space to create a web URL.<br />
URL				Drug IDs<br />
https://www.drugbank.ca/drugs/	DB00303<br />
https://www.drugbank.ca/drugs/	DB00114<br />
https://www.drugbank.ca/drugs/	DB00142<br />
https://www.drugbank.ca/drugs/	DB01839<br />
https://www.drugbank.ca/drugs/	DB00125</p>

<h1 id="a-large-scale-example-summary">A large scale example summary</h1>
<p>I do not suggest this for a small project, but if I was to automate subsection requests for real:</p>
<ul>
  <li>[1] Download the whole database (probably a big table sized &gt;100MB) and</li>
  <li>[2] For every query (the Drug ID) extract the sections of interest (indication,  Biologic Classification, Description,  FDA label, etc.)</li>
  <li>[3] Display each section as additional columns in candidate genes table.<br />
<br /></li>
  <li>[1] Would be here: <a href="https://www.drugbank.ca/releases/latest">https://www.drugbank.ca/releases/latest</a></li>
  <li>[2] Would be like this: <a href="https://www.w3schools.com/xml/default.asp">https://www.w3schools.com/xml/default.asp</a><br />
Look at example 2, your database request might be something like:
[get food name = Belgian Waffles, description] or
[get drug ID = DB11595, indication,  Biologic Classification, Description,  FDA label.]
The database request problem can be tricky to optimise but not especially difficult with some experience in SQL-type management.</li>
  <li>[3] For every line in the gene candidate table, do this query request and output the result into the same row.<br />
The final table would be something that includes colunm headers like:<br />
Gene, consequence, variant, amino acid, genome position, CADD, DrugBank ID, Description, Indication, FDA label PDF link, etc.
This table could be ranked based on consequence, CADD score.
The top couple of rows then might be converted into a more readable format like a PDF.</li>
</ul>

<h1 id="funding-strategy">Funding strategy</h1>
<p>University-based start-ups ususally follow a plan with three or four funding stages before coming to market. 
It is also possible to get investors from day 1, but it is more usual to follow 
the steps outlined here.
It is also possible to partner with early investors for their guidance rather 
than for financial investment.</p>

<h3 id="example-funding-stages">Example funding stages:</h3>

<ul>
  <li>A - Fundamental research (e.g. SNSF)</li>
  <li>B - Tech development (gap between the basic research and usable product)</li>
  <li>C - Product development (e.g. industry, Innosuisse funding)</li>
</ul>

<h3 id="example-funding-sources-per-stage">Example funding sources per stage:</h3>

<ol>
  <li>[A] Ignite grant
    <ul>
      <li><a href="https://www.epfl.ch/innovation/startup/grants/ignition-grants/">https://www.epfl.ch/innovation/startup/grants/ignition-grants/</a></li>
      <li>30K - 6 month, salary/consumables</li>
    </ul>
  </li>
  <li>[A] Innogrant
    <ul>
      <li>&lt;https://www.epfl.ch/innovation/startup/grants/innogrants/</li>
      <li>100K - 1 year, salary for startup founder.</li>
    </ul>
  </li>
  <li>[B] BRIDGE Proof of concept
    <ul>
      <li><a href="https://www.bridge.ch/en/proof-of-concept/">https://www.bridge.ch/en/proof-of-concept/</a></li>
      <li>130K - 1 year</li>
    </ul>
  </li>
  <li>[B] BRIDGE Discovery
    <ul>
      <li><a href="https://www.bridge.ch/en/discovery/">https://www.bridge.ch/en/discovery/</a></li>
      <li>(alternative to Proof of concept, more for experienced researchers).</li>
    </ul>
  </li>
  <li>[C] Innosuisse
    <ul>
      <li>Federal funding for startups of social benefit, etc.</li>
      <li><a href="https://www.innosuisse.ch/inno/en/home/start-and-grow-your-business/startup-coaching.html">https://www.innosuisse.ch/inno/en/home/start-and-grow-your-business/startup-coaching.html</a></li>
    </ul>
  </li>
  <li>[C] Other investors, venture capital, or investing from big companies like J&amp;J, Pfizer, &amp;c. For example, in a recent J&amp;J meeting the start-up-related experts discussed how they work with startups.</li>
</ol>
<p><a href="https://advancesindrugdiscovery.splashthat.com">https://advancesindrugdiscovery.splashthat.com</a>
Basically, the “innovation” department experts help you to figure out what stage you are at. You can contact them as soon as you can disclose your tech non-confidentially (either you have patent protection or do not need it). If you were taking this path you would probably have competed steps 1-4.</p>

<h1 id="legal-requirements">Legal requirements</h1>
<h3 id="swiss-law">Swiss law</h3>
<p>I include both the English and French translations here as the original source does not always include full English translation. 
Swiss law contains specific provisions on genetic testing in humans.
The Federal Council is divided into <a href="https://www.admin.ch/gov/fr/accueil.html">7 departments</a> and one chancellory. 
Each department contains their relevant offices (usually fewer than 10). 
For our interests, the governing hierarchy order is as follows:</p>

<ul>
  <li>Le Conseil fédéral</li>
  <li>The Federal Council
    <ul>
      <li>Département fédéral de l’intérieur (DFI),</li>
      <li>Federal Department of Home Affair (FDHI),
        <ul>
          <li>Office fédéral de la santé publique (OFSP).</li>
          <li>Federal Office of Public Health (FOPH).</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>

<p>This office is then responsible for their relevant ordinances as organised under internal law:
<a href="https://www.fedlex.admin.ch/en/cc/internal-law/8">https://www.fedlex.admin.ch/en/cc/internal-law/8</a>.
A direct weblink to our area of interest is available on <a href="https://www.bag.admin.ch/bag/fr/home/gesetze-und-bewilligungen/gesetzgebung/gesetzgebung-mensch-gesundheit/gesetzgebung-genetische-untersuchungen.html">Législation Analyses génétiques</a>, 
but it is useful to view the legal framework in context instead of abstractly.</p>

<ul>
  <li>Internal law (1-9 sec)</li>
  <li>Droite interne (1-9 sec)</li>
  <li>Sec 8..: (81-86 subsections) Health - Employment - Social security</li>
  <li>Sec 8..: (81-86 subsections) Santé - Travail - Sécurité sociale</li>
  <li>Sec 81.: Health: (810-819 subsubsections)</li>
  <li>Sec 81.: Santé: (810-819 subsubsections)</li>
  <li>Sec 810: Medicine and human dignity</li>
  <li>Sec 810: Médecine et dignité humaine</li>
  <li>Sec 810.1: Medically assisted reproduction and genetic engineering in the human field</li>
  <li>Sec 810.1: Procréation médicalement assistée et génie génétique dans le domaine humain</li>
  <li>Sec 810.12: Federal Act of 8 October 2004 on Human Genetic Testing (HGTA)</li>
  <li>Sec 810.12: Loi fédérale du 8 octobre 2004 sur l’analyse génétique humaine (LAGH)<br />
This contains section (810.12) contains 10 sections with 44 articles covering the initial regulations.</li>
</ul>

<p>Three ordinance then include further details (with several sections, articles, or annexes each):</p>

<ul>
  <li>Sec 810.122.1: Ordinance of 14 February 2007 on Human Genetic Analysis (OAGH)</li>
  <li>Sec 810.122.1: Ordonnance du 14 février 2007 sur l’analyse génétique humaine (OAGH)</li>
  <li>Sec 810.122.122: Ordinance of the Federal Department of Home Affairs of February 14, 2007 on Human Genetic Analysis (OAGH-DFI)</li>
  <li>Sec 810.122.122: Ordonnance du DFI du 14 février 2007 sur l’analyse génétique humaine (OAGH-DFI)</li>
  <li>Sec 810.122.2: Ordinance of February 14, 2007 on DNA profiling in civil and administrative matters (OACA)</li>
  <li>Sec 810.122.2: Ordonnance du 14 février 2007 sur l’établissement de profils d’ADN en matière civile et administrative (OACA)</li>
</ul>

<p>The details are then listed individually at:
<a href="https://www.fedlex.admin.ch/fr/cc/internal-law/81#810.12">https://www.fedlex.admin.ch/fr/cc/internal-law/81#810.12</a>
and as stated, includes authorisation of “Pharmacogenetic tests performed to determine the effects of a planned therapy”, 
“analyses pharmacogénétiques effectuées dans le but de déterminer les effets d’une thérapie prévue”.</p>

<h3 id="accreditation">Accreditation</h3>
<p><a href="https://www.iso.org/standard/56115.html">ISO 15189</a> is a commonly sought standard accreditation for genetic analysis labs, 
which is carried out by recognized accreditation services like <a href="https://www.finas.fi/Sivut/default.aspx">FINAS</a>.
Here it is mentioned for the Geneva health 2030 genome center for clinical grade sequencing:
<a href="https://www.health2030genome.ch/dna-sequencing-platform/">https://www.health2030genome.ch/dna-sequencing-platform/</a>.
Other additional ISO accreditation standard concern 
<a href="https://www.iso.org/search.html?q=Genomic%20information%20representation&amp;hPP=10&amp;idx=all_en&amp;p=0&amp;hFR%5Bcategory%5D%5B0%5D=standard">Genomic information representation</a>,
including
<a href="https://www.iso.org/standard/75859.html">23092-4 Reference software</a> or 
<a href="https://www.iso.org/standard/79882.html">23092 Transport and storage of genomic information</a>.</p>

<p>GA4GH provides other information about many <a href="https://www.ga4gh.org/genomic-data-toolkit/regulatory-ethics-toolkit/">legal and ethic topics</a>.
BlueprintGenomics is a good example company for comparison:
<a href="https://blueprintgenetics.com/certifications/">https://blueprintgenetics.com/certifications/</a>.</p>

<h1 id="more-questions">More questions</h1>

<p>Q: Do we have to infer that in the future everybody will have his genome sequenced ?<br />
You do not have to assume this. It may become true. There could be privacy concerns or social problems arising from genetic prejudice, etc.</p>

<p>Q: Before using our algorithm, patient will have to sequence a part of their genome and thus this is a weakness of our algorithm?<br />
Your tool will provide a service based on genetics.<br />
Option [1] One has their genetics already and will use it for personal medicine.<br />
Option [2] they are prescribed a drug and want to only sequence the genes of interest that could affect this drug.<br />
Option [3] they do not want any genetic info and therefor your product is irrelevant.<br />
Option [4] they do not want their personal genetics, but are willing to estimate their relatedness to others in a genetic database and therefore calculate a probability of accuracy for this drug-gene information. e.g. both parents are Swiss and therefore based on the population they have probability of X that their genotype is Y.</p>

<p>Q: Is it realistic to assume that it will be feasible based on the fact that the sequencing cost is decreasing ?<br />
Irrelevant in this case, but yes, whole genome seq is sometimes below 200CHF will likely be common soon.</p>

<p>Q:  We plan to use polyphen and/or sift in order to discriminate between those types of variants. Is that a good idea?<br />
That is a good start. CADD score is also pretty well known among physicians.
In my opinion, I do not trust the scores often.
However, it is common that for processing a large amount of data, such prediction tools are useful in general.
For example, I might [1] rank first on VEP variant “consequences”; stop mutations with most importance.
[2] Then rank secondly with these values since you cannot interpret most with consequence = missense variant.</p>

<h1 id="references">References</h1>
<ul>
  <li><a href="https://www.fda.gov/drugs/science-research-drugs/table-pharmacogenomic-biomarkers-drug-labeling">https://www.fda.gov/drugs/science-research-drugs/</a></li>
  <li><a href="https://www.pharmgkb.org/view/drug-labels.do">https://www.pharmgkb.org/view/drug-labels.do</a></li>
  <li>Mary V. Relling &amp; William E. Evans. Pharmacogenomics in the clinic. <em>Nature</em> 2015; 526, 343–350. doi: 10.1038/nature15817</li>
  <li>Yip VL, Hawcutt DB, Pirmohamed M. Pharmacogenetic Markers of Drug Efficacy and Toxicity. <em>Clin Pharmacol Ther.</em> 2015;98(1):61-70. doi: 10.1002/cpt.135.</li>
  <li>David R. Adams, M.D., Ph.D.,  and Christine M. Eng, M.D. Next-Generation Sequencing to Diagnose Suspected Genetic Disorders N Engl J Med Oct 2018 doi: 10.1056/NEJMra1711801</li>
</ul>


</div>

<!--

<div id="related">
  <h2>Related Topics</h2>
  <ul class="topics">
    
  </ul>
</div>

-->


<div class="footer">
	<div class="contact">
	<p>
	<img src="/images/logos/logo_3.0.png" 
	alt="Logo image" width="99" height="67.1" />
	</p>
	</div>
	<div class="contact">
		<p>
		Built by Dylan Lawless
		<br />
		Postdoc at
		<a href="https://people.epfl.ch/dylan.lawless">EPFL</a>
		<br />
		<!--dylan( . )lawless( at )epfl( . )ch-->
		</p>
	</div>
	
	<div class="contact">
		<p>
		<a href="https://scholar.google.com/citations?user=RPBxP1wAAAAJ&hl=en&oi=ao">google scholar</a><br />
		<a href="https://dylan-lawless.github.io">Resume</a><br />
		<!-- <img src="https://hitcounter.pythonanywhere.com/count/tag.svg" alt="Hits"> -->
	</div>
	
	</div>
</div>

<!-- Banner on top right corner linking to github
<a href="https://github.com/mojombo"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_red_aa0000.png" alt="Fork me on GitHub" /></a>
-->

<!-- ClickTale Bottom part -->
<div id="ClickTaleDiv" style="display: none;"></div>
<script type="text/javascript">
if(document.location.protocol!='https:')
  document.write(unescape("%3Cscript%20src='https://s.clicktale.net/WRb.js'%20type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
if(typeof ClickTale=='function') ClickTale(206,0.3,"www03");
</script>
<!-- ClickTale end of Bottom part -->

<!-- Google Analytics -->
<!--
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "https://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
var pageTracker = _gat._getTracker("UA-6016902-1");
pageTracker._trackPageview();
</script>
-->
<!-- Google Analytics end -->

</body>
</html>
