
Predicting chaos: protein-protein intereaction with AI

## In silico protein structure prediction

One of the major immediately practical achievments in biology has been the 
advances made in highly accurate protein structure prediction.

The most well-known of these methods is AlphaFold,
<!-- {% cite jumper2021highly %}. -->
AlphaFold was seen to be highly successful in 
Critical Assessment of protein Structure Prediction (CASP) session 14, 
hosted by the Protein Structure Prediction Center
<https://www.predictioncenter.org>.
While there are several other very good methods, AlphaFold's success and popularity has massively driven the adoption of in silico-predicted protein structure use.

## Why rare disease genomics is one of the greatest discovery tools

Uncovering the mysteries of biological mechanisms is an extremely slow and incremental process.
With each step, a small piece of knowledge is gained about how 
proteins, carbohydrates and fats combine and react to make up single cells, transfer energy, build complex cellular structures that make life possible.

The scientific process allows us to take ground truths which have been previously uncovered, and by making a small incremental change to the situation, observe what effect occurs. 
If one is careful the changing cause and effect can be quantified and adequately explained.
The majority of biology labs worldwide conduct experiments this way:
undergraduate and postgraduate students, senior scientists, all making logical incremental modifications to experimental conditions to observe (usually expected) outcomes.

For example, if we have previously observed that two proteins are known to bind together, 
using [co-immunoprecipitation](https://en.wikipedia.org/wiki/Immunoprecipitation#Protein_complex_immunoprecipitation_(Co-IP)),
we might next want to identify the interaction sites on the surface of each protein.
We might already know some of the major [protein domains](https://en.wikipedia.org/wiki/Protein_domain) for these proteins.
Therefore, we might already predict a specific part of the protein responsible for this protein-protein interaction (PPI).
To test our hypothesis, a common approach would be to perform [site-directed mutagenesis](https://en.wikipedia.org/wiki/Site-directed_mutagenesis) (SDM) to induce a DNA point-mutation at the predicted PPI site.
With the mutated protein expression plasmid transfected in an in vitro expression system, we can finally observe whether the PPI has been disrupted. 
A loss of PPI might be due to directly mutation the interaction region or be due to some other protein misfolding - but those details can be ignored under the assumption that the test conditions resolve this.

Although a massive repertoire of biological experiment tools exist, this example is fine for summarising them.
I think of this as the traditional, sequential, standard approach.
If you were to start and PhD in wet-lap biology this is the kind of work you might be doing for several years, although all the grunt work has been glossed over here (Sanger sequencing, microbiology, autoclaving, tissue culturing, PCR, western-blotting, on and on...).

The _other_ approach to wet-lab biology is what I think of as the reverse, non-standard approach.
That is, find unusual _occurrences_ of something in nature and quantify what _makes_ them unusual.
Using this approach, we first observe a "successful" disruption experiment and secondly try to figure out the "modified experiment conditions" which nature has provided.

I believe that this second approach is one of the greatest discovery tools in biology.
However, it is a waiting game and only opportunistic events can be discovered. 
Therefore the majority of scientific work time has to be spent on the, first,  traditional approach.

As an example of this, say we are interested in (may favourite topic, the center of all medicine) human immunology.
Using the traditional incremental approach, we can slowly uncover how some proteins in humans are responsible for detecting foreign pathogenic patterns, thus indication a dangerous infection.
We could discover a long, complex series of events whereby a pathogen-recognition receptor binds to the pathogen, trigging downstream partner proteins to bind, create a phosphorylation chain-reaction, and induce the expression of pro-inflammatory proteins, thereby initiating the immune response.
Now imagine that since we already now _so much_, if we observe a very ill patient in hospital who has all of the symptoms of this type of infection response but no actual infection.
In fact, this patient chronically suffers from the autoinflammatory event. 
Using genomic and molecular biology discovery tools, we might uncover a rare genetic variant in this patients' DNA.
This DNA variant occurs on the same gene which is translated into the pathogen-recognition receptor protein, causing it to slightly misfold into the configuration which mimics a binding to a pathogen.
Since this variant occurs on every copy of the patients DNA, it means that this process occurs continuously, inducing a constant pro-inflammatory disease.

By discovering the cause of disease for the patient, we may be able to treat them and in some very luckly cases a cure might be possible by hematopoietic stem cell transplantation 
<!-- {% cite signa2022hematopoietic %}. -->
Furthermore, we may have just discovered a new mechanism if extremely important protein function.
This experiment designed by nature has provided us with a discovery, we just needed to uncover the experimental conditions that were used. 
Using the traditional, sequential approach, it may be unlikely that this particular DNA variant would every be mutated in vitro to observe and effect.
By genetic drift, de novo mutation, etc. nature only provides us with successful experiments. 
Every other genetic variation is either benign, emrbyonic lethal, or so unlikely that has simply not occurred.
In this case, 
[survivorship bias](https://en.wikipedia.org/wiki/Survivorship_bias) is extremely useful.
In the other cases (benign, embryonic lethal) we have 
[other ways to predict if they are also informative](https://lawlessgenomics.com/topic/acgm-criteria-table-main).

## What good is the protein structure in precision medicine?

A major source of evidence for interpreting gene to protein to function.

## Protein pathways: an discovery goldmine


