<!DOCTYPE html>
<html xmlns="https://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
	<!-- link to MathJax for LaTeX stlye eqn -->
	 <script type="text/javascript" async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-MML-AM_CHTML">
</script>

	 
	 <!-- https://realfavicongenerator.net -->
	 <link rel="apple-touch-icon" sizes="180x180" href="/images/logos/favicon_package_v0.16_bw/apple-touch-icon.png">
	 <link rel="icon" type="image/png" sizes="32x32" href="/images/logos/favicon_package_v0.16_bw/favicon-32x32.png">
	 <link rel="icon" type="image/png" sizes="16x16" href="/images/logos/favicon_package_v0.16_bw/favicon-16x16.png">
	 <link rel="manifest" href="/images/logos/favicon_package_v0.16_bw/site.webmanifest">
	 <meta name="msapplication-TileColor" content="#da532c">
	 <meta name="theme-color" content="#ffffff">
	
	 <meta name="viewport" content="width=device-width, initial-scale=1">
	
	<meta http-equiv="content-type" content="text/html; charset=utf-8">
	<meta name="description" content="A home for topics in human precision medicine, genomic analysis, and data visualisation.">
	<title>STAAR framework - variant-set test using annotation</title>
	<meta name="author" content="LawlessGenomics">
	<link href="https://lawlessgenomics.com/feed.xml" rel="alternate" title="LawlessGenomics" type="application/atom+xml">
	<!-- <meta name="readability-verification" content="QCzSs992GxmRYRKVpPeZ6LE2tS8aYKxsSSQKV8YM"/> -->
	
	<!-- include a google font -->
	<link href="https://fonts.googleapis.com/css?family=Roboto+Mono%7CRoboto+Slab:300%7CRoboto:500" rel="stylesheet">
	<link href="https://fonts.googleapis.com/css?family=Lora" rel="stylesheet">
	
	<!-- syntax highlighting CSS -->
	<link rel="stylesheet" href="/css/syntax.css" type="text/css">
	
	<!-- Homepage CSS -->
	<link rel="stylesheet" href="/css/screen.css" type="text/css" media="screen, projection">
	
	<!-- Google font -->
	<link rel="preconnect" href="https://fonts.googleapis.com"> 
	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin> 
	<link href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,400;0,600;0,700;1,400;1,600;1,700&amp;display=swap" rel="stylesheet">
	
	<!-- Typekit -->
	<script type="text/javascript" src="https://use.typekit.com/jpd0pfm.js"></script>
	<script type="text/javascript">try{Typekit.load();}catch(e){}</script>
</head>
<body>

<!-- ClickTale Top part -->
<script type="text/javascript">
	var WRInitTime=(new Date()).getTime();
</script>
<!-- ClickTale end of Top part -->

<!-- Logo and navigation bar -->



<!-- SITE START --------------------------------------------------------- -->
<div class="site">

<!-- NAV START --------------------------------------------------------- -->
<div class="sitewidthnav">
	<div class="title">
		<div class="nav-container">
			<nav>

			<div class="logoimage">
				<a href="/">
				<img style="margin-bottom:-16px" src="/images/logos/logo4.2_150ppi.png" alt="Logo image" width="25" height="38.5"></a>
				<!-- alt="Logo image" width="35.7" height="55" /></a> -->
			</div>

			<div class="logo">
				<a href="/">LawlessGenomics</a>
			</div>

				<input type="checkbox" id="nav-toggle">
					<label for="nav-toggle" class="gg-menu"></label>
					<div class="right-menu">
						<a href="/resume">About</a>
						<a href="/">Home</a>
					</div>
			</nav>
		</div>
	</div>



</div>
<!-- NAV END ----------------------------------------------------------- -->

<!-- MAIN START -------------------------------------------------------- -->
<div class="sitewidth">
	<div id="topic">
Reading time  9 minutes
<h1 id="staar-framework---variant-set-test-using-annotation">STAAR framework - variant-set test using annotation</h1>
<ul id="markdown-toc">
  <li><a href="#abbreviations" id="markdown-toc-abbreviations">Abbreviations</a></li>
  <li><a href="#overview" id="markdown-toc-overview">Overview</a></li>
  <li><a href="#method" id="markdown-toc-method">Method</a></li>
  <li><a href="#type-i-error-rate-simulations" id="markdown-toc-type-i-error-rate-simulations">Type I error rate simulations</a></li>
  <li><a href="#using-acat-in-staar" id="markdown-toc-using-acat-in-staar">Using ACAT in STAAR</a></li>
  <li><a href="#noncoding-rare-variant-association-tests" id="markdown-toc-noncoding-rare-variant-association-tests">Noncoding rare-variant association tests</a></li>
  <li><a href="#applying-staar-o-for-multiple-annotation-weights" id="markdown-toc-applying-staar-o-for-multiple-annotation-weights">Applying STAAR-O for multiple annotation weights</a></li>
  <li><a href="#non-gene-centric-analysis-using-dynamic-windows-with-scang-staar" id="markdown-toc-non-gene-centric-analysis-using-dynamic-windows-with-scang-staar">Non-gene-centric analysis using dynamic windows with SCANG-STAAR</a></li>
  <li><a href="#multi-weight-annotation-analysis" id="markdown-toc-multi-weight-annotation-analysis">Multi-weight annotation analysis</a></li>
</ul>

<p class="meta">26 Apr 2023 - last update</p>

<h2 id="abbreviations">Abbreviations</h2>

<ul>
  <li>COSI: Calibration Coalescent Model</li>
  <li>FAVOR: Functional Annotations of Variants - Online Resource</li>
  <li>GWAS: Genome-Wide Association Study</li>
  <li>MAF: Minor Allele Frequency</li>
  <li>RVAT: Rare-Variant Association Test</li>
  <li>SKAT: Sequence Kernel Association Test</li>
  <li>STAAR: variant-set test for association using annotation Information</li>
  <li>STAAR-O: STAAR with multiple annotation weights (Omnibus Test)</li>
  <li>SCANG-STAAR: SCANG (scan the genome) method combined with STAAR</li>
  <li>WGS: Whole-Genome Sequencing</li>
  <li>WES: Whole-Exome Sequencing</li>
</ul>

<h2 id="overview">Overview</h2>

<p>The SKAT (Sequence Kernel Association Test) method is fundamental to the STAAR (variant-Set Test for Association using Annotation information) framework. 
STAAR builds upon the SKAT method by incorporating multiple functional annotations for genetic variants to improve the power of the association tests.
The integration of annotation information in STAAR allows prioritization of functional variants using multidimensional variant biological functions.</p>

<p>The authors target:</p>

<ol>
  <li>
<strong>Limitations of existing methods</strong>: Current approaches for RV analysis of WGS/WES studies have limited ability to define analysis units in the noncoding genome.</li>
  <li>
<strong>Proposed framework</strong>: Introduce a computationally efficient and robust noncoding RV association detection framework for WGS data.</li>
  <li>
<strong>Gene-centric analysis</strong>: Propose additional strategies for grouping noncoding variants based on functional annotations within STAAR.</li>
  <li>
<strong>Non-gene-centric analysis</strong>: Propose SCANG-STAAR, a flexible data-adaptive window size RVAT method that incorporates multiple functional annotations and accounts for relatedness and population structure.</li>
  <li>
<strong>STAARpipeline</strong>: Develop a pipeline that functionally annotates both noncoding and coding variants of a WGS study and performs RVATs using the proposed methods for both gene-centric and non-gene-centric analysis.</li>
</ol>

<p>First, the rare-variant association tests (RVATs) are defined to analyse rare variants in association with phenotypes. 
Then, STAAR is applied, which incorporates multiple functional annotations for genetic variants to boost power in RVATs. STAAR can be applied in different ways, such as STAAR-O, which uses multiple annotation weights. Additionally, for non-gene-centric analysis, SCANG-STAAR is proposed, which uses dynamic windows with data-adaptive sizes and incorporates multidimensional functional annotations.</p>

<h2 id="method">Method</h2>

<ol>
  <li>Notations and model:
    <ul>
      <li>Generalized linear model for unrelated samples (Equation 1):<br>
\(g(\mu_i) = \alpha_0 + \boldsymbol{X}_i^T\alpha + \boldsymbol{G}_i^T\beta\)</li>
      <li>Generalized linear mixed model for related samples (Equation 2):<br>
\(g(\mu_i) = \alpha_0 + \boldsymbol{X}_i^T\alpha + \boldsymbol{G}_i^T\beta + b_i\)</li>
    </ul>
  </li>
  <li>Variant-set test using STAAR:
    <ul>
      <li>Burden test statistic:<br>
\(Q_{\mathrm{Burden},l,k} = \left(\sum_{j=1}^p \hat{\pi}_{jk}w_{jl}S_j\right)^2\)</li>
      <li>SKAT test statistic:<br>
\(Q_{\mathrm{SKAT},l,k} = \sum_{j=1}^p \hat{\pi}_{jk}w_{jl}^2S_j^2\)</li>
      <li>ACAT-V test statistic:<br>
\(Q_{\mathrm{ACAT-V},l,k} = \overline{\hat{\pi}_{\cdot k}w_{\cdot l}^2\mathrm{MAF}(1 - \mathrm{MAF})}\tan\left((0.5 - p_{0,k})\pi\right) + \sum_{j=1}^{p'} \hat{\pi}_{jk}w_{jl}^2\mathrm{MAF}_j(1 - \mathrm{MAF}_j)\tan\left((0.5 - p_j)\pi\right)\)</li>
    </ul>
  </li>
  <li>STAAR tests:
    <ul>
      <li>STAAR-Burden (STAAR-B):<br>
\(T_{\mathrm{STAAR-B}} = \sum_{l=1}^2\sum_{k=0}^K\frac{\tan\{(0.5 - p_{\mathrm{Burden},l,k})\pi\}}{2(K+1)}\)</li>
      <li>STAAR-S:<br>
\(T_{\mathrm{STAAR-S}} = \sum_{l=1}^2\sum_{k=0}^K\frac{\tan\{(0.5 - p_{\mathrm{SKAT},l,k})\pi\}}{2(K+1)}\)</li>
      <li>STAAR-ACAT-V (STAAR-A):<br>
\(T_{\mathrm{STAAR-A}} = \sum_{l=1}^2\sum_{k=0}^K\frac{\tan\{(0.5 - p_{\mathrm{ACAT-V},l,k})\pi\}}{2(K+1)}\)</li>
      <li>STAAR-O test statistic:<br>
\(T_{\mathrm{STAAR-O}} = \frac{1}{3}\left[\tan\{(0.5 - p_{\mathrm{STAAR-B}})\pi\} + \tan\{(0.5 - p_{\mathrm{STAAR-S}})\pi\} + \tan\{(0.5 - p_{\mathrm{STAAR-A}})\pi\}\right]\)</li>
    </ul>
  </li>
  <li>Dynamic window analysis using SCANG-STAAR:
    <ul>
      <li>Minimum P value of all candidate moving windows:<br>
\(p_{\mathrm{min}} = \min_{L_{\mathrm{min}} \le |I| \le L_{\mathrm{max}}}p(I)\)</li>
    </ul>
  </li>
  <li>Conditional analysis
    <ul>
      <li>The STAARpipeline performs conditional analysis to identify RV association independent of known variants. We first select a list of known variants by including the trait-associated variants identified in literature, for example, variants indexed in the GWAS Catalog or significant variants in large-scale GWAS. The significant variants detected in individual analyses using the same data could also be added into the known variants list to ensure the RV signals are not captured by the significant individual variants. We then use the following stepwise selection strategy to select a subset of independent variants representing the known variant list as the variants adjusted in the conditional analysis:
        <ol>
          <li>Calculate the individual P value of all variants in the known variants list and select the most significant variant.</li>
          <li>For each step, calculate the P values of all the remaining variants conditional on the variant(s) that have already been selected. For each variant, we only condition on the selected variants within a specified region of that variant, such as the \(\pm\)1-Mb window.</li>
          <li>Select the variant with minimum conditional P value that is lower than the cutoff P value, for example, \(1 \times 10^{-4}\).</li>
          <li>Repeat steps 2–3 until no variants can be selected.
Finally, we calculate the conditional P value of each significant RV analysis unit by adjusting for the selected variants residing in an extended region (for example, \(\pm\)1-Mb window) of the analysis unit.</li>
        </ol>
      </li>
    </ul>
  </li>
</ol>

<h2 id="type-i-error-rate-simulations">Type I error rate simulations</h2>
<p>The authors perform this as follows:</p>

<ol>
  <li>100,000 sequencing chromosomes in a 10-Mb region</li>
  <li>African American population linkage disequilibrium structure (COSI)</li>
  <li>Total sample sizes (n = 50,000)</li>
  <li>Continuous traits from a linear model:<br>
\(Y_i = 0.5X_{1i} + 0.5X_{2i} + {\it{\epsilon }}_i\),<br>
\(X_{1i} \sim N(0,1)$, $X_{2i} \sim \mathrm{Bernoulli}(0.5)\), \(\epsilon_i \sim N(0,1)\)</li>
  <li>Dichotomous traits from a logistic model:<br>
\(\mathrm{logit}\,P(Y_i = 1) = \alpha_0 + 0.5X_{1i} + 0.5X_{2i}\),<br>
\(X_{1i}\) and \(X_{2i}\) defined as in continuous traits, \(\alpha_0\) for 1% prevalence</li>
  <li>Case–control sampling</li>
  <li>Ten annotations generated: \(A_1, \ldots, A_{10}\) as i.i.d. \(N(0,1)\) for each variant</li>
  <li>SCANG-STAAR-S, SCANG-STAAR-B, SCANG-STAAR-O with MAF and ten annotations as weights</li>
  <li>10,000 replicates for genome-wise (family-wise) type I error rates at α = 0.05 and 0.01</li>
</ol>

<h2 id="using-acat-in-staar">Using ACAT in STAAR</h2>

<p>The Aggregated Cauchy Association Test (ACAT) is a statistical method used for rare-variant association tests (RVATs) in genetic studies. 
ACAT is designed to aggregate the association signals of multiple rare genetic variants within a genomic region or a gene, while accounting for the directions of the effects of these variants on the phenotype of interest. 
The ACAT method utilizes a Cauchy distribution, which allows for improved performance in identifying true associations, especially when the directions and magnitudes of variant effects are heterogeneous.</p>

<p>In a separate page, I discuss the ACAT method. 
Some of the following passages are included in both pages since they related.</p>

<p><img src="/images/acat_fig_2.png" width="80%" style="display: block; margin: auto;"></p>

<p><strong>Figure 1.</strong> Slide from presentation of ACAT method by Dr. Xihong Lin.</p>

<h2 id="noncoding-rare-variant-association-tests">Noncoding rare-variant association tests</h2>

<p>The authors propose the following:</p>

<ol>
  <li>
<strong>Noncoding RVAT framework</strong>: Propose a computationally efficient and robust noncoding rare-variant association test (RVAT) framework for phenotype-genotype association analyses of WGS data.</li>
  <li>
<strong>Regression-based</strong>: Allows adjustment for covariates, population structure, and relatedness by fitting linear and logistic mixed models for quantitative and dichotomous traits.</li>
  <li>
<strong>Gene-centric approach</strong>: Group noncoding rare variants for each gene using eight functional categories of regulatory regions and apply STAAR, which incorporates multiple in silico variant functional annotation scores.</li>
  <li>
<strong>Non-gene-centric approach</strong>: Propose SCANG-STAAR, using dynamic windows with data-adaptive sizes and incorporating multidimensional functional annotations instead of fixed-size sliding windows.</li>
  <li>
<strong>Conditional analysis</strong>: Perform analytical follow-up to dissect RV association signals independent of known variants.</li>
</ol>

<h2 id="applying-staar-o-for-multiple-annotation-weights">Applying STAAR-O for multiple annotation weights</h2>

<p>In the STAAR Nature Methods paper, the section <em>Gene-centric analysis of the noncoding genome</em> 
shows how the STAAR method can indeed be used to capitalize on the ACAT method to obtain a combined p-value from a set of annotations for a single variant. The STAAR framework incorporates multiple functional annotation scores into the RVATs (rare-variant association tests) to increase the power of association analysis. In this context, it uses the STAAR-O test, an omnibus test that aggregates annotation-weighted burden test, SKAT, and ACAT-V within the STAAR framework.</p>

<p>By incorporating multiple functional annotation scores, such as CADD, LINSIGHT, FATHMM-XF, and annotation principal components (aPCs), the STAAR method enhances the ability to detect associations between variants and traits of interest. Therefore, the STAAR framework can be used to leverage the strengths of the ACAT method and obtain a combined p-value from a set of annotations for a single variant or a set of variants.</p>

<h2 id="non-gene-centric-analysis-using-dynamic-windows-with-scang-staar">Non-gene-centric analysis using dynamic windows with SCANG-STAAR</h2>

<p>The SCANG-STAAR method is an improvement over the fixed-size sliding window RVAT in the STAAR framework. It proposes a dynamic window-based approach called SCANG-STAAR, which extends the SCANG procedure by incorporating multidimensional functional annotations. This method allows for flexible detection of locations and sizes of signal windows across the genome, as the locations of regions associated with a disease or trait are often unknown in advance, and their sizes may vary across the genome. Using a prespecified fixed-size sliding window for RVAT can lead to power loss if the prespecified window sizes do not align with the true locations of the signals.</p>

<p>The SCANG-STAAR method has two main procedures: SCANG-STAAR-S and SCANG-STAAR-B. SCANG-STAAR-S extends the SCANG-SKAT (SCANG-S) procedure by calculating the STAAR-SKAT (STAAR-S) p-value in each overlapping window by incorporating multiple variant functional annotations, instead of using just the MAF-weight-based SKAT p-value. SCANG-STAAR-B is based on the STAAR-Burden p-value. SCANG-STAAR-S has two advantages over SCANG-STAAR-B in detecting noncoding associations using dynamic windows: first, the effects of causal variants in a neighborhood in the noncoding genome tend to be in different directions, especially in intergenic regions; second, due to the different correlation structures of the two test statistics for overlapping windows, the genome-wide significance threshold of SCANG-STAAR-B is lower than that of SCANG-STAAR-S.</p>

<p>SCANG-STAAR also provides the SCANG-STAAR-O procedure, based on an omnibus p-value of SCANG-STAAR-S and SCANG-STAAR-B calculated by the ACAT method. However, unlike STAAR-O, the ACAT-V test is not incorporated into the omnibus test because it is designed for sparse alternatives, and as a result, it tends to detect the region with the smallest size that contains the most significant variant in the dynamic window procedure.</p>

<p><img src="/images/acat_fig_3.png" width="80%" style="display: block; margin: auto;"></p>

<p><strong>Figure 2.</strong> Slide from presentation of ACAT application in STAAR by Dr. Xihong Lin.</p>

<h2 id="multi-weight-annotation-analysis">Multi-weight annotation analysis</h2>

<p>The STAAR framework can be used to combine the p-values associated with each of the 5 annotation columns (CADD_score, MAF, GnomAD_AF, REVEL_score, ClinVar_score) for a single variant. STAAR incorporates multiple functional annotation scores as weights when constructing its statistics, making it suitable for combining p-values from different annotation columns to obtain a single combined p-value for that variant.</p>

<p><img src="/images/staar_fig_1.png" width="80%" style="display: block; margin: auto;"></p>

<p><strong>Figure 3</strong>: From Li et al NatMeth 2022: <em>a, Prepare the input data of STAARpipeline, including genotypes, phenotypes and covariates. b, Annotate all variants in the genome using FAVORannotator through FAVOR database and calculate the (sparse) GRM. c, Define analysis units in the noncoding genome: eight functional categories of regulatory regions, sliding windows and dynamic windows using SCANG. d, Obtain genome-wide significant associations and perform analytical follow-up via conditional analysis.</em></p>


</div>

<!--

<div id="related">
  <h2>Related Topics</h2>
  <ul class="topics">
    
  </ul>
</div>

-->

















</div>
<!-- # MAIN END ---------------------------------------------------------- -->

<!-- FOOT START -------------------------------------------------------- -->
<div class="sitewidthnav">
	<div class="footer">
		<div class="foot-container">
				<a href="/" class="logo-foot">
				<img style="margin-bottom:-16px" src="/images/logos/logo4.2_150ppi.png" alt="Logo image" width="25" height="38.5"></a>
					<div class="right-root">
						<p><a href="https://github.com/DylanLawless/DylanLawless.github.io/tree/source">Maintained</a> by Dylan Lawless. Scientist at <a href="https://people.epfl.ch/dylan.lawless">EPFL</a>.
						<a href="/resume">Resume</a>
						and <a href="https://scholar.google.com/citations?user=RPBxP1wAAAAJ&hl=en&oi=ao">Google scholar</a>.
						<a href="https://lawlessgenomics.com/feed.xml">RSS feed</a>
						</p>
					</div>
		</div>
	</div>
</div>
<!-- FOOT END ---------------------------------------------------------- -->

</div>
<!-- SITE END ----------------------------------------------------------- -->


<!-- Banner on top right corner linking to github
<a href="https://github.com/mojombo"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_red_aa0000.png" alt="Fork me on GitHub" /></a>
-->

<!-- ClickTale Bottom part -->
<div id="ClickTaleDiv" style="display: none;"></div>
<script type="text/javascript">
if(document.location.protocol!='https:')
  document.write(unescape("%3Cscript%20src='https://s.clicktale.net/WRb.js'%20type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
if(typeof ClickTale=='function') ClickTale(206,0.3,"www03");
</script>
<!-- ClickTale end of Bottom part -->

<!-- Google Analytics -->
<!--
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "https://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
var pageTracker = _gat._getTracker("UA-6016902-1");
pageTracker._trackPageview();
</script>
-->
<!-- Google Analytics end -->

</body>
</html>
