<!DOCTYPE html>
<html xmlns="https://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
	<!-- link to MathJax for LaTeX stlye eqn -->
	 <script type="text/javascript" async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-MML-AM_CHTML">
</script>

	 
	 <!-- https://realfavicongenerator.net -->
	 <link rel="apple-touch-icon" sizes="180x180" href="/images/logos/favicon_package_v0.16_bw/apple-touch-icon.png">
	 <link rel="icon" type="image/png" sizes="32x32" href="/images/logos/favicon_package_v0.16_bw/favicon-32x32.png">
	 <link rel="icon" type="image/png" sizes="16x16" href="/images/logos/favicon_package_v0.16_bw/favicon-16x16.png">
	 <link rel="manifest" href="/images/logos/favicon_package_v0.16_bw/site.webmanifest">
	 <meta name="msapplication-TileColor" content="#da532c">
	 <meta name="theme-color" content="#ffffff">
	
	 <meta name="viewport" content="width=device-width, initial-scale=1">
	
	<meta http-equiv="content-type" content="text/html; charset=utf-8">
	<meta name="description" content="A home for topics in human precision medicine, genomic analysis, and data visualisation.">
	<title>Variant annotation</title>
	<meta name="author" content="LawlessGenomics">
	<link href="https://lawlessgenomics.com/feed.xml" rel="alternate" title="LawlessGenomics" type="application/atom+xml">
	<!-- <meta name="readability-verification" content="QCzSs992GxmRYRKVpPeZ6LE2tS8aYKxsSSQKV8YM"/> -->
	
	<!-- include a google font -->
	<link href="https://fonts.googleapis.com/css?family=Roboto+Mono%7CRoboto+Slab:300%7CRoboto:500" rel="stylesheet">
	<link href="https://fonts.googleapis.com/css?family=Lora" rel="stylesheet">
	
	<!-- syntax highlighting CSS -->
	<link rel="stylesheet" href="/css/syntax.css" type="text/css">
	
	<!-- Homepage CSS -->
	<link rel="stylesheet" href="/css/screen.css" type="text/css" media="screen, projection">
	
	<!-- Google font -->
	<link rel="preconnect" href="https://fonts.googleapis.com"> 
	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin> 
	<link href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,400;0,600;0,700;1,400;1,600;1,700&amp;display=swap" rel="stylesheet">
	
	<!-- Typekit -->
	<script type="text/javascript" src="https://use.typekit.com/jpd0pfm.js"></script>
	<script type="text/javascript">try{Typekit.load();}catch(e){}</script>
</head>
<body>

<!-- ClickTale Top part -->
<script type="text/javascript">
	var WRInitTime=(new Date()).getTime();
</script>
<!-- ClickTale end of Top part -->

<!-- Logo and navigation bar -->



<!-- SITE START --------------------------------------------------------- -->
<div class="site">

<!-- NAV START --------------------------------------------------------- -->
<div class="sitewidthnav">
	<div class="title">
		<div class="nav-container">
			<nav>

			<div class="logoimage">
				<a href="/">
				<img style="margin-bottom:-16px" src="/images/logos/logo4.1_150ppi.png" alt="Logo image" width="35.7" height="55"></a>
			</div>

			<div class="logo">
				<a href="/">LawlessGenomics</a>
			</div>

				<input type="checkbox" id="nav-toggle">
					<label for="nav-toggle" class="gg-menu"></label>
					<div class="right-menu">
						<a href="/resume">About</a>
						<a href="/">Home</a>
					</div>
			</nav>
		</div>
	</div>
</div>
<!-- NAV END ----------------------------------------------------------- -->

<!-- MAIN START -------------------------------------------------------- -->
<div class="sitewidth">
	<div id="topic">
Reading time  6 minutes
<h1 id="variant-annotation">Variant annotation</h1>
<p class="meta">23 Sept 2021</p>

<p>Unfinished article</p>

<h2 id="annotation">Annotation</h2>
<p><a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/">Variant Effect Predictor (VEP)</a> is very useful.
A real-world example of VEP in use can be read in <a href="https://lawlessgenomics.com/topic/pharmacogenomics">this pharmacogenomics article</a>.
During variant annotation, VEP supplies a “consequence” column.
Consequences are general and based on translation of genetic code in humans. 
The Loss-of-function (LoF) consequence is the simplest example (splice, stop mutations).
The variant consequence may be one of the defining criteria by which variants can 
be included in analysis since they are <em>interpretable</em> or of ostensibly <em>known significance</em>.
<em>Note: Using this alone could introduce spurious results so it is  best to have a solid criteria 
for selecting consequences of interest</em>.
The consequences provided by VEP are too long to discuss in detail here.
The table from the ensembl website is worth reading; the HIGH impact variants 
might be a simple method for selecting candidates:
<a href="https://grch37.ensembl.org/info/genome/variation/prediction/predicted_data.html#consequences">Ensembl Variation - Calculated variant consequences</a>.<br>
<img src="/images/VEP_consequences.jpg" width="100%"></p>

<p><em>Note: For a real product, the code should be run offline (a perl program with a few local library dependencies). The databases/cache that it uses are a bit too large to include on in a user software. In the real world you would have to send anonymised packets from the user via an API for accessing the genomic databases hosted on your servers. Make sure to check their license to see if you can use oftware and databases in a commercial product</em>.
<a href="http://www.ensembl.org/info/about/legal/code_licence.html">http://www.ensembl.org/info/about/legal/code_licence.html</a></p>

<p>Running the software:</p>

<ul>
  <li>Using VEP is a vital part of converting the DNA variant information (genome position and nucleotide change) into annotated variant effects (protein coding change, gene name, predicted pathogenicity).</li>
  <li>It requires the VEP code to run and requires a copy of the database files (reference genome, gene information, etc.).</li>
  <li>You can upload a small number of variants to the online VEP web server to do this, or you can download the database and code to run on your own computer/server.</li>
</ul>

<p>The terms in the table below are shown in order of severity (more severe to less severe) as estimated by Ensembl.
In single case studies, we often find rare disease due to variants of “IMPACT” that is “HIGH or MODERATE”.
However, there may be a bias in that these variants are often most likely to be functionally interpretable.
Variants that are more difficult to predict function might only be detected in large-scale association studies where a  is statistically valid explanation for correlation/cause is possible.
Other tools may provide alternative estimates that the ones shown here from VEP. 
Missense variants may have further annotation on their effect on the protein function, using a number of algorithms
(see <a href="https://m.ensembl.org/info/genome/variation/prediction/predicted_data.html">VEP Ensembl Variation - Calculated variant consequences</a>).</p>

<div class="table-wrapper">

  <table>
    <thead>
      <tr>
        <th>SO term</th>
        <th>SO description</th>
        <th>SO accession</th>
        <th>Display term</th>
        <th>IMPACT</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>transcript_ablation</td>
        <td>A feature ablation whereby the deleted region includes a transcript feature.</td>
        <td>SO:0001893</td>
        <td>Transcript ablation</td>
        <td>HIGH</td>
      </tr>
      <tr>
        <td>splice_acceptor_variant</td>
        <td>A splice variant that changes the 2 base region at the 3’ end of an intron.</td>
        <td>SO:0001574</td>
        <td>Splice acceptor variant</td>
        <td>HIGH</td>
      </tr>
      <tr>
        <td>splice_donor_variant</td>
        <td>A splice variant that changes the 2 base region at the 5’ end of an intron.</td>
        <td>SO:0001575</td>
        <td>Splice donor variant</td>
        <td>HIGH</td>
      </tr>
      <tr>
        <td>stop_gained</td>
        <td>A sequence variant whereby at least one base of a codon is changed, resulting in a premature stop codon, leading to a shortened transcript.</td>
        <td>SO:0001587</td>
        <td>Stop gained</td>
        <td>HIGH</td>
      </tr>
      <tr>
        <td>frameshift_variant</td>
        <td>A sequence variant which causes a disruption of the translational reading frame, because the number of nucleotides inserted or deleted is not a multiple of three.</td>
        <td>SO:0001589</td>
        <td>Frameshift variant</td>
        <td>HIGH</td>
      </tr>
      <tr>
        <td>stop_lost</td>
        <td>A sequence variant where at least one base of the terminator codon (stop) is changed, resulting in an elongated transcript.</td>
        <td>SO:0001578</td>
        <td>Stop lost</td>
        <td>HIGH</td>
      </tr>
      <tr>
        <td>start_lost</td>
        <td>A codon variant that changes at least one base of the canonical start codon.</td>
        <td>SO:0002012</td>
        <td>Start lost</td>
        <td>HIGH</td>
      </tr>
      <tr>
        <td>transcript_amplification</td>
        <td>A feature amplification of a region containing a transcript.</td>
        <td>SO:0001889</td>
        <td>Transcript amplification</td>
        <td>HIGH</td>
      </tr>
      <tr>
        <td>inframe_insertion</td>
        <td>An inframe non synonymous variant that inserts bases into in the coding sequence.</td>
        <td>SO:0001821</td>
        <td>Inframe insertion</td>
        <td>MODERATE</td>
      </tr>
      <tr>
        <td>inframe_deletion</td>
        <td>An inframe non synonymous variant that deletes bases from the coding sequence.</td>
        <td>SO:0001822</td>
        <td>Inframe deletion</td>
        <td>MODERATE</td>
      </tr>
      <tr>
        <td>missense_variant</td>
        <td>A sequence variant, that changes one or more bases, resulting in a different amino acid sequence but where the length is preserved.</td>
        <td>SO:0001583</td>
        <td>Missense variant</td>
        <td>MODERATE</td>
      </tr>
      <tr>
        <td>protein_altering_variant</td>
        <td>A sequence_variant which is predicted to change the protein encoded in the coding sequence.</td>
        <td>SO:0001818</td>
        <td>Protein altering variant</td>
        <td>MODERATE</td>
      </tr>
      <tr>
        <td>splice_region_variant</td>
        <td>A sequence variant in which a change has occurred within the region of the splice site, either within 1-3 bases of the exon or 3-8 bases of the intron.</td>
        <td>SO:0001630</td>
        <td>Splice region variant</td>
        <td>LOW</td>
      </tr>
      <tr>
        <td>incomplete_terminal_codon_variant</td>
        <td>A sequence variant where at least one base of the final codon of an incompletely annotated transcript is changed.</td>
        <td>SO:0001626</td>
        <td>Incomplete terminal codon variant</td>
        <td>LOW</td>
      </tr>
      <tr>
        <td>start_retained_variant</td>
        <td>A sequence variant where at least one base in the start codon is changed, but the start remains.</td>
        <td>SO:0002019</td>
        <td>Start retained variant</td>
        <td>LOW</td>
      </tr>
      <tr>
        <td>stop_retained_variant</td>
        <td>A sequence variant where at least one base in the terminator codon is changed, but the terminator remains.</td>
        <td>SO:0001567</td>
        <td>Stop retained variant</td>
        <td>LOW</td>
      </tr>
      <tr>
        <td>synonymous_variant</td>
        <td>A sequence variant where there is no resulting change to the encoded amino acid.</td>
        <td>SO:0001819</td>
        <td>Synonymous variant</td>
        <td>LOW</td>
      </tr>
      <tr>
        <td>coding_sequence_variant</td>
        <td>A sequence variant that changes the coding sequence.</td>
        <td>SO:0001580</td>
        <td>Coding sequence variant</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>mature_miRNA_variant</td>
        <td>A transcript variant located with the sequence of the mature miRNA.</td>
        <td>SO:0001620</td>
        <td>Mature miRNA variant</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>5_prime_UTR_variant</td>
        <td>A UTR variant of the 5’ UTR.</td>
        <td>SO:0001623</td>
        <td>5 prime UTR variant</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>3_prime_UTR_variant</td>
        <td>A UTR variant of the 3’ UTR.</td>
        <td>SO:0001624</td>
        <td>3 prime UTR variant</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>non_coding_transcript_exon_variant</td>
        <td>A sequence variant that changes non-coding exon sequence in a non-coding transcript.</td>
        <td>SO:0001792</td>
        <td>Non coding transcript exon variant</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>intron_variant</td>
        <td>A transcript variant occurring within an intron.</td>
        <td>SO:0001627</td>
        <td>Intron variant</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>NMD_transcript_variant</td>
        <td>A variant in a transcript that is the target of NMD.</td>
        <td>SO:0001621</td>
        <td>NMD transcript variant</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>non_coding_transcript_variant</td>
        <td>A transcript variant of a non coding RNA gene.</td>
        <td>SO:0001619</td>
        <td>Non coding transcript variant</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>upstream_gene_variant</td>
        <td>A sequence variant located 5’ of a gene.</td>
        <td>SO:0001631</td>
        <td>Upstream gene variant</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>downstream_gene_variant</td>
        <td>A sequence variant located 3’ of a gene.</td>
        <td>SO:0001632</td>
        <td>Downstream gene variant</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>TFBS_ablation</td>
        <td>A feature ablation whereby the deleted region includes a transcription factor binding site.</td>
        <td>SO:0001895</td>
        <td>TFBS ablation</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>TFBS_amplification</td>
        <td>A feature amplification of a region containing a transcription factor binding site.</td>
        <td>SO:0001892</td>
        <td>TFBS amplification</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>TF_binding_site_variant</td>
        <td>A sequence variant located within a transcription factor binding site.</td>
        <td>SO:0001782</td>
        <td>TF binding site variant</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>regulatory_region_ablation</td>
        <td>A feature ablation whereby the deleted region includes a regulatory region.</td>
        <td>SO:0001894</td>
        <td>Regulatory region ablation</td>
        <td>MODERATE</td>
      </tr>
      <tr>
        <td>regulatory_region_amplification</td>
        <td>A feature amplification of a region containing a regulatory region.</td>
        <td>SO:0001891</td>
        <td>Regulatory region amplification</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>feature_elongation</td>
        <td>A sequence variant that causes the extension of a genomic feature, with regard to the reference sequence.</td>
        <td>SO:0001907</td>
        <td>Feature elongation</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>regulatory_region_variant</td>
        <td>A sequence variant located within a regulatory region.</td>
        <td>SO:0001566</td>
        <td>Regulatory region variant</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>feature_truncation</td>
        <td>A sequence variant that causes the reduction of a genomic feature, with regard to the reference sequence.</td>
        <td>SO:0001906</td>
        <td>Feature truncation</td>
        <td>MODIFIER</td>
      </tr>
      <tr>
        <td>intergenic_variant</td>
        <td>A sequence variant located in the intergenic region, between genes.</td>
        <td>SO:0001628</td>
        <td>Intergenic variant</td>
        <td>MODIFIER</td>
      </tr>
    </tbody>
  </table>

</div>


</div>

<!--

<div id="related">
  <h2>Related Topics</h2>
  <ul class="topics">
    
  </ul>
</div>

-->

















</div>
<!-- # MAIN END ---------------------------------------------------------- -->

<!-- FOOT START -------------------------------------------------------- -->
<div class="sitewidthnav">
	<div class="footer">
		<div class="foot-container">
				<a href="/" class="logo-foot">
				<img style="margin-bottom:-16px" src="/images/logos/logo4.2_150ppi.png" alt="Logo image" width="35.7" height="55"></a>
					<div class="right-root">
						<p><a href="https://github.com/DylanLawless/DylanLawless.github.io/tree/source">Maintained</a> by Dylan Lawless. Postdoc at <a href="https://people.epfl.ch/dylan.lawless">EPFL</a>.
						<a href="/resume">Resume</a>
						and <a href="https://scholar.google.com/citations?user=RPBxP1wAAAAJ&hl=en&oi=ao">Google scholar</a>.
						<a href="https://lawlessgenomics.com/feed.xml">RSS feed</a>
						</p>
					</div>
		</div>
	</div>
</div>
<!-- FOOT END ---------------------------------------------------------- -->

</div>
<!-- SITE END ----------------------------------------------------------- -->


<!-- Banner on top right corner linking to github
<a href="https://github.com/mojombo"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_red_aa0000.png" alt="Fork me on GitHub" /></a>
-->

<!-- ClickTale Bottom part -->
<div id="ClickTaleDiv" style="display: none;"></div>
<script type="text/javascript">
if(document.location.protocol!='https:')
  document.write(unescape("%3Cscript%20src='https://s.clicktale.net/WRb.js'%20type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
if(typeof ClickTale=='function') ClickTale(206,0.3,"www03");
</script>
<!-- ClickTale end of Bottom part -->

<!-- Google Analytics -->
<!--
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "https://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
var pageTracker = _gat._getTracker("UA-6016902-1");
pageTracker._trackPageview();
</script>
-->
<!-- Google Analytics end -->

</body>
</html>
