%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Document type      %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\documentclass[preprint,11pt,fleqn]{elsarticle}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Document structure %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\usepackage[margin=3.6cm]{geometry} % boarder size
\usepackage{lineno} % used along with \linenumbers after begin document. 
\usepackage{setspace} % line spacing
    \setstretch{1.4}
\makeatletter % The following lines get rid of footer stating pre-preint to elsevier.
	\def\ps@pprintTitle{%
	\let\@oddhead\@empty
	\let\@evenhead\@empty
	\def\@oddfoot{}%
	\let\@evenfoot\@oddfoot}
\makeatother
\graphicspath{ {images/} } % sets the path to image files (Figures)

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Bibliography       %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%\usepackage{natbib} \setcitestyle{comma,authoryear}
\usepackage{natbib}
	\setcitestyle{numbers,sort&compress}
	\setcitestyle{sort&compress}
	\usepackage{hypernat} % hypernat also required to allow citations to compress. 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Aesthetics         %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\usepackage{microtype} % Creates better spaced text
\RequirePackage{times} % Font
\usepackage{siunitx} % SI units 
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{nameref}% this allows a reference be named, to print unnumbered references by their section name (used here for linking to Supplemental text in this case).
\usepackage{xcolor} % Setting colours and their usage
	\definecolor{natureblue}{RGB}{5,110,210}
    \usepackage[colorlinks]{hyperref} % Colour for hyperlinks, (URLs, citations, cross reference)
	\AtBeginDocument{%this allows colours to chage from the defined elsearticle template.
	\hypersetup{
	colorlinks=true,
    linkcolor={natureblue},
	citecolor={natureblue},
    filecolor=blue!50!black,
    urlcolor=cyan,
	}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Supplemental labels%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%Define command to start a supplemental section
%set the supplemental letter used for figures (e.g. Figure E1)
\newcommand{\beginsupplement}{%
        \setcounter{table}{0}
        \renewcommand{\thetable}{E\arabic{table}}%
        \setcounter{figure}{0}
        \renewcommand{\thefigure}{E\arabic{figure}}%
     }

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Building tables    %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\usepackage{booktabs} % required for tables
\usepackage{rotating,tabularx} % tabularx is the table style used, rotating can also be used
\newcolumntype{Z}{ >{\centering\arraybackslash}X } % defining table content layout per box
\usepackage{ltablex} % allow page break between lines in tabularx
\usepackage{caption} \captionsetup{font=normalsize} % to set the caption size as normal even when table is tiny.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Begin content      %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}
%\linenumbers
\small

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Title and author   %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frontmatter}

\title{\mbox{Predicting the occurrence of variants in \textit{RAG1} and \textit{RAG2}}}

\author[add1,add2]{Dylan Lawless\corref{cor1}}
	\ead{Dylan.Lawless@epfl.ch}
\author[add3,add4]{Hana Lango Allen}
\author[add5]{James Thaventhiran}
\author[add3]{NIHR BioResourceâ€“Rare Diseases Consortium}
\author[add2]{Flavia Hodel}
\author[add1]{Rashida Anwar}
\author[add2,add6]{Jacques Fellay}
\author[add7,add8]{Jolan E. Walter}
\author[add9,add10]{Sinisa Savic\corref{cor1}}
	\ead{S.Savic@leeds.ac.uk}

\address[add1]{Leeds Institute of Biomedical and Clinical Sciences, University of Leeds, Wellcome Trust Brenner Building, St James's University Hospital, Beckett Street, Leeds, UK.}
\address[add2]{Global Health Institute, School of Life Sciences, \'Ecole Polytechnique F\'ed\'erale de Lausanne, 1015, Lausanne, Switzerland.}
\address[add3]{NIHR BioResource, Cambridge University Hospitals, Cambridge Biomedical Campus, Cambridge, CB20QQ, UK.}
\address[add4]{Department of Haematology, University of Cambridge, Cambridge Biomedical Campus, Cambridge, CB20XY, UK.}
\address[add5]{MRC Toxicology Unit, School of Biological Sciences, University of Cambridge, UK.}
\address[add6]{Swiss Institute of Bioinformatics, Lausanne, 1015, Switzerland.}
\address[add7]{University of South Florida and Johns Hopkins All Children's Hospital, Saint Petersburg, Florida, USA.}
\address[add8]{Division of Allergy Immunology, Massachusetts General Hospital for Children, Boston, Massachusetts, USA.}
\address[add9]{Department of Clinical Immunology and Allergy, St James's University Hospital, Beckett Street, Leeds, UK.}
\address[add10]{National Institute for Health Research Leeds Musculoskeletal Biomedical Research Centre and Leeds Institute of Rheumatic and Musculoskeletal Medicine, Wellcome Trust Brenner Building, St James's University Hospital, Beckett Street, Leeds, UK.}
\cortext[cor1]{Addresses for correspondence}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Abstract           %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstract}
\setstretch{1.2}
While widespread genome sequencing ushers in a new era of preventive medicine, the tools for predictive genomics are still lacking. 
Time and resource limitations mean that human diseases remain uncharacterised because of an inability to predict clinically relevant genetic variants.
A strategy of targeting highly conserved protein regions is used commonly in functional studies. 
However, this benefit is lost for rare diseases where the attributable genes are mostly conserved.
An immunological disorder exemplifying this challenge occurs through damaging mutations in \textit{RAG1} and \textit{RAG2} which presents at an early age with a distinct phenotype of life-threatening immunodeficiency or autoimmunity.
Many tools exist for variant pathogenicity prediction but these cannot account for the probability of variant occurrence.
Here, we present a method that predicts the likelihood of mutation for every amino acid residue in the RAG1 and RAG2 proteins.
Population genetics data from approximately 146,000 individuals was used for rare variant analysis.
Forty-four known pathogenic variants reported in patients and recombination activity measurements from 110 RAG1/2 mutants were used to validate calculated scores.
Probabilities were compared with 98 currently known human cases of disease.
A genome sequence dataset of 558 patients who have primary immunodeficiency but that are negative for RAG deficiency were also used as validation controls.
We compared the difference between mutation likelihood and pathogenicity prediction.
Our method builds a map of most probable mutations allowing pre-emptive functional analysis. 
This method may be applied to other diseases with hopes of improving preparedness for clinical diagnosis.
\end{abstract}
\end{frontmatter}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Formalities        %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Funding}
\noindent This work is funded by the University of Leeds 110 Anniversary Research Scholarship and by the National Institute for Health Research (NIHR, grant number RG65966).
This work was also supported by the National Institutes of Health (sub-R01AI100887-05 to J.E.W.) and Robert A. Good Endowment, University of South Florida (to J.E.W.).

\section*{Acknowledgements}
\noindent We gratefully acknowledge the participation of all NIHR BioResource volunteers, and thank the NIHR BioResource centres and staff for their contribution.
We thank the National Institute for Health Research and NHS Blood and Transplant.
The views expressed are those of the author(s) and not necessarily those of the NHS, the NIHR or the Department of Health.

\section*{Ethics statement}
\noindent The study was performed in accordance with the Declaration of Helsinki.
The NIHR BioResource projects were approved by Research Ethics Committees in the UK and appropriate national ethics authorities in non-UK enrolment centres.

\section*{Key words}
\noindent Recombination activating genes 1 and 2 (RAG1, RAG2), pathogenic variant, genomics, predictive.

\section*{Abbreviations}
\noindent 
BCR (B-cell receptor),
CADD (combined annotation dependent depletion),
CID-G/A (combined immunodeficiency with granuloma and/or autoimmunity),
GWAS (genome-wide association studies),
HGMD (human gene mutation database),
${M}_{r}$ (mutation rate),
MRF (mutation rate residue frequency),
PID (primary immunodeficiency),
pLI (probability of being loss-of-function intolerant),
\textit{RAG1} (recombination activating gene 1),
${R}_{f}$ (residue frequency),
%${rf}$-${igf}$ (residue frequency, inverse gene frequency),
RNH (RNase H),
RSS (recombination signal sequence),
SCID (severe combined immunodeficiency),
TCR (T-cell receptor).
%${tf}$-${idf}$ (term frequency, inverse document frequency).
\pagebreak

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Introduction       %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Introduction}
\noindent Costs associated with genomic investigations continue to reduce \citep{payne2018cost} while the richness of data generated increases. 
Globally, the adoption of wide scale genome sequencing implies that all new-born infants may receive screening for pathogenic genetic variants in an asymptomatic stage, pre-emptively \citep{kwan2014newborn}. 
The one dimensionality of individual genomes is now being expanded by the possibility of massive parallel sequencing for somatic variant analysis and by single-cell or lineage-specific genotyping; culminating in a genotype spectrum. 
In whole blood, virtually every nucleotide position may be mutated across $10^5$ cells \citep{Liggett208066}. 
Mapping one's genotype across multiple cell types and at several periods during a person's life may soon be feasible \citep{clark2018scnmt}. 
Such genotype snapshots might allow for prediction and tracking of somatic, epigenetic, and transcriptomic profiling. 

The predictive value of genomic screening highly depends on the computation tools used for data analysis and its correlation with functional assays or prior clinical experience.
Interpretation of that data is especially challenging for rare human genetic disorders; candidate disease-causing variants that are predicted as pathogenic often require complex functional investigations to confirm their significance.
There is a need for predictive genomic modelling with aims to provide a reliable guidance for therapeutic intervention for patients harbouring genetic defects for life-threatening disease before the illness becomes clinically significant. 


The study of predictive genomics is exemplified by consideration of gene essentiality, accomplished by observing intolerance to loss-of-function variants. 
Several gene essentiality scoring methods are available for both the coding and non-coding genome  \citep{bartha2017human}. 
Approximately 3,000 human genes cannot tolerate the loss of one allele \citep{bartha2017human}. 
The greatest hurdle in monogenic disease is the interpretation of variants of unknown significance while functional validation is a major time and cost investment for laboratories investigating rare disease. 

Severe, life-threatening immune diseases are caused by genetic variations in almost 300 genes  \citep{picard2018international, conley2014discovery} however, only a small percentage of disease causing variants have been characterised using functional studies. 
Several robust tools are in common usage for predicting variant pathogenicity. 
Compared to methods for pathogenicity prediction, a void remains for predicting mutation probability, essential for efficient pre-emptive validation.
Our investigation aims to apply predictive genomics as a tool to identify genetic variants that are most likely to be seen in patient cohorts.

We present the first application of our novel approach of predictive genomics using Recombination activating gene 1 (RAG1) and RAG2 deficiency as a model for a rare primary immunodeficiency (PID) caused by autosomal recessive variants. 
\textit{RAG1} and \textit{RAG2} encode lymphoid-specific proteins that are essential for V(D)J recombination. 
This genetic recombination mechanism is essential for a robust immune response by diversification the T and B cell repertoire in the thymus and bone marrow, respectively \citep{schatz1989v,oettinger1990rag}. 
Deficiency of RAG1 
\citep{mombaerts1992rag} 
and RAG2 \citep{shinkai1992rag} 
in mice causes inhibition of B and T cell development.
\citet{schwarz1996rag} formed the first publication reporting that RAG mutations in humans causes severe combined immunodeficiency (SCID), and deficiency in peripheral B and T cells.
Patient studies identified a form of immune dysregulation known as Omenn syndrome
\citep{de1991restricted, rieux1998highly}. 
The patient phenotype includes multi-organ infiltration with oligoclonal, activated T cells.
The first reported cases of Omenn syndrome identified infants with hypomophic RAG variants which retained partial recombination activity
\citep{villa1998partial}.
RAG deficiency can be measured by in vitro quantification of recombination activity  
\citep{lawless2018prevalence, lee2014systematic, tirosh2018recombination}.
Hypomorphic \textit{RAG1} and \textit{RAG2} mutations, responsible for residual V(D)J recombination activity (on average 5-30\%), result in a distinct phenotype of combined immunodeficiency with granuloma and/or autoimmunity (CID-G/A) \citep{kwan2014newborn,walter2015broad,schuetz2008immunodeficiency}. 

Human RAG deficiency has traditionally been identified at very early ages due to the rapid drop of maternally-acquired antibody in the first six months of life. 
A loss of adequate lymphocyte development quickly results in compromised immune responses. 
More recently, we have found that RAG deficiency is also found for some adults living with PID
\citep{lawless2018prevalence}.

\textit{RAG1} and \textit{RAG2} are highly conserved genes but disease is only reported with autosomal recessive inheritance. 
Only 44\% of amino acids in RAG1 and RAG2 are reported as mutated on GnomAD and functional validation of candidate variants is difficult
\citep{lek2016analysis}. 
Pre-emptive selection of residues for functional validation is a major challenge; a selection based on low allele frequency alone is infeasible since the majority of each gene is highly conserved.
A shortened time between genetic analysis and diagnosis means that treatments may be delivered earlier. 
RAG deficiency may present with diverse phenotypes and treatment strategies vary.
With such tools, early intervention may be prompted. Some patients could benefit from hematopoietic stem cell transplant \citep{john2016unrelated} when necessary while others may be provided mechanism-based treatment \citep{casanova2014guidelines}. 
Here, we provide a new method for predictive scoring that was validated against groups of functional assay values, human disease cases, and population genetics data.
We present the list of variants most likely seen as future determinants of RAG deficiency, meriting functional investigation.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Methods            %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Methods}
\subsection*{\textbf{Population genetics and data sources}}
\noindent GnomAD (version r2.0.2) \citep{lek2016analysis} was queried for the canonical transcripts of \textit{RAG1} and \textit{RAG2} from population genetics data of approximately 146,000 individuals; ENST00000299440 (\textit{RAG1}) 1586 variants, GRCh37 11:36532259-36614706 and ENST00000311485 (\textit{RAG2}) 831 variants, GRCh37 11:36597124 - 36619829. 
Data was filtered to contain the variant effect identifiers: frameshift, inframe deletion, inframe insertion, missense, stop lost, or stop gained. 
Reference transcripts were sourced from Ensembl in the FASTA format amino acid sequence for transcript RAG1-201 ENST00000299440.5 [HGNC:9831] and transcript RAG2-201 ENST00000311485.7 [HGNC:9832]. 
These sequences were converted to their three-letter code format using \textit{One to Three} from the Sequence Manipulation Suite (SMS2)
\citep{stothard2000sequence}.
Combined Annotation Dependent Depletion (CADD) scores were sourced from 
\hyperref[https://cadd.gs.washington.edu/download]{https://cadd.gs.washington.edu/download} (Nov 2018)
and are reported by \citet{kircher2014general}.
The dataset used was "All possible SNVs" from whole genome data, from which we extracted the data for coding regions of \textit{RAG1} and \textit{RAG2}.
We used the Human Gene Mutation Database (HGMD) from the Institute of Medical Genetics in Cardiff as a pre-defined source of known RAG deficiency cases 
\hyperref[http://www.hgmd.cf.ac.uk/ac/index.php]{http://www.hgmd.cf.ac.uk/ac/index.php} (Feb 2019, free access version to NM\_000448.2.)
\citep{stenson2014human}.
Data was formatted into CSV and imported into R for combined analysis with PHRED-scaled CADD scores and the main dataframe. 
The crystal structure render of DNA bound RAG complex was produced with data from RCSB Protein Data Bank (3jbw.pdb)
\citep{ru2015molecular}.
Structures were visualised using the software VMD from the Theoretical and Computational Biophysics Group
\citep{Humphrey1996vmd},
imaged with Tachyon rendering 
\citep{Stone1998Ane}, and colour mapped using our scoring method. 

\subsection*{\textbf{Data processing}}
\noindent The population genetics input dataset used GnomAD variant allele frequencies and reference sequences processed as CSV files, cleaned and sorted to contain only amino acid codes, residue numbers, alternate residues, alternate allele frequencies, and a score of 0 or 1 to indicate presence or absence of variants where 1 represented none reported. 
An annotation column was also provided to label where multiple alternate variants existed. Statistics and calculation steps are listed in order in 
\textbf{Supplemental Tables \ref{table:E3}-\ref{table:E8}}.

The percentage of conserved residues was calculated (55.99\% of amino acids contained no reported variants in RAG1, 55.98\% in RAG2 
(\textbf{Table \ref{table:E4}})) . 
Basic protein statistics were generated using canonical reference transcript sequences of \textit{RAG1} and \textit{RAG2} with the SMS2 tool \textit{Protein Stats}
\citep{stothard2000sequence}. 
The resulting pattern percentage value was converted to a frequency (decimal 0-1) based on the number of residues per protein to generate the residue frequency (${R}_{f}$).
The ${R}_{f}$ values were found for both proteins as shown in 
\textbf{Table \ref{table:E5}} and summarised in \textbf{Table \ref{table:E6}}. 

The count of variants per residue were found for both proteins and the  mutation rates (${M}_{r}$) per residue were calculated as shown in \textbf{Table \ref{table:E7}}. 
${M}_{r}$ was found by counting the number of mutations per residue in a window, sized to contain each protein individually.
For genome-wide application the window size may be increased or decreased. 
In this case the window consisted of only the coding regions.
The ${M}_{r}$ values were then converted to the frequencies based on the number of residues per protein.
Separate, and overlapping windows could also be used based on genome phase data and regions of linkage disequilibrium to account for non-random association of alleles at different loci; this might be particularly important for disorders with multiple genetic determinants. 

The ${M}_{r}$ and ${R}_{f}$  multiply to give the raw mutation rate residue frequency (MRF) value 
(\textbf{Table \ref{table:E8}}).
This value is also shown in 
\textbf{Tables \ref{table:mrf.assay} and \ref{Table:1}}. 
Our investigation used a Boolean score $C$ to account for the presence or absence of a mutation in the general population; 0 for any variant existing in the population and 1 for conserved residues. 
$C \times {M}_{r} \times  {R}_{f}$, in our case, produced the MRF score for conserved residues.
\textbf{Figure \ref{fig:Figure_main} (ii)} illustrates the raw MRF as a histogram and the MRF, after applying $C$, as a heatmap.

An important consideration for future application is whether to use this Boolean score or instead use a discrete variable which accounts for the true allele frequency in the general population. 
In the clinical setting, the likelihood of \textit{de novo} mutations and inherited mutations have different impacts when considering recessive and dominant diseases. 
A patient is more likely to inherit a variant that exists even at a very low frequency than to acquire a random \textit{de novo} mutation. 
Therefore, a value representing an allele frequency may be used to replace $C$ in many investigations, particularity when considering variants that exist at low rates. 
PRHED-scaled CADD score data consisted of nucleotide level values. 
For comparison with MRF, the median CADD scores were averaged per codon as demonstrated in 
\nameref{sec:Supplemental_text} text.
A summary of data processing and analysis is illustrated in \textbf{Figure 
\ref{fig:method_map}}.

\subsection*{\textbf{Raw data availability and analysis script}}
\noindent The supplemental files \textit{``Raw\_data\_R\_analysis\_for\_figures''} contains all raw data and analysis methods used to produce figures (except illustrations in Figures \ref{fig:Figure_main} and \ref{fig:RAG_CADD}).
\textit{``data\_analysis.R''} is an R script that contains the methods used to produce figures. 
Each of the input data CSV files are explained on first usage within the analysis script.
Running \textit{``data\_analysis.R''} from within the same directory as the associated input data CSV files will replicate analysis.

\subsection*{\textbf{Data visualisation}}
\noindent For our visualisation of MRF scores, small clusters of high MRF values were of more appealing than individual highly conserved residues. 
Therefore, we applied a 1\% average filter where values were averaged over a sliding window of N number of residues (10 in the case of RAG1, 6 in the case of RAG2). 
For a clear distinction of MRF clusters, a cut-off threshold was applied at the 75$^{th}$ percentile (e.g. 0.0168 in RAG1) as shown in heatmaps in \textbf{Figure \ref{fig:Figure_main} (iii)} and \textbf{\ref{fig:RAG_CADD}}. 
The gene heatmaps for coding regions in \textit{RAG1} and \textit{RAG2} (\textbf{Fig. \ref{fig:Figure_main}}) were populated with (i) Boolean $C$ score from population genetics data, (ii) raw MRF scores, and (iii) MRF clusters with 1\% average and cut-off threshold. 
GraphPad Prism was used for heatmaps. 
The data used for heatmaps is available in \textbf{Table \ref{Table:1}} and in the supplemental R source to allow for alternative visualisations.
An example of alternative output for non-R users is shown in \textbf{Figure
\ref{fig:RAG_MRF_map}}. 
Adobe Illustrator and Photoshop were used for protein domain illustrations in \textbf{Figure \ref{fig:Figure_main} (iv)}.

\subsection*{\textbf{Validation of MRF against functional data}}
\noindent The recombination activity of RAG1 and RAG2 was previously measured on known or candidate pathogenic variants \citep{lee2014systematic, lawless2018prevalence, tirosh2018recombination}. 
Briefly, the pathogenicity of variants in \textit{RAG1} and \textit{RAG2} was measured functionally \textit{in vitro} by either expression of RAG1 and RAG2 in combination with a recombination substrate plasmid containing recombination signal sequence (RSS) sites which are targeted by RAG complex during normal V(D)J recombination, or  Abelson virus-transformed Rag2-/- pro-B cells with an RSS-flanked inverted GFP cassette.  
Recombination events were assessed by quantitative real-time PCR using comparative CT or expression of GFP evaluated by flow cytometry, respectively.
The inverse score of recombination activity (0-100\%) was used to quantify pathogenicity of variants in our study. 
Comparison between known pathogenicity scores and MRF was done by scaling MRF scores from 0-100\% (100\% being highest probability of occurring as damaging). 
A data and analysis is summarised in \textbf{Figure 
\ref{fig:method_map}}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Results            %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Results}
\subsection*{\textbf{RAG1 and RAG2 conservation and mutation rate residue frequency}}
\noindent Variant probability prediction is dependent on population genetics data.
Our study queried GnomAD 
\citep{lek2016analysis}
to identify conserved residues using a Boolean score $C$ of 0 (present in population) or 1 (conserved). 
The gene-specific mutation rate ${M}_{r}$ of each residue was calculated from allele frequencies. 
The gene-specific residue frequency ${R}_{f}$ represented the frequency of a residue occurring per gene, acquired by converting gene residue percentage (from the SMS2 tool \textit{Protein stats}) to a frequency (decimal 0-1)
\citep{stothard2000sequence}.
Together the values were used to calculate the most probable disease-causing variants which have not yet been identified in patients. 
We termed the resulting score a mutation rate residue frequency, where $MRF = {C} \times {M}_{r} \times {R}_f$. 
This score represents the likelihood that a clinically relevant mutation will occur. 

\textbf{Figure \ref{fig:Figure_main}} 
presents the most probable unidentified disease-causing variants in \textit{RAG1/2}. 
Variants with a low MRF may still be damaging but resources for functional validation are best spent on gene regions with high MRF. 
Clusters of conserved residues are shown in 
\textbf{Figure \ref{fig:Figure_main} (i)} and are generally considered important for protein structure or function.
However, these clusters do not predict the likelihood of mutation. 
Raw MRF scores are presented in 
\textbf{Figure \ref{fig:Figure_main} (ii)}. 
Histograms illustrates the MRF without Boolean scoring applied and 
\textbf{Figure \ref{fig:Figure_main} (iii)} 
provides a clearer illustration of top MRF score clusters. 
For visualisation, a noise reduction method was applied; a sliding window was used to find the average MRF per 1\% interval of each gene.
The resulting scores displayed in
\textbf{Figure \ref{fig:Figure_main} (iii)}
contain a cut-off threshold to highlight the top scoring residues (using the 75$^{th}$ percentile).
Variant sites most likely to present in disease cases are identified by high MRF scoring.
This model may be expanded by the addition of phenotypic or epigenetic data
(\textbf{Supplemental; \nameref{sec:Supplemental_Bayesian}}).

\begin{figure}[h]
\hspace*{0cm}   % This line is adjusting position on page left or right. 
	\includegraphics[scale=0.74]{Figure_main}
	\caption{RAG1 (red, left) and RAG2 (blue, right) conservation and mutation rate residue frequency. 
(i) Gene conservation score; non-conserved 0 and conserved 1. Colour indicates no known mutations in humans. 
(ii) Histogram; raw MRF score. 
Heatmap; MRF prediction for conserved residues, graded 0 to 0.05 (scale of increasing mutation likelihood with human disease). 
(iii) Coloured bars indicate most likely clinically relevant variant clusters.
MRF score averaged with 1\% intervals for each gene and cut-off below 75th percentile, graded 0 to 0.03 (noise reduction method). 
(iv) Gene structure with functional domains. Full list of residues and scores available in
\textbf{Table \ref{Table:1}}.}
	\label{fig:Figure_main}
\end{figure}

\textbf{Table
\ref{Table:1}}
provides all MRF scores for both proteins. Raw data used for calculations and the list of validated residues of RAG1 and RAG2 are available in 
\textbf{Tables \ref{table:E3}-\ref{table:E8}}.
\textbf{\nameref{table:mrf.assay}}
shows the MRF mutation likelihood score for mutations that have also been reported as tested for recombination activity in functional assays.
The likelihood of mutation does not correlate with pathogenicity; \textbf{figures}
\textbf{\ref{fig:MRF_categories_and_assayed}} and \textbf{\ref{fig:Accuracy_MRF_vs_functional}}
show that most mutations tested had severe loss of protein function, while the likelihood each mutation occurring in humans varied significantly. 
Analysis-ready files are also available in Supplemental data along with the associated R source file to allow for alternative visualisations as shown in \textbf{Figure
\ref{fig:RAG_MRF_map}}. 

\begin{tiny}
%\setlength\tabcolsep{0.6pt}
\begin{tabularx}{\textwidth}{*{8}{Z}}
\multicolumn{8}{c}{\textbf{RAG1}} \\
\midrule
{MRF} & {Residue} & {Assayed} & {Recombition activity (\%)} & {MRF} & {Residue} & {Assayed} & {Recombition activity (\%)} \\
\midrule
{0.030} & {56} & {I56T} & {3.5 $\pm$ 0.2} & {0.022} & {539} & {D539V} & {3.2 $\pm$ 0.2} \\
{0.030} & {86} & {K86VfsX33} & {2.7 $\pm$ 0.3} & {0.025} & {541} & {L541CfsX30} & {1.2 $\pm$ 0.9} \\
{0.014} & {99} & {G99S} & {113.2 $\pm$ 3.7} & {0.043} & {559} & {R559S} & {1.0 $\pm$ 0.4} \\
{0.012} & {106} & {N106K} & {80.4 $\pm$ 16.4} & {0.043} & {561} & {R561H} & {2.0 $\pm$ 0.6} \\
{0.043} & {108} & {R108X} & {1.8 $\pm$ 0.3} & {0.041} & {601} & {S601P} & {0.0 $\pm$ 0.0} \\
{0.043} & {142} & {R142X} & {9.0 $\pm$ 4.0} & {0.026} & {612} & {H612R} & {121.6 $\pm$ 0.9} \\
{0.032} & {174} & {E174SfsX27} & {0.5 $\pm$ 0.2} & {0.043} & {624} & {R624H} & {0.0 $\pm$ 0.4} \\
{0.027} & {246} & {A246TfsX17} & {0.8 $\pm$ 0.1} & {0.041} & {626} & {S626X} & {0.0 $\pm$ 0.0} \\
{0.012} & {248} & {Q248X} & {1.2 $\pm$ 0.2} & {0.041} & {651} & {S651P} & {0.5 $\pm$ 0.5} \\
{0.026} & {249} & {H249R} & {112.2 $\pm$ 3.5} & {0.043} & {699} & {R699Q,W} & {45.9 $\pm$ 1.5, 19.3 $\pm$ 1.8} \\
{0.043} & {314} & {R314W} & {24.3 $\pm$ 5.2} & {0.032} & {722} & {E722K} & {0.0 $\pm$ 0.2} \\
{0.012} & {328} & {C328Y} & {16.0 $\pm$ 2.9} & {0.012} & {730} & {C730F} & {0.0 $\pm$ 0.0} \\
{0.030} & {383} & {K383RfsX7} & {0.1 $\pm$ 0.0} & {0.025} & {732} & {L732P} & {0.0 $\pm$ 0.0} \\
{0.013} & {386} & {F386CfsX4} & {0.2 $\pm$ 0.1} & {0.043} & {737} & {R737H} & {0.2 $\pm$ 0.0} \\
{0.030} & {391} & {K391E} & {6.5 $\pm$ 1.6} & {0.043} & {759} & {R759C} & {17.2 $\pm$ 3.3} \\
{0.043} & {394} & {R394Q} & {0.1 $\pm$ 0-0.1} & {0.043} & {764} & {R764P} & {0.0 $\pm$ 0.0} \\
{0.043} & {396} & {R396C} & {0.4-0.6 $\pm$ 0-0.1} & {0.008} & {768} & {Y768X} & {0.0 $\pm$ 0.0} \\
{0.041} & {401} & {S401P} & {0.0 $\pm$ 0.0} & {0.032} & {770} & {E770K} & {21.0 $\pm$ 0.4} \\
{0.020} & {403} & {T403P} & {0.0 $\pm$ 0.0} & {0.043} & {778} & {R778Q,W} & {8.6 $\pm$ 1.0, 4.6 $\pm$ 0.6} \\
{0.043} & {404} & {R404Q} & {1.2 $\pm$ 0.1} & {0.028} & {786} & {P786L} & {0.0 $\pm$ 0.1} \\
{0.043} & {410} & {R410Q} & {0.0 $\pm$ 0.0} & {0.030} & {820} & {K820R} & {117.9 $\pm$ 6.3} \\
{0.025} & {411} & {L411P} & {0.0 $\pm$ 0.0} & {0.025} & {836} & {L836V} & {75.0 $\pm$ 1.3} \\
{0.022} & {429} & {D429G} & {0.1 $\pm$ 0.0} & {0.043} & {841} & {R841Q,W} & {0.0 $\pm$ 0.0, 10.0 $\pm$ 0.5} \\
{0.028} & {433} & {V433M} & {0.2 $\pm$ 0.0} & {0.027} & {868} & {A868V} & {100.0 $\pm$ 5.0} \\
{0.019} & {435} & {M435V} & {23.6 $\pm$ 4.8} & {0.005} & {896} & {W896R} & {0.9 $\pm$ 0.1} \\
{0.027} & {444} & {A444V} & {1.4 $\pm$ 0.2} & {0.008} & {912} & {Y912C} & {6.9 $\pm$ 0.4} \\
{0.043} & {449} & {R449K} & {92.1 $\pm$ 3.6} & {0.005} & {959} & {W959X} & {0.0 $\pm$ 0.0} \\
{0.025} & {454} & {L454Q} & {5.4 $\pm$ 0.7} & {0.032} & {965} & {E965X} & {0.0 $\pm$ 0.0} \\
{0.019} & {458} & {M458SfsX34} & {0.0 $\pm$ 0.0} & {0.043} & {973} & {R973C} & {0.0 $\pm$ 0.2} \\
{0.027} & {472} & {A472V} & {0.4 $\pm$ 0.0} & {0.013} & {974} & {F974L} & {56.5 $\pm$ 0.8} \\
{0.043} & {474} & {R474C} & {125.4 $\pm$ 2.6} & {0.043} & {975} & {R975W,Q} & {57.9 $\pm$ 1.6, 53.5 $\pm$ 3.6} \\
{0.028} & {475} & {V475AfsX17} & {0.1 $\pm$ 0.0} & {0.012} & {981} & {Q981P} & {7.2 $\pm$ 0.1} \\
{0.025} & {506} & {L506F} & {1.0 $\pm$ 0.1} & {0.030} & {983} & {K983NfsX9} & {0.1 $\pm$ 0.0} \\
{0.043} & {507} & {R507W} & {15.9 $\pm$ 0.8} & {0.030} & {992} & {K992E} & {9.1 $\pm$ 1.2} \\
{0.014} & {516} & {G516A} & {40.2 $\pm$ 1.3} &	{0.030} & {1006} & {M1006V} & {105.6 $\pm$ 6.8} \\
 {0.005} & {522} & {W522C} & {41.6 $\pm$ 1.9}	& {} {} {} {} \\						
\midrule
\multicolumn{8}{c}{\textbf{RAG2}} \\
\midrule
{MRF} & {Residue} & {Assayed} & {Recombition activity (\%)} & {MRF} & {Residue} & {Assayed} & {Recombition activity (\%)} \\
\midrule
{0.013} & {1} & {M1T} & {65.3 $\pm$ 2.2} & {0.013} & {285} & {M285R} & {24.7 $\pm$ 0.8} \\
{0.006} & {16} & {Q16X} & {1.7 $\pm$ 0.4} & {0.004} & {307} & {W307X} & {0.2 $\pm$ 0.2} \\
{0.038} & {35} & {G35A,V} & {22.1 $\pm$ 3.1, 0.4 $\pm$ 0.3} & {0.017} & {386} & {F386L} & {109.1 $\pm$ 5} \\
{0.023} & {39} & {R39G} & {0.2 $\pm$ 0.1} & {0.025} & {407} & {E407X} & {2.9 $\pm$ 0.4} \\
{0.011} & {41} & {C41W} & {0.2 $\pm$ 0.4} & {0.004} & {416} & {W416L} & {1.4 $\pm$ 0.2} \\
{0.017} & {62} & {F62L} & {19.6 $\pm$ 3} & {0.025} & {437} & {E437K} & {0.9 $\pm$ 0.2} \\
{0.028} & {65} & {D65Y} & {6.8 $\pm$ 1.2} & {0.017} & {440} & {K440N} & {26.7 $\pm$ 2.4} \\
{0.023} & {73} & {R73H} & {12.4 $\pm$ 1.4} & {0.013} & {443} & {M443I} & {0.4 $\pm$ 0.2} \\
{0.034} & {77} & {T77N} & {42.6 $\pm$ 2.7} & {0.027} & {444} & {I444M} & {2.7 $\pm$ 0.3} \\
{0.038} & {95} & {G95R} & {0.3 $\pm$ 0.2} & {0.011} & {446} & {C446W} & {2.9 $\pm$ 0.1} \\
{0.013} & {110} & {M110L} & {74.6 $\pm$ 1.8} & {0.038} & {451} & {G451A} & {66.3 $\pm$ 4.8} \\
{0.017} & {127} & {K127X} & {0.1 $\pm$ 0} & {0.004} & {453} & {W453R} & {0.6 $\pm$ 0.1} \\
{0.038} & {157} & {G157V} & {0.4 $\pm$ 0.2} & {0.017} & {456} & {A456T} & {16 $\pm$ 2.9} \\
{0.030} & {160} & {S160L} & {5.8 $\pm$ 0.6} & {0.013} & {459} & {M459L} & {30.8 $\pm$ 0.6} \\
{0.023} & {180} & {P180H} & {31.1 $\pm$ 0.5} & {0.034} & {474} & {N474S} & {97.5 $\pm$ 5.9} \\
{0.019} & {195} & {Y195D} & {2 $\pm$ 0.3} & {0.011} & {478} & {C478Y} & {0.2 $\pm$ 0.1} \\
{0.034} & {215} & {T215I} & {67.2 $\pm$ 1} & {0.025} & {480} & {E480X} & {2.8 $\pm$ 0.6} \\
{0.023} & {229} & {R229Q,W} & {8.9 $\pm$ 1, 10.5 $\pm$ 0.5} & {0.017} & {481} & {H481P} & {23.8 $\pm$ 3.9} \\
{0.023} & {253} & {P253R} & {95.4 $\pm$ 2.3} & {0.013} & {502} & {M502V} & {99.6 $\pm$ 3.4} \\
{0.006} & {278} & {Q278X} & {0.1 $\pm$ 0.1} & {} {} {} {}\\
\bottomrule
\caption[Summary table 1]{MRF likelihood scores for variants functionally assayed to date 
\citep{lee2014systematic, tirosh2018recombination, lawless2018prevalence}.
Increased MRF score indicate higher likelihood of occurrence.
Recombination activity measured by functional assay is shown as a percentage of wildtype (\% SEM). Residues with multiple mutations are shown with both alternative variants and values.
$MRF_{max} = 0.043$ and $MRF_{min} = 0.004$.
The full table of all protein positions can be found in Supplemental Table
\ref{Table:1}.}
\label{table:mrf.assay}
\end{tabularx}
\end{tiny}

\subsection*{\textbf{MRF scores select for confirmed variants in human disease}}
\noindent We have applied MRF scores to known damaging mutations from other extensive reports in cases of human disease
\citep{
schuetz2008immunodeficiency, lee2014systematic, 
villa2001v, abolhassani2014hypomorphic, 
kutukculer2012novel, sobacchi2006rag, 
villa1998partial, noordzij2002immunophenotypic, 
crestani2014rag1, dalal2005evolution, 
kuijpers2011idiopathic, gruber2009clinical, 
de2010hypomorphic, buchbinder2015identification, 
felgentreff2011clinical, schwarz1996rag, reiff2013exome, 
corneo2001identical, asai2011analysis, kato2015rag1, 
yu2014human, de2005novel, zhang2005novel, 
henderson2013expanding, avila2010highly, 
riccetto2014compound, walter2015broad, 
gomez2000mutations, chou2012novel}
 [\emph{originally compiled by} \citet{notarangelo2016human}]. 
This dataset compares a total of 44 variants. 
We expected that functionally damaging variants (resulting in low recombination activity in vitro) that have the highest probability of occurrence would be identified with high MRF scores. 
MRF prediction correctly identified clinically relevant mutations in \textit{RAG1} and \textit{RAG2} 
(\textbf{Fig. \ref{fig:Human_cases} (i)}). 
Variants reported on GnomAD which are clinically found to cause disease had significantly higher MRF scores than variants which have not been reported to cause disease.
We observed that rare and likely mutations provided high scores while rare but unlikely or common variants had low scores
(\textbf{Fig. \ref{fig:Human_cases} (i)}). 

\begin{figure}[h]
\hspace*{0cm}   % This line is adjusting position on page left or right. 
	\includegraphics[scale=0.7]{Human_cases}
	\caption{RAG1 and RAG2 MRF score predict the likelihood of mutations that are clinically relevant. 
(i) Known damaging variants (clinically diagnosed with genetic confirmation) reported on GnomAD have significantly higher MRF scores than unreported variants. 
(ii) GnomAD rare variant allele frequency <0.0001. 
No significant difference in allele frequency is found between known damaging and non-clinically reported variants. Unpaired t-test. 
RAG1 P-value 0.002** RAG2 P-value 0.0339*. MRF; mutation rate residue frequency, ns; non-significant.}
	\label{fig:Human_cases}
\end{figure}

Allele frequency is generally the single most important filtering method for rare disease in whole genome (and exome) sequencing experiments. 
Variants under pressure from purifying selection are more likely to cause disease than common variants.
However, most RAG mutations are rare.
Therefore, allele frequencies of rare variants reported on GnomAD cannot differentially predict the likelihood of causing disease
(\textbf{Fig. \ref{fig:Human_cases} (ii)}). 
As such, we found no significant difference between known damaging variants and those that have not yet been reported as disease-causing. 
The comparison between \textbf{Figure \ref{fig:Human_cases} (i) and (ii)} illustrates the reasoning for the design of our method. 

Many non-clinically-reported rare variants may cause disease; the MRF score identifies the top clinically relevant candidates. 
Based on the frequency of protein-truncating variants in the general population, \textit{RAG1} and \textit{RAG2} are considered to be tolerant to the loss of one allele, as indicated by their low probability of being loss-of-function intolerant (pLI) scores of 0.00 and 0.01, respectively 
\citep{lek2016analysis}. 
This is particularly important for recessive diseases such as RAG deficiency where most new missense variants will be of unknown significance until functionally validated.

\subsection*{\textbf{Top candidate variants require validation}}
\noindent Functionally characterising protein activity is both costly and time consuming.
RAG1 and RAG2 have now been investigated by multiple functional assays for at least 110 coding variants 
\citep{lee2014systematic, tirosh2018recombination, lawless2018prevalence}.
In each case, researchers selected variants in \textit{RAG1} and \textit{RAG2} that were potentially damaging or were identified from PID patients as the most probable genetic determinant of disease. 
Functional assays for RAG deficiency in those cases, and generally, measured a loss of recombination activity as a percentage of wild type function (0-100\%). 

Pre-emptively performing functional variant studies benefits those who will be identified with the same variants in the future, before the onset of disease complications.
While more than 100 variants have been assayed in vitro, we calculated that only one-quarter of them are most probable candidates for clinical presentation. 
\textbf{Figure 
\ref{fig:MRF_categories_and_assayed}}
illustrates that while functional work targeted ``hand picked'' variants that were ultimately confirmed as damaging, many of them may be unlikely to arise based on population genetics data. 
\textbf{Figure 
\ref{fig:MRF_categories_and_assayed}}
presents, in increasing order, the number of potential variants based on likelihood of presentation and stacked by the number of variants per score category.
Variants that have been measured for their loss of protein activity are coloured by severity.
Potential variants that remain untested are coloured in grey.
Only 21 of the top 66 most probable clinically relevant variants have been assayed in \textit{RAG1}.  
 
\begin{figure}[h]
\centering
	\includegraphics[scale=0.55]{MRF_categories_and_assayed}
	\caption{RAG1 and RAG2 MRF score categories and variants assayed to date.  Protein residues are ranked and stacked into categories based on their MRF score. High scores (0.043 and 0.038 in RAG1 and RAG2, respectively) represent a greater mutation likelihood. Functional assays have measured recombination activity (as its inverse; \% loss of activity) in a total of 110 mutants. The severity of protein loss of function is represented by a red gradient. Residues that have not been functionally tested are shown in grey. While many protein residues are critical to protein function, their mutation is less probable than many of the top MRF candidates. Data further expanded in Figure
	\ref{fig:Accuracy_MRF_vs_functional}.
	MRF; mutation rate residue frequency.}
	\label{fig:MRF_categories_and_assayed}
\end{figure}

Supplemental Figure 
\textbf{\ref{fig:Accuracy_MRF_vs_functional}} 
further illustrates the individual variants which have been tested functionally (the coloured \textit{recombination activity} subset of Fig \ref{fig:MRF_categories_and_assayed}). 
We compared predicted MRF scores to assay measurements for 71 \textit{RAG1} and 39 \textit{RAG2} mutants.
Most mutations tested showed severe loss of protein function (bottom panel of Supplemental Figure 
\textbf{\ref{fig:Accuracy_MRF_vs_functional}}), while the likelihood each mutation occurring in humans varied significantly (top panels).

If MRF scoring was used in the same cases pre-emptively, the loss of investment  would be minimal; only 8 variants out of 71 mutants tested had an above-average MRF score while being measured as functionally benign (a rate of 11.27\%). 
RAG2 had only 3 out of 39 variants (7.69\%) with an above-average MRF score while functionally benign.
For the expended resources, approximately 30\% more top candidates would have been tested in place of unlikely and functionally non-damaging mutations. 
However, the true measurement of accuracy is limited in that very few of the most likely clinically relevant variants predicted by MRF scoring have been tested to date. 

\subsection*{\textbf{False positives in \textit{Transib} domains do not negatively impact prediction}}
\noindent Adaptive immunity is considered to have evolved through jawed vertebrates after integration of the RAG transposon into an ancestral antigen receptor gene 
\citep{agrawal1998transposition, hiom1998dna}.
The \textit{Transib} transposon is a 600 amino acid core region of RAG1 that targets RSS-like sequences in many invertebrates.
A linked \textit{RAG1/RAG2} was shown in the lower dueterosome (sea urchin), indicating an earlier common ancestor than the invertebrate
\citep{fugmann2006ancient}, 
and more recently,
a recombinatorially active RAG transposon (ProtoRAG) was found in the lower chordate amphioxus (or lancelet); the most basal extant chordate and a ``living fossil of RAG'' 
\citep{huang2016discovery}.

\begin{figure}[h]
\hspace*{0cm}   % This line is adjusting position on page left or right. 
	\includegraphics[scale=0.45]{False_positives_in_Transip_does_not_worsen_probability}
	\caption{False positives in \textit{Transib} domains do not worsen probability prediction. 
The \textit{Transib} domains contain critical conserved protein residues. 
(i) False positives were simulated by scoring \textit{Transib} domain MRF without omitting Boolean conservation weight $C=0$. 
(ii) Allele frequencies on GnomAD had conservation levels inversely proportional to simulated false-positive MRF scoring. 
(iii) When testing for all Boolean component $C>0$ after MRF calculation the effect of false positives remained non-significant, illustrating the non-negative impact of MRF for prediciting the mutation. Unpaired t-test,  * P = 0.0195, *** P < 0.0001. MRF; mutation rate residue frequency, ns; non-significant.} 	\label{fig:False_positives_in_Transip_does_not_worsen_probability}
\end{figure}

A set of conserved motifs in core \textit{RAG1} are shared with the \textit{Transib} transposase, including the critical DDE residue catalytic triad (residues 603, 711, and 965) 
\citep{kapitonov2005rag1}. 
Ten \textit{RAG1} core  motifs are conserved amongst a set of diverse species including human \citep{kapitonov2005rag1}. 
This evolutionarily conserved region is considered as most important to protein function.
Therefore, we chose this region to determine if MRF scoring would have a negative impact if mutations were falsely predicted as clinically important.
To assess the influence of a false positive effect on prediction, the MRF scores for conserved residues in this group were compared to GnomAD allele frequencies.
\textbf{Figure \ref{fig:False_positives_in_Transip_does_not_worsen_probability} (i)} 
plots the MRF (without omitting the Boolean component $C=0$) for conserved \textit{Transib} motif residues, non-conserved \textit{Transib} motif residues, and non-\textit{Transib} residues. 
\textbf{Figure \ref{fig:False_positives_in_Transip_does_not_worsen_probability} (ii)}
shows the percentage of these which were reported as mutated on GnomAD. 
By accounting for unreported variants by applying $C>0$, the resulting effect on incorrectly scoring MRF in the conserved \textit{Transib} motifs remained neutral. 

\subsection*{\textbf{MRF predicts RAG deficiency amongst PID patients harbouring rare variants}}
\noindent 
We have previously measured the recombination activity of RAG1 and RAG2 disease-causing variants in several patients \citep{lawless2018prevalence}. 
We have compiled our own and other functional assay data from \citet{lee2014systematic} and 
\citet{tirosh2018recombination} to produce a panel of recombination activity  measurements for coding variants in both \textit{RAG1} and \textit{RAG2}. 
RAG deficiency was measured as the level of recombination potential produced by the protein complex. 
Each method of investigation simulated the efficiency of wild-type or mutant proteins expressed by patients for their ability to produce a diverse repertoire of T-cell receptor (TCR) and B-cell receptor (BCR) and coding for immunoglobulins. 
In functional experiments, mutant proteins were assayed for their ability to perform recombination on a substrate which mimics the RSS of TCR and BCR in comparison to wild-type protein complex (as \% SEM). 

By gathering confirmed RAG deficiency cases, we compiled the MRF scores for 43 damaging \textit{RAG1} variants in 77 PID cases and 14 damaging \textit{RAG2} variants in 21 PID cases (MRF scores spanning over 22 categories). 
To test our method against a strong control group, we identified coding variants in patients with PID where RAG deficiency due to coding variants has been ruled out as the cause of disease.  
We obtained \textit{RAG1/2} variants in 558 PID patients who had their genomes sequenced as part of the NIHR BioResource - Rare Diseases study 
\citep{lawless2018prevalence}.
Filtering initially identified 32 variants in 166 people. 
This set was trimmed to contain only rare variants; 29 variants over 26 MRF scoring categories from 72 cases of non-RAG-deficient PID.
The scatterplot in Figure 5 shows that most PID cases had damaging variants with a high MRF score, while PID cases carried benign variants in RAG1/2 with lower MRF scores; i.e. an MRF >0.04 was seen for 31 cases of a damaging variant and only 2 cases of a non-damaging variant. 
Linear regression on this control group produced negative or near-zero slopes for \textit{RAG1} and \textit{RAG2}, respectively. 
The same analysis for known-damaging mutations in disease cases had a significant prediction accuracy for \textit{RAG1}.
Analysis for \textit{RAG2} was not significant.  However, the sample size to date may be too small to significantly measure \textit{RAG2} MRF scoring although a positive correlation was inferred in 
\textbf{Figure \ref{fig:RAG_mrf_linear_regression}} 
\citep{altman1995statistics}.
R source and raw data can be found in supplemental material.

\begin{figure}[h]
\hspace*{1cm}   % This line is adjusting position on page left or right. 
	\includegraphics[scale=0.7]{RAG_mrf_linear_regression}
	\caption{A linear regression model of RAG1/2 MRF scoring in cases of primary immune deficiency. 
MRF prediction correlates with clinical presentation.
Damaging variants identified in confirmed RAG deficiency cases. 
Non-damaging variants sourced from cases of PID with rare variants but not responsible for disease.
An MRF >0.04 was seen for 31 cases of a damaging RAG1 variants.
(Slopes of RAG1: Damaging: 0.0008* ($\pm$ 0.0004) P<0.05, intercept 5.82e-05 ***, Non-damaging: -0.0007 ($\pm$ 0.001).  Slopes of RAG2; Damaging: 0.0023 ($\pm$ 0.0018),  intercept 0.0312 *, Non-damaging 0.0001 ($\pm$ 0.0008). Source data and script in supplemental material).}
	\label{fig:RAG_mrf_linear_regression}
\end{figure}
\clearpage 

\subsection*{\textbf{MRF supplements pathogenicity prediction tools for translational research}}
\noindent CADD scoring \citep{kircher2014general} is an important bioinformatics tool that exemplifies pathogenicity prediction.
While CADD is a valuable scoring method, its purpose is not to predict likelihood of variation.
Similarly, MRF scoring is not a measure of pathogenicity. 
MRF scoring may be complemented by tools for scoring variant deleteriousness.
We compare MRF to the PHRED-scaled CADD scores for all possible SNV positions in \textit{RAG1}
(\textbf{Fig. \ref{fig:RAG_CADD}}) illustrating that pathogenicity prediction cannot account for mutation probability.
Combining both methods allows researchers to identify highly probable mutations before querying predicted pathogenicity.

\begin{figure}[h]
\hspace*{0.4cm}   % This line is adjusting position on page left or right. 
	\includegraphics[scale=0.75]{RAG_CADD}
	\caption{\textit{RAG1} PHRED-scaled CADD score versus GnomAD conservation rate and MRF score. 
Allele frequency conservation rate (\textbf{top}) is vastly important for identifying critical structural and functional protein regions. 
The impact of mutation in one of these conserved regions is often estimated using CADD scoring (\textbf{middle}). 
CADD score heatmap is aligned by codon and separated into three layers for individual nucleotide positions.
The MRF score (\textbf{bottom})(visualised using the 75$^{th}$ percentile with 1\% averaging) highlights protein regions which are most likely to present clinically and may require pre-emptive functional investigation.}
	\label{fig:RAG_CADD}
\end{figure}

To further develop this concept, we first annotated variants with  MRF likelihood scores and pathogenic prediction PHRED-scaled CADD scores 
(\textbf{Figure \ref{fig:hgmd_combined_methods}}), 
and secondly, performed a manual investigation of the clinical relevance of top candidates 
(\textbf{Table \ref{table:hgmd_data}}). 
We used HGMD as an unbiased source of known RAG deficiency cases in both instances. 
CADD score was very successful at predicting the pathogenicity of a variant, (a high-density cluster of variants with CADD scores >25) as shown in \textbf{red} in 
\textbf{Figure \ref{fig:hgmd_combined_methods} (i)}. 
At about the same rate, CADD score also predicted variants as pathogenic that are, to date, unreported 
(as \textbf{pink} in \textbf{Fig. \ref{fig:hgmd_combined_methods} (i)}). 
Indeed, those unreported variants may very well be pathogenic. 
However, the likelihood of each mutation varies. 
As such, we developed the MRF score to account for that likelihood. 
As expected, the likelihood of mutations occurring that were unreported was low according to MRF 
(\textbf{Fig.  
\ref{fig:hgmd_combined_methods} (ii)}, \textbf{pink}), 
while the mutations which did occur were highly enriched in at high MRF scores 
(\textbf{Fig.
\ref{fig:hgmd_combined_methods} (ii)}, \textbf{red} high-density cluster >0.043). 
Combining mutation prediction (MRF) with pathogenicity prediction (tools like CADD) increases the accuracy of pre-emptively targeting clinically relevant variants.  
\textbf{Figure \ref{fig:hgmd_combined_methods} (iii)} 
shows that while the number of variants presented to date is relatively small, they already account for 36\% of the top MRF score candidates. 

\begin{figure}[h]
\hspace*{1cm}   % This line is adjusting position on page left or right. 
	\includegraphics[scale=0.6]{hgmd_combined_methods}
	\caption{\textit{RAG1} PHRED-scaled CADD score versus MRF score against HGMD data.
	(i) A high CADD score is a predictor of deleteriousness. Both reported (red) and non-reported residues (pink) have a high density of high CADD score. 
	(ii) MRF scores only show a high-density cluster for high-likelihood variants, reflected by the high MRF score observed for known RAG deficiency variants.
	The number of pathogenic variants is outweighed by conserved residues; (i-ii) shows density of scores to normalise between groups.
	AUC overlap difference in CADD score of 21.43\% and MRF score of 74.28\% (above intersects >22.84 and >0.0409, in \textit{(i-ii)} respectively).
	(iii) The number of residues per MRF category shows that disease reported on HGMD accounts for 36\% of top MRF candidates. AUC; area under curve, CADD; Combined Annotation Dependent Depletion, HGMD; Human Gene Mutation Database.}
	\label{fig:hgmd_combined_methods}
\end{figure}
\clearpage

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Discussion         %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Discussion}
\noindent Determining disease-causing variants for functional analysis typically aims to target conserved gene regions. 
On GnomAD 56\% of \textit{RAG1} (approx. 246,000 alleles) is conserved with no reported variants. 
Functional validation of unknown variants in genes with this level purifying selection is generally infeasible. 
Furthermore, we saw that a vast number of candidates are ``predicted pathogenic'' by commonly used pathogenicity tools, which may indeed be damaging but unlikely to occur.
To overcome the challenge of manual selection we quantified the likelihood of mutation for each candidate variant.

Targeting clearly defined regions with high MRF scores allows for functional validation studies tailored to the most clinically relevant protein regions. 
An example of high MRF score clustering occurred in the RAG1 catalytic RNase H (RNH) domain at p.Ser638-Leu658 which is also considered a conserved \textit{Transib} motif. 

While many hypothetical variants with low MRF scores may be uncovered as functionally damaging, our findings suggest that human genomic studies will benefit by first targeting variants with the highest probability of occurrence (gene regions with high MRF). 
\textbf{Table 
\ref{Table:1}} lists the values for calculated MRFs for RAG1 and RAG2.

We have presented a basic application of MRF scoring for RAG deficiency. 
The method can be applied genome-wide.
This can include phenotypically derived weights to target candidate genes or tissue-specific epigenetic features. 
In the state presented here, MRF scores are used for pre-clinical studies. 
A more advanced development may allow for use in single cases. 
During clinical investigations using personalised analysis of patient data, further scoring methods may be applied based on disease features. 
A patient phenotype can contribute a weight based on known genotype correlations separating primary immunodeficiencies or autoinflammatory diseases \citep{picard2018international}. 
For example, a patient with autoinflammatory features may require a selection that favors genes associated with proinflammatory disease such as \textit{MEFV} and \textit{TNFAIP3}, whereas a patient with mainly immunodeficiency may have preferential scoring for genes such as \textit{BTK} and \textit{DOCK8}. 
In this way, a check-list of most likely candidates can be confirmed or excluded by whole genome or panel sequencing. 
However, validation of these expanded implementations requires a deeper consolidation of functional studies than is currently available. 

\citet{Havrilla220814} have recently developed a method with similar possible applications for human health mapping constrained coding regions.  
Their study employed a method that included weighting by sequencing depth. 
Similarly, genome-wide scoring may benefit from mutation significance cut-off, which is applied for tools such as CADD, PolyPhen-2, and SIFT 
\citep{itan2016mutation}.
We have not included an adjustment method as our analysis was gene-specific but implementation is advised when calculating genome-wide MRF scores. 

The MRF score was developed to identify the top most probable variants that have the potential to cause disease. 
It is not a predictor of pathogenicity. However, MRF may contribute to disease prediction; a clinician may ask for the likelihood of RAG deficiency (or any other Mendelian disease of interest) prior to examination 
(\textbf{\nameref{sec:Supplemental_text}}).

Predicting the likelihood of discovering novel mutations has implications in genome-wide association studies (GWAS). 
Variants with low minor allele frequencies have a low discovery rate and low probability of disease association \citep{kido2018minor}, an important consideration for rare diseases such as RAG deficiency.
An analysis of the NHGRI-EBI catalogue data highlighted diseases whose average risk allele frequency was low
\citep{kido2018minor}.
Autoimmune diseases had risk allele frequencies considered low at approximately 0.4. 
Without a method to rank most probable novel disease-causing variants, it is unlikely that GWAS will identify very rare disease alleles (with frequencies <0.001). 
It is conceivable that a number of rare immune diseases are attributable to polygenic rare variants. 
However, evidence for low-frequency polygenic compounding mutations will not be available until large, accessible genetics databases are available, exemplified by the NIHR BioResource Rare Diseases study 
\citep{lawless2018prevalence}. 
An interesting consideration when predicting probabilities of variant frequency, is that of protective mutations. 
Disease risk variants are quelled at low frequency by negative selection, while protective variants may drift at higher allele frequencies 
\citep{chan2014excess}.

The cost-effectiveness of genomic diagnostic tests is already outperforming traditional, targeted sequencing \citep{payne2018cost}. 
Even with substantial increases in data sharing capabilities and adoption of clinical genomics, rare diseases due to variants of unknown significance and low allele frequencies will remain non-actionable until reliable predictive genomics practices are developed. 
Bioinformatics as a whole has made staggering advances in the field of genetics 
\citep{libbrecht2015machine}. 
Challenges that remain unsolved, hindering the benefit of national or global genomics databases, include DNA data storage and random access retrieval 
\citep{Organick114553}, 
data privacy management 
\citep{Huang:224980}, 
and predictive genomics analysis methods. 
Variant filtration in rare disease is based on reference allele frequency, yet the result is not clinically actionable in many cases. 
Development of predictive genomics tools may provide a critical role for single patient studies and timely diagnosis \citep{casanova2014guidelines}. 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Conclusion         %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Conclusion}
\noindent We provide a list of amino acid residues for RAG1 and RAG2 that have not been reported to date, but are most likely to present clinically as RAG deficiency. 
This method may be applied to other diseases with hopes of improving preparedness for clinical diagnosis.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Authorship Contributions %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Authorship Contributions}
\noindent Dylan Lawless analysed data and conceived and wrote the manuscript, 
Hana Allen Lango analysed NIHR genomic data, 
James Thaventhiran provided genomic data, 
NIHR BioResourceâ€“Rare Diseases Consortium provided genomic data, 
Rashida Anwar wrote the manuscript, 
Jolan E. Walter provided clinical genomics data and wrote the manuscript, 
Jacques Fellay wrote the manuscript, 
Sinisa Savic conceived and wrote the manuscript.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Conflict of Interest%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Conflict of Interest}
\noindent The authors declare no conflict of interest.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% References         %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{\refname}
	\bibliographystyle{unsrtnat}
%	\bibliographystyle{plainnat}
	\bibliography{references}
\pagebreak 


%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% Supplemental       %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\beginsupplement
\section*{Supplemental}
 \label{sec:Supplemental_text}
 
 \begin{figure}[h]
	\includegraphics[scale=0.31]{method_map}
	\caption{Data analysis summary map.
	Raw data and analysis scripts are provided in the supplemental. Analysis steps and data sources for each procedure described in \textit{methods}. 
	 MRF; mutation rate residue frequency, PID; primary immunodeficiency.}
	\label{fig:method_map}
\end{figure}

 \begin{figure}[h]
%\hspace*{-1cm}   % This line is adjusting position
	\includegraphics[scale=0.8]{RAG_MRF_map}
	\caption{An alternative visualisation of MRF scores for RAG1 and RAG2 proteins.
	The data from Table \ref{Table:1} in column ``Average over 1\%'' is displayed on both the y-axis and colour scale. 
	An analysis-friendly long form CSV of the Table \ref{Table:1} data is also provided in the compressed supplemental R data ``mrf.csv''.}
  \label{fig:RAG_MRF_map}
\end{figure}

\begin{figure}[h]
	\hspace*{0cm} \includegraphics[scale=0.62]{MRF_vs_functional}
	\caption{MRF likelihood score versus known functional activity. 
We compiled all variants that we know to have been assayed for protein function to date.
The inverse of functional assay measurements were used, where 0\% activity represents 100\% loss of activity.
MRF scores are presented as a percentage of the maximum score per gene (i.e., for RAG1 $MRF_{max} = 0.043$ (100\%) and $MRF_{min} = 0.0048$ (0\%)). 	
\textbf{Top panels} show how likely each mutation is predicted to occur in humans. 
\textbf{Bottom panels} show the loss of protein activity  as a percentage compared to wild-type (\% SEM); most mutations tested produced severe loss of protein function, regardless of their mutation likelihood. Subset of \textit{Recombination activity} data from Figure \ref{fig:MRF_categories_and_assayed}.}
	\label{fig:Accuracy_MRF_vs_functional}
\end{figure}
\clearpage

\subsection*{\textbf{Main supplemental data table}} 
\begin{table}[h]
	\caption{MRF data tables. The complete scores are listed for each protein. 
Both the wild type and alternative variants reported on GnomAD are shown \citep{lek2016analysis}. 
Multiple alternative variants are reported for some residues which are annotated in a separate column. 
A column listing the 1\% average of MRF scores using a sliding window is present; this is used in Figure \ref{fig:Figure_main} (iii) with a cutoff threshold at the 75th percentile to clearly visualise high scoring clusters.
The Boolean conservation score is based on population genetics data, where 1 represents no known variant at a residue site. The percentile values for 0.9 and 0.75 are provided to allow for alternative visualisation methods using the R source supplemental \textit{``mrf.csv''} file.}
	\label{Table:1}
\end{table}

\subsection*{\textbf{Clinical relevance of top candidates}}
\noindent The top scoring candidates in RAG1 were assessed for potential clinical relevance 
(\textbf{Table \ref{table:hgmd_data}}). 
HGMD was chosen as a reliable, curated source of identifying pathogenic variants.
45\% of RAG1 variants reported on HGMD (23 of 51) were predicted by our model as the most likely candidates seen clinically (the top scoring MRF group of had 66 residues total).
The remaining variants in the top MRF group, which were not reported by HGMD (43 of 66), were assessed manually for their likelihood as potentially disease causing.
21 (49\%) were highly conserved, not reported on GnomAD, and would be considered probable RAG deficiency on presentation as homozygous or compound heterozygous with a second damaging variant.
The remainder had allele frequencies <0.0006, were only found as low frequency heterozygous in the general population, and justify functional validation.
We expect that none of the top candidate mutations are benign.

\begin{table}[!h]
\centering
\begin{tabular}{l p{3cm}}
\toprule
\multicolumn{2}{c}{\textbf{Most likely mutation candidates}}  \\
\midrule
{\textbf{Variant type}} & {\textbf{Number variants}} \\
{Top MRF score candidates total} & {66} \\
{(i) Of which are reported on HGMD} & {23} \\
{(ii) Not reported on HGMD to date} & {43}  \\
\bottomrule
\end{tabular}\\
\bigskip
\begin{tabular}{p{0.15\linewidth}p{0.2\linewidth}p{0.18\linewidth}p{0.18\linewidth}}
\toprule
\multicolumn{4}{c}{\textbf{Top MRF score candidates not reported on HGMD}}  \\
\midrule
{\textbf{Frequency in Population}} & {\textbf{Number variants}} & {\textbf{Unreported top candidate (\%)}} & {\textbf{GnomAD allele frequency}} \\
{Not found} & {21 of 43} & {49\%} & {0} \\
{Very rare} & {15 of 43} & {35\%} & {<0.00002*} \\
{Very rare} & {7 of 43} & {16\%} & {<0.00006**} \\
\bottomrule
\end{tabular}
\caption{Clinical relevance of top candidates.
23 top MRF score variants were reported as pathogenic on HGMD to date.
The remaining variants (the 43 not reported) were assessed by their frequency in population based on GnomAD (allele frequencies vary between individual variants but equate to approximately <6* and 9-77** heterozygous per 125,000 individuals).
Therefore, no top candidates should be considered benign without functional validation.
HGMD; Human Gene Mutation Database, MRF; mutation rate residue frequency.}
\label{table:hgmd_data}
\end{table} 
\clearpage

\subsection*{\textbf{Supplemental analysis tables}}
\begin{table}[!h]
	\caption{Known damaging in human cases. Amino acids and residue numbers are listed along with their basic mutation rate residue frequency value for  known RAG deficiency 
	\cite{notarangelo2016human}. Supplemental CSV file.}
	\label{table:E3}
\end{table}
\begin{table}[!h]
	\caption{Percentage of variants per gene.
	Percentage of mutated versus  non-mutated amino acids in RAG1 and RAG2 based on GnomAD population genetics data 
	\citep{lek2016analysis}. Supplemental CSV file.}
	\label{table:E4}
\end{table}
\begin{table}[!h]
	\caption{Residue frequency.
	Basic statistics for RAG1/2 were produced using SMS2
	\citep{stothard2000sequence}.			
Results are shown for the canonical sequences, a 1043-residue sequence of RAG1-201 peptide ENSP00000299440 and for a 527-residue sequence of RAG2-201 peptide ENSP00000308620.  Both percentage and frequency of residue usage are provided. Supplemental CSV file.}
	\label{table:E5}
\end{table}
\begin{table}[!h]
	\caption{Simplified residue frequency. The data from Table \ref{table:E5} is simplified for use in data analysis to only include residues, count, total, and frequency per protein expressed. Supplemental CSV file.}
	\label{table:E6}
\end{table}
\begin{table}[!h]
	\caption{Ratio of mutation per gene. The number of times each residue is mutated was found in population genetic data. The number of mutant versus wild-type is shown, from which the rate of each is derived. Supplemental CSV file.}
	\label{table:E7}
\end{table} 
\begin{table}[!h]
	\caption{Basic MRF scores. The mutation rate and frequency is shown which were used to calculate the basic mutation rate residue frequency for the main analysis dataframe and Table \ref{Table:1}. Supplemental CSV file.}
		\label{table:E8}
	\end{table}

\clearpage

\subsection*{\textbf{Protein structure application}}

\noindent With the availability of a structured protein complex, modelling can be carried out prior to functional assays. Residues with the highest MRF for both RAG1 and RAG2 were mapped in
\textbf{Figure \ref{fig:Structure}}.

\begin{figure}[!h]
\hspace*{1cm}   % This line is adjusting position on page left or right. 
	\includegraphics[scale=0.21]{Structure}
	\caption{The RAG1 (blue) and RAG2 (grey) protein structure with top candidate MRF scores. 
(i) Protein dimers and (ii=iv) monomers illustrating the three highest category MRF scores for predicted clinically relevant variants. 
Increasing in score the top three MRF categories (illustrated in 
\textbf{Figure \ref{fig:MRF_categories_and_assayed}})
for each protein are highlighted; yellow, orange, red. 
DNA (green) is bound by the RAG protein complex at recombination signal sequences (PDB:3jbw).}
	\label{fig:Structure}
\end{figure}
\clearpage

\subsection*{\textbf{Median CADD score per residue}}
\noindent The sourced PHRED-scaled CADD score data consisted of nucleotide level values. 
We were interested in CADD scores averaged per codon.
For every nucleotide position there were three alternative variants to consider, e.g.
\begin{tabularx}{\textwidth}{*{9}{Z}}
{Chrom} & {Pos} & {Ref} & {Alt1} & {Alt2} & {Alt3} & {PHRED1} & {PHRED2} & {PHRED3} \\
{11} & {36594855} & {A} & {C} & {G} & {T} & {22.3} & {18.81} & {22.4}
\end{tabularx}
\noindent The PHRED-scaled scores are listed here; raw CADD scores are also included in the original database.
To produce a working input we used the median score per codon, that is three scores per nucleotide and three nucleotides per codon.
This produced median PHRED-scaled score per codon / residue, e.g.:
\begin{tabularx}{\textwidth}{*{5}{Z}}
{Chrom} & {Pos} & {PHRED1} & {PHRED2} & {PHRED3} \\
{11} & {36594855} & {22.3} & {18.81} & {22.4} \\
{11} & {36594856} & {25.3} & {23.6} & {24.6} \\
{11} & {36594857} & {24.8} & {24.3} & {24.5} \\
\end{tabularx}

\noindent Median PHRED = 24.3 \\
\textbf{Supplemental file} \textit{`RAG1.cadd.amino.csv'} within the analysis data \textit{`Raw\_data\_R\_analysis\_for\_figures'} contains the  median values over a three-nucleotide window, starting at nucleotide 1 to produce input data with the correct reading frame. 
The ``PHRED-scaled'' values are used as a normalised and externally comparable unit of analysis, rather than raw CADD scores. 
The area under the curve was calculated for density plots to quantify the difference between pathogenic and unreported variants with high scores, above the intersects >0.0409 and >22.84 for MRF and CADD, respectively, using score value ($x$) versus density ($y$) (Fig.
\ref{fig:hgmd_combined_methods} (i-ii)) with 
$\int_a^b \! f(x) \, \mathrm{d}x \approx\ \left(b-a\right) \left[\frac{f\left(a\right)\ + f\left(b\right)}{2}\right].$

\subsection*{\textbf{Genome-wide and disease-specific application}}
\noindent Weighting data can also be applied to the MRF score model to amplify the selectivity. 
The mutation rate can be applied genome wide with a process common in the study of information retrieval; term frequency, inverse document frequency ($tf-idf$).
In this case the â€œtermâ€ and â€œdocumentâ€ are replaced by amino acid residue $r$ and gene $g$ , respectively such that,
\newcommand{\dd}[1]{\mathrm{d}#1}
\begin{equation}
{rf-igf}_{r,g} ={rf}_{r,g} \times {igf}_r
\end{equation}
We may view each gene as a vector with one component corresponding to each residue mutation in the gene, together with a weight for each component that is given by (1). 
Therefore, we can find the overlap score measure with the ${rf-igf}$ weight of each term in $g$, for a query $q$;

\begin{displaymath}
\mbox{Score}(q,g)=\sum_{r\in q} \mbox{rf-igf}_{r,g}.
\end{displaymath}
In respect to MRF scoring, this information retrieval method might be applied as follows; the ${rf-igf}$ weight of a term is the product of its $rf$ weight and its $igf$ weight (${W}_{r,g}={rf}_{r,g} \times \log \frac{N}{{gf}_{r}}$) or (${W}_{r,g}=(1 + \log {rf}_{r,g}) \times \log \frac{N}{{gf}_{r}}$). 
That is, firstly, the number of times a residue mutates in a gene ($rf={rf}_{r,g}$) and secondly, the rarity of the mutation genome-wide in $N$ number of genes ($igf=N/{gf}_{r}$). 
Finally, ranking the score of genes for a mutation query $q$ by; 
\begin{displaymath}
\mbox{Score}(q,g)=\sum_{r\in q\cap g} \mbox{rf-igf}_{r,g}
\end{displaymath}
\noindent The score of the query (Score($q,g$)) equals the mutations (terms) that appear in both the query and the gene ($r\in q\cap g$). 
Working out the $rf-igf$ weight for each of those variants (${rf.igf}_{r,g}$) and then summing them ($\sum$) to give the score for the specific gene with respect to the query.

\subsection*{\textbf{Bayesian probability}}
 \label{sec:Supplemental_Bayesian}
\noindent MRF score may provide a limiting component required for applying Bayesian probability to disease prediction.
A clinician may ask for the likelihood of RAG deficiency (or any Mendelian disease of interest) for a patient given a set of gene variants $P(H|E)$ using Bayes' theorem,
\begin{displaymath}
P(H|E) = \frac{P(E|H) P(H)}{P(E)}
\end{displaymath}
where $P(H)$ is the probability of a patient having RAG deficiency, $P(E | H)$  is the probability of RAG deficiency due to a set of variants that have been pre-emptively assayed, and $P(E)$ is the probability of having a set of gene variants.

$P(H)$ is known since the rate of RAG deficiency is
estimated at an incidence of 1:181,000
\citep{kumanovics2017estimated},
SCID at a rate of 1:330,000 \citep{kwan2014newborn},
and we also recently show the rate of RAG deficiency in adults with PID
\citep{lawless2018prevalence}.
Being a recessive disease, $P(E)$ must account for biallelic variants and is the most difficult value to determine. 
This may be found from population genetics data for (i) the rate of two separate, compound heterozygous variants, (ii) the rate of a homozygous variant or potential consanguinity, or (iii) the rate of de novo variation
\citep{lek2016analysis}. 
$P(E|H)$ would be identified where all variants are functionally validated. 
This requires a major investment, however the MRF score provides a good approximation.

\end{document}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% End of Document    %%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%
