---
layout: topic
title: Proteomics - Proteome Identification and Quantification
created: 19 Aug 2023
bibliography:
- '../tail/bibliography.bib'
tags: omics
subject: omics
---
{{ page.title }}
================

<!-- * TOC -->
<!-- {:toc} -->

There are five proteomics applications from FGCZ.
1. Proteome Identification and Quantification
2. Amino Acid Analysis
3. Glycan / Glycoprotein analysis
4. Interaction Proteomics
5. Characterization of purified biomolecules

## Introduction

FGCZ supports a wide range of protein analysis and proteomics applications, such as:

- Identification of the protein content in purified or complex samples (i.e., gel bands)
- Quantification of thousands of proteins upon different treatments or conditions (i.e., discovery of biomarkers in clinical samples)
- Analysis of post-translational modifications (i.e., phosphorylation)
- Identification of drug targets
- Analysis of low-input samples (i.e., a few 100s or 1000s of cells)

## The Method

The FGCZ supports various sample preparation workflows for different sample types, such as tissues, FFPE samples, cells, protein extracts, and biofluids (i.e., plasma and urine). The methods include:

- Automation in workflows
- Processing samples on the latest generation instruments using data dependent (DDA) and data independent (DIA) acquisition methods
- Analysis using the most appropriate software tool

## The Workflow

Most workflows follow these steps:

1. Protein extraction with specific protocols
2. Enzymatic digestion of proteins to generate peptides (i.e., SP3, PreOmics iST, PCT, FASP, etc.)
3. Desalting of the peptide mixture
4. Injection of the peptides to an online LC-MS/MS system
5. Bioinformatics analysis
6. Statistical analysis and data visualization

## The Output

Possible output formats vary depending on the requested analysis. All analyses include:

- A report with details of the applied methods
- The results of the analysis
- Delivery of the raw data

## List of applications and services

FGCZ routinely supports the following Proteome Identification and Quantification services, but it offers many additional services not listed below.

| Applications (examples) | Which service to order |
| --- | --- |
| Protein identification in simple or complex mixtures (i.e., gels, extracts) | Protein/Peptide Identification |
| Biomarker discovery in body fluids, tissues, cells, FFPEs | Proteome Quantification - Label-Free |
| High-throughput proteomics quantification | Proteome Quantification - Label-Free |
| Deep coverage proteome quantification | Proteome Quantification - TMT |
| Identification/quantification of proteins from low-input material (ideally 500-5000 cells) | Proteome Quantification - Label-Free (Low input analysis info - please contact us before ordering) |
| Large-scale phosphoproteomics analysis (with label-free approach) | Phosphopeptide Quantification - Label-Free |
| High coverage phosphoproteomics analysis (with label-based approach) | Phosphopeptide Quantification - TMT |
| Analysis of post-translational modifications (i.e., phospho, ubiquitin, acetyl, methyl) | Analysis of selected PTMs |
| Proteome-wide identification of protein N-Termini (TAILS) | N-Terminal characterization - TAILS |
| Setup of a quantification assay for a set of target proteins (panels, validation studies) | Targeted Quantification - Assay Setup |
| Quantification of target proteins (panels, validation studies) | Targeted Quantification |
| Any analysis not covered in the list above | Other (please contact us before ordering) |

In some cases, FGCZ also offers:

| Applications (examples) | Which service to order |
| --- | --- |
| Custom data analysis on FGCZ datasets | Bioinformatics analysis (only in special cases - please contact us before ordering) |
| Acquisition of LC-MS data | Data acquisition (LC-MS/MS) (only in special cases - please contact us before ordering) |
| Peptide/proteins HPLC fractionation | Other (please contact us before ordering) |

## References

This content is largely based on information from ETH Zurich functional genomics center 
<https://fgcz.ch/omics_areas.html>.
