<!DOCTYPE html>
<html xmlns="https://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
   <!-- link to MathJax for LaTeX stlye eqn -->
    <script type="text/javascript" async
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-MML-AM_CHTML">
</script>

	
	<!-- favicon -->
	<!--
	<link rel="icon" type="image/png" sizes="32x32" href="/images/logos/logo_2.8.png">
	-->
	<!-- https://realfavicongenerator.net -->
	<link rel="apple-touch-icon" sizes="180x180" href="/images/logos/favicon_package_v0.16_color/apple-touch-icon.png">
	<link rel="icon" type="image/png" sizes="32x32" href="/images/logos/favicon_package_v0.16_color/favicon-32x32.png">
	<link rel="icon" type="image/png" sizes="16x16" href="/images/logos/favicon_package_v0.16_color/favicon-16x16.png">
	<link rel="manifest" href="/images/logos/favicon_package_v0.16_color/site.webmanifest">
	<meta name="msapplication-TileColor" content="#da532c">
	<meta name="theme-color" content="#ffffff">

	<meta name="viewport" content="width=device-width, initial-scale=1">

   <!--  -->
   <meta http-equiv="content-type" content="text/html; charset=utf-8" />
   <title>SARS-CoV-2 variants of concern</title>
   <meta name="author" content="LawlessGenomics" />
   <link href="https://feeds.feedburner.com/tom-preston-werner" rel="alternate" title="Dyaln feed title" type="application/atom+xml" />
   <meta name="readability-verification" content="QCzSs992GxmRYRKVpPeZ6LE2tS8aYKxsSSQKV8YM"/>

   <!-- include a google font -->
   <link href="https://fonts.googleapis.com/css?family=Roboto+Mono|Roboto+Slab:300|Roboto:500" rel="stylesheet">
   <link href="https://fonts.googleapis.com/css?family=Lora" rel="stylesheet">

   <!-- syntax highlighting CSS -->
   <link rel="stylesheet" href="/css/syntax.css" type="text/css" />

   <!-- Homepage CSS -->
   <link rel="stylesheet" href="/css/screen.css" type="text/css" media="screen, projection" />

   <!-- Typekit -->
   <script type="text/javascript" src="https://use.typekit.com/jpd0pfm.js"></script>
   <script type="text/javascript">try{Typekit.load();}catch(e){}</script>
</head>
<body>

<!-- ClickTale Top part -->
<script type="text/javascript">
var WRInitTime=(new Date()).getTime();
</script>
<!-- ClickTale end of Top part -->

<div class="site">
  <div class="title">
    <img style="margin-bottom:-8px" 
	src="/images/logos/logo_2.8.png" 
	alt="Logo image" width="55" height="37.3"/>
    <a href="/">LawlessGenomics</a>
    <a class="extra" href="/">Home</a>
  </div>

  <div id="post">
<h1 id="sars-cov-2-variants-of-concern">SARS-CoV-2 variants of concern</h1>

<p class="meta">07 June 2021</p>

<h1 id="variants-of-concern-aligned-to-vaccine-coding-sequences">Variants of concern aligned to vaccine coding sequences</h1>

<p>This work has not been peer reviewed.</p>

<p><a href="https://github.com/DylanLawless/variants_of_concern_to_vaccine_SARS-CoV2/blob/master/variants_of_concern.pdf">Open PDF visualisation - variants_of_concern.pdf</a></p>

<p>Variants of concern for five SARS-CoV-2 strains 
(<a href="https://www.cdc.gov/coronavirus/2019-ncov/variants/variant-info.html?CDC_AA_refVal=https%3A%2F%2Fwww.cdc.gov%2Fcoronavirus%2F2019-ncov%2Fcases-updates%2Fvariant-surveillance%2Fvariant-info.html">CDC 4 Jun 2021</a>)
are illustrated against the 
translated amino acid sequences of the vaccines; Moderna mRNA-1273 and 
Pfizer/BioNTech BNT-162b2, and reference sequence NC_045512.</p>

<p>The variants are shown here, illustrated on the protein structure;
<a href="https://www.rcsb.org/structure/6ZOX">6ZOX.pdb DOI: 10.2210/pdb6ZOX/pdb</a>
<em>Structure of Disulphide-stabilized SARS-CoV-2 Spike Protein Trimer (x2 disulphide-bond mutant, G413C, V987C, single Arg S1/S2 cleavage site),</em>
provided by Xiong et al., 2020 <a href="https://www.nature.com/articles/s41594-020-0478-5">10.1038/s41594-020-0478-5</a>.</p>

<ul>
  <li><a href="https://github.com/DylanLawless/variants_of_concern_to_vaccine_SARS-CoV2/blob/master/pdb/renders/6zox_voc_20210604_front.jpg">6zox_voc_20210604_front.jpg</a></li>
  <li><a href="https://github.com/DylanLawless/variants_of_concern_to_vaccine_SARS-CoV2/blob/master/pdb/renders/6zox_voc_20210604_side.jpg">6zox_voc_20210604_side.jpg</a></li>
  <li><a href="https://github.com/DylanLawless/variants_of_concern_to_vaccine_SARS-CoV2/blob/master/pdb/renders/6zox_voc_20210604_top.jpg">6zox_voc_20210604_top.jpg</a></li>
</ul>

<p><img src="/images/posts/6zox_voc_20210604_front.jpg" width="100%" /></p>

<p>Visual alignment is shown against translated coding sequence for spike glycoprotein,
illustrated here via 
<a href="https://nextstrain.org/sars-cov-2">nextstrain.org</a>.</p>

<p><img src="/images/posts/nextstrain_spike.png" width="100%" /></p>

<ul>
  <li>Variants of Concern B.1.1.7</li>
  <li>Variants of Concern B.1.351</li>
  <li>Variants of Concern B.1.427</li>
  <li>Variants of Concern B.1.429</li>
  <li>Variants of Concern P.1</li>
  <li>Ref NC_045512.2.21563-25384</li>
  <li>mRNA-1273 vaccine translated</li>
  <li>BNT-162b2 vaccine translated</li>
</ul>

<p>The assemblies of putative SARS-CoV-2 spike encoding mRNA sequences for vaccines BNT-162b2 and mRNA-1273 have been provided by NAalytics; 
<a href="https://github.com/NAalytics/Assemblies-of-putative-SARS-CoV2-spike-encoding-mRNA-sequences-for-vaccines-BNT-162b2-and-mRNA-1273">Assemblies-of-putative-SARS-CoV2-spike-encoding-mRNA-sequences-for-vaccines-BNT-162b2-and-mRNA-1273</a>.
Their experimental sequence information for the RNA components of the initial 
Moderna (<a href="https://pubmed.ncbi.nlm.nih.gov/32756549/">https://pubmed.ncbi.nlm.nih.gov/32756549/</a>) 
and Pfizer/BioNTech (<a href="https://pubmed.ncbi.nlm.nih.gov/33301246/">https://pubmed.ncbi.nlm.nih.gov/33301246/</a>) 
COVID-19 vaccines, allowed them to produce a working assembly of the former 
and a confirmation of previously reported sequence information for the latter RNA.</p>

<p>From this data:</p>

<ol>
  <li>The translated amino acid sequences for both vaccines were derived.</li>
  <li>These were aligned to the SARS-CoV-2 reference amino acid sequence of spike glycoprotein.</li>
  <li>Known of variants-of-concern were then annotated and visualised.</li>
</ol>

<p>For visual simplicity, an X symbol was used to illustrate amino acid deletions.
All other amino acid changes use their correct symbol.</p>

<h1 id="files">Files</h1>

<ol>
  <li>README.md</li>
  <li><a href="https://github.com/DylanLawless/variants_of_concern_to_vaccine_SARS-CoV2/blob/master/sarscov2_vaccine_sequences_nucleotide.md">sarscov2_vaccine_sequences_nucleotide.md</a></li>
  <li><a href="https://github.com/DylanLawless/variants_of_concern_to_vaccine_SARS-CoV2/blob/master/sarscov2_vaccine_sequences_translated.md">sarscov2_vaccine_sequences_translated.md</a></li>
  <li><a href="https://github.com/DylanLawless/variants_of_concern_to_vaccine_SARS-CoV2/blob/master/reference_sequence_NC_045512.2.21563-25384.md">reference_sequence_NC_045512.2.21563-25384.md</a></li>
  <li><a href="https://github.com/DylanLawless/variants_of_concern_to_vaccine_SARS-CoV2/blob/master/variants_of_concern.xlsx">variants_of_concern.xlsx</a></li>
  <li><a href="https://github.com/DylanLawless/variants_of_concern_to_vaccine_SARS-CoV2/blob/master/variants_of_concern.csv">variants_of_concern.csv</a></li>
  <li><a href="https://github.com/DylanLawless/variants_of_concern_to_vaccine_SARS-CoV2/blob/master/variants_of_concern_to_vaccine.fa">variants_of_concern_to_vaccine.fa</a></li>
  <li><a href="https://github.com/DylanLawless/variants_of_concern_to_vaccine_SARS-CoV2/blob/master/variants_of_concern_to_vaccine.praln">variants_of_concern_to_vaccine.praln</a></li>
  <li><a href="https://github.com/DylanLawless/variants_of_concern_to_vaccine_SARS-CoV2/blob/master/variants_of_concern.pdf">variants_of_concern_to_vaccine.pdf</a></li>
</ol>

<h1 id="vaccine-sequences-and-amino-acid-translations">Vaccine sequences and amino acid translations</h1>
<p>The vaccine nucleotide sequence data
for vaccines BNT-162b2 and mRNA-1273 (sourced from <a href="https://github.com/NAalytics/Assemblies-of-putative-SARS-CoV2-spike-encoding-mRNA-sequences-for-vaccines-BNT-162b2-and-mRNA-1273">NAalytics</a>.
was converted to text and slightly formatted to select the coding sequences,
as shown in file:</p>

<p>[2] sarscov2_vaccine_sequences_nucleotide.md.</p>

<p>The nucleotide sequences were then translated into amino acid coding sequences,
as shown in file:</p>

<p>[3] sarscov2_vaccine_sequences_translated.md.</p>

<p>The nucleotide sequence was translated via <a href="https://web.expasy.org/translate/">https://web.expasy.org/translate/</a>.</p>

<h1 id="reference-genome-sequence">Reference genome sequence</h1>
<p>The reference data was sourced from:</p>
<ul>
  <li>Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome</li>
  <li>NCBI Reference Sequence: NC_045512.2</li>
</ul>

<p>via <a href="https://www.ncbi.nlm.nih.gov/nuccore/NC_045512">https://www.ncbi.nlm.nih.gov/nuccore/NC_045512</a>,
as shown in file:</p>

<p>[4] reference_sequence_NC_045512.2.21563-25384.md.</p>

<h1 id="vaccine-multiple-sequence-alignment">Vaccine Multiple Sequence Alignment</h1>
<p>The amino acid sequences of the coding region from each of the vaccine sequences 
and the reference sequence were used for multiple sequence alignment
via <a href="https://www.ebi.ac.uk/Tools/msa/clustalo/">https://www.ebi.ac.uk/Tools/msa/clustalo/</a>.</p>

<p>Variants of concern were then formatted to be used for annotation on the 
aligned sequences.</p>

<h1 id="variants-of-concern">Variants of concern</h1>
<p>SARS-CoV-2 Variant Classifications and Definitions were derived from
<a href="https://www.cdc.gov/coronavirus/2019-ncov/variants/variant-info.html?CDC_AA_refVal=https%3A%2F%2Fwww.cdc.gov%2Fcoronavirus%2F2019-ncov%2Fcases-updates%2Fvariant-surveillance%2Fvariant-info.html">https://www.cdc.gov/coronavirus/2019-ncov/variants/variant-info.html?CDC_AA_refVal=https%3A%2F%2Fwww.cdc.gov%2Fcoronavirus%2F2019-ncov%2Fcases-updates%2Fvariant-surveillance%2Fvariant-info.html</a></p>

<p>This dataset includes:</p>

<ul>
  <li>Variants of Interest (VOI)</li>
  <li>Variants of Concern (VOC)</li>
  <li>Variants of High Consequence (VOHC)</li>
</ul>

<p>The reformatted tables are presented in files:</p>

<p>[5] variants_of_concern.xlsx</p>

<p>[6] variants_of_concern.csv</p>

<p>There are currently no VOHC. 
VOC (but not VOI) were presented in the final visualisation.</p>

<h1 id="aligned-variants-of-concern-to-vaccine">Aligned variants of concern to vaccine</h1>

<p>The variants of concern were formatted such that one pseudo-fasta format entry 
contains the amino acid change for each strain. 
This data was then added to the multiple sequence alignment file to allow for
aligned annotations,
as shown in file:</p>

<p>[7] variants_of_concern_to_vaccine.fa</p>

<p>The file contains the list the variants-of-concern for five Sars-CoV-2 strains,
1 reference sequence, and
2 vaccine sequences:</p>

<ul>
  <li>Variants of Concern B.1.1.7</li>
  <li>Variants of Concern B.1.351</li>
  <li>Variants of Concern B.1.427</li>
  <li>Variants of Concern B.1.429</li>
  <li>Variants of Concern P.1</li>
  <li>Ref NC_045512.2:21563-25384</li>
  <li>mRNA-1273 vaccine translated</li>
  <li>BNT-162b2 vaccine translated</li>
</ul>

<h1 id="other-notes">Other notes</h1>
<p>Different strains will contain benign variants.
Typically, full sequences are used for alignment. 
However, this can be visually distracting.
Instead, only the variants-of-concern are annotated for the strain sequences.</p>

<p>The final illustration was made using <a href="https://www.snapgene.com">https://www.snapgene.com</a> software.
The snapgene-software formatted output can be loaded with the file:</p>

<p>[8] variants_of_concern_to_vaccine.praln</p>

<p>The final PDF version is shown in file:</p>

<p>[9] variants_of_concern_to_vaccine.pdf</p>

<p><a href="https://github.com/DylanLawless/variants_of_concern_to_vaccine_SARS-CoV2/blob/master/variants_of_concern.pdf">Open PDF visualisation - variants_of_concern.pdf</a></p>

</div>

<div id="related">
  <h2>Related Posts</h2>
  <ul class="posts">
    
      <li><span>07 Jun 2021</span> &raquo; <a href="/2021/06/07/weeknotes.html">Weeknotes</a></li>
    
      <li><span>31 May 2021</span> &raquo; <a href="/2021/05/31/weeknotes.html">Weeknotes</a></li>
    
      <li><span>28 May 2021</span> &raquo; <a href="/2021/05/28/pathway_analysis.html">Historical review of rare immune pathway analysis</a></li>
    
  </ul>
</div>



<div class="footer">
	<div class="contact">
	<p>
	<img src="/images/logos/logo_3.0.png" 
	alt="Logo image" width="99" height="67.1" />
	</p>
	</div>
	<div class="contact">
		<p>
		Built by Dylan Lawless
		<br />
		Postdoc at
		<a href="https://people.epfl.ch/dylan.lawless">EPFL</a>
		<br />
		<!--dylan( . )lawless( at )epfl( . )ch-->
		</p>
	</div>
	
	<div class="contact">
		<p>
		<a href="https://scholar.google.com/citations?user=RPBxP1wAAAAJ&hl=en&oi=ao">google scholar</a><br />
		<a href="https://dylan-lawless.github.io">Resume</a><br />
		<!-- <img src="https://hitcounter.pythonanywhere.com/count/tag.svg" alt="Hits"> -->
	</div>
	
	</div>
</div>

<!-- Banner on top right corner linking to github
<a href="https://github.com/mojombo"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_red_aa0000.png" alt="Fork me on GitHub" /></a>
-->

<!-- ClickTale Bottom part -->
<div id="ClickTaleDiv" style="display: none;"></div>
<script type="text/javascript">
if(document.location.protocol!='https:')
  document.write(unescape("%3Cscript%20src='https://s.clicktale.net/WRb.js'%20type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
if(typeof ClickTale=='function') ClickTale(206,0.3,"www03");
</script>
<!-- ClickTale end of Bottom part -->

<!-- Google Analytics -->
<!--
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "https://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
var pageTracker = _gat._getTracker("UA-6016902-1");
pageTracker._trackPageview();
</script>
-->
<!-- Google Analytics end -->

</body>
</html>
