---
layout: post
title: Unnecessary complexity in precision medicine
created: 2020-05-05
tags: genomics, journal club
status: unfinished
---

{{ page.title }}
================
<p class="meta">26 Apr 2020</p>
<br/>
* TOC
{:toc}
<br/>
Singh_and_Gupta.2020.GenoMed;  
Singh, R.S., Gupta, B.P. Genes and genomes and unnecessary complexity in precision medicine. npj Genom. Med. 5, 21 (2020). https://doi.org/10.1038/s41525-020-0128-1 Open Access
[https://doi.org/10.1038/s41525-020-0128-1](https://doi.org/10.1038/s41525-020-0128-1)

# Abstract

Risk factors, gene to phenotype, are more complex that anticipated in early genomics. 
Genotype - phenotype
Molecular and evolutionary complexity.
Molecular contigency - chance driven mutation, redundancy, molectular pathway cross-over (shared phenotypes).
Necessary complexity versus unnecessary - evolutionary baggage due to molecular constraint and blind evolution. 

# Introduction

"Two individuals bearing the same set of risk factors may not present (or exhibit the symptoms of) the same disease. This is the kind of problem that precision medicine is expected to overcome."
From familial studies, we often see that even clear monogenic, dominant disease does not always manifest equally among family members. "The lack of genetic determinism in disease is the result of molecular contingency."

An interesting point is made that I haven't though much about; "Whether evolution is deterministic, i.e., predictable and repeatable, or contingent and unpredictable is an interesting topic in evolutionary biology (ref 1) (for a recent review see ref. 2)"
Although, this is just a smaller version of a question in physics that I do think about frequently: if every irreducibly small particle/force in the universe could be detected with a mapped trajectory, then we could "see" both forward and reverse in time; causal determinism. Since the particle physics answer will not come anytime soon, a discussion on the evolutionary version is reasonable. 

"..phenotypic evolutionary ‘repeatability’ is common when the founding populations are closely related, perhaps resulting from shared genes and developmental pathways, whereas different outcomes become more likely as historical divergences increase."

I believe that in some cases we can produces actionable levels of confidence in mutation prediction to make this a worthwhile endeavour. Knowing that a mutation is likely to occur (and be evolutionarily selected against) and result in disease means that we can prepare a treatment / cure for when it does occur naturally. 
For example, each nucleotide within each individual cell will have a particular probability for the energy required to mutate. 
Many factors affect this value; nucleotide type, methylation, DNA/RNA folding, bound proteins, chemical/UV insult, etc. 
Some of these factors can be estimated based on large studies, and some may never be quantifiable in the real world.
However, we can at least quantify some values to produce mutation predictor value X and and response Y and therefore quantify estimate coefficients, and therefore predict that mutation 1 is more likely to occur than mutation 2. 

Reference is made to "‘Evolution and Tinkering,’ Jacob (ref 4)" who compares natural selection to a tinkerer as opposed to an engineer.
Singh and Gupta propose "a theory of molecular complexity, consisting of necessary and unnecessary complexity in living systems", that "exceedingly high level of unnecessary complexity" is more due to blind evolution rather than randomness of mutation. 
"Unecessary and unnecessary complexity is relevant to current discussions on genomics and precision medicine."
 
# Complexity in living systems
# Is complexity necessary?
# Evolution of complexity: genic, genomic, and developmental
# Unnecessary complexity in precision medicine
# Unnecessary complexity and the etiology of cancer
# Unnecessary complexity beyond precision medicine
# Conclusions


