<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
<!-- --- --><!-- layout: post --><!-- title: Historical review of rare immune pathway analysis --><!-- created: 28 May 2021 --><!-- tags: genomics --><!-- status: finished --><!-- --- --><!-- Pathway_analysis --><!-- ================ --><!-- <p class="meta">28 May 2021</p> --><!-- # Itan & Casanova contributions --><!-- We are tackling this problem of protein pathway analysis from the viewpoint of --><!-- rare immune disease and infection. --><!-- Historically, several topics in bioinformatic and functional analysis have --><!-- been required before we could achieve our current position of --><!-- statistically-robust genetic discovery for rare disease: --><!-- 1. Candidate variant select for individual genomes --><!-- 2. Compiling reliable cohorts of patients with shared phenotypes --><!-- 3. Protein-protein interactions --><!-- 4. Variant collapse --><!-- 5. Protein pathway analysis --><!-- 6. Functional validation --><!-- Therefore, a historical review of the timeline is beneficial to illustrate the --><!-- technical successes that allow us to reliably produce candidate variants by --><!-- genome sequencing and to validate statistically-driven results by _"traditional"_ --><!-- functional validation. --><!-- These steps [1, 2 and 6 in our list] are exemplified by the following --><!-- historical review. --><!-- The complete list of steps 1-6 are touched on, --><!-- but full validation of each step is the culmination of what we are currently --><!-- working on and will be explicitly reviewed when we have completed our study. --><!-- As one of the leaders in this field, Casanova lab has provided a lot of insider --><!-- history to the story in a great twitter thread --><!-- <https://twitter.com/casanova_lab/status/1397539593608695808>. --><!-- The literature for discussion is first listed here to facilitate downloading --><!-- but sources are referenced as usual throughout. --><!-- --- --><!-- 2013 PNAS. The human gene connectome as a map of short cuts for morbid allele discovery. --><!-- <a class="citation" href="#itan2013human">(Itan et al., 2013)</a> --><!-- <https://pubmed.ncbi.nlm.nih.gov/23509278/> --><!-- 2014 BMC Gen. HGCS: an online tool for prioritizing disease-causing gene variants by biological distance. --><!-- <a class="citation" href="#itan2014hgcs">(Itan et al., 2014)</a> --><!-- <https://pubmed.ncbi.nlm.nih.gov/24694260/> --><!-- 2015 Front. Novel primary immunodeficiency candidate genes predicted by the human gene connectome. --><!-- <a class="citation" href="#itan2015novel">(Itan &amp; Casanova, 2015)</a> --><!-- <https://pubmed.ncbi.nlm.nih.gov/25883595/>, --><!-- 2015 PNAS. The human gene damage index as a gene-level approach to prioritizing exome variants. --><!-- <a class="citation" href="#itan2015human">(Itan et al., 2015)</a> --><!-- <https://pubmed.ncbi.nlm.nih.gov/26483451/> --><!-- 2016 NatMet. The mutation significance cutoff: gene-level thresholds for variant predictions. --><!-- <a class="citation" href="#itan2016mutation">(Itan et al., 2016)</a> --><!-- <https://pubmed.ncbi.nlm.nih.gov/26820543/> --><!-- 2015 PNAS. Can the impact of human genetic variations be predicted? --><!-- <a class="citation" href="#itan2015can">(Itan &amp; Casanova, 2015)</a> --><!-- <https://pubmed.ncbi.nlm.nih.gov/26351682/> --><!-- 2018 Bioinf. PopViz: a webserver for visualizing minor allele frequencies and damage prediction scores of human genetic variations. --><!-- <a class="citation" href="#zhang2018popviz">(Zhang et al., 2018)</a> --><!-- <https://pubmed.ncbi.nlm.nih.gov/30535305/> --><!-- 2019 PNAS. Blacklisting variants common in private cohorts but not in public databases optimizes human exome analysis. --><!-- <a class="citation" href="#maffucci2019blacklisting">(Maffucci et al., 2019)</a> --><!-- <https://pubmed.ncbi.nlm.nih.gov/30591557/> --><!-- 2019 NAR. SeqTailor: a user-friendly webserver for the extraction of DNA or protein sequences from next-generation sequencing data. --><!-- <a class="citation" href="#zhang2019seqtailor">(Zhang et al., 2019)</a> --><!-- <https://pubmed.ncbi.nlm.nih.gov/31045209/>. --><!-- 2020 Hum Gen. The human genetic determinism of life-threatening infectious diseases: genetic heterogeneity and physiological homogeneity? --><!-- <a class="citation" href="#casanova2020human">(Casanova &amp; Abel, 2020)</a> --><!-- <https://pubmed.ncbi.nlm.nih.gov/32462426/> --><!-- 2021 JCI. Herpes simplex encephalitis in a patient with a distinctive form of inherited IFNAR1 deficiency. --><!-- <a class="citation" href="#bastard2021herpes">(Bastard et al., 2021)</a> --><!-- <https://pubmed.ncbi.nlm.nih.gov/32960813/> --><!-- 2021 JCI. TLR3 controls constitutive IFN-β antiviral immunity in human fibroblasts and cortical neurons. --><!-- <a class="citation" href="#gao2021tlr3">(Gao et al., 2021)</a> --><!-- <https://pubmed.ncbi.nlm.nih.gov/33393505/> --><!-- 2021 AJHG. A computational approach for detecting physiological homogeneity in the midst of genetic heterogeneity --><!-- <a class="citation" href="#zhang2021computational">(Zhang et al., 2021)</a> --><!-- <https://pubmed.ncbi.nlm.nih.gov/34015270/> --><!-- --- --><!-- To date, the main paper that implements protein pathway analysis for rare --><!-- immune disease is that by --><!-- Peng Zhang and --><!-- Yuval Itan <a class="citation" href="#zhang2021computational">(Zhang et al., 2021)</a>. --><!-- The history begins with determining methods for candidate variant selection - --><!-- the main challenge in human genomics, --><!-- especially for individual patients who can benefit from precision medicine. --><!-- Over the last decade, we have reached a point where we can now reasonably --><!-- discern individual candidate-causal variants from the background noise of --><!-- genomic variability. --><!-- From the authors' perspective, the project began around 2011 and its first --><!-- step was concluded in 2013 with --><!-- Yuval Itan's --><!-- first "Human Gene Connectome" paper --><!-- while he was a post-doc with Casanova lab. --><!-- <a class="citation" href="#itan2013human">(Itan et al., 2013)</a>. --><!-- This software connected genes like streets in a map, --><!-- based on their physiological relatedness. --><!-- It was soon followed by methodological development --><!-- <a class="citation" href="#itan2014hgcs">(Itan et al., 2014)</a> --><!-- and application to inborn errors of immunity --><!-- <a class="citation" href="#itan2015novel">(Itan &amp; Casanova, 2015)</a> --><!-- or both --><!-- <a class="citation" href="#itan2015human">(Itan et al., 2015)</a>, --><!-- and a couple of necessary detours --><!-- <a class="citation" href="#itan2016mutation">(Itan et al., 2016)</a> --><!-- and --><!-- <a class="citation" href="#maffucci2019blacklisting">(Maffucci et al., 2019)</a>. --><!-- A review was also written by two of the main authors during the same period --><!-- <a class="citation" href="#itan2015can">(Itan &amp; Casanova, 2015)</a>. --><!-- When Peng Zhang joined the Casanova lab as post-doc, --><!-- Yuval Itan had started his own lab. --><!-- However, the pair worked together to continue producing the papers on variant --><!-- interpretation and data processing --><!-- <a class="citation" href="#zhang2018popviz">(Zhang et al., 2018)</a> --><!-- and --><!-- <a class="citation" href="#zhang2019seqtailor">(Zhang et al., 2019)</a>. --><!-- After completing this period of work, --><!-- they renamed "Human Gene Connectome II" the --><!-- "Network-based Heterogeneity Clustering". --><!-- At this point, their aims were defined as being generally indistinguishable --><!-- from ours. --><!-- That is, _"the detection of physiological homogeneity in a cohort of patients --><!-- sharing a clinical phenotype but with high genetic heterogeneity - --><!-- a hallmark of severe infectious diseases"_ (Casanova via twitter), --><!-- as presented in their next paper on this topic --><!-- <a class="citation" href="#casanova2020human">(Casanova &amp; Abel, 2020)</a>. --><!-- Shen-Ying Zhang came on board as senior author on the next two papers. --><!-- With an excellent database of immune disorders and infections, --><!-- the team could gradually build their software. --><!-- Exomes from patients with HSV-1 encephalitis were used for testing successive --><!-- versions in --><!-- <a class="citation" href="#bastard2021herpes">(Bastard et al., 2021)</a> --><!-- and --><!-- <a class="citation" href="#gao2021tlr3">(Gao et al., 2021)</a>. --><!-- Quoting Casanova _"When they were capable of detecting the known TLR3-IFN needles in the HSE stack, they installed camp 1, rested a bit and reported to me on the radio, while I was watching them from the basecamp with binoculars. I encouraged them to push for the final ascent and they did."_ --><!-- With the same goal as our own - --><!-- producing unbiased methods for detection of biologically-connected causal --><!-- genetic variation - --><!-- they found new gene variants that interact via the TLR3-IFN protein pathway, --><!-- in individual patients. --><!-- Shen-Ying Zhang found them to be biochemically deleterious, --><!-- an important factor for validation of genetic-first aproaches. --><!-- In this case, Zhang _et al_ get as close to the _"gold-standard"_ --><!-- as anyone to date. --><!-- The functional validation of candidate variants in disease then provided a proof-of-principle indication that they could detect physiological homogeneity in the midst of genetic heterogeneity --><!-- <a class="citation" href="#zhang2021computational">(Zhang et al., 2021)</a>. --><!-- Quoting Casanova --><!-- _"A computational approach for detecting physiological homogeneity in the midst of genetic heterogeity. --><!-- That was terrific!"_. --><!-- --- --><!-- > An aside on what I call the _"gold-standard"_ for our field should be: --><!-- > 1. Unbiased statistical detection of a genetic phenomenon. --><!-- > 2. Validation by functional models under systematic control. --><!-- > --><!-- > Part [1] Depends on patient cohorts that are large enough to detect the effect based on the phenotype strength - difficult for _rare disease_.\ --><!-- > Part [2] Depends on independently testing biological mechanisms. --><!-- > --><!-- > This second step generally consists of two hurdles: --><!-- > * If the same researchers perform (1) genetic stats and (2) functional work, --><!-- > there is a bias that is difficult to avoid when trying to functionally --><!-- > validate statistically positive results. --><!-- > * If the statistical genetic associations happen to contain a false positive --><!-- > for something like severe rare immune disease, --><!-- > the sensitive functional models may detect a damaging response. --><!-- > One might find a truly damaging biological mechanism, --><!-- > but if the statistical genetic association is not correct then this --><!-- > biological mechanism should not be deemed causal; --><!-- > back-tracking at this stage would be very difficult due to self-imposed bias. --><!-- > --><!-- > Ideally, in the future we hope to see a separation of the two steps --><!-- > (stat genetics and wet-lab) such that each are performed independently. --><!-- > The wet-lab would also ideally focus their routines on a particular protein --><!-- > pathway/system with SOPs that improve accuracy and precision --><!-- > (e.g. clinical diagnostics labs, clinical trials) rather than setting up --><!-- > models for each new study. --><!-- --- --><!-- Returning to our historical review, --><!-- we have been producing our protocols similarly in parallel. --><!-- With patient cohorts of comparable sizes and phenotypes we will soon --><!-- have a complementary validation of protocols. --><!-- However, great care is also being taken to test and select the most reliable --><!-- statistical methods for association testing - --><!-- an improvement to the fine work by --><!-- <a class="citation" href="#zhang2021computational">(Zhang et al., 2021)</a>. --><!-- Best practices in candidate variant selection protocols are basically --><!-- standardised as of 2021, --><!-- so the main remaining task is standardisation of the protein-pathway --><!-- annotation and association testing methods - --><!-- steps which we will soon be ready to publish after peer-review. --><!-- # References --><!-- <ol class="bibliography"><li><span id="itan2013human">Itan, Y., Zhang, S.-Y., Vogt, G., Abhyankar, A., Herman, M., Nitschke, P., Fried, D., Quintana-Murci, L., Abel, L., &amp; Casanova, J.-L. (2013). The human gene connectome as a map of short cuts for morbid allele discovery. <i>Proceedings of the National Academy of Sciences</i>, <i>110</i>(14), 5558–5563.</span></li>
<li><span id="itan2014hgcs">Itan, Y., Mazel, M., Mazel, B., Abhyankar, A., Nitschke, P., Quintana-Murci, L., Boisson-Dupuis, S., Boisson, B., Abel, L., Zhang, S.-Y., &amp; others. (2014). HGCS: an online tool for prioritizing disease-causing gene variants by biological distance. <i>BMC Genomics</i>, <i>15</i>(1), 1–8.</span></li>
<li><span id="itan2015novel">Itan, Y., &amp; Casanova, J.-L. (2015). Novel primary immunodeficiency candidate genes predicted by the human gene connectome. <i>Frontiers in Immunology</i>, <i>6</i>, 142.</span></li>
<li><span id="itan2015human">Itan, Y., Shang, L., Boisson, B., Patin, E., Bolze, A., Moncada-Vélez, M., Scott, E., Ciancanelli, M. J., Lafaille, F. G., Markle, J. G., &amp; others. (2015). The human gene damage index as a gene-level approach to prioritizing exome variants. <i>Proceedings of the National Academy of Sciences</i>, <i>112</i>(44), 13615–13620.</span></li>
<li><span id="itan2016mutation">Itan, Y., Shang, L., Boisson, B., Ciancanelli, M. J., Markle, J. G., Martinez-Barricarte, R., Scott, E., Shah, I., Stenson, P. D., Gleeson, J., &amp; others. (2016). The mutation significance cutoff: gene-level thresholds for variant predictions. <i>Nature Methods</i>, <i>13</i>(2), 109–110.</span></li>
<li><span id="itan2015can">Itan, Y., &amp; Casanova, J.-L. (2015). Can the impact of human genetic variations be predicted? <i>Proceedings of the National Academy of Sciences</i>, <i>112</i>(37), 11426–11427.</span></li>
<li><span id="zhang2018popviz">Zhang, P., Bigio, B., Rapaport, F., Zhang, S.-Y., Casanova, J.-L., Abel, L., Boisson, B., &amp; Itan, Y. (2018). PopViz: a webserver for visualizing minor allele frequencies and damage prediction scores of human genetic variations. <i>Bioinformatics</i>, <i>34</i>(24), 4307–4309.</span></li>
<li><span id="maffucci2019blacklisting">Maffucci, P., Bigio, B., Rapaport, F., Cobat, A., Borghesi, A., Lopez, M., Patin, E., Bolze, A., Shang, L., Bendavid, M., &amp; others. (2019). Blacklisting variants common in private cohorts but not in public databases optimizes human exome analysis. <i>Proceedings of the National Academy of Sciences</i>, <i>116</i>(3), 950–959.</span></li>
<li><span id="zhang2019seqtailor">Zhang, P., Boisson, B., Stenson, P. D., Cooper, D. N., Casanova, J.-L., Abel, L., &amp; Itan, Y. (2019). SeqTailor: a user-friendly webserver for the extraction of DNA or protein sequences from next-generation sequencing data. <i>Nucleic Acids Research</i>, <i>47</i>(W1), W623–W631.</span></li>
<li><span id="casanova2020human">Casanova, J.-L., &amp; Abel, L. (2020). The human genetic determinism of life-threatening infectious diseases: genetic heterogeneity and physiological homogeneity? <i>Human Genetics</i>, <i>139</i>, 681–694.</span></li>
<li><span id="bastard2021herpes">Bastard, P., Manry, J., Chen, J., Rosain, J., Seeleuthner, Y., AbuZaitun, O., Lorenzo, L., Khan, T., Hasek, M., Hernandez, N., &amp; others. (2021). Herpes simplex encephalitis in a patient with a distinctive form of inherited IFNAR1 deficiency. <i>The Journal of Clinical Investigation</i>, <i>131</i>(1).</span></li>
<li><span id="gao2021tlr3">Gao, D., Ciancanelli, M. J., Zhang, P., Harschnitz, O., Bondet, V., Hasek, M., Chen, J., Mu, X., Itan, Y., Cobat, A., &amp; others. (2021). TLR3 controls constitutive IFN-βantiviral immunity in human fibroblasts and cortical neurons. <i>The Journal of Clinical Investigation</i>, <i>131</i>(1).</span></li>
<li><span id="zhang2021computational">Zhang, P., Cobat, A., Lee, Y.-S., Wu, Y., Bayrak, C. S., Boccon-Gibod, C., Matuozzo, D., Lorenzo, L., Jain, A., Boucherit, S., &amp; others. (2021). A computational approach for detecting physiological homogeneity in the midst of genetic heterogeneity. <i>The American Journal of Human Genetics</i>.</span></li></ol> -->
