<!DOCTYPE html>
<html xmlns="https://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
	<!-- link to MathJax for LaTeX stlye eqn -->
	 <script type="text/javascript" async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-MML-AM_CHTML">
</script>

	 
	 <!-- https://realfavicongenerator.net -->
	 <link rel="apple-touch-icon" sizes="180x180" href="/images/logos/favicon_package_v0.16_bw/apple-touch-icon.png">
	 <link rel="icon" type="image/png" sizes="32x32" href="/images/logos/favicon_package_v0.16_bw/favicon-32x32.png">
	 <link rel="icon" type="image/png" sizes="16x16" href="/images/logos/favicon_package_v0.16_bw/favicon-16x16.png">
	 <link rel="manifest" href="/images/logos/favicon_package_v0.16_bw/site.webmanifest">
	 <meta name="msapplication-TileColor" content="#da532c">
	 <meta name="theme-color" content="#ffffff">
	
	 <meta name="viewport" content="width=device-width, initial-scale=1">
	
	<meta http-equiv="content-type" content="text/html; charset=utf-8">
	<meta name="description" content="A home for topics in human precision medicine, genomic analysis, and data visualisation.">
	<title>Worlds simplest GWAS</title>
	<meta name="author" content="LawlessGenomics">
	<link href="https://lawlessgenomics.com/feed.xml" rel="alternate" title="LawlessGenomics" type="application/atom+xml">
	<!-- <meta name="readability-verification" content="QCzSs992GxmRYRKVpPeZ6LE2tS8aYKxsSSQKV8YM"/> -->
	
	<!-- include a google font -->
	<link href="https://fonts.googleapis.com/css?family=Roboto+Mono%7CRoboto+Slab:300%7CRoboto:500" rel="stylesheet">
	<link href="https://fonts.googleapis.com/css?family=Lora" rel="stylesheet">
	
	<!-- syntax highlighting CSS -->
	<link rel="stylesheet" href="/css/syntax.css" type="text/css">
	
	<!-- Homepage CSS -->
	<link rel="stylesheet" href="/css/screen.css" type="text/css" media="screen, projection">
	
	<!-- Google font -->
	<link rel="preconnect" href="https://fonts.googleapis.com"> 
	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin> 
	<link href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,400;0,600;0,700;1,400;1,600;1,700&amp;display=swap" rel="stylesheet">
	
	<!-- Typekit -->
	<script type="text/javascript" src="https://use.typekit.com/jpd0pfm.js"></script>
	<script type="text/javascript">try{Typekit.load();}catch(e){}</script>
</head>
<body>

<!-- ClickTale Top part -->
<script type="text/javascript">
	var WRInitTime=(new Date()).getTime();
</script>
<!-- ClickTale end of Top part -->

<!-- Logo and navigation bar -->



<!-- SITE START --------------------------------------------------------- -->
<div class="site">

<!-- NAV START --------------------------------------------------------- -->
<div class="sitewidthnav">
	<div class="title">
		<div class="nav-container">
			<nav>

			<div class="logoimage">
				<a href="/">
				<img style="margin-bottom:-16px" src="/images/logos/logo4.2_150ppi.png" alt="Logo image" width="25" height="38.5"></a>
				<!-- alt="Logo image" width="35.7" height="55" /></a> -->
			</div>

			<div class="logo">
				<a href="/">LawlessGenomics</a>
			</div>

				<input type="checkbox" id="nav-toggle">
					<label for="nav-toggle" class="gg-menu"></label>
					<div class="right-menu">
						<a href="/resume">About</a>
						<a href="/">Home</a>
					</div>
			</nav>
		</div>
	</div>



</div>
<!-- NAV END ----------------------------------------------------------- -->

<!-- MAIN START -------------------------------------------------------- -->
<div class="sitewidth">
	<div id="post">
Reading time 3 minutes
<h1 id="worlds-simplest-gwas">Worlds simplest GWAS</h1>
<p class="meta">18 Jan 2022</p>

<ul id="markdown-toc">
  <li>
<a href="#datasets" id="markdown-toc-datasets">Datasets</a>    <ul>
      <li><a href="#genotype-matrix" id="markdown-toc-genotype-matrix">Genotype matrix</a></li>
      <li><a href="#phenotype-matrix" id="markdown-toc-phenotype-matrix">Phenotype matrix</a></li>
    </ul>
  </li>
  <li><a href="#experiment" id="markdown-toc-experiment">Experiment</a></li>
  <li>
<a href="#coded-example" id="markdown-toc-coded-example">Coded example</a>    <ul>
      <li>
<a href="#glm_loopr" id="markdown-toc-glm_loopr">glm_loop.R</a>        <ul>
          <li><a href="#result" id="markdown-toc-result">Result</a></li>
        </ul>
      </li>
    </ul>
  </li>
  <li><a href="#real-world" id="markdown-toc-real-world">Real world</a></li>
  <li><a href="#other-code" id="markdown-toc-other-code">Other code</a></li>
</ul>

<p>To answer a Q about how to start with GWAS, examples are helpful.
For a GWAS we need some genotype data and some phenotype data.</p>

<h2 id="datasets">Datasets</h2>
<ul>
  <li>For each patient, the genotype data might consist of 500’000 variant positions in the genome (SNPs).</li>
  <li>You could imagine this as a file containing a matrix with 500’000 columns (SNPs) and one row per patient (e.g. 500 patients).</li>
  <li>At each SNP the data will say either:
    <ul>
      <li>“reference” (wild-type) encode as 0 (e.g. A/A)</li>
      <li>“variant” (heterozygous variant) encoded as 1 (e.g. A/T)</li>
      <li>“variant” (homozygous variant) encoded as 2 (e.g. T/T).</li>
    </ul>
  </li>
  <li>A second file might have phenotype data. For example it could be protein abundace measured by mass spectrometry (MS), with one column per protein (1000 proteins) and one row per patient (e.g. 500 patients).</li>
</ul>

<h3 id="genotype-matrix">Genotype matrix</h3>

<table>
  <thead>
    <tr>
      <th>Sample</th>
      <th>SNP 1</th>
      <th>SNP 2</th>
      <th>SNP 3</th>
      <th>SNP 4</th>
      <th>SNP 5</th>
      <th>SNP 6</th>
      <th>SNP 7</th>
      <th>SNP 8</th>
      <th>SNP 9</th>
      <th>SNP 10</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Sample1</td>
      <td>2</td>
      <td>2</td>
      <td>0</td>
      <td>0</td>
      <td>1</td>
      <td>2</td>
      <td>2</td>
      <td>0</td>
      <td>1</td>
      <td>1</td>
    </tr>
    <tr>
      <td>Sample2</td>
      <td>0</td>
      <td>1</td>
      <td>2</td>
      <td>2</td>
      <td>0</td>
      <td>2</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
    </tr>
    <tr>
      <td>Sample3</td>
      <td>2</td>
      <td>1</td>
      <td>0</td>
      <td>0</td>
      <td>1</td>
      <td>2</td>
      <td>0</td>
      <td>0</td>
      <td>1</td>
      <td>2</td>
    </tr>
    <tr>
      <td>Sample4</td>
      <td>1</td>
      <td>2</td>
      <td>1</td>
      <td>0</td>
      <td>0</td>
      <td>2</td>
      <td>0</td>
      <td>2</td>
      <td>2</td>
      <td>1</td>
    </tr>
    <tr>
      <td>Sample5</td>
      <td>2</td>
      <td>1</td>
      <td>1</td>
      <td>0</td>
      <td>0</td>
      <td>2</td>
      <td>0</td>
      <td>1</td>
      <td>1</td>
      <td>0</td>
    </tr>
    <tr>
      <td>Sample6</td>
      <td>1</td>
      <td>2</td>
      <td>0</td>
      <td>0</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>0</td>
      <td>0</td>
      <td>2</td>
    </tr>
    <tr>
      <td>Sample7</td>
      <td>2</td>
      <td>1</td>
      <td>2</td>
      <td>2</td>
      <td>2</td>
      <td>2</td>
      <td>2</td>
      <td>1</td>
      <td>1</td>
      <td>2</td>
    </tr>
    <tr>
      <td>Sample8</td>
      <td>0</td>
      <td>2</td>
      <td>1</td>
      <td>2</td>
      <td>1</td>
      <td>0</td>
      <td>2</td>
      <td>1</td>
      <td>2</td>
      <td>2</td>
    </tr>
    <tr>
      <td>Sample9</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>1</td>
      <td>1</td>
      <td>2</td>
      <td>0</td>
      <td>0</td>
      <td>1</td>
      <td>0</td>
    </tr>
    <tr>
      <td>Sample10</td>
      <td>1</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>2</td>
      <td>0</td>
      <td>1</td>
      <td>1</td>
      <td>0</td>
      <td>2</td>
    </tr>
  </tbody>
</table>

<h3 id="phenotype-matrix">Phenotype matrix</h3>

<table>
  <thead>
    <tr>
      <th>Sample</th>
      <th>protein 1</th>
      <th>protein 2</th>
      <th>protein 3</th>
      <th>protein 4</th>
      <th>protein 5</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Sample1</td>
      <td>7</td>
      <td>5</td>
      <td>1</td>
      <td>9</td>
      <td>1</td>
    </tr>
    <tr>
      <td>Sample2</td>
      <td>5</td>
      <td>4</td>
      <td>6</td>
      <td>3</td>
      <td>8</td>
    </tr>
    <tr>
      <td>Sample3</td>
      <td>8</td>
      <td>7</td>
      <td>8</td>
      <td>8</td>
      <td>7</td>
    </tr>
    <tr>
      <td>Sample4</td>
      <td>3</td>
      <td>8</td>
      <td>5</td>
      <td>3</td>
      <td>8</td>
    </tr>
    <tr>
      <td>Sample5</td>
      <td>2</td>
      <td>4</td>
      <td>9</td>
      <td>6</td>
      <td>2</td>
    </tr>
    <tr>
      <td>Sample6</td>
      <td>7</td>
      <td>2</td>
      <td>8</td>
      <td>4</td>
      <td>2</td>
    </tr>
    <tr>
      <td>Sample7</td>
      <td>2</td>
      <td>2</td>
      <td>3</td>
      <td>7</td>
      <td>2</td>
    </tr>
    <tr>
      <td>Sample8</td>
      <td>8</td>
      <td>4</td>
      <td>7</td>
      <td>2</td>
      <td>8</td>
    </tr>
    <tr>
      <td>Sample9</td>
      <td>1</td>
      <td>1</td>
      <td>7</td>
      <td>4</td>
      <td>9</td>
    </tr>
    <tr>
      <td>Sample10</td>
      <td>1</td>
      <td>3</td>
      <td>7</td>
      <td>6</td>
      <td>3</td>
    </tr>
  </tbody>
</table>

<h2 id="experiment">Experiment</h2>
<ul>
  <li>At SNP1, and protein 1, you might run a linear regression using R or <a href="https://zzz.bwh.harvard.edu/plink/index.shtml">plink</a> software, between the outcome “SNP” and the predictor “protein level”.</li>
  <li>That will result in a p-value and other stats about the association between that SNP1 genotype (predictor) and protein 1 level (outcome <em>or dependent variable</em>) for all patients in the study.</li>
  <li>It will answer the Q: <em>is this nucleotide (genotype) associated with an increase or decrease in this protein?</em>
</li>
  <li>This one test is then repeated for every SNP (500’000 times) <em>and</em> for every protein (1000 times).</li>
  <li>With plink software it may take ~2hours to complete. But most of the researcher’s time is spent in preparation by looking at the summary information about the SNPs and proteins (frequency of SNPs, level range of proteins, etc.).</li>
</ul>

<h2 id="coded-example">Coded example</h2>

<p>Here is an example using R. It has</p>
<ul>
  <li>6 patients</li>
  <li>2 SNPs (predictor)</li>
  <li>protein measurement (outcome <em>or dependent variable</em>)</li>
</ul>

<p>It runs an association test (linear regression) between the SNP (nucleotide A or T) and protein level (0.2-0.9).</p>

<h3 id="glm_loopr">glm_loop.R</h3>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span><span class="w">

</span><span class="c1"># Test each SNP 1 and 2 for status/outcome</span><span class="w">
</span><span class="c1"># status 0 = A/A nucleotide, status 1 = A/T nucleotide variant.</span><span class="w">
</span><span class="c1"># outcome = phenotype measurement (clinical feature, protein level, etc,)</span><span class="w">
</span><span class="n">SNP</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w">
</span><span class="n">status</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">
</span><span class="n">outcome</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0.2</span><span class="p">,</span><span class="w"> </span><span class="m">0.2</span><span class="p">,</span><span class="w"> </span><span class="m">0.2</span><span class="p">,</span><span class="w"> </span><span class="m">0.2</span><span class="p">,</span><span class="w"> </span><span class="m">0.9</span><span class="p">,</span><span class="w"> </span><span class="m">0.9</span><span class="p">,</span><span class="w"> </span><span class="m">0.2</span><span class="p">,</span><span class="w"> </span><span class="m">0.2</span><span class="p">,</span><span class="w"> </span><span class="m">0.2</span><span class="p">,</span><span class="w"> </span><span class="m">0.2</span><span class="p">,</span><span class="w"> </span><span class="m">0.9</span><span class="p">,</span><span class="w"> </span><span class="m">0.9</span><span class="p">)</span><span class="w">

</span><span class="c1"># make a dataframe</span><span class="w">
</span><span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">SNP</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">,</span><span class="w"> </span><span class="n">outcome</span><span class="p">,</span><span class="w"> </span><span class="n">stringsAsFactors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">

</span><span class="c1"># status is a factor rather than a number</span><span class="w">
</span><span class="n">df</span><span class="o">$</span><span class="n">status</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.factor</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">status</span><span class="p">)</span><span class="w">

</span><span class="c1"># Test association between SNP1 and outcome by linear regression</span><span class="w">
</span><span class="n">df_1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">filter</span><span class="p">(</span><span class="n">SNP</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">

</span><span class="n">linear_reg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">glm</span><span class="p">(</span><span class="n">outcome</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">status</span><span class="p">,</span><span class="w">
				 </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">df_1</span><span class="p">,</span><span class="w">
				 </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"gaussian"</span><span class="p">)</span><span class="w">

</span><span class="n">summary</span><span class="p">(</span><span class="n">linear_reg</span><span class="p">)</span><span class="w">
</span><span class="n">coef</span><span class="p">(</span><span class="n">summary</span><span class="p">(</span><span class="n">linear_reg</span><span class="p">))</span><span class="w"> </span><span class="c1"># show the main info only</span><span class="w">

</span><span class="c1"># Test association between SNP2 and outcome by linear regression</span><span class="w">
</span><span class="n">df_2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">filter</span><span class="p">(</span><span class="n">SNP</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w">

</span><span class="n">linear_reg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">glm</span><span class="p">(</span><span class="n">outcome</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">status</span><span class="p">,</span><span class="w">
				 </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">df_2</span><span class="p">,</span><span class="w">
				 </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"gaussian"</span><span class="p">)</span><span class="w">

</span><span class="n">summary</span><span class="p">(</span><span class="n">linear_reg</span><span class="p">)</span><span class="w">
</span><span class="n">coef</span><span class="p">(</span><span class="n">summary</span><span class="p">(</span><span class="n">linear_reg</span><span class="p">))</span><span class="w">

</span><span class="c1"># Instead of testing each SNP manually</span><span class="w">
</span><span class="c1"># Test SNP1 and SNP2 using a loop directly from df</span><span class="w">
</span><span class="n">coef</span><span class="p">(</span><span class="n">summary</span><span class="p">(</span><span class="n">linear_reg</span><span class="p">))</span><span class="w">

</span><span class="n">lapply</span><span class="p">(</span><span class="n">split</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="w"> </span><span class="n">df</span><span class="o">$</span><span class="n">SNP</span><span class="p">),</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w">
	</span><span class="n">coef</span><span class="p">(</span><span class="n">summary</span><span class="p">(</span><span class="w">
		</span><span class="n">glm</span><span class="p">(</span><span class="n">outcome</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">status</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"gaussian"</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x</span><span class="p">)</span><span class="w">
	</span><span class="p">)))</span><span class="w">

</span><span class="c1"># Maybe just save the P-values to know which SNPs have a significant association (Pval &lt; 0.05)</span><span class="w">
</span><span class="c1"># [,'Pr(&gt;|t|)'] shows the Pval only in row "status1".  </span><span class="w">
</span><span class="n">l</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">split</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="w"> </span><span class="n">df</span><span class="o">$</span><span class="n">SNP</span><span class="p">),</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w">
					</span><span class="n">coef</span><span class="p">(</span><span class="n">summary</span><span class="p">(</span><span class="w">
						</span><span class="n">glm</span><span class="p">(</span><span class="n">outcome</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">status</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"gaussian"</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x</span><span class="p">)</span><span class="w">
					</span><span class="p">))[,</span><span class="w"> </span><span class="s1">'Pr(&gt;|t|)'</span><span class="p">])</span><span class="w">

</span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">l</span><span class="p">)</span><span class="w"> </span><span class="c1"># X1 = SNP1. </span><span class="w">

</span></code></pre></div></div>

<h4 id="result">Result</h4>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">df</span><span class="w">
</span><span class="n">SNP</span><span class="w"> </span><span class="n">status</span><span class="w"> </span><span class="n">outcome</span><span class="w">
</span><span class="m">1</span><span class="w">    </span><span class="m">1</span><span class="w">      </span><span class="m">0</span><span class="w">     </span><span class="m">0.2</span><span class="w">
</span><span class="m">2</span><span class="w">    </span><span class="m">1</span><span class="w">      </span><span class="m">0</span><span class="w">     </span><span class="m">0.2</span><span class="w">
</span><span class="m">3</span><span class="w">    </span><span class="m">1</span><span class="w">      </span><span class="m">1</span><span class="w">     </span><span class="m">0.2</span><span class="w">
</span><span class="m">4</span><span class="w">    </span><span class="m">1</span><span class="w">      </span><span class="m">0</span><span class="w">     </span><span class="m">0.2</span><span class="w">
</span><span class="m">5</span><span class="w">    </span><span class="m">1</span><span class="w">      </span><span class="m">1</span><span class="w">     </span><span class="m">0.9</span><span class="w">
</span><span class="m">6</span><span class="w">    </span><span class="m">1</span><span class="w">      </span><span class="m">1</span><span class="w">     </span><span class="m">0.9</span><span class="w">
</span><span class="m">7</span><span class="w">    </span><span class="m">2</span><span class="w">      </span><span class="m">0</span><span class="w">     </span><span class="m">0.2</span><span class="w">
</span><span class="m">8</span><span class="w">    </span><span class="m">2</span><span class="w">      </span><span class="m">1</span><span class="w">     </span><span class="m">0.2</span><span class="w">
</span><span class="m">9</span><span class="w">    </span><span class="m">2</span><span class="w">      </span><span class="m">1</span><span class="w">     </span><span class="m">0.2</span><span class="w">
</span><span class="m">10</span><span class="w">   </span><span class="m">2</span><span class="w">      </span><span class="m">0</span><span class="w">     </span><span class="m">0.2</span><span class="w">
</span><span class="m">11</span><span class="w">   </span><span class="m">2</span><span class="w">      </span><span class="m">1</span><span class="w">     </span><span class="m">0.9</span><span class="w">
</span><span class="m">12</span><span class="w">   </span><span class="m">2</span><span class="w">      </span><span class="m">1</span><span class="w">     </span><span class="m">0.9</span><span class="w">

</span><span class="c1"># Result</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">split</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="w"> </span><span class="n">df</span><span class="o">$</span><span class="n">SNP</span><span class="p">),</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w">
</span><span class="o">+</span><span class="w"> 	</span><span class="n">coef</span><span class="p">(</span><span class="n">summary</span><span class="p">(</span><span class="w">
</span><span class="o">+</span><span class="w"> 		</span><span class="n">glm</span><span class="p">(</span><span class="n">outcome</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">status</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"gaussian"</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x</span><span class="p">)</span><span class="w">
</span><span class="o">+</span><span class="w"> 	</span><span class="p">)))</span><span class="w">
</span><span class="o">$</span><span class="n">`1`</span><span class="w">
             </span><span class="n">Estimate</span><span class="w"> </span><span class="n">Std.</span><span class="w"> </span><span class="n">Error</span><span class="w">  </span><span class="n">t</span><span class="w"> </span><span class="n">value</span><span class="w">  </span><span class="n">Pr</span><span class="p">(</span><span class="o">&gt;|</span><span class="n">t</span><span class="o">|</span><span class="p">)</span><span class="w">
</span><span class="p">(</span><span class="n">Intercept</span><span class="p">)</span><span class="w"> </span><span class="m">0.2000000</span><span class="w">  </span><span class="m">0.1649916</span><span class="w"> </span><span class="m">1.212183</span><span class="w"> </span><span class="m">0.2921436</span><span class="w">
</span><span class="n">status1</span><span class="w">     </span><span class="m">0.4666667</span><span class="w">  </span><span class="m">0.2333333</span><span class="w"> </span><span class="m">2.000000</span><span class="w"> </span><span class="m">0.1161165</span><span class="w">

</span><span class="o">$</span><span class="n">`2`</span><span class="w">
            </span><span class="n">Estimate</span><span class="w"> </span><span class="n">Std.</span><span class="w"> </span><span class="n">Error</span><span class="w">  </span><span class="n">t</span><span class="w"> </span><span class="n">value</span><span class="w">  </span><span class="n">Pr</span><span class="p">(</span><span class="o">&gt;|</span><span class="n">t</span><span class="o">|</span><span class="p">)</span><span class="w">
</span><span class="p">(</span><span class="n">Intercept</span><span class="p">)</span><span class="w">     </span><span class="m">0.20</span><span class="w">  </span><span class="m">0.2474874</span><span class="w"> </span><span class="m">0.808122</span><span class="w"> </span><span class="m">0.4643386</span><span class="w">
</span><span class="n">status1</span><span class="w">         </span><span class="m">0.35</span><span class="w">  </span><span class="m">0.3031089</span><span class="w"> </span><span class="m">1.154701</span><span class="w"> </span><span class="m">0.3125000</span><span class="w">
</span></code></pre></div></div>

<h2 id="real-world">Real world</h2>
<p>After this example, go read:</p>

<p>Uffelmann, E., Huang, Q.Q., Munung, N.S. et al. Genome-wide association studies. Nat Rev Methods Primers 1, 59 (2021).
<a href="https://doi.org/10.1038/s43586-021-00056-9">https://doi.org/10.1038/s43586-021-00056-9</a></p>

<h2 id="other-code">Other code</h2>
<p>The tables above were generated as follows:</p>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Genotype matrix</span><span class="w">
</span><span class="c1"># Create a dataframe with 10 rows and 10 columns</span><span class="w">
</span><span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">matrix</span><span class="p">(</span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">))</span><span class="w">

</span><span class="c1"># Fill the dataframe with values 0, 1, and 2</span><span class="w">
</span><span class="k">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
	</span><span class="k">for</span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
		</span><span class="n">df</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sample</span><span class="p">(</span><span class="m">0</span><span class="o">:</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">
	</span><span class="p">}</span><span class="w">
</span><span class="p">}</span><span class="w">

</span><span class="c1"># Set the column names to SNP1-SNP10</span><span class="w">
</span><span class="n">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="s2">"SNP"</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">)</span><span class="w">

</span><span class="c1"># Set the row names to Sample1-Sample10</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="s2">"Sample"</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">)</span><span class="w">

</span><span class="c1"># View the resulting dataframe</span><span class="w">
</span><span class="n">df</span><span class="w">
</span></code></pre></div></div>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Phenotype matrix</span><span class="w">
</span><span class="c1"># Create a dataframe with 10 rows and 5 columns</span><span class="w">
</span><span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">matrix</span><span class="p">(</span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">))</span><span class="w">

</span><span class="c1"># Fill the dataframe with random values between 1 and 9</span><span class="w">
</span><span class="k">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
	</span><span class="k">for</span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
		</span><span class="n">df</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sample</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">9</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">
	</span><span class="p">}</span><span class="w">
</span><span class="p">}</span><span class="w">

</span><span class="c1"># Set the column names to protein1-protein5</span><span class="w">
</span><span class="n">colnames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="s2">"protein"</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">)</span><span class="w">

</span><span class="c1"># Set the row names to Sample1-Sample10</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="s2">"Sample"</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">)</span><span class="w">

</span><span class="c1"># View the resulting dataframe</span><span class="w">
</span><span class="n">df</span><span class="w">
</span></code></pre></div></div>

</div>

<div id="related">
  <h2>Related Posts</h2>
  <ul class="posts">
    
      <li>
<span>13 Aug 2023</span> » <a href="/2023/08/13/inf_causal_metab_sem.html">Summary - inference of causal metabolite networks</a>
</li>
    
      <li>
<span>09 May 2023</span> » <a href="/2023/05/09/what_i_want_from_genome_center.html">What I want from genomic data centers</a>
</li>
    
      <li>
<span>18 Jan 2023</span> » <a href="/2023/01/18/ai_from_complex_requests_to_simple_fixes.html">AI - From Complex Requests to Simple Fixes</a>
</li>
    
  </ul>
</div>


</div>
<!-- # MAIN END ---------------------------------------------------------- -->

<!-- FOOT START -------------------------------------------------------- -->
<div class="sitewidthnav">
	<div class="footer">
		<div class="foot-container">
				<a href="/" class="logo-foot">
				<img style="margin-bottom:-16px" src="/images/logos/logo4.2_150ppi.png" alt="Logo image" width="25" height="38.5"></a>
					<div class="right-root">
						<p><a href="https://github.com/DylanLawless/DylanLawless.github.io/tree/source">Maintained</a> by Dylan Lawless. Scientist at <a href="https://people.epfl.ch/dylan.lawless">EPFL</a>.
						<a href="/resume">Resume</a>
						and <a href="https://scholar.google.com/citations?user=RPBxP1wAAAAJ&hl=en&oi=ao">Google scholar</a>.
						<a href="https://lawlessgenomics.com/feed.xml">RSS feed</a>
						</p>
					</div>
		</div>
	</div>
</div>
<!-- FOOT END ---------------------------------------------------------- -->

</div>
<!-- SITE END ----------------------------------------------------------- -->


<!-- Banner on top right corner linking to github
<a href="https://github.com/mojombo"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_red_aa0000.png" alt="Fork me on GitHub" /></a>
-->

<!-- ClickTale Bottom part -->
<div id="ClickTaleDiv" style="display: none;"></div>
<script type="text/javascript">
if(document.location.protocol!='https:')
  document.write(unescape("%3Cscript%20src='https://s.clicktale.net/WRb.js'%20type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
if(typeof ClickTale=='function') ClickTale(206,0.3,"www03");
</script>
<!-- ClickTale end of Bottom part -->

<!-- Google Analytics -->
<!--
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "https://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
var pageTracker = _gat._getTracker("UA-6016902-1");
pageTracker._trackPageview();
</script>
-->
<!-- Google Analytics end -->

</body>
</html>
