<!DOCTYPE html>
<html xmlns="https://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
	<!-- link to MathJax for LaTeX stlye eqn -->
	 <script type="text/javascript" async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-MML-AM_CHTML">
</script>

	 
	 <!-- https://realfavicongenerator.net -->
	 <link rel="apple-touch-icon" sizes="180x180" href="/images/logos/favicon_package_v0.16_bw/apple-touch-icon.png">
	 <link rel="icon" type="image/png" sizes="32x32" href="/images/logos/favicon_package_v0.16_bw/favicon-32x32.png">
	 <link rel="icon" type="image/png" sizes="16x16" href="/images/logos/favicon_package_v0.16_bw/favicon-16x16.png">
	 <link rel="manifest" href="/images/logos/favicon_package_v0.16_bw/site.webmanifest">
	 <meta name="msapplication-TileColor" content="#da532c">
	 <meta name="theme-color" content="#ffffff">
	
	 <meta name="viewport" content="width=device-width, initial-scale=1">
	
	<meta http-equiv="content-type" content="text/html; charset=utf-8">
	<meta name="description" content="A home for topics in human precision medicine, genomic analysis, and data visualisation.">
	<title>Genomic analysis intro</title>
	<meta name="author" content="LawlessGenomics">
	<link href="https://lawlessgenomics.com/feed.xml" rel="alternate" title="LawlessGenomics" type="application/atom+xml">
	<!-- <meta name="readability-verification" content="QCzSs992GxmRYRKVpPeZ6LE2tS8aYKxsSSQKV8YM"/> -->
	
	<!-- include a google font -->
	<link href="https://fonts.googleapis.com/css?family=Roboto+Mono%7CRoboto+Slab:300%7CRoboto:500" rel="stylesheet">
	<link href="https://fonts.googleapis.com/css?family=Lora" rel="stylesheet">
	
	<!-- syntax highlighting CSS -->
	<link rel="stylesheet" href="/css/syntax.css" type="text/css">
	
	<!-- Homepage CSS -->
	<link rel="stylesheet" href="/css/screen.css" type="text/css" media="screen, projection">
	
	<!-- Google font -->
	<link rel="preconnect" href="https://fonts.googleapis.com"> 
	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin> 
	<link href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,400;0,600;0,700;1,400;1,600;1,700&amp;display=swap" rel="stylesheet">
	
	<!-- Typekit -->
	<script type="text/javascript" src="https://use.typekit.com/jpd0pfm.js"></script>
	<script type="text/javascript">try{Typekit.load();}catch(e){}</script>
</head>
<body>

<!-- ClickTale Top part -->
<script type="text/javascript">
	var WRInitTime=(new Date()).getTime();
</script>
<!-- ClickTale end of Top part -->

<!-- Logo and navigation bar -->



<!-- SITE START --------------------------------------------------------- -->
<div class="site">

<!-- NAV START --------------------------------------------------------- -->
<div class="sitewidthnav">
	<div class="title">
		<div class="nav-container">
			<nav>

			<div class="logoimage">
				<a href="/">
				<img style="margin-bottom:-16px" src="/images/logos/logo4.1_150ppi.png" alt="Logo image" width="35.7" height="55"></a>
			</div>

			<div class="logo">
				<a href="/">LawlessGenomics</a>
			</div>

				<input type="checkbox" id="nav-toggle">
					<label for="nav-toggle" class="gg-menu"></label>
					<div class="right-menu">
						<a href="/resume">About</a>
						<a href="/">Home</a>
					</div>
			</nav>
		</div>
	</div>
</div>
<!-- NAV END ----------------------------------------------------------- -->

<!-- MAIN START -------------------------------------------------------- -->
<div class="sitewidth">
	<div id="topic">
Reading time 2 minutes
<h1 id="genomic-analysis-intro">Genomic analysis intro</h1>


<p>Like any data science, bioinformatics is a discipline of data
manipulation. The majority of jobs could be accomplished simply with a
method for sequence alignment and data mining using grep, sed, and awk.
However, the development of specialised genomics-based tools allows us
to standardise procedures and expand the avenues of exploration. One of
the greatest single, collaborative, sources of genomics analysis tool is
the Genome Analysis Toolkit developed by The Broad Institute.</p>

<p>While not every tool was used in this study, a synopsis of analysis
options is worthwhile; an overview of GATK provides a good example of
the current trends. The software provided by GATK includes methods for
data manipulation. As of writing, there are 291 packages in this
software suite. 
These are divided into major topics of genomic data
handling that include:</p>

<ul>
  <li>
    <p>Tools dedicated to managing read data in SAM, BAM or CRAM formats.</p>
  </li>
  <li>
    <p>Diagnostics and QC to collect sequencing quality and comparative
metrics;</p>
  </li>
  <li>
    <p>Interval manipulation to process genomic intervals in various formats.
For example, converting a BED file to a Picard interval list;</p>
  </li>
  <li>
    <p>Metagenomics. For example, microbial community composition and pathogen
detection using read filtering, microbe reference alignment, and
abundance scoring;</p>
  </li>
  <li>
    <p>Tools that manipulate FASTA format references. For example, creating a
custom capture library relies on oligonucleotide baits for hybrid
selection reactions, or making BWA-MEM index image files, or a sequence
dictionary to accompany a reference.</p>
  </li>
  <li>
    <p>Variant calling and genotyping for variants such as SNVs, SNPs, and
Indels. For example, haplotype calling of germline SNPs and indels by
performing a local re-assembly of haplotypes, such HaplotypeCaller gVCF
files are generally merged into batches of single gVCFs to manage
databases, and joint genotyping is a common approach on these databases.
Some tools also specialise in calling somatic SNVs and indels also by
local assembly of haplotypes.</p>
  </li>
  <li>
    <p>Variant manipulation software for handling variant call format (VCF)
data.</p>
  </li>
  <li>
    <p>Base calling. This is software that is used at the early stage of
sequence data interpretation to process the raw data, i.e. base calls,
and other attributes such as the adapters used.</p>
  </li>
  <li>
    <p>Read filters which can be applied by the engine to select reads for
analysis.</p>
  </li>
  <li>
    <p>Variant annotations is a software that can be used during critical
stages of analysis by other tools, i.e. HaplotypeCaller, Mutect2,
VariantAnnotator and GenotypeGVCFs.</p>
  </li>
  <li>
    <p>Copy number variant discovery using read coverage to detect copy number
variants.</p>
  </li>
  <li>
    <p>Coverage analysis using allele depths as the metric.</p>
  </li>
  <li>
    <p>Structural variant discovery.</p>
  </li>
  <li>
    <p>Variant evaluation and refinement. For example, variant calls can be
further detailed using annotations which are not offered by the base
software.</p>
  </li>
  <li>
    <p>Variant filtering that allows annotation of the FILTER column in a
dataset.</p>
  </li>
</ul>

</div>

<!--

<div id="related">
  <h2>Related Topics</h2>
  <ul class="topics">
    
  </ul>
</div>

-->

















</div>
<!-- # MAIN END ---------------------------------------------------------- -->

<!-- FOOT START -------------------------------------------------------- -->
<div class="sitewidthnav">
	<div class="footer">
		<div class="foot-container">
				<a href="/" class="logo-foot">
				<img style="margin-bottom:-16px" src="/images/logos/logo4.2_150ppi.png" alt="Logo image" width="35.7" height="55"></a>
					<div class="right-root">
						<p><a href="https://github.com/DylanLawless/DylanLawless.github.io/tree/source">Maintained</a> by Dylan Lawless. Postdoc at <a href="https://people.epfl.ch/dylan.lawless">EPFL</a>.
						<a href="/resume">Resume</a>
						and <a href="https://scholar.google.com/citations?user=RPBxP1wAAAAJ&hl=en&oi=ao">Google scholar</a>.
						<a href="https://lawlessgenomics.com/feed.xml">RSS feed</a>
						</p>
					</div>
		</div>
	</div>
</div>
<!-- FOOT END ---------------------------------------------------------- -->

</div>
<!-- SITE END ----------------------------------------------------------- -->


<!-- Banner on top right corner linking to github
<a href="https://github.com/mojombo"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_red_aa0000.png" alt="Fork me on GitHub" /></a>
-->

<!-- ClickTale Bottom part -->
<div id="ClickTaleDiv" style="display: none;"></div>
<script type="text/javascript">
if(document.location.protocol!='https:')
  document.write(unescape("%3Cscript%20src='https://s.clicktale.net/WRb.js'%20type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
if(typeof ClickTale=='function') ClickTale(206,0.3,"www03");
</script>
<!-- ClickTale end of Bottom part -->

<!-- Google Analytics -->
<!--
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "https://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
var pageTracker = _gat._getTracker("UA-6016902-1");
pageTracker._trackPageview();
</script>
-->
<!-- Google Analytics end -->

</body>
</html>
