<!DOCTYPE html>
<html xmlns="https://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
	<!-- link to MathJax for LaTeX stlye eqn -->
	 <script type="text/javascript" async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-MML-AM_CHTML">
</script>

	 
	 <!-- https://realfavicongenerator.net -->
	 <link rel="apple-touch-icon" sizes="180x180" href="/images/logos/favicon_package_v0.16_bw/apple-touch-icon.png">
	 <link rel="icon" type="image/png" sizes="32x32" href="/images/logos/favicon_package_v0.16_bw/favicon-32x32.png">
	 <link rel="icon" type="image/png" sizes="16x16" href="/images/logos/favicon_package_v0.16_bw/favicon-16x16.png">
	 <link rel="manifest" href="/images/logos/favicon_package_v0.16_bw/site.webmanifest">
	 <meta name="msapplication-TileColor" content="#da532c">
	 <meta name="theme-color" content="#ffffff">
	
	 <meta name="viewport" content="width=device-width, initial-scale=1">
	
	<meta http-equiv="content-type" content="text/html; charset=utf-8">
	<meta name="description" content="A home for topics in human precision medicine, genomic analysis, and data visualisation.">
	<title>Genomic analysis code examples</title>
	<meta name="author" content="LawlessGenomics">
	<link href="https://lawlessgenomics.com/feed.xml" rel="alternate" title="LawlessGenomics" type="application/atom+xml">
	<!-- <meta name="readability-verification" content="QCzSs992GxmRYRKVpPeZ6LE2tS8aYKxsSSQKV8YM"/> -->
	
	<!-- include a google font -->
	<link href="https://fonts.googleapis.com/css?family=Roboto+Mono%7CRoboto+Slab:300%7CRoboto:500" rel="stylesheet">
	<link href="https://fonts.googleapis.com/css?family=Lora" rel="stylesheet">
	
	<!-- syntax highlighting CSS -->
	<link rel="stylesheet" href="/css/syntax.css" type="text/css">
	
	<!-- Homepage CSS -->
	<link rel="stylesheet" href="/css/screen.css" type="text/css" media="screen, projection">
	
	<!-- Google font -->
	<link rel="preconnect" href="https://fonts.googleapis.com"> 
	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin> 
	<link href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,400;0,600;0,700;1,400;1,600;1,700&amp;display=swap" rel="stylesheet">
	
	<!-- Typekit -->
	<script type="text/javascript" src="https://use.typekit.com/jpd0pfm.js"></script>
	<script type="text/javascript">try{Typekit.load();}catch(e){}</script>
</head>
<body>

<!-- ClickTale Top part -->
<script type="text/javascript">
	var WRInitTime=(new Date()).getTime();
</script>
<!-- ClickTale end of Top part -->

<!-- Logo and navigation bar -->



<!-- SITE START --------------------------------------------------------- -->
<div class="site">

<!-- NAV START --------------------------------------------------------- -->
<div class="sitewidthnav">
	<div class="title">
		<div class="nav-container">
			<nav>

			<div class="logoimage">
				<a href="/">
				<img style="margin-bottom:-16px" src="/images/logos/logo4.2_150ppi.png" alt="Logo image" width="25" height="38.5"></a>
				<!-- alt="Logo image" width="35.7" height="55" /></a> -->
			</div>

			<div class="logo">
				<a href="/">LawlessGenomics</a>
			</div>

				<input type="checkbox" id="nav-toggle">
					<label for="nav-toggle" class="gg-menu"></label>
					<div class="right-menu">
						<a href="/resume">About</a>
						<a href="/">Home</a>
					</div>
			</nav>
		</div>
	</div>



</div>
<!-- NAV END ----------------------------------------------------------- -->

<!-- MAIN START -------------------------------------------------------- -->
<div class="sitewidth">
	<div id="topic">
30 seconds
<h1 id="genomic-analysis-code-examples">Genomic analysis code examples</h1>

<h1 id="command-line-example-code">Command line example code</h1>
<p><strong>Whole exome analysis</strong></p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c">#!/bin/bash </span>
<span class="c">############################################ </span>
<span class="c">#### The basic protocol for analysis. </span>
<span class="c">#### It is best to set up a loop that </span>
<span class="c">#### can run the protocol on all samples. </span>
<span class="c">############################################ </span>
<span class="c"># Make project organisation folders </span>
<span class="nb">mkdir</span> ~/1.fastq/ <span class="o">&amp;&amp;</span> <span class="se">\</span>
<span class="nb">mkdir</span> ~/2.trim/ <span class="o">&amp;&amp;</span> <span class="se">\</span>
<span class="nb">mkdir</span> ~/3.sort/ <span class="o">&amp;&amp;</span> <span class="se">\</span>
<span class="nb">mkdir</span> ~/4.dedup/ <span class="o">&amp;&amp;</span> <span class="se">\</span>
<span class="nb">mkdir</span> ~/5.realtar/ <span class="o">&amp;&amp;</span> <span class="se">\</span>
<span class="nb">mkdir</span> ~/6.indelrealn/ <span class="o">&amp;&amp;</span> <span class="se">\</span>
<span class="nb">mkdir</span> ~/7.baserecal/ <span class="o">&amp;&amp;</span> <span class="se">\</span>
<span class="nb">mkdir</span> ~/9.printbam/ <span class="o">&amp;&amp;</span> <span class="se">\</span>
<span class="nb">mkdir</span> ~/10.gvcf/ <span class="o">&amp;&amp;</span> <span class="se">\</span>
<span class="nb">mkdir</span> ~/geno/ <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### Typical workflow </span>
<span class="c">############################################ </span>

<span class="c">############################################ </span>
<span class="c">#### [command_cut] </span>
<span class="c">############################################ </span>
trim_galore <span class="nt">-q</span> 20 <span class="nt">-fastqc_args</span> <span class="se">\</span>
<span class="nt">-outdir</span> ~/2.trim/QC_reports <span class="nt">-illumina</span> <span class="nt">-gzip</span> <span class="se">\</span>
<span class="nt">-o</span> ~/2.trim/ <span class="nt">-length</span> 20 <span class="nt">-paired</span> <span class="se">\</span>
~/1.fastq/Sequencing_ID_L001_R1_001.fastq.gz <span class="se">\</span>
~/1.fastq/Sequencing_ID_L001_R2_001.fastq.gz <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### [command_align] </span>
<span class="c">############################################ </span>
bwa mem <span class="nt">-t</span> 12 <span class="nt">-M</span> ~/ref/human_g1k_v37.fasta <span class="se">\</span>
~/2.trim/Sequencing_ID_L001_R1_001_val_1.fq.gz <span class="se">\</span>
~/2.trim/Sequencing_ID_L001_R2_001_val_2.fq.gz <span class="se">\</span>
<span class="nt">-v</span> 1 <span class="nt">-R</span> <span class="s1">'@RG\tID:Sample_ID\tSM:Sample_ID \
tPL:ILLUMINA\tLB:Sample_ID'</span> <span class="se">\</span>
<span class="nt">-M</span> | samtools view <span class="nt">-Sb</span> - <span class="o">&gt;</span> ~/2.trim/Sample_ID.bam <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### [command_sort] </span>
<span class="c">############################################ </span>
java <span class="nt">-Xmx8g</span> <span class="nt">-jar</span> ~/picard/picard-tools-2.5.0/picard.jar <span class="se">\</span>
SortSam <span class="se">\</span>
<span class="nv">I</span><span class="o">=</span> ~/2.trim/Sample_ID.bam <span class="se">\</span>
<span class="nv">O</span><span class="o">=</span> ~/3.sort/Sample_ID.sort.bam <span class="se">\</span>
<span class="nv">SO</span><span class="o">=</span>coordinate <span class="nv">CREATE_INDEX</span><span class="o">=</span>TRUE <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### [command_dedup] </span>
<span class="c">############################################ </span>
java <span class="nt">-Xmx8g</span> <span class="nt">-jar</span> ~/picard/picard-tools-2.5.0/picard.jar <span class="se">\</span>
MarkDuplicates <span class="se">\</span>
<span class="nv">I</span><span class="o">=</span> ~/3.sort/Sample_ID.sort.bam <span class="se">\</span>
<span class="nv">O</span><span class="o">=</span> ~/4.dedup/Sample_ID.sort.dedup.bam <span class="se">\</span>
<span class="nv">M</span><span class="o">=</span> ~/4.dedup/Sample_ID.sort.dedup.metrics <span class="se">\</span>
<span class="nv">CREATE_INDEX</span><span class="o">=</span>TRUE <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### [command_realtar] </span>
<span class="c">############################################ </span>
java <span class="nt">-Xmx6g</span> <span class="nt">-jar</span> ~/GATK/GenomeAnalysisTK.jar <span class="se">\</span>
<span class="nt">-T</span> RealignerTargetCreator <span class="se">\</span>
<span class="nt">-R</span> ~/ref/human_g1k_v37.fasta <span class="se">\</span>
<span class="nt">-known</span> ~/ref/1000G_phase1.indels.b37.vcf <span class="se">\</span>
<span class="nt">-known</span> ~/ref/Mills_and_1000G_gold_standard.<span class="se">\ </span>indels.b37.sites.vcf <span class="se">\</span>
<span class="nt">-I</span> ~/4.dedup/Sample_ID.sort.dedup.bam <span class="se">\</span>
<span class="nt">-o</span> ~/5.realtar/Sample_ID.sort.dedup.bam.intervals <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### [command_indelrealign]</span>
<span class="c">############################################ </span>
java <span class="nt">-Xmx6g</span> <span class="nt">-jar</span> ~/GATK/GenomeAnalysisTK.jar <span class="se">\</span>
<span class="nt">-T</span> IndelRealigner <span class="se">\</span>
<span class="nt">-R</span> ~/ref/human_g1k_v37.fasta <span class="se">\</span>
<span class="nt">-known</span> ~/ref/1000G_phase1.indels.b37.vcf <span class="se">\</span>
<span class="nt">-known</span> ~/ref/Mills_and_1000G_gold_standard.<span class="se">\ </span>indels.b37.sites.vcf <span class="se">\</span>
<span class="nt">-I</span> ~/4.dedup/Sample_ID.sort.dedup.bam <span class="se">\</span>
<span class="nt">-targetIntervals</span> <span class="se">\</span>
~/5.realtar/Sample_ID.sort.dedup.bam.intervals <span class="se">\</span>
<span class="nt">-o</span> ~/6.indelrealn/Sample_ID.sort.dedup.indelrealn.bam <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### [command_bsqr] </span>
<span class="c">############################################ </span>
java <span class="nt">-Xmx8g</span> <span class="nt">-jar</span> ~/GATK/GenomeAnalysisTK.jar <span class="se">\</span>
<span class="nt">-T</span> BaseRecalibrator <span class="se">\</span>
<span class="nt">-R</span> ~/ref/human_g1k_v37.fasta <span class="se">\</span>
<span class="nt">-knownSites</span> ~/ref/dbSnp146.b37.vcf.gz <span class="se">\</span>
<span class="nt">-knownSites</span> ~/ref/1000G_phase1.indels.b37.vcf <span class="se">\</span>
<span class="nt">-knownSites</span> ~/ref/Mills_and_1000G_gold_standard.<span class="se">\ </span>indels.b37.sites.vcf <span class="se">\</span>
<span class="nt">-o</span> ~/7.baserecal/Sample_ID.sort.dedup.indelrealn.recal.grp <span class="se">\</span>
<span class="nt">-I</span> ~/6.indelrealn/Sample_ID.sort.dedup.indelrealn.bam <span class="se">\</span>
<span class="nt">-nct</span> 6 <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### Optional check for base recalibration </span>
<span class="c">############################################ </span>
<span class="c">####  Print reads</span>
<span class="c">############################################ </span>
java <span class="nt">-Xmx12g</span> <span class="nt">-jar</span> ~/GATK/GenomeAnalysisTK.jar <span class="se">\</span>
<span class="nt">-T</span> PrintReads <span class="se">\</span>
<span class="nt">-R</span> ~/ref/human_g1k_v37.fasta <span class="se">\</span>
<span class="nt">-I</span> ~/6.indelrealn/Sample_ID.sort.dedup.indelrealn.bam <span class="se">\</span>
<span class="nt">-BQSR</span> ~/7.baserecal/Sample_ID.sort.dedup.indelrealn.recal.grp <span class="se">\</span>
<span class="nt">-o</span> ~/9.printbam/Sample_ID.sort.dedup.indelrealn.recal.bam <span class="se">\</span>
<span class="nt">-disable_indel_quals</span> <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### [command_hc] </span>
<span class="c">############################################ </span>
java <span class="nt">-Xmx8g</span> <span class="nt">-jar</span> ~/GATK/GenomeAnalysisTK.jar <span class="se">\</span>
<span class="nt">-T</span> HaplotypeCaller <span class="nt">-emitRefConfidence</span> GVCF <span class="se">\</span>
<span class="nt">-R</span> ~/ref/human_g1k_v37.fasta <span class="se">\</span>
<span class="nt">-D</span> ~/ref/dbSnp146.b37.vcf.gz <span class="se">\</span>
<span class="nt">-stand_call_conf</span> 30 <span class="se">\</span>
<span class="nt">-stand_emit_conf</span> 10 <span class="se">\</span>
<span class="nt">-o</span> ~/10.gvcf/Sample_ID.sort.dedup.indelrealn.recal.HC.g.vcf <span class="se">\</span>
<span class="nt">-L</span> ~/ref/SureSelectAllExonV6/S07604514_Regions_b37.bed <span class="se">\</span>
<span class="nt">-ip</span> 30 <span class="o">&amp;&amp;</span> <span class="se">\</span>
<span class="c"># deprecated -I ~/9.printbam/Sample_ID.sort.dedup.indelrealn.recal.bam</span>

<span class="c">############################################ </span>
<span class="c">#### [command_joint] </span>
<span class="c">############################################ </span>
java <span class="nt">-Xmx12g</span> <span class="nt">-jar</span> ~/GATK/GenomeAnalysisTK.jar <span class="se">\</span>
<span class="nt">-T</span> GenotypeGVCFs <span class="se">\</span>
<span class="nt">-R</span> ~/ref/human_g1k_v37.fasta <span class="se">\</span>
<span class="nt">-D</span> ~/ref/dbSnp146.b37.vcf.gz <span class="nt">-stand_call_conf</span> 30 <span class="se">\</span>
<span class="nt">-stand_emit_conf</span> 10 <span class="se">\</span>
<span class="nt">-V</span> ~/10.gvcf/Sample_ID.sort.dedup.indelrealn.recal.HC.g.vcf <span class="se">\</span>
<span class="nt">-V</span> ~/10.gvcf/Sample_ID.sort.dedup.indelrealn.recal.HC.g.vcf <span class="se">\</span>
<span class="nt">-V</span> ~/10.gvcf/Sample_ID.sort.dedup.indelrealn.recal.HC.g.vcf <span class="se">\</span>
<span class="nt">-V</span> ~/10.gvcf/Sample_ID.sort.dedup.indelrealn.recal.HC.g.vcf <span class="se">\</span>
<span class="nt">-V</span> ~/10.gvcf/Sample_ID.sort.dedup.indelrealn.recal.HC.g.vcf <span class="se">\</span>
<span class="nt">-V</span> ~/10.gvcf/Sample_ID.sort.dedup.indelrealn.recal.HC.g.vcf <span class="se">\</span>
<span class="nt">-V</span> ~/10.gvcf/Sample_ID.sort.dedup.indelrealn.recal.HC.g.vcf <span class="se">\</span>
<span class="nt">-V</span> ~/10.gvcf/Sample_ID.sort.dedup.indelrealn.recal.HC.g.vcf <span class="se">\</span>
<span class="nt">-o</span> ~/geno/genotype.vcf <span class="nt">-nda</span> <span class="nt">-showFullBamList</span> <span class="nt">-nt</span> 12 <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### Hard filter selecting SNVs </span>
<span class="c">############################################ </span>
java <span class="nt">-Xmx12g</span> <span class="nt">-jar</span> ~/GATK/GenomeAnalysisTK.jar <span class="se">\</span>
<span class="nt">-T</span> SelectVariants <span class="se">\</span>
<span class="nt">-R</span> ~/ref/human_g1k_v37.fasta <span class="se">\</span>
<span class="nt">-selectType</span> SNP <span class="se">\</span>
<span class="nt">-variant</span> ~/geno/genotype.vcf <span class="se">\</span>
<span class="nt">-o</span> ~/geno/genotype.raw-snps.vcf <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### Hard filter selecting INDELs </span>
<span class="c">############################################ </span>
java <span class="nt">-Xmx12g</span> <span class="nt">-jar</span> ~/GATK/GenomeAnalysisTK.jar <span class="se">\</span>
<span class="nt">-T</span> SelectVariants <span class="se">\</span>
<span class="nt">-R</span> ~/ref/human_g1k_v37.fasta <span class="se">\</span>
<span class="nt">-variant</span> ~/geno/genotype.vcf <span class="se">\</span>
<span class="nt">-selectType</span> INDEL <span class="nt">-selectType</span> MNP <span class="se">\</span>
<span class="nt">-o</span> ~/geno/genotype.raw-indels.vcf <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### Applying hard filter for SNVs </span>
<span class="c">############################################ </span>
java <span class="nt">-Xmx8g</span> <span class="nt">-jar</span> ~/GATK/GenomeAnalysisTK.jar <span class="se">\</span>
<span class="nt">-T</span> VariantFiltration <span class="se">\</span>
<span class="nt">-R</span> ~/ref/human_g1k_v37.fasta <span class="se">\</span>
<span class="nt">-V</span> ~/geno/genotype.raw-snps.vcf <span class="se">\</span>
<span class="nt">-filterExpression</span> QD &lt; 2.0 <span class="o">||</span> FS <span class="o">&gt;</span> 60.0 <span class="o">||</span> MQ &lt; 40.0 <span class="o">||</span><span class="se">\</span>
MappingQualityRankSum &lt; <span class="nt">-12</span>.5 <span class="o">||</span> ReadPosRankSum &lt; <span class="nt">-8</span>.0 <span class="se">\</span>
<span class="nt">-filterName</span> snp_hard_filter <span class="se">\</span>
<span class="nt">-o</span> ~/geno/genotype.raw-snps.filtered.snvs.vcf <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### Applying hard filter for INDELs </span>
<span class="c">############################################ </span>
java <span class="nt">-Xmx8g</span> <span class="nt">-jar</span> ~/GATK/GenomeAnalysisTK.jar <span class="se">\</span>
<span class="nt">-T</span> VariantFiltration <span class="se">\</span>
<span class="nt">-R</span> ~/ref/human_g1k_v37.fasta <span class="se">\</span>
<span class="nt">-V</span> ~/geno/genotype.raw-indels.vcf <span class="se">\</span>
<span class="nt">-filterExpression</span> QD &lt; 2.0 <span class="o">||</span> FS <span class="o">&gt;</span> 200.0 <span class="o">||</span><span class="se">\</span>
ReadPosRankSum &lt; <span class="nt">-20</span>.0 <span class="se">\</span>
<span class="nt">-filterName</span> indel_hard_filter <span class="se">\</span>
<span class="nt">-o</span> ~/geno/genotype.raw-indels.filtered.indels.vcf <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### Combine filtered results </span>
<span class="c">############################################ </span>
java <span class="nt">-Xmx8g</span> <span class="nt">-jar</span> ~/GATK/GenomeAnalysisTK.jar <span class="se">\</span>
<span class="nt">-T</span> CombineVariants <span class="nt">-R</span> ~/ref/human_g1k_v37.fasta <span class="se">\</span>
<span class="nt">-variant</span> ~/geno/genotype.raw-snps.filtered.snvs.vcf <span class="se">\</span>
<span class="nt">-variant</span> ~/geno/genotype.raw-indels.filtered.indels.vcf <span class="se">\</span>
<span class="nt">-o</span> ~/geno/genotype.fltd-combinedvars.vcf <span class="se">\</span>
<span class="nt">-genotypemergeoption</span> UNSORTED <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### Filter variants in EdbSNP &gt;/= 1% \</span>
<span class="c">#### and not listed as pathogenic by ClinVar </span>
<span class="c">############################################ </span>
perl ~/vcfhacks-v0.2.0/annotateSnps.pl <span class="se">\</span>
<span class="nt">-d</span> ~/ref/dbSnp146.b37.vcf.gz ~/ref/clinvar_20160531.vcf.gz <span class="se">\</span>
<span class="nt">-f</span> 1 <span class="nt">-pathogenic</span> <span class="se">\</span>
<span class="nt">-i</span> ~/geno/genotype.fltd-combinedvars.vcf <span class="se">\</span>
<span class="nt">-o</span> ~/geno/genotype.fltd-combinedvars.1pcdbsnp.vcf <span class="se">\</span>
<span class="nt">-t</span> 12 <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### Filter variants in EVS greater &gt;/= 1% </span>
<span class="c">############################################ </span>
perl ~/vcfhacks-v0.2.0/filterOnEvsMaf.pl <span class="nt">-d</span> ~/ref/evs/ <span class="se">\</span>
<span class="nt">-f</span> 1 <span class="nt">-progress</span> <span class="se">\</span>
<span class="nt">-i</span> ~/geno/genotype.fltd-combinedvars.1pcdbsnp.vcf <span class="se">\</span>
<span class="nt">-o</span> ~/geno/genotype.fltd-combinedvars.1pcdbsnp.1pcEVS.vcf <span class="se">\</span>
<span class="nt">-t</span> 12 <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### Exac filter for population frequency </span>
<span class="c">############################################ </span>
perl ~/vcfhacks-v0.2.0/filterVcfOnVcf.pl <span class="se">\</span>
<span class="nt">-i</span> ~/geno/genotype.fltd-combinedvars.1pcdbsnp.1pcEVS.vcf <span class="se">\</span>
<span class="nt">-f</span> ~/ref/ExAC/ExAC.r0.3.sites.vep.vcf.gz <span class="se">\</span>
<span class="nt">-o</span> ~/geno/genotype.fltd-combinedvars.1pcdbsnp.1pcEVS.exac.vcf <span class="se">\</span>
<span class="nt">-w</span> <span class="nt">-y</span> 0.01 <span class="se">\</span>
<span class="nt">-b</span> <span class="se">\</span>
<span class="c"># progress bar \</span>
<span class="nt">-t</span>  <span class="o">&amp;&amp;</span> <span class="se">\</span>
<span class="c"># number of threads</span>

<span class="c">############################################ </span>
<span class="c">#### Annotate with variant effect predictor </span>
<span class="c">############################################ </span>
perl ~/variant_effect_predictor/variant_effect_predictor.pl <span class="se">\</span>
<span class="nt">-offline</span> <span class="nt">-vcf</span> <span class="nt">-everything</span> <span class="se">\</span>
<span class="nt">-dir_cache</span> ~/variant_effect_predictor/vep_cache <span class="se">\</span>
<span class="nt">-dir_plugins</span> ~/variant_effect_predictor/vep_cache/Plugins <span class="se">\</span>
<span class="nt">-plugin</span> Condel,<span class="se">\</span>
~/variant_effect_predictor/vep_cache/Plugins/config/Condel/config/ <span class="se">\</span>
<span class="nt">-plugin</span> ExAC,~/ref/ExAC/ExAC.r0.3.sites.vep.vcf.gz <span class="se">\</span>
<span class="nt">-plugin</span> SpliceConsensus <span class="se">\</span>
<span class="nt">-fasta</span> <span class="se">\</span>
~/variant_effect_predictor/fasta/<span class="se">\</span>
Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz <span class="se">\</span>
<span class="nt">-i</span> ~/geno/genotype.fltd-combinedvars.1pcdbsnp.1pcEVS.exac.vcf <span class="se">\</span>
<span class="nt">-o</span> ~/geno/genotype.fltd-combinedvars.1pcdbsnp.1pcEVS.exac.vep.vcf <span class="se">\</span>
<span class="nt">-fork</span> 12 <span class="o">&amp;&amp;</span> <span class="se">\</span>

<span class="c">############################################ </span>
<span class="c">#### Confirm samples names </span>
<span class="c">############################################ </span>
perl ~/vcfhacks-v0.2.0/getSampleNames.pl <span class="se">\</span>
<span class="nt">-i</span> ~/geno/genotype.fltd-combinedvars.1pcdbsnp.1pcEVS.exac.vep.vcf <span class="o">&amp;&amp;</span> <span class="se">\</span>


<span class="c">############################################ </span>
<span class="c">## Data extraction </span>
<span class="c">############################################ </span>
<span class="c">#### Extract columns </span>
<span class="c">############################################ </span>
<span class="c">#### A list of files for</span>
<span class="c">#### where the data from</span>
<span class="c">#### one column is compiled into</span>
<span class="c">#### a master table </span>
<span class="c">############################################ </span>

<span class="c">############################################ </span>
<span class="c">#### This takes column 3 from every file</span>
<span class="c">#### and appends to the output file.</span>
<span class="c">#### Space delimited. </span>
<span class="c">############################################ </span>
<span class="nb">awk </span><span class="nv">FNR</span><span class="o">==</span>1<span class="o">{</span>f++<span class="o">}{</span>a[f,FNR]<span class="o">=</span><span class="nv">$3</span><span class="o">}</span>END<span class="o">{</span><span class="k">for</span><span class="o">(</span><span class="nv">x</span><span class="o">=</span>1<span class="p">;</span>x&lt;<span class="o">=</span>FNR<span class="p">;</span>x++<span class="o">)</span><span class="se">\ </span>
<span class="o">{</span><span class="k">for</span><span class="o">(</span><span class="nv">y</span><span class="o">=</span>1<span class="p">;</span>y&lt;ARGC<span class="p">;</span>y++<span class="o">)</span><span class="nb">printf</span><span class="o">(</span>%s ,a[y,x]<span class="o">)</span><span class="p">;</span>print <span class="o">}}</span> <span class="se">\</span>
./../<span class="k">*</span>/pheno.txt <span class="o">&gt;</span> master.txt

<span class="c">############################################ </span>
<span class="c">#### The spacer method can be changed; tab, space, comma, etc.</span>
<span class="c">#### Another way to convert later is with the following command.</span>
<span class="c">#### [The tab character (after s/) must be removed</span>
<span class="c">#### and printed to the command line using ctrl+v then tab.] </span>
<span class="c">############################################ </span>
<span class="nb">sed</span> <span class="s1">'s/ /,/g'</span> input.tsv <span class="o">&gt;</span> output.csv

<span class="c">############################################ </span>
<span class="c">#### Candidate filter</span>
<span class="c">############################################ </span>
<span class="c">#### Filter a VCF on a candidate gene list. </span>
<span class="c">############################################ </span>
<span class="c">#### List format as X:1-2000,</span>
<span class="c">#### or -b for a bed file or</span>
<span class="c">#### a list file with 1 per line. </span>
<span class="c">############################################ </span>
<span class="k">for </span>f <span class="k">in</span> ~/immune.panel/vep/<span class="k">*</span>.vcf <span class="k">do 
</span>perl ~/vcfhacks-v0.2.0/filterVcfOnLocation.pl <span class="se">\</span>
<span class="nt">-i</span> ~/immune.panel/vep/<span class="nv">$f</span> <span class="se">\</span>
<span class="nt">-b</span> ~/deep.panel.bed <span class="se">\</span>
<span class="nt">-o</span> ~/immune.panel/filter/<span class="si">$(</span> <span class="nb">basename</span> <span class="nt">-s</span> .vcf <span class="nv">$f</span> <span class="si">)</span>.panel.vcf <span class="se">\</span>
<span class="k">done</span>

<span class="c">############################################ </span>
<span class="c">#### Post-routine analysis candidate filtering.</span>
<span class="c">#### Similar filtering can be done without going back</span>
<span class="c">#### to analysis stages to create a virtual panel. </span>
<span class="c">############################################ </span>
<span class="c">#### Export all gene names and give the count.</span>
<span class="nb">sort </span>list.txt | <span class="se">\</span>
<span class="nb">uniq</span> <span class="nt">-c</span> <span class="o">&gt;</span> InflammatoryDisorderCohortHitCount.txt
<span class="c">#### Format to csv.</span>
<span class="c">#### Cross against a master list of immune genes.</span>

<span class="c">############################################ </span>
<span class="c">#### In R, import data </span>
<span class="c">############################################ </span>
master &lt;- read.csv<span class="o">(</span>./master.csv, <span class="nv">stringsAsFactors</span><span class="o">=</span>FALSE<span class="o">)</span>
InflammatoryDisorderCohortHitCount &lt;- <span class="se">\</span>
read.csv<span class="o">(</span>./ InflammatoryDisorderCohortHitCount.csv, <span class="nv">stringsAsFactors</span><span class="o">=</span>FALSE<span class="o">)</span>

<span class="c">############################################ </span>
<span class="c">#### Merge the master immune gene list</span>
<span class="c">#### with the Inflammatory disorder cohort hits. </span>
<span class="c">############################################ </span>
combine &lt;- merge<span class="o">(</span>master, InflammatoryDisorderCohortHitCount, 
        by <span class="o">=</span> Gene, all <span class="o">=</span> TRUE<span class="o">)</span>

<span class="c">############################################ </span>
<span class="c">#### Remove the genes that happen to overlap</span>
<span class="c">#### gene of interest and remove anything from</span>
<span class="c">#### the master list that is not in the cohort list. </span>
<span class="c">############################################ </span>
clean &lt;- na.omit<span class="o">(</span>combine<span class="o">)</span>

<span class="c">############################################ </span>
<span class="c">#### Write out the table. </span>
<span class="c">############################################ </span>
write.csv<span class="o">(</span>clean, ./GenesOfInterest.csv, row.names <span class="o">=</span> FALSE<span class="o">)</span>

<span class="c">############################################ </span>
<span class="c">#### The output can be sorted as of interest</span>
<span class="c">#### e.g. autosomal dominant autoinflammatory gene. </span>
<span class="c">############################################ </span>

<span class="c">############################################ </span>
<span class="c">#### Tailored filtering</span>
<span class="c">############################################ </span>
<span class="c">#### Filter on sample.</span>
<span class="c">#### May need use a -freq option</span>
<span class="c">#### to account for index hopping.</span>
<span class="c">#### Filter on sample removes anything shared</span>
<span class="c">#### with cases (-s) that are not listed but not others (-x). </span>
<span class="c">############################################ </span>
perl /home/vcfhacks-v0.2.0/filterOnSample.pl <span class="se">\</span>
<span class="nt">-i</span> ~/samples.vep.vcf <span class="se">\</span>
<span class="nt">-s</span> <span class="k">case</span>.1 <span class="k">case</span>.2 <span class="k">case</span>.3 <span class="nt">-x</span> <span class="se">\</span>
<span class="nt">-o</span> ~/samples.getFunctionalVariantsVep.vcf

<span class="c">############################################ </span>
<span class="c">#### Get variants. </span>
<span class="c">############################################ </span>
<span class="c">#### Getting functional variants. The -n flag allows</span>
<span class="c">#### selections only when &gt;2 samples</span>
<span class="c">#### have variants in a shared gene. </span>
perl /home/vcfhacks-v0.2.0/getFunctionalVariants.pl <span class="se">\</span>
<span class="nt">-s</span> <span class="k">case</span>.1 <span class="k">case</span>.2 <span class="k">case</span>.3 <span class="se">\</span>
<span class="nt">-i</span> ~/samples.vep.vcf <span class="se">\</span>
<span class="nt">-f</span> <span class="nt">-n</span> 2 <span class="se">\</span>
<span class="nt">-o</span> ~/samples.getFunctionalVariantsVep.SharedGenes.vcf

<span class="c">#### Candidate compound heterozygous.</span>
<span class="c">#### Only variants that are common in ALL -s are considered.</span>
<span class="c">#### Flag -n specifies the number of cases</span>
<span class="c">#### required to return a genotype. </span>
perl /home/vcfhacks-v0.2.0/findBiallelic.pl <span class="se">\</span>
<span class="nt">-i</span> ~/samples.vep.vcf <span class="se">\</span>
<span class="nt">-s</span> <span class="k">case</span>.1 <span class="k">case</span>.2 <span class="k">case</span>.3 <span class="se">\</span>
<span class="nt">-n</span> 1 <span class="se">\</span>
<span class="nt">-o</span> ~/samples.findBiallelic.all.vcf

<span class="c">############################################ </span>
<span class="c">#### Rank, annontate, and simplify </span>
<span class="c">############################################ </span>
perl /home/vcfhacks-v0.2.0/rankOnCaddScore.pl <span class="se">\</span>
<span class="nt">-c</span> /data/shared/cadd/v1.3/<span class="k">*</span>.gz <span class="se">\</span>
<span class="nt">-i</span> ~/samples.getFunctionalVariantsVep.SharedGenes.vcf <span class="se">\</span>
<span class="nt">-o</span> ~/samples.getFunctionalVariantsVep.SharedGenes.cadd.ranked.vcf <span class="se">\</span>
<span class="nt">-progress</span>

perl /home/vcfhacks-v0.2.0/geneAnnotator.pl <span class="se">\</span>
<span class="nt">-d</span> /home/vcfhacks-v0.2.0/data/geneAnnotatorDb <span class="se">\</span>
<span class="nt">-i</span> ~/samples.findBiallelic.all.vcf <span class="se">\</span>
<span class="nt">-o</span> ~/samples.findBiallelic.all.gene.anno

perl /home/vcfhacks-v0.2.0/annovcfToSimple.pl <span class="se">\</span>
<span class="nt">-i</span> ~/samples.findBiallelic.all.gene.anno <span class="se">\</span>
<span class="nt">-vep</span> <span class="nt">-gene_anno</span> <span class="se">\</span>
<span class="nt">-canonical_only</span> <span class="se">\</span>
<span class="nt">-u</span> <span class="nt">-contains_variant</span> <span class="se">\</span>
<span class="nt">-o</span> ~/samples.findBiallelic.all.gene.anno.simple.xlsx

</code></pre></div></div>

</div>

<!--

<div id="related">
  <h2>Related Topics</h2>
  <ul class="topics">
    
  </ul>
</div>

-->

















</div>
<!-- # MAIN END ---------------------------------------------------------- -->

<!-- FOOT START -------------------------------------------------------- -->
<div class="sitewidthnav">
	<div class="footer">
		<div class="foot-container">
				<a href="/" class="logo-foot">
				<img style="margin-bottom:-16px" src="/images/logos/logo4.2_150ppi.png" alt="Logo image" width="25" height="38.5"></a>
					<div class="right-root">
						<p><a href="https://github.com/DylanLawless/DylanLawless.github.io/tree/source">Maintained</a> by Dylan Lawless. Scientist at <a href="https://people.epfl.ch/dylan.lawless">EPFL</a>.
						<a href="/resume">Resume</a>
						and <a href="https://scholar.google.com/citations?user=RPBxP1wAAAAJ&hl=en&oi=ao">Google scholar</a>.
						<a href="https://lawlessgenomics.com/feed.xml">RSS feed</a>
						</p>
					</div>
		</div>
	</div>
</div>
<!-- FOOT END ---------------------------------------------------------- -->

</div>
<!-- SITE END ----------------------------------------------------------- -->


<!-- Banner on top right corner linking to github
<a href="https://github.com/mojombo"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_red_aa0000.png" alt="Fork me on GitHub" /></a>
-->

<!-- ClickTale Bottom part -->
<div id="ClickTaleDiv" style="display: none;"></div>
<script type="text/javascript">
if(document.location.protocol!='https:')
  document.write(unescape("%3Cscript%20src='https://s.clicktale.net/WRb.js'%20type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
if(typeof ClickTale=='function') ClickTale(206,0.3,"www03");
</script>
<!-- ClickTale end of Bottom part -->

<!-- Google Analytics -->
<!--
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "https://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
var pageTracker = _gat._getTracker("UA-6016902-1");
pageTracker._trackPageview();
</script>
-->
<!-- Google Analytics end -->

</body>
</html>
