<!DOCTYPE html>
<html xmlns="https://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
	<!-- link to MathJax for LaTeX stlye eqn -->
	 <script type="text/javascript" async
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-MML-AM_CHTML">
</script>

	 
	 <!-- https://realfavicongenerator.net -->
	 <link rel="apple-touch-icon" sizes="180x180" href="/images/logos/favicon_package_v0.16_bw/apple-touch-icon.png">
	 <link rel="icon" type="image/png" sizes="32x32" href="/images/logos/favicon_package_v0.16_bw/favicon-32x32.png">
	 <link rel="icon" type="image/png" sizes="16x16" href="/images/logos/favicon_package_v0.16_bw/favicon-16x16.png">
	 <link rel="manifest" href="/images/logos/favicon_package_v0.16_bw/site.webmanifest">
	 <meta name="msapplication-TileColor" content="#da532c">
	 <meta name="theme-color" content="#ffffff">
	
	 <meta name="viewport" content="width=device-width, initial-scale=1">
	
	<meta http-equiv="content-type" content="text/html; charset=utf-8" />
	<title>Burden testing with variant collapse</title>
	<meta name="author" content="LawlessGenomics" />
	<!-- <link href="https://feeds.feedburner.com/tom-preston-werner" rel="alternate" title="Dyaln feed title" type="application/atom+xml" /> -->
	<!-- <meta name="readability-verification" content="QCzSs992GxmRYRKVpPeZ6LE2tS8aYKxsSSQKV8YM"/> -->
	
	<!-- include a google font -->
	<link href="https://fonts.googleapis.com/css?family=Roboto+Mono|Roboto+Slab:300|Roboto:500" rel="stylesheet">
	<link href="https://fonts.googleapis.com/css?family=Lora" rel="stylesheet">
	
	<!-- syntax highlighting CSS -->
	<link rel="stylesheet" href="/css/syntax.css" type="text/css" />
	
	<!-- Homepage CSS -->
	<link rel="stylesheet" href="/css/screen.css" type="text/css" media="screen, projection" />
	
	<!-- Typekit -->
	<script type="text/javascript" src="https://use.typekit.com/jpd0pfm.js"></script>
	<script type="text/javascript">try{Typekit.load();}catch(e){}</script>
</head>
<body>

<!-- ClickTale Top part -->
<script type="text/javascript">
	var WRInitTime=(new Date()).getTime();
</script>
<!-- ClickTale end of Top part -->

<!-- Logo and navigation bar -->
<div class="site">
	<div class="title">
		<div class="nav-container">
			<nav>

			<div class="logoimage">
				<a href="/">
				<img style="margin-bottom:-16px" 
				src="/images/logos/logo4.1_150ppi.png" 
				alt="Logo image" width="35.7" height="55" /></a>
			</div>

			<div class="logo">
				<a href="/">LawlessGenomics</a>
			</div>

				<input type="checkbox" id="nav-toggle">
					<label for="nav-toggle" class="gg-menu"></label>
					<div class="right-menu">
						<a href="/resume">About</a>
						<a href="/">Home</a>
					</div>
			</nav>
		</div>
	</div>

	<div id="topic">
<h1 id="burden-testing-with-variant-collapse">Burden testing with variant collapse</h1>
<ul id="markdown-toc">
  <li><a href="#burden-testing-with-variant-collapse" id="markdown-toc-burden-testing-with-variant-collapse">Burden testing with variant collapse</a></li>
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a>    <ul>
      <li><a href="#to-do" id="markdown-toc-to-do">To do</a></li>
    </ul>
  </li>
  <li><a href="#main-papers-in-order" id="markdown-toc-main-papers-in-order">Main papers in order</a>    <ul>
      <li><a href="#major-classes-of-tests" id="markdown-toc-major-classes-of-tests">Major Classes of Tests</a></li>
      <li><a href="#burden-tests-so-far" id="markdown-toc-burden-tests-so-far">Burden Tests So Far</a></li>
    </ul>
  </li>
  <li><a href="#li-and-leal-2008" id="markdown-toc-li-and-leal-2008">Li and Leal 2008</a>    <ul>
      <li><a href="#background" id="markdown-toc-background">Background</a></li>
      <li><a href="#genetic-model" id="markdown-toc-genetic-model">Genetic model</a></li>
      <li><a href="#single-marker-test" id="markdown-toc-single-marker-test">Single-marker test</a></li>
      <li><a href="#multiple-marker-test" id="markdown-toc-multiple-marker-test">Multiple-marker test</a></li>
      <li><a href="#basic-collapsing-method" id="markdown-toc-basic-collapsing-method">Basic collapsing method</a></li>
      <li><a href="#cmc-method" id="markdown-toc-cmc-method">CMC method</a></li>
      <li><a href="#power-of-cmc-method" id="markdown-toc-power-of-cmc-method">Power of CMC method</a></li>
      <li><a href="#misclassification" id="markdown-toc-misclassification">Misclassification</a></li>
      <li><a href="#effects-of-linkage-disequilibrium-ld" id="markdown-toc-effects-of-linkage-disequilibrium-ld">Effects of Linkage Disequilibrium (LD)</a></li>
      <li><a href="#evaluation-of-type-i-error-rate" id="markdown-toc-evaluation-of-type-i-error-rate">Evaluation of Type I Error Rate</a></li>
      <li><a href="#problems-with-this-paper" id="markdown-toc-problems-with-this-paper">Problems with this paper</a></li>
    </ul>
  </li>
  <li><a href="#madsen-and-browning-2009" id="markdown-toc-madsen-and-browning-2009">Madsen and Browning 2009</a>    <ul>
      <li><a href="#background-1" id="markdown-toc-background-1">Background</a></li>
      <li><a href="#main-theory" id="markdown-toc-main-theory">Main theory</a></li>
      <li><a href="#methods" id="markdown-toc-methods">Methods</a></li>
      <li><a href="#encode-data" id="markdown-toc-encode-data">ENCODE data</a></li>
      <li><a href="#results" id="markdown-toc-results">Results</a></li>
      <li><a href="#discussion" id="markdown-toc-discussion">Discussion</a></li>
      <li><a href="#summary" id="markdown-toc-summary">Summary</a></li>
    </ul>
  </li>
  <li><a href="#price-2010" id="markdown-toc-price-2010">Price 2010</a>    <ul>
      <li><a href="#methods-1" id="markdown-toc-methods-1">Methods</a></li>
      <li><a href="#results-and-discussion" id="markdown-toc-results-and-discussion">Results and Discussion</a></li>
    </ul>
  </li>
  <li><a href="#neale-2011" id="markdown-toc-neale-2011">Neale 2011</a>    <ul>
      <li><a href="#c-alpha-tests" id="markdown-toc-c-alpha-tests">C-alpha tests</a></li>
    </ul>
  </li>
  <li><a href="#skat" id="markdown-toc-skat">SKAT</a>    <ul>
      <li><a href="#introduction-1" id="markdown-toc-introduction-1">Introduction</a></li>
      <li><a href="#methods-2" id="markdown-toc-methods-2">Methods</a></li>
      <li><a href="#skat-is-a-generalization-of-the-c-alpha-test" id="markdown-toc-skat-is-a-generalization-of-the-c-alpha-test">SKAT Is a Generalization of the C-Alpha Test</a></li>
    </ul>
  </li>
  <li><a href="#lee-2012" id="markdown-toc-lee-2012">Lee 2012</a>    <ul>
      <li><a href="#introduction-2" id="markdown-toc-introduction-2">Introduction</a></li>
      <li><a href="#methods-3" id="markdown-toc-methods-3">Methods</a></li>
    </ul>
  </li>
  <li><a href="#protein-pathway-analysis" id="markdown-toc-protein-pathway-analysis">Protein pathway analysis</a>    <ul>
      <li><a href="#zhang-2021-ashg" id="markdown-toc-zhang-2021-ashg">Zhang 2021 ASHG</a></li>
      <li><a href="#problems-with-this-paper-1" id="markdown-toc-problems-with-this-paper-1">Problems with this paper</a></li>
    </ul>
  </li>
  <li><a href="#itan--casanova-contributions" id="markdown-toc-itan--casanova-contributions">Itan &amp; Casanova contributions</a>    <ul>
      <li><a href="#historical-review" id="markdown-toc-historical-review">Historical review</a></li>
    </ul>
  </li>
  <li><a href="#burden-test-power-calculation" id="markdown-toc-burden-test-power-calculation">Burden test power calculation</a></li>
  <li><a href="#references" id="markdown-toc-references">References</a></li>
</ul>

<h1 id="introduction">Introduction</h1>

<p>In this series I will go over methods and theory for gene burden testing.
I am publishing this live as I work since there are several heavy papers to get through.
<!--For the working writeup see: pri_net_manuscript/pre_print_md/detailed_methods/statistics.md--></p>

<h2 id="to-do">To do</h2>

<p>Running code on test data: Remake the gVCF as BP_RESOLUTION to list a site for every position. 
Filter: If variants are MAF &gt;1.1 in gnomAD then run a filter to remove variants that are not sequenced in controls.
Justification: If we ran that filter for every singleton it would remove the very rare variants that would have been a true positive if it had been sequenced in the controls.</p>

<h1 id="main-papers-in-order">Main papers in order</h1>

<ul>
  <li>Methods for detecting associations with rare variants for common diseases: application to analysis of sequence data
<a class="citation" href="#li2008methods">(Li &amp; Leal, 2008)</a>.</li>
  <li>A groupwise association test for rare mutations using a weighted sum statistic.
(missing reference).</li>
  <li>An evaluation of statistical approaches to rare variant analysis in genetic association studies.
<a class="citation" href="#morris2010evaluation">(Morris &amp; Zeggini, 2010)</a>.</li>
  <li>Pooled association tests for rare variants in exon-resequencing studies.
<a class="citation" href="#price2010pooled">(Price et al., 2010)</a>.</li>
  <li>Testing for an unusual distribution of rare variants.
(missing reference).</li>
  <li>Rare-Variant Association Testing for Sequencing Data with the Sequence Kernel Association Test.
(missing reference).</li>
  <li>Optimal tests for rare variant effects in sequencing association studies.
<a class="citation" href="#Lee2012Optimal">(Lee et al., 2012)</a>.</li>
  <li>Optimal Unified Approach for Rare-Variant Association Testing with Application to Small-Sample Case-Control Whole-Exome Sequencing Studies.
<a class="citation" href="#Lee2012Optimalunified">(Lee et al., 2012)</a>.</li>
  <li>Sequence Kernel Association Tests for the Combined Effect of Rare and Common Variants.
<a class="citation" href="#IonitaLaza2013Sequence">(Ionita-Laza et al., 2013)</a></li>
</ul>

<h2 id="major-classes-of-tests">Major Classes of Tests</h2>

<ul>
  <li>Burden/Collapsing tests</li>
  <li>Supervised/Adaptive Burden/Collapsing tests</li>
  <li>Variance component (similarity) based tests</li>
  <li>Omnibus tests: hedge against difference scenarios</li>
</ul>

<h2 id="burden-tests-so-far">Burden Tests So Far</h2>

<h3 id="tests">Tests</h3>

<ul>
  <li>Binary Collapsing: CAST</li>
  <li>CMC</li>
  <li>Count Collapsing: MZ (GRANVIL)</li>
  <li>Weighted Sum Test</li>
</ul>

<h3 id="power-of-burden-tests-depends-on">Power of burden tests depends on</h3>

<ul>
  <li>Number of associated variants</li>
  <li>Number of non-associated variants</li>
  <li>Direction of the effects.</li>
  <li>Powerful if most variants are causal and have effects in the same direction.</li>
</ul>

<h1 id="li-and-leal-2008">Li and Leal 2008</h1>
<ul>
  <li>
    <p>Methods for detecting associations with rare variants for common diseases: application to analysis of sequence data
<a class="citation" href="#li2008methods">(Li &amp; Leal, 2008)</a></p>
  </li>
  <li>The collapsing method, which involves collapsing genotypes across variants and applying a univariate test, is powerful for analyzing rare variants.</li>
  <li>The multivariate analysis is robust against inclusion of non-causal variants.</li>
  <li>Both methods are superior to analyzing each variant individually with univariate tests.</li>
  <li>Combined Multivariate and Collapsing (CMC) method unifies the advantages of both collapsing and multiple-marker tests.</li>
</ul>

<h2 id="background">Background</h2>

<p>Genotype phenotype association may be tested by:</p>

<ul>
  <li>A single market test (i.e. GWAS) may use standard univariate statistical tests
(e.g., chi-squared test, Fisher’s exact test, or Cochran Armitage test for trend) 
and with the family-wise error rate (FWER) controlled by a multiple-comparison correction (e.g., Bonferroni, permutation).</li>
  <li>A multiple-marker test,
    <ul>
      <li>which tests multiple variant sites simultaneously with the use of multivariate methods,</li>
      <li>such as the Fisher product method,</li>
      <li>Hotelling’s T test 
or logistic regression.</li>
    </ul>
  </li>
  <li>Collapsing variants can increase power.</li>
  <li>Classification of variants can be a major problem.</li>
  <li>If classified correct power of causal detection is higher than single- and multiple-marker testing.</li>
  <li>If miss-classified, non-functional variants could introduce error.</li>
</ul>

<h2 id="genetic-model">Genetic model</h2>

<ul>
  <li>In a locus M variants can independently cause disease.</li>
  <li>A = allele</li>
  <li>1-M = allele label</li>
  <li>p = allele frequency</li>
  <li>Gk = 0/1/2 genotype</li>
  <li>Fki = penertance of geno for ith variant.</li>
  <li>Relative risk of each variant i in models: additive, multiplicative, dominant, and recessive.</li>
  <li>Prevalence of disease caused by each individual variant is calculated.</li>
  <li>Prevalence of disease caused by entire locus can therefore be calculated.</li>
  <li>Total prevalence can be approximated by the sum of the individual prev.</li>
  <li>Same pheno can be due to different causal variants.</li>
  <li>The proportion of cases as a result of variant i can be ascertained.</li>
  <li>These cases are members of group i.</li>
  <li>There is a total of M groups.</li>
  <li>For group i the frequency of variant i can also be produced for 0,1,2.</li>
  <li>Then, the expected freq of genotype G at variant i across all M groups of cases can be found.</li>
  <li>Controls expected genotype freq at i variant can be found.</li>
  <li>The expected freqs are used to calculate the power to detect assoc in this study.</li>
  <li>Their focus is the omnibus test, which provides an association test of the entire locus and is not focused on any specific variant within the locus.</li>
</ul>

<h2 id="single-marker-test">Single-marker test</h2>

<ul>
  <li>One approach of association studies is to test each variant site individually with the use of a univariate test and assess the significance of the <em>omnibus test</em> (chi-squared test or F test) after correction for multiple comparisons.</li>
  <li>For univariate tests, a contingency table can be constructed to compare genotype frequencies at each variant site in cases and controls.</li>
  <li>Because an observation of individuals that are homozygous for the high-risk rare allele is extremely rare, AA genotypes are collapsed with Aa genotypes, and a table is constructed.</li>
  <li>For an equal number of cases and controls, the classical Pearson chi-squared statistic for testing equal genotype frequencies in cases and controls is used.</li>
</ul>

<h2 id="multiple-marker-test">Multiple-marker test</h2>

<p>Another option is to test all variants simultaneously by multivariate test</p>

<ul>
  <li>e.g. Fisher product method, Hotelling’s T-squared test, or multiple logistic regression.</li>
  <li>(This is not the same as one regression run for every single variant with multiple covariates, as done in GWAS sometimes).</li>
  <li>The authors give an example using Xiong et. al.,  (2002). Generalized T2 test for genome association studies. Am. J. Hum. Genet.</li>
  <li><em>Note: Hotellings T-squared tests the differences between the (multivariate) means of different populations, where tests for univariate problems would make use of a t-test. The distribution is named for Harold Hotelling, who developed it as a generalization of Student’s t-distribution.</em></li>
  <li>Rejecting the null hypothesis would indicate that at least one of the variants is assocciated with disease.</li>
</ul>

<h2 id="basic-collapsing-method">Basic collapsing method</h2>

<p>Since variants are expected to be very rare, 
genotypes across all variants are collapsed.
An individual is coded once if they have any one variant for the locus.</p>

<ul>
  <li>1 = rare variants present.</li>
  <li>0 = no rare variants.
The classic Pearson chi-squared statistic can be used to test the summed proportion of variants in cases vs controls. 
The power can be calculated as shown by the authors.</li>
</ul>

<h2 id="cmc-method">CMC method</h2>

<p>Unified method that combines collapsing and multivariate testing.</p>

<ul>
  <li>Markers are divided into subgroups (e.g. allele frequency).</li>
  <li>Within group, markers are collapsed.</li>
  <li>A multivariate test is then used (Hotelling’s T-squared).</li>
</ul>

<p>e.g. M markers in a locus are split into K groups.
There are n markers in group g, and so on for each group.
The multivariate test is done where for each group, the  individuals are coded with either 1 (a carrier of one or more variants) or 0 (wild-type).</p>

<h2 id="power-of-cmc-method">Power of CMC method</h2>

<p>Results:</p>

<ul>
  <li>Freq below 0.01 collapsed.</li>
  <li>Freq above 0.01 not collapsed.</li>
  <li>With misclassification, power is much better with CMC method compared to collapsing method
    <ul>
      <li>particularly when with a high frequency of the non-causal variant.</li>
    </ul>
  </li>
  <li>Slight (very slight I think) loss of power when <em>causal</em> variants are high frequency compared to collapsing.
    <ul>
      <li>This scenario is unlikely otherwise single-marker test may be possible to detect.</li>
    </ul>
  </li>
</ul>

<h2 id="misclassification">Misclassification</h2>

<p>Two types of miss-classifications are considered:</p>

<ul>
  <li>inclusion of non-functional variants and</li>
  <li>exclusion of functional variants.</li>
  <li>There is a large section on results of <em>Excluding Functional Variants</em> and <em>Inclusion of Nonfunctional Variants</em>.</li>
</ul>

<p><em>This section can be read for detail but is probably familiar enough for us to skip.</em></p>

<h2 id="effects-of-linkage-disequilibrium-ld">Effects of Linkage Disequilibrium (LD)</h2>

<p>A simulation is done in this section:</p>

<ul>
  <li>The locus has six variants, with a total allele frequency of 0.05.</li>
  <li>Four of the variants have an allele frequency of 0.01 and are on different haplotypes.</li>
  <li>Each of the remaining two variants, with allele frequencies of 0.005, is on one of the haplotypes where a variant with allele frequency of 0.01 resides;
    <ul>
      <li>there is complete LD between these variants (r 2 z0:5).</li>
    </ul>
  </li>
  <li>For comparison purposes, a second simulation was carried out, in which all variants were on separate haplotypes.</li>
  <li>One thousand replicates were generated, and the power was evaluated for an a level of 0.001.</li>
</ul>

<p>Results:</p>

<ul>
  <li>In the presence of LD, the power for the single-marker test, Hotelling’s T2 test, and the collapsing method is
    <ul>
      <li>0.075, 0.63, and 0.85, respectively.</li>
    </ul>
  </li>
  <li>Data generated with each variant on a separate haplotype, the corresponding powers are
    <ul>
      <li>0.011, 0.451, and 0.737, respectively.</li>
    </ul>
  </li>
</ul>

<h2 id="evaluation-of-type-i-error-rate">Evaluation of Type I Error Rate</h2>

<ul>
  <li>Simulation was used to generate data under the null hypothesis of no association between variants and disease status.</li>
  <li>This process was repeated for 5000 replicates.</li>
  <li>It was then evaluated whether or not each replicate had a p value % 0.05.</li>
  <li>The type I error rate was estimated by the proportion of replicates with a p value below 0.05.</li>
  <li>A type I error rate above 0.05 signifies a higher false-positive rate, and conversely, a type I error rate below 0.05 indicates a conservative test.</li>
</ul>

<p>Result:</p>

<ul>
  <li>Hotelling’s T2 test and the collapsing method:
    <ul>
      <li><em>well controlled and slightly conservative.</em></li>
    </ul>
  </li>
  <li>Logistic regression for multiple-marker test and the likelihood-ratio test performed on the basis of an asymptotic chi-squared distribution:
    <ul>
      <li><em>not well controlled.</em></li>
      <li><em>The inflation increases with decreasing allele frequencies.</em></li>
    </ul>
  </li>
  <li>CMC method, with multivariate Hotelling’s T2 test or logistic regression:
    <ul>
      <li><em>well controlled</em>.</li>
    </ul>
  </li>
</ul>

<h2 id="problems-with-this-paper">Problems with this paper</h2>

<ul>
  <li>The main problem with collapsing method is when single individuals have more than one causal variant being only counted as “1”. This more likely to occur for our protein-pathway collapse than a single gene/locus collapse.</li>
  <li>Reason 1: dosage is not counted. Someone may have homozygous and only counts for 1 rather than 2 (or some other weight).</li>
  <li>Reason 2: someone with more than one variant is only counted as 1 (if it is in the same allele frequency group for collapse). This may be rare in a single gene but is more likely in a protein network collapse. Usually it would be OK particularly if the second variant is in LD with the first and therefore <em>should</em> only be counted once.</li>
  <li>CMC code - maf threshold: In the R code, those variants with minor allele frequency below the specified maf threshold are collapsed into a single super variant. Can we group data into several frequency groups or is one <em>above/below</em> threshold only possible?</li>
</ul>

<h1 id="madsen-and-browning-2009">Madsen and Browning 2009</h1>

<ul>
  <li>A groupwise association test for rare mutations using a weighted sum statistic
<a class="citation" href="#madsen2009groupwise">(Madsen &amp; Browning, 2009)</a></li>
</ul>

<h2 id="background-1">Background</h2>

<ul>
  <li>
    <p>CAST: Cohort allelic sums test - an existing grouping method in which the number of individuals with one or more mutations in a group (e.g. gene) is compared between affected and unaffected individuals.</p>
  </li>
  <li>Cites three papers:
    <ul>
      <li>[Cohen JC, Kiss RS, Pertsemlidis A, Marcel YL, McPherson R, et al. (2004) Multiple rare alleles contribute to low plasma levels of HDL cholesterol. Science 305: 869–872.]</li>
      <li>[Li and Leal 2008]</li>
      <li>[Morgenthaler S, Thilly WG (2007) A strategy to discover genes that carry multi- allelic or mono-allelic risk for common diseases: A cohort allelic sums test (CAST). Mutation Research/Fundamental and Molecular Mechanisms of Mutagenesis 615: 28–56.]</li>
    </ul>
  </li>
  <li>
    <p>CMC: Combined Multivariate and Collapsing - the alternative method for grouping variants.
All rare variants are collapsed, 
as in the CAST method, 
and the collapsed variants are treated as a single common variant which is analysed together with the other common variants using multivariate analysis. 
In the CMC version used in [Li and Leal], 
rare variants are defined as those having a minor allele frequency (MAF) of at most 1%.</p>
  </li>
  <li>
    <p>In this study, they presume a group of candidate functional variants.</p>
  </li>
  <li>Propose a weighted-sum method in which mutations are
    <ul>
      <li>grouped according to function (e.g. gene),</li>
      <li>and each individual is scored by a weighted sum of the mutation counts.</li>
    </ul>
  </li>
  <li>To test for an excess of mutations in affected individuals, they use permutation of disease status among affected and unaffected individuals.
    <ul>
      <li>Using permutation adjusts for the weighting of the mutations and the requirement that a mutation must be observed to be included in the study.</li>
    </ul>
  </li>
</ul>

<p><em>Note that permutation of disease status results in correct type I error even in the presence of LD [Cheverud 2001, Churchill 1994], 
although relatively low LD is expected between rare variants [Li and Leal 2008, Pritchard 2001,2001].</em></p>

<h2 id="main-theory">Main theory</h2>

<ul>
  <li>
    <p>The weighted-sum method deviates from the CAST method [5,27]:
weighting the variants differently when determining the genetic load of an individual. 
By weighting the signals from each mutation, 
the weighted sum method accentuates mutations that
are rare in the unaffected individuals, 
so that the test is not completely dominated by common mutations.</p>
  </li>
  <li>In the CAST method,
    <ul>
      <li>common variants have a high impact on the group signal,</li>
      <li>and if many common mutations are present in a group, 
almost all individuals will have one or more mutations.</li>
      <li>To avoid this effect a threshold on the mutation-frequencies may be used,
as suggested in the CMC method [Li and Leal 2008].</li>
      <li>May be difficulty to select biological meaningful threshold, affecting test outcome.</li>
    </ul>
  </li>
  <li>In the weighted-sum method,
    <ul>
      <li>include mutations of all frequencies,</li>
      <li>but mutations are weighted according to their frequency in the unaffected individuals.</li>
    </ul>
  </li>
</ul>

<h2 id="methods">Methods</h2>
<h3 id="weighted-sum-method">Weighted-Sum Method</h3>
<p>Compares the number of mutations in a group of variants between case/control. 
Designed to identify an excess of mutations in case, compared to controls. 
Each variant belongs to a group (gene, pathway, etc.) and, 
for a group with L variants, the method is comprised of the following steps:</p>

<ul>
  <li>(A)
    <ul>
      <li>For each variant, choose ALT/minor allele</li>
      <li>define number in cases/controls.</li>
      <li>calculate the weight for a variant.</li>
      <li>It is used to down-weight mutation counts in constructing the weighted-sum score; see (B) and (C) below</li>
    </ul>
  </li>
  <li>Eqn 1: Weight
    <ul>
      <li>qi = ( ((no. mut allele in control) + 1)/(2(no. controls) +2) )</li>
      <li>Wi = root (total samples x qi(1-qi))</li>
      <li>For 10 case (7 mut), 10 control (5 mut),
weight for variant is 0.447</li>
    </ul>
  </li>
  <li>Mutation freq of controls is used so that true signal in cases is not deflated by using to total number.</li>
  <li>A drawback when variance of scaled mut-freq, hence loss of power when mutation freq is high.</li>
  <li>
    <p>The <em>one</em> and <em>two</em> to the numerator and denominator, respectively, avoids zero estimates - used on next steps.</p>
  </li>
  <li>(B)
    <ul>
      <li>Genetic score of each individual calculated.</li>
      <li>j - Genetic score of individual.</li>
      <li>Iij - number of mutations for variant i in individual j.</li>
      <li>Generic model uses 0/1/2.</li>
      <li>Recessive model only homozygous are assigned 1.</li>
      <li>Dominant model both het and homo are assigned 1.</li>
    </ul>
  </li>
  <li>(C)
    <ul>
      <li>All sample combined are ranked according to genetic scores.</li>
      <li>The sum of ranks for cases is calculated.</li>
      <li>Under null-hypothesis, normal distribution of random variables using a ranking procedure equivalent to that in Wilcoxon test.</li>
    </ul>
  </li>
  <li>(D)
    <ul>
      <li>Case/control status is permuted</li>
      <li>steps (A)-(C) repeated k times to sampled X1,..Xk.</li>
    </ul>
  </li>
  <li>(E)
    <ul>
      <li>The average and sample standard deviation of X1,…,Xk are calculated and standardized score-sum is found (z).</li>
      <li>Under the null hypothesis, this has an approximately standard normal distribution.</li>
      <li>Thus, a p-value for association test can be obtained by comparing z in the quantiles of standard normal.</li>
    </ul>
  </li>
  <li>
    <p>An alternative p-value calculation is also provided using a standard permutation with a stopping rule (slighly slower for their power similations). That also gives uniform (0,1) distributed p-value, which might be useful for further analysis.</p>
  </li>
  <li>Permutation of case/control labels maintains the LD structure.</li>
</ul>

<h3 id="power-simulations">Power simulations</h3>
<p>Comparison of:</p>

<ul>
  <li>Weighted-sum method,</li>
  <li>CAST,</li>
  <li>CMC,</li>
  <li>and variant-by-variant methods,</li>
</ul>

<p>For each set of parameters,</p>

<ul>
  <li>100 datasets are simulated,</li>
  <li>the four methods are applied,</li>
  <li>and the proportions of significant outcomes used as the power estimates.</li>
</ul>

<p>To mimic a genome wide study of about 20,000 fairly independent human genes, 
we calculate a p-value for each gene, 
and use a significance threshold of 0.05/20000 = 2.561026 in all power simulations.</p>

<p>Genetic Models:</p>

<ul>
  <li>Recessive, Additive and Dominant models:
    <ul>
      <li>the disease-related variants act independently,</li>
    </ul>
  </li>
  <li>Recessive-Set model:
    <ul>
      <li>outcome of a mutation at one variant depends on the presence of a mutation at another variant.</li>
    </ul>
  </li>
</ul>

<p>Frequency spectra:</p>

<ul>
  <li>Wright’s formula used to sample the unaffected population frequency spectrum of the mutations at each variant
for the Recessive, Additive and Dominant models. 
e.g. <em>The mutation probability is calculated such that the probability (pM) that a haplotype contains at least one disease-risk mutation is fixed in unaffected individuals</em>.</li>
</ul>

<p>A method for sampling is provided and ignored in this summary.</p>

<p>A method is provided to compare models with damaging variants and benign variants.</p>

<p>NB. Tested Variants.</p>

<ul>
  <li>The mutation probabilities (p) can be very low for some of the sampled variants.
This means that some variants contain no mutations in any of the sampled individuals, and these variants are hence omitted in the tests.
    <h2 id="encode-data">ENCODE data</h2>
    <p>The method was tested on rare variants with the frequency-spectrum of a naturally occurring population using the ENCODE resequencing data.</p>
  </li>
</ul>

<h2 id="results">Results</h2>

<h3 id="proportion-of-variants-containing-mutations">Proportion of Variants Containing Mutations</h3>
<p>The mutation frequencies are sampled according to Wright’s formula, 
and hence mutations are very rare for some variants. 
Using 1000 affected and 1000 unaffected individuals, 
mutations are on average observed at only 49.4% of the variants (sd: 4.9%). 
This level is in concordance with the level from human resequencing studies.</p>

<p>Skipping the section: Power under Varying Model Parameters.</p>

<h2 id="discussion">Discussion</h2>

<p>Analysis of pathways can be done in two different ways. 
One way is to use the pathway as a group, and run the test on the entire pathway. 
On the other hand, for large pathways, it may be beneficial to use a method that allows a gene with a strong signal to have a high impact on the combined pathway test-statistic (T). 
If a pathway contains G non-overlapping genes, a method to do this is to use the weighted-sum method on each gene, and combine the resulting p-values (p1,…,pG) with the Fisher product test statistic.</p>

<h2 id="summary">Summary</h2>

<ul>
  <li>Weight based on frequency of variat in the controls.</li>
  <li>Sum of ranks in cases.</li>
  <li>Case/control labels permuted n times, mean, SD, standardised score-sum found (z).</li>
  <li>p-value for association test obtained by comparing z in the quantiles of standard normal.</li>
  <li>Alternative p-value using a standard permutation with a stopping rule gives uniform (0,1) distributed p-value, which might be useful for further analysis.</li>
</ul>

<h1 id="price-2010">Price 2010</h1>

<ul>
  <li>Pooled association tests for rare variants in exon-resequencing studies.
<a class="citation" href="#price2010pooled">(Price et al., 2010)</a></li>
  <li>The approach is based on the regression of
    <ul>
      <li>phenotypic values</li>
      <li>on individuals’ genotype scores</li>
      <li>subject to a variable allele-frequency threshold,</li>
      <li>incorporating computational predictions of the functional effects of missense variants.</li>
    </ul>
  </li>
  <li>Statistical significance is assessed by permutation testing with variable thresholds.</li>
</ul>

<h2 id="methods-1">Methods</h2>

<h3 id="simulation-framework">Simulation Framework</h3>
<p>Simplified summary:</p>

<ul>
  <li>Model of European ancestry to recapitulate the site-frequency spectrum of non-synonymous human SNPs.</li>
  <li>9kb for 10K individuals.</li>
  <li>Missense only.</li>
  <li>Phenotypes
    <ul>
      <li>quantitative normal distribution</li>
      <li>carrier and non-carrier same variance</li>
      <li>carriers shifted by \(\delta\) SD</li>
    </ul>
  </li>
</ul>

<h3 id="weighted-approaches-correspond-to-implicit-assumptions-about-log-odds-ratios">Weighted Approaches Correspond to Implicit Assumptions about Log Odds Ratios</h3>
<p>(missing reference)
	- Based on the frequency in controls.
	- Relationship between log odds ratio and allele frequency the same as Madsen 2009.</p>

<h3 id="fixed-threshold-approach">Fixed-Threshold Approach</h3>
<p>(missing reference)</p>

<ul>
  <li>i indexes SNPs,</li>
  <li>Ci is the reference allele count of SNP i in cases,</li>
  <li>and xi (\(\xi_i\)) is equal to 1 if the frequency of SNP i is below a specified threshold (1% or 5%)</li>
  <li>
    <p>and is equal to 0 otherwise.</p>
  </li>
  <li>They generalise this for quantitative phenotypes.
    <ul>
      <li>add the term: pj (\(\pi_j\)) - the phenotype of sample j.
  -
I need to compare these two papers here in more detail 
(genetic score section).
(missing reference)</li>
      <li>Generic model uses 0/1/2.</li>
      <li>Recessive model only homozygous are assigned 1.</li>
      <li>Dominant model both het and homo are assigned 1.</li>
    </ul>
  </li>
</ul>

<h3 id="weighted-approach">Weighted Approach</h3>
<p>(missing reference)</p>
<ul>
  <li>Also generalise by adding \(\pi_j\) for a quantitative phenotype value.</li>
</ul>

<h3 id="variable-threshold-approach">Variable-Threshold Approach</h3>
<ul>
  <li>There exists some (unknown) threshold T for which variants with a minor allele frequency (MAF) below T are substantially more likely to be functional than are variants with an MAF above T.</li>
  <li>z-score of a regression across samples of phenotypes versus counts of mutations meeting the allele-frequency threshold T.</li>
</ul>

<h3 id="cheating-approach-to-incorporating-varphip-phip">Cheating Approach to Incorporating \(\varphi(p)\) (“phi(p)”)</h3>
<ul>
  <li>Weight variants according to the probability \(\varphi(p)\)
    <ul>
      <li>that an allele of frequency p is functional,</li>
      <li>as inferred by using the same simulated data used to evaluate power.</li>
    </ul>
  </li>
</ul>

<h3 id="incorporation-of-computational-predictions-of-functional-effects">Incorporation of Computational Predictions of Functional Effects</h3>
<ul>
  <li>Tested if incorporation of PolyPhen-2 scores improves statistical test.</li>
  <li>To asses, simulated PolyPhen-2 predictions of damaging and neutral mutations.</li>
</ul>

<h3 id="application-to-empirical-data-sets">Application to Empirical Data Sets</h3>
<p>Applied to some real data.</p>

<h2 id="results-and-discussion">Results and Discussion</h2>
<ul>
  <li>Skipping this.</li>
  <li>The methods are an improvement.
(missing reference)</li>
  <li>and additionaly adds a method for the quantitative variable.</li>
</ul>

<h1 id="neale-2011">Neale 2011</h1>

<ul>
  <li>Testing for an unusual distribution of rare variants.
(missing reference)</li>
</ul>

<h2 id="c-alpha-tests">C-alpha tests</h2>
<ul>
  <li>They propose here the C-alpha test statistic as a novel approach for testing for
    <ul>
      <li>the presence of this mixture of effects across a set of rare variants.</li>
    </ul>
  </li>
  <li>Unlike existing burden tests, C-alpha, by testing the variance rather than the mean, maintains consistent power when the target set contains both risk and protective variants.</li>
</ul>

<h1 id="skat">SKAT</h1>

<p>Rare-Variant Association Testing for Sequencing Data with the Sequence Kernel Association Test
(missing reference)</p>

<h2 id="introduction-1">Introduction</h2>

<ul>
  <li>They propose the sequence kernel association test (SKAT),
    <ul>
      <li>regression method to test for</li>
      <li>association between genetic variants (common and rare)</li>
      <li>in a region</li>
      <li>and a continuous or dichotomous trait</li>
      <li>while easily adjusting for covariates.</li>
    </ul>
  </li>
  <li>As a score-based variance-component test, SKAT can
    <ul>
      <li>quickly calculate p values analytically by</li>
      <li>fitting the null model containing only the covariates,</li>
      <li>and so can easily be applied to genome-wide data.</li>
    </ul>
  </li>
  <li>A limitation for previous burden tests is that they implicitly assume that 
all rare variants influence the phenotype in the same direction 
and with the same magnitude of effect (after incorporating known weights).</li>
  <li>However, one would expect most variants (common or rare) within a sequenced region 
to have little or no effect on phenotype,</li>
  <li>whereas some variants are protective and others deleterious,</li>
  <li>and the magnitude of each variant’s effect is likely to vary (e.g., rarer variants might have larger effects).</li>
  <li>Collapsing across all variants is likely to introduce substantial noise.</li>
  <li>Burden tests require either specification of thresholds for collapsing</li>
  <li>or the use of permutation to estimate the threshold.</li>
</ul>

<h2 id="methods-2">Methods</h2>

<h3 id="sequencing-kernel-association-test">Sequencing Kernel Association Test</h3>
<p>*SKAT is a supervised test 
	- for the joint effects of multiple variants 
	- in a region 
	- on a phenotype. 
	-</p>
<ul>
  <li>For each region, SKAT analytically calculates a p value for association while adjusting for covariates.</li>
  <li>Adjustments for multiple comparisons are necessary for analyzing multiple regions,</li>
  <li>for example with the Bonferroni correction or FDR control.</li>
</ul>

<p>Notation:</p>

<ul>
  <li>n subjects</li>
  <li>p variants</li>
  <li>covariates</li>
  <li>\(y_i\) phenotype for sample i</li>
  <li>\(X_i\) covariats</li>
  <li>\(G_i\) genotypes (0,1,2 general)</li>
</ul>

<h3 id="skat-model-and-test-for-linear-snp-effects">SKAT Model and Test for Linear SNP Effects</h3>
<p>Eqn 1 and 2:</p>

<ul>
  <li>linear regression and</li>
  <li>logistic regression</li>
</ul>

<p>Eqn 3:</p>

<ul>
  <li>
    <p>variance-component score statistic</p>
  </li>
  <li>The weight is adjusted based on the MAF
    <ul>
      <li>if rarer variants are expected to be more likely to have larger effects.</li>
      <li>still putting decent nonzero weights for variants with MAF 1%–5%.</li>
    </ul>
  </li>
  <li>A special case of SKAT arises when the outcome is dichotomous, no covariates are included, and all \(W_j\) = 1.</li>
  <li>Under these conditions, we show in Appendix A that the SKAT test statistic Q is equivalent to the C-alpha test statistic T.</li>
</ul>

<h3 id="relationship-between-linear-skat-and-individual-variant-test-statistics">Relationship between Linear SKAT and Individual Variant Test Statistics</h3>
<ul>
  <li>One needs to fit the null model only a single time to be able to compute the Sj for all individual variants j as well as all regions to be tested.</li>
  <li>Similarly, if multiple regions are under consideration, then the same mb0 can be used to compute the SKAT Q statistics for each region.</li>
</ul>

<h3 id="accommodating-epistatic-effects-and-prior-information-under-the-skat">Accommodating Epistatic Effects and Prior Information under the SKAT</h3>
<p>Ability to model the epistatic effects of sequence variants on the phenotype within the flexible kernel machine regression framework.</p>

<p>They replace Gi’b by a more flexible function f(Gi) in the linear and logistic models (1) and (2) where f(Gi) allows for rare variant by rare variant and common variant by rare-variant interactions</p>

<h2 id="skat-is-a-generalization-of-the-c-alpha-test">SKAT Is a Generalization of the C-Alpha Test</h2>

<ul>
  <li>The recently proposed the C-alpha test has advantages over burden tests in that
    <ul>
      <li>it explicitly models the possibility that minor alleles can be deleterious or protective.</li>
    </ul>
  </li>
  <li>However, it does not currently allow for the analysis of quantitative outcomes or the inclusion of covariates and p value calculation requires permutation.</li>
  <li>They demonstrate that for a dichotomous trait in the absence of covariates,
    <ul>
      <li>the C-alpha test statistic is equivalent to the SKAT statistic with unweighted linear kernel,</li>
      <li>which is the same as the kernel machine test in Wu et al.</li>
    </ul>
  </li>
</ul>

<h1 id="lee-2012">Lee 2012</h1>
<h2 id="introduction-2">Introduction</h2>
<p>SKAT-O</p>

<ul>
  <li>
    <p>Optimal tests for rare variant effects in sequencing association studies.
<a class="citation" href="#Lee2012Optimal">(Lee et al., 2012)</a></p>
  </li>
  <li>
    <p>A class of tests that include burden tests and SKAT as special cases, and derive an optimal test within this class that maximizes power.</p>
  </li>
</ul>

<h2 id="methods-3">Methods</h2>
<ul>
  <li>They show the SKAT method</li>
  <li>They show the buden testing methods</li>
  <li>The show a method for:
    <ul>
      <li>\(Q_p\) = (1 - \(p\))\(Q_{SKAT}\) + \(pQ_{burden}\)</li>
      <li>The resulting optimal test corresponds to a best linear combination of SKAT and burden tests that maximizes power.</li>
    </ul>
  </li>
</ul>

<h1 id="protein-pathway-analysis">Protein pathway analysis</h1>
<h2 id="zhang-2021-ashg">Zhang 2021 ASHG</h2>

<p>Zhang et al. provide and excellent example of the same problem that we are
assessing. 
<a class="citation" href="#zhang2021computational">(Zhang et al., 2021)</a>
<a href="https://www.sciencedirect.com/science/article/pii/S0002929721001543?dgcid=author">A computational approach for detecting physiological homogeneity in the midst of genetic heterogeneity</a></p>

<p>I have never had a paper so similar to a project that we are currently working on.
Basically the only difference is:</p>

<ol>
  <li>the cluster size method (theirs might work better, using total edge weights instead of protein count),</li>
  <li>cluster separation by NHC instead of MCL,</li>
  <li>using KEGG+REACTOME to additionally define the description of the cluster (but this can be done at the end).</li>
  <li>The method in main analysis is carrier count instead of SKAT style (we have compared both).</li>
</ol>

<p>Same cohort size. Really nice to see it in ASHG. Will have to do a comparison to per-gene SKAT-O to show the improvement, as they do.</p>

<h2 id="problems-with-this-paper-1">Problems with this paper</h2>

<h3 id="cluster-method">Cluster method</h3>
<p>I surprised to see them apply their own novel clustering method (NHC) without some formal methods paper first - so many already exist, including 10 that can be used with their String data. 
In theory it sound good but there is no real evidence.</p>

<h3 id="carrier-counts-instead-of-allele-dosage">Carrier counts instead of allele dosage</h3>
<p>Their main analysis is:</p>

<ul>
  <li>(a) with PC-adj = glm(Pheno ~ Carrier-status + PCs, family=binomial).
    <ul>
      <li>The information for carrier status is missing, presumably 0/1 resulting in a sort of burden test.</li>
    </ul>
  </li>
  <li>(b) without PC-adj = fisher exact test.
    <ul>
      <li>When doing this we got great results with for our earlier tests but thought it might be too simplistic.</li>
    </ul>
  </li>
  <li>Using pathway SKAT-O (accounts for frequency/dosage) they got poorer results.</li>
</ul>

<p>Main problems here:</p>

<p>One count per gene (as data$CARRIER in analysis) presumes that any variant (homozygous or heterozygous, compound het, etc) is deleterious in the individual.
In restricted to LoF then OK, but missense variants are included in the study (and will usually be the majority of candidates).
Filters are applied to get a high quality list, but there will still be a majority of missense VUS.</p>

<p>In the section “Application to an HSE cohort (V): 
Comparison with a burden test on the HSE cohort”, SKAT-O is not a burden test; 
while their method is actually more similar to a burden test since they are labelling an individual if they carry a qualifying variant in a gene. 
This is just a terminology error but it indicates that maybe the key differences in pathway testing have not been considered, as written up here in our review.</p>

<p>One could argue that only counting once per gene could help prevent problems due to LD.
Although, the WES/WGS data would have the phasing information included to prune out LD if that was the argument.
Basically, I think this is just a simplification rather than an advantage. 
If SKAT had happened to work better they would have used those results.</p>

<p>e.g. In Fig 3, TLR3 variants = 7 cases, 1 Hom, 9 Het.
This seems to amount to a carrier status of 7 in their dataset for the glm and fisher exact.</p>

<p>Their main analysis:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># call R command to run pc-adj enrichment</span>
ro.r<span class="o">(</span><span class="s2">"data &lt;- read.table('temp_pc.txt', header=T, sep='</span><span class="se">\t</span><span class="s2">')"</span><span class="o">)</span>
ro.r<span class="o">(</span><span class="s2">"fit &lt;- glm(data</span><span class="nv">$PHENOTYPE</span><span class="s2"> ~ data</span><span class="nv">$CARRIER</span><span class="s2">+data</span><span class="nv">$PC1</span><span class="s2">+data</span><span class="nv">$PC2</span><span class="s2">+data</span><span class="nv">$PC3</span><span class="s2">, family='binomial')"</span><span class="o">)</span>
ro.r<span class="o">(</span><span class="s2">"adjusted.pval &lt;- anova(fit, test='LRT')[2, 5]"</span><span class="o">)</span>
r_pvalue <span class="o">=</span> ro.r<span class="o">(</span><span class="s2">"adjusted.pval"</span><span class="o">)</span>
pvalue <span class="o">=</span> r_pvalue[0]
pvalue <span class="o">=</span> float<span class="o">(</span><span class="s1">'%.3E'</span> % Decimal<span class="o">(</span>pvalue<span class="o">))</span>
file_out.write<span class="o">(</span>str<span class="o">(</span>pvalue<span class="o">)</span> + <span class="s1">'\t'</span><span class="o">)</span>

<span class="c"># pathway enrichment</span>
this_case_pathway_hit <span class="o">=</span> dict<span class="o">()</span>
<span class="k">for </span>each_pathway <span class="k">in </span>pathway_gene_set_dict.keys<span class="o">()</span>:
	pathway_gene_set <span class="o">=</span> pathway_gene_set_dict[each_pathway]
	case_pathway_overlap <span class="o">=</span> this_case_gene_set &amp; pathway_gene_set
	case_in <span class="o">=</span> len<span class="o">(</span>case_pathway_overlap<span class="o">)</span>
	case_out <span class="o">=</span> len<span class="o">(</span>this_case_gene_set<span class="o">)</span> - case_in
	pathway_in <span class="o">=</span> len<span class="o">(</span>pathway_gene_set<span class="o">)</span>
	pathway_out <span class="o">=</span> len<span class="o">(</span>pathway_genes<span class="o">)</span> - pathway_in
	<span class="k">if </span>case_in <span class="o">!=</span> 0:
		odd, pvalue <span class="o">=</span> stats.fisher_exact<span class="o">([[</span>case_in, case_out], 
		<span class="o">[</span>pathway_in, pathway_out]], <span class="nv">alternative</span><span class="o">=</span><span class="s1">'two-sided'</span><span class="o">)</span>
		adj_pvalue <span class="o">=</span> pvalue <span class="k">*</span> pathway_size
		<span class="k">if </span>adj_pvalue &lt; 0.00001:
			adj_pvalue <span class="o">=</span> float<span class="o">(</span><span class="s1">'%.3E'</span> % Decimal<span class="o">(</span>adj_pvalue<span class="o">))</span>
			this_case_pathway_hit[each_pathway] <span class="o">=</span> adj_pvalue

<span class="c"># They do not show any code with the SKAT parameters, presumably defaults using R.</span>
</code></pre></div></div>

<h1 id="itan--casanova-contributions">Itan &amp; Casanova contributions</h1>
<h2 id="historical-review">Historical review</h2>

<p>We are tackling this problem of protein pathway analysis from the viewpoint of 
rare immune disease and infection.
Historically, several topics in bioinformatic and functional analysis have 
been required before we could achieve our current position of 
statistically-robust genetic discovery for rare disease:</p>

<ol>
  <li>Candidate variant select for individual genomes</li>
  <li>Compiling reliable cohorts of patients with shared phenotypes</li>
  <li>Protein-protein interactions</li>
  <li>Variant collapse</li>
  <li>Protein pathway analysis</li>
  <li>Functional validation</li>
</ol>

<p>Therefore, a historical review of the timeline is beneficial to illustrate the 
technical successes that allow us to reliably produce candidate variants by
genome sequencing and to validate statistically-driven results by <em>“traditional”</em>
functional validation. 
These steps [1, 2 and 6 in our list] are exemplified by the following 
historical review. 
The complete list of steps 1-6 are touched on, 
but full validation of each step is the culmination of what we are currently 
working on and will be explicitly reviewed when we have completed our study.</p>

<p>As one of the leaders in this field, Casanova lab has provided a lot of insider
history to the story in a great twitter thread
<a href="https://twitter.com/casanova_lab/status/1397539593608695808">https://twitter.com/casanova_lab/status/1397539593608695808</a>.
The literature for discussion is first listed here to facilitate downloading 
but sources are referenced as usual throughout.</p>

<hr />

<p>2013 PNAS. The human gene connectome as a map of short cuts for morbid allele discovery.
<a class="citation" href="#itan2013human">(Itan et al., 2013)</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/23509278/">https://pubmed.ncbi.nlm.nih.gov/23509278/</a></p>

<p>2014 BMC Gen. HGCS: an online tool for prioritizing disease-causing gene variants by biological distance.
<a class="citation" href="#itan2014hgcs">(Itan et al., 2014)</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/24694260/">https://pubmed.ncbi.nlm.nih.gov/24694260/</a></p>

<p>2015 Front. Novel primary immunodeficiency candidate genes predicted by the human gene connectome.
<a class="citation" href="#itan2015novel">(Itan &amp; Casanova, 2015)</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/25883595/">https://pubmed.ncbi.nlm.nih.gov/25883595/</a>,</p>

<p>2015 PNAS. The human gene damage index as a gene-level approach to prioritizing exome variants.
<a class="citation" href="#itan2015human">(Itan et al., 2015)</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/26483451/">https://pubmed.ncbi.nlm.nih.gov/26483451/</a></p>

<p>2016 NatMet. The mutation significance cutoff: gene-level thresholds for variant predictions.
<a class="citation" href="#itan2016mutation">(Itan et al., 2016)</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/26820543/">https://pubmed.ncbi.nlm.nih.gov/26820543/</a></p>

<p>2015 PNAS. Can the impact of human genetic variations be predicted?
<a class="citation" href="#itan2015can">(Itan &amp; Casanova, 2015)</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/26351682/">https://pubmed.ncbi.nlm.nih.gov/26351682/</a></p>

<p>2018 Bioinf. PopViz: a webserver for visualizing minor allele frequencies and damage prediction scores of human genetic variations.
<a class="citation" href="#zhang2018popviz">(Zhang et al., 2018)</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/30535305/">https://pubmed.ncbi.nlm.nih.gov/30535305/</a></p>

<p>2019 PNAS. Blacklisting variants common in private cohorts but not in public databases optimizes human exome analysis.
<a class="citation" href="#maffucci2019blacklisting">(Maffucci et al., 2019)</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/30591557/">https://pubmed.ncbi.nlm.nih.gov/30591557/</a></p>

<p>2019 NAR. SeqTailor: a user-friendly webserver for the extraction of DNA or protein sequences from next-generation sequencing data.
<a class="citation" href="#zhang2019seqtailor">(Zhang et al., 2019)</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/31045209/">https://pubmed.ncbi.nlm.nih.gov/31045209/</a>.</p>

<p>2020 Hum Gen. The human genetic determinism of life-threatening infectious diseases: genetic heterogeneity and physiological homogeneity?
<a class="citation" href="#casanova2020human">(Casanova &amp; Abel, 2020)</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/32462426/">https://pubmed.ncbi.nlm.nih.gov/32462426/</a></p>

<p>2021 JCI. Herpes simplex encephalitis in a patient with a distinctive form of inherited IFNAR1 deficiency.
<a class="citation" href="#bastard2021herpes">(Bastard et al., 2021)</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/32960813/">https://pubmed.ncbi.nlm.nih.gov/32960813/</a></p>

<p>2021 JCI. TLR3 controls constitutive IFN-β antiviral immunity in human fibroblasts and cortical neurons.
<a class="citation" href="#gao2021tlr3">(Gao et al., 2021)</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/33393505/">https://pubmed.ncbi.nlm.nih.gov/33393505/</a></p>

<p>2021 AJHG. A computational approach for detecting physiological homogeneity in the midst of genetic heterogeneity
<a class="citation" href="#zhang2021computational">(Zhang et al., 2021)</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/34015270/">https://pubmed.ncbi.nlm.nih.gov/34015270/</a></p>

<hr />

<p>To date, the main paper that implements protein pathway analysis for rare 
immune disease is that by
Peng Zhang and 
Yuval Itan <a class="citation" href="#zhang2021computational">(Zhang et al., 2021)</a>.</p>

<p>The history begins with determining methods for candidate variant selection - 
the main challenge in human genomics, 
especially for individual patients who can benefit from precision medicine. 
Over the last decade, we have reached a point where we can now reasonably 
discern individual candidate-causal variants from the background noise of 
genomic variability.</p>

<p>From the authors’ perspective, the project began around 2011 and its first 
step was concluded in 2013 with 
Yuval Itan’s
first “Human Gene Connectome” paper 
while he was a post-doc with Casanova lab. 
<a class="citation" href="#itan2013human">(Itan et al., 2013)</a>.
This software connected genes like streets in a map, 
based on their physiological relatedness.
It was soon followed by methodological development 
<a class="citation" href="#itan2014hgcs">(Itan et al., 2014)</a>
and application to inborn errors of immunity 
<a class="citation" href="#itan2015novel">(Itan &amp; Casanova, 2015)</a>
or both 
<a class="citation" href="#itan2015human">(Itan et al., 2015)</a>,
and a couple of necessary detours 
<a class="citation" href="#itan2016mutation">(Itan et al., 2016)</a>
and 
<a class="citation" href="#maffucci2019blacklisting">(Maffucci et al., 2019)</a>.
A review was also written by two of the main authors during the same period
<a class="citation" href="#itan2015can">(Itan &amp; Casanova, 2015)</a>.</p>

<p>When Peng Zhang joined the Casanova lab as post-doc, 
Yuval Itan had started his own lab. 
However, the pair worked together to continue producing the papers on variant
interpretation and data processing
<a class="citation" href="#zhang2018popviz">(Zhang et al., 2018)</a>
and
<a class="citation" href="#zhang2019seqtailor">(Zhang et al., 2019)</a>.</p>

<p>After completing this period of work, 
they renamed “Human Gene Connectome II” the 
“Network-based Heterogeneity Clustering”.
At this point, their aims were defined as being generally indistinguishable 
from ours. 
That is, <em>“the detection of physiological homogeneity in a cohort of patients 
sharing a clinical phenotype but with high genetic heterogeneity - 
a hallmark of severe infectious diseases”</em> (Casanova via twitter), 
as presented in their next paper on this topic
<a class="citation" href="#casanova2020human">(Casanova &amp; Abel, 2020)</a>.</p>

<p>Shen-Ying Zhang came on board as senior author on the next two papers.
With an excellent database of immune disorders and infections, 
the team could gradually build their software.
Exomes from patients with HSV-1 encephalitis were used for testing successive 
versions in
<a class="citation" href="#bastard2021herpes">(Bastard et al., 2021)</a> 
and 
<a class="citation" href="#gao2021tlr3">(Gao et al., 2021)</a>.</p>

<p>Quoting Casanova <em>“When they were capable of detecting the known TLR3-IFN needles in the HSE stack, they installed camp 1, rested a bit and reported to me on the radio, while I was watching them from the basecamp with binoculars. I encouraged them to push for the final ascent and they did.”</em></p>

<p>With the same goal as our own - 
producing unbiased methods for detection of biologically-connected causal 
genetic variation - 
they found new gene variants that interact via the TLR3-IFN protein pathway,
in individual patients. 
Shen-Ying Zhang found them to be biochemically deleterious, 
an important factor for validation of genetic-first aproaches. 
In this case, Zhang <em>et al</em> get as close to the <em>“gold-standard”</em> 
as anyone to date.</p>

<p>The functional validation of candidate variants in disease then provided a proof-of-principle indication that they could detect physiological homogeneity in the midst of genetic heterogeneity 
<a class="citation" href="#zhang2021computational">(Zhang et al., 2021)</a>.</p>

<p>Quoting Casanova 
<em>“A computational approach for detecting physiological homogeneity in the midst of genetic heterogeity.
That was terrific!”</em>.</p>

<hr />

<blockquote>
  <p>An aside on what I call the <em>“gold-standard”</em> for our field should be:</p>
  <ol>
    <li>Unbiased statistical detection of a genetic phenomenon.</li>
    <li>Validation by functional models under systematic control.</li>
  </ol>

  <p>Part [1] Depends on patient cohorts that are large enough to detect the effect based on the phenotype strength - difficult for <em>rare disease</em>.<br />
Part [2] Depends on independently testing biological mechanisms.</p>

  <p>This second step generally consists of two hurdles:</p>
  <ul>
    <li>If the same researchers perform (1) genetic stats and (2) functional work, 
there is a bias that is difficult to avoid when trying to functionally 
validate statistically positive results.</li>
    <li>If the statistical genetic associations happen to contain a false positive 
for something like severe rare immune disease, 
the sensitive functional models may detect a damaging response. 
One might find a truly damaging biological mechanism, 
but if the statistical genetic association is not correct then this 
biological mechanism should not be deemed causal; 
back-tracking at this stage would be very difficult due to self-imposed bias.</li>
  </ul>

  <p>Ideally, in the future we hope to see a separation of the two steps 
(stat genetics and wet-lab) such that each are performed independently. 
The wet-lab would also ideally focus their routines on a particular protein 
pathway/system with SOPs that improve accuracy and precision 
(e.g. clinical diagnostics labs, clinical trials) rather than setting up 
models for each new study.</p>
</blockquote>

<hr />

<p>Returning to our historical review, 
we have been producing our protocols similarly in parallel.
With patient cohorts of comparable sizes and phenotypes we will soon 
have a complementary validation of protocols.
However, great care is also being taken to test and select the most reliable 
statistical methods for association testing - 
an improvement to the fine work by
<a class="citation" href="#zhang2021computational">(Zhang et al., 2021)</a>.</p>

<p>Best practices in candidate variant selection protocols are basically 
standardised as of 2021, 
so the main remaining task is standardisation of the protein-pathway 
annotation and association testing methods - 
steps which we will soon be ready to publish after peer-review.</p>

<h1 id="burden-test-power-calculation">Burden test power calculation</h1>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">This</span><span class="w"> </span><span class="n">code</span><span class="w"> </span><span class="n">seems</span><span class="w"> </span><span class="n">to</span><span class="w"> </span><span class="n">work</span><span class="w">
</span><span class="c1"># This script tests then the minimum number of heterozygous mutations that can exist in a test set to reach significance.</span><span class="w">

</span><span class="c1"># Required inputs, can be modified to read data variable</span><span class="w">
</span><span class="c1"># SAMPLES (equal case and control)</span><span class="w">
</span><span class="c1"># SNPs (the number of SNPs in the test pathway, can modify to loop every pathway)</span><span class="w">
</span><span class="c1"># CMC settings may need adjustment to match your model</span><span class="w">

</span><span class="n">library</span><span class="p">(</span><span class="n">AssotesteR</span><span class="p">)</span><span class="w">

</span><span class="c1"># Pathway size of 5 SNPs</span><span class="w">
</span><span class="c1"># 100 case and 100 controls</span><span class="w">

</span><span class="c1"># for (SNPs in 1:3) {</span><span class="w">
 </span><span class="k">for</span><span class="w"> </span><span class="p">(</span><span class="n">SNPs</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">10</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">

  </span><span class="n">SAMPLES</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="w">
  </span><span class="n">TOTAL</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">SAMPLES</span><span class="o">*</span><span class="n">SNPs</span><span class="w">

</span><span class="k">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">TOTAL</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
  </span><span class="c1"># cases</span><span class="w">
    </span><span class="n">val_1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">replicate</span><span class="p">(</span><span class="n">i</span><span class="p">,</span><span class="m">0</span><span class="p">)</span><span class="w">
    </span><span class="n">x</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="p">(</span><span class="n">TOTAL</span><span class="o">-</span><span class="n">i</span><span class="p">)</span><span class="w">
    </span><span class="n">val_2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">replicate</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="m">0</span><span class="p">)</span><span class="w">
    
    </span><span class="c1"># cases</span><span class="w">
    </span><span class="n">val_3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">replicate</span><span class="p">(</span><span class="n">i</span><span class="p">,</span><span class="m">0</span><span class="p">)</span><span class="w">
    </span><span class="n">y</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="p">(</span><span class="n">TOTAL</span><span class="o">-</span><span class="n">i</span><span class="p">)</span><span class="w">
    </span><span class="n">val_4</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">replicate</span><span class="p">(</span><span class="n">y</span><span class="p">,</span><span class="m">1</span><span class="p">)</span><span class="w">

    </span><span class="c1"># combine</span><span class="w">
    </span><span class="n">mat1.data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w">  </span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">val_1</span><span class="p">,</span><span class="w"> </span><span class="n">val_2</span><span class="p">,</span><span class="w"> </span><span class="n">val_3</span><span class="p">,</span><span class="w"> </span><span class="n">val_4</span><span class="p">))</span><span class="w">
    
    </span><span class="c1"># contols then cases</span><span class="w">
    </span><span class="n">genotype</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="n">mat1.data</span><span class="p">,</span><span class="n">nrow</span><span class="o">=</span><span class="p">(</span><span class="n">SAMPLES</span><span class="o">*</span><span class="m">2</span><span class="p">),</span><span class="n">ncol</span><span class="o">=</span><span class="n">SNPs</span><span class="p">,</span><span class="n">byrow</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
    </span><span class="c1">#print(genotype)</span><span class="w">
    
    </span><span class="n">stat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="n">genotype</span><span class="p">,</span><span class="w"> </span><span class="n">dimnames</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">outer</span><span class="p">(</span><span class="n">colnames</span><span class="p">(</span><span class="n">genotype</span><span class="p">),</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">genotype</span><span class="p">),</span><span class="w"> </span><span class="n">FUN</span><span class="o">=</span><span class="n">paste</span><span class="p">)),</span><span class="w"> </span><span class="kc">NULL</span><span class="p">))</span><span class="w">
    </span><span class="n">print</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="s2">"for"</span><span class="p">,</span><span class="w"> 
                </span><span class="n">SAMPLES</span><span class="p">,</span><span class="w"> </span><span class="s2">"cases and "</span><span class="p">,</span><span class="w">
                </span><span class="n">SAMPLES</span><span class="p">,</span><span class="w"> </span><span class="s2">"controls:"</span><span class="p">))</span><span class="w">
    </span><span class="n">print</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="s2">"with"</span><span class="p">,</span><span class="w"> </span><span class="n">SNPs</span><span class="p">,</span><span class="w"> </span><span class="s2">"candidate pathway SNPs"</span><span class="p">))</span><span class="w">
    </span><span class="n">print</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="nf">sum</span><span class="p">(</span><span class="n">stat</span><span class="p">),</span><span class="w">
                 </span><span class="s2">"mutations per"</span><span class="p">,</span><span class="w"> </span><span class="n">TOTAL</span><span class="p">,</span><span class="w"> </span><span class="s2">"total nucleotides in cases"</span><span class="p">))</span><span class="w">
    </span><span class="n">print</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="w">
                 </span><span class="s2">"0 mutation per"</span><span class="p">,</span><span class="w"> </span><span class="n">TOTAL</span><span class="p">,</span><span class="w"> </span><span class="s2">"total nucleotides in controls."</span><span class="w">
                 </span><span class="p">))</span><span class="w">

    </span><span class="c1"># number of cases</span><span class="w">
    </span><span class="n">cases</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">SAMPLES</span><span class="w">
    
    </span><span class="c1"># number of controls</span><span class="w">
    </span><span class="n">controls</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">SAMPLES</span><span class="w">
    
    </span><span class="c1"># total (cases + controls)</span><span class="w">
    </span><span class="n">total</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cases</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">controls</span><span class="w">
    
    </span><span class="c1"># phenotype vector</span><span class="w">
    </span><span class="n">phenotype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="n">controls</span><span class="p">),</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="n">cases</span><span class="p">))</span><span class="w">
    </span><span class="c1">#print(phenotype)</span><span class="w">
    
    </span><span class="c1">#mycmc = CMC(phenotype, genotype, maf=0.0000001, perm=100)</span><span class="w">
    </span><span class="c1">#print(mycmc)</span><span class="w">
    </span><span class="n">myskat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">SKAT</span><span class="p">(</span><span class="n">phenotype</span><span class="p">,</span><span class="w"> </span><span class="n">genotype</span><span class="p">,</span><span class="w"> </span><span class="n">kernel</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"linear"</span><span class="p">)</span><span class="w">
    </span><span class="n">print</span><span class="p">(</span><span class="n">myskat</span><span class="p">[</span><span class="s2">"asym.pval"</span><span class="p">])</span><span class="w">
</span><span class="p">}</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>

<p>However, I am filling up SNPs columns so that all cases get the variant mutation. 
I believe SKAT will give a better score to 5 unique variants rather than 5 shared variants.
Therefore, I am currently revising the code.</p>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># SKAT: Sequence Kernel Association Test </span><span class="w">
</span><span class="c1"># </span><span class="w">
</span><span class="c1"># [1] "for 100 cases and  100 controls:"</span><span class="w">
</span><span class="c1"># [1] "with 5 candidate pathway SNPs"</span><span class="w">
</span><span class="c1"># [1] "18 mutations per 500 total nucleotides in cases"</span><span class="w">
</span><span class="c1"># [1] "0 mutation per 500 total nucleotides in controls."</span><span class="w">
</span><span class="c1"># $asym.pval</span><span class="w">
</span><span class="c1"># [1] 0.04799332</span><span class="w">
</span><span class="c1"># </span><span class="w">
</span><span class="c1"># </span><span class="w">
</span><span class="c1"># [1] "for 100 cases and  100 controls:"</span><span class="w">
</span><span class="c1"># [1] "with 10 candidate pathway SNPs"</span><span class="w">
</span><span class="c1"># [1] "36 mutations per 1000 total nucleotides in cases"</span><span class="w">
</span><span class="c1"># [1] "0 mutation per 1000 total nucleotides in controls."</span><span class="w">
</span><span class="c1"># $asym.pval</span><span class="w">
</span><span class="c1"># [1] 0.04799332</span><span class="w">
</span></code></pre></div></div>
<p>You can view to genotype matrix constructions by uncommenting</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="c1"># print(genotype).</span><span class="w">
</span></code></pre></div></div>

<p>Exmple:</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">[</span><span class="m">198</span><span class="p">,]</span><span class="w">    </span><span class="m">0</span><span class="w">    </span><span class="m">0</span><span class="w">    </span><span class="m">0</span><span class="w">    </span><span class="m">0</span><span class="w">    </span><span class="m">0</span><span class="w">
</span><span class="p">[</span><span class="m">199</span><span class="p">,]</span><span class="w">    </span><span class="m">0</span><span class="w">    </span><span class="m">0</span><span class="w">    </span><span class="m">0</span><span class="w">    </span><span class="m">0</span><span class="w">    </span><span class="m">0</span><span class="w">
</span><span class="p">[</span><span class="m">200</span><span class="p">,]</span><span class="w">    </span><span class="m">0</span><span class="w">    </span><span class="m">0</span><span class="w">    </span><span class="m">0</span><span class="w">    </span><span class="m">1</span><span class="w">    </span><span class="m">1</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s2">"for 100 cases and  100 controls:"</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s2">"with 5 candidate pathway SNPs"</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s2">"2 mutations per 500 total nucleotides in cases"</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s2">"0 mutation per 500 total nucleotides in controls."</span><span class="w">
</span><span class="o">$</span><span class="n">asym.pval</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="m">0.3165047</span><span class="w">
</span></code></pre></div></div>

<p>I will instead try to construct the test set with uniq variants rather than shared. I only realised the problem at the end.
An example of what we want instead is - checking the P-value and combinations of variants. 
Then we will find the real minimum variant count required to pass significance:</p>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># With a small test set</span><span class="w">
</span><span class="c1"># 2 case, 2 control, 4 positions</span><span class="w">
</span><span class="c1"># 1 control SNP, 3 case SNP</span><span class="w">

</span><span class="n">cases</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="w">
</span><span class="n">controls</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="w">
</span><span class="n">phenotype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="n">controls</span><span class="p">),</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="n">cases</span><span class="p">))</span><span class="w">

</span><span class="n">x</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">expand.grid</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">1</span><span class="p">),</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">1</span><span class="p">),</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">1</span><span class="p">))</span><span class="w">
</span><span class="n">geno1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">x</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">3</span><span class="p">,</span><span class="m">4</span><span class="p">),</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">)]</span><span class="w">
</span><span class="n">geno2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">x</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">9</span><span class="p">,</span><span class="m">13</span><span class="p">),</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">)]</span><span class="w">
</span><span class="n">geno3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">x</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="m">14</span><span class="p">),</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">)]</span><span class="w">
</span><span class="n">geno4</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">x</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="m">15</span><span class="p">),</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">)]</span><span class="w">

</span><span class="n">geno1</span><span class="w">
</span><span class="p">(</span><span class="n">SKAT</span><span class="p">(</span><span class="n">phenotype</span><span class="p">,</span><span class="w"> </span><span class="n">geno1</span><span class="p">,</span><span class="w"> </span><span class="n">kernel</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"linear"</span><span class="p">)[</span><span class="s2">"asym.pval"</span><span class="p">])</span><span class="w">
</span><span class="n">geno2</span><span class="w">
</span><span class="p">(</span><span class="n">SKAT</span><span class="p">(</span><span class="n">phenotype</span><span class="p">,</span><span class="w"> </span><span class="n">geno2</span><span class="p">,</span><span class="w"> </span><span class="n">kernel</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"linear"</span><span class="p">)[</span><span class="s2">"asym.pval"</span><span class="p">])</span><span class="w">
</span><span class="n">geno3</span><span class="w">
</span><span class="p">(</span><span class="n">SKAT</span><span class="p">(</span><span class="n">phenotype</span><span class="p">,</span><span class="w"> </span><span class="n">geno3</span><span class="p">,</span><span class="w"> </span><span class="n">kernel</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"linear"</span><span class="p">)[</span><span class="s2">"asym.pval"</span><span class="p">])</span><span class="w">
</span><span class="n">geno4</span><span class="w">
</span><span class="p">(</span><span class="n">SKAT</span><span class="p">(</span><span class="n">phenotype</span><span class="p">,</span><span class="w"> </span><span class="n">geno3</span><span class="p">,</span><span class="w"> </span><span class="n">kernel</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"linear"</span><span class="p">)[</span><span class="s2">"asym.pval"</span><span class="p">])</span><span class="w">


</span><span class="c1"># &gt; geno1</span><span class="w">
</span><span class="c1">#   Var1 Var2 Var3 Var4</span><span class="w">
</span><span class="c1"># 1    0    0    0    0</span><span class="w">
</span><span class="c1"># 2    1    0    0    0</span><span class="w">
</span><span class="c1"># 3    0    1    0    0</span><span class="w">
</span><span class="c1"># 4    1    1    0    0</span><span class="w">
</span><span class="c1"># $asym.pval [1] 0.0619688</span><span class="w">
</span><span class="c1"># </span><span class="w">
</span><span class="c1"># &gt; geno2</span><span class="w">
</span><span class="c1">#    Var1 Var2 Var3 Var4</span><span class="w">
</span><span class="c1"># 1     0    0    0    0</span><span class="w">
</span><span class="c1"># 2     1    0    0    0</span><span class="w">
</span><span class="c1"># 9     0    0    0    1</span><span class="w">
</span><span class="c1"># 13    0    0    1    1</span><span class="w">
</span><span class="c1"># $asym.pval [1] 0.0311075</span><span class="w">
</span><span class="c1"># </span><span class="w">
</span><span class="c1"># &gt; geno3</span><span class="w">
</span><span class="c1">#     Var1 Var2 Var3 Var4</span><span class="w">
</span><span class="c1"># 1      0    0    0    0</span><span class="w">
</span><span class="c1"># 2      1    0    0    0</span><span class="w">
</span><span class="c1"># 1.1    0    0    0    0</span><span class="w">
</span><span class="c1"># 14     1    0    1    1</span><span class="w">
</span><span class="c1"># $asym.pval [1] 0.0311075</span><span class="w">
</span><span class="c1"># </span><span class="w">
</span><span class="c1"># &gt; geno4</span><span class="w">
</span><span class="c1">#     Var1 Var2 Var3 Var4</span><span class="w">
</span><span class="c1"># 1      0    0    0    0</span><span class="w">
</span><span class="c1"># 2      1    0    0    0</span><span class="w">
</span><span class="c1"># 1.1    0    0    0    0</span><span class="w">
</span><span class="c1"># 15     0    1    1    1</span><span class="w">
</span><span class="c1"># $asym.pval [1] 0.0311075</span><span class="w">
</span></code></pre></div></div>

<h1 id="references">References</h1>

<ol class="bibliography"><li><span id="li2008methods">Li, B., &amp; Leal, S. M. (2008). Methods for detecting associations with rare variants for common diseases: application to analysis of sequence data. <i>The American Journal of Human Genetics</i>, <i>83</i>(3), 311–321.</span></li>
<li><span id="morris2010evaluation">Morris, A. P., &amp; Zeggini, E. (2010). An evaluation of statistical approaches to rare variant analysis in genetic association studies. <i>Genetic Epidemiology</i>, <i>34</i>(2), 188–193.</span></li>
<li><span id="price2010pooled">Price, A. L., Kryukov, G. V., de Bakker, P. I. W., Purcell, S. M., Staples, J., Wei, L.-J., &amp; Sunyaev, S. R. (2010). Pooled association tests for rare variants in exon-resequencing studies. <i>The American Journal of Human Genetics</i>, <i>86</i>(6), 832–838.</span></li>
<li><span id="Lee2012Optimal">Lee, S., Wu, M. C., &amp; Lin, X. (2012). Optimal tests for rare variant effects in sequencing association studies. <i>Biostatistics</i>, <i>13</i>(4), 762–775. https://doi.org/10.1093/biostatistics/kxs014</span></li>
<li><span id="Lee2012Optimalunified">Lee, S., Emond, M. J., Bamshad, M. J., Barnes, K. C., Rieder, M. J., Nickerson, D. A., Christiani, D. C., Wurfel, M. M., &amp; Lin, X. (2012). Optimal Unified Approach for Rare-Variant Association Testing with Application to Small-Sample Case-Control Whole-Exome Sequencing Studies. <i>The American Journal of Human Genetics</i>, <i>91</i>(2), 224–237. https://doi.org/https://doi.org/10.1016/j.ajhg.2012.06.007</span></li>
<li><span id="IonitaLaza2013Sequence">Ionita-Laza, I., Lee, S., Makarov, V., Buxbaum, J. D., &amp; Lin, X. (2013). Sequence Kernel Association Tests for the Combined Effect of Rare and Common Variants. <i>The American Journal of Human Genetics</i>, <i>92</i>(6), 841–853. https://doi.org/https://doi.org/10.1016/j.ajhg.2013.04.015</span></li>
<li><span id="madsen2009groupwise">Madsen, B. E., &amp; Browning, S. R. (2009). A groupwise association test for rare mutations using a weighted sum statistic. <i>PLoS Genet</i>, <i>5</i>(2), e1000384.</span></li>
<li><span id="zhang2021computational">Zhang, P., Cobat, A., Lee, Y.-S., Wu, Y., Bayrak, C. S., Boccon-Gibod, C., Matuozzo, D., Lorenzo, L., Jain, A., Boucherit, S., &amp; others. (2021). A computational approach for detecting physiological homogeneity in the midst of genetic heterogeneity. <i>The American Journal of Human Genetics</i>.</span></li>
<li><span id="itan2013human">Itan, Y., Zhang, S.-Y., Vogt, G., Abhyankar, A., Herman, M., Nitschke, P., Fried, D., Quintana-Murci, L., Abel, L., &amp; Casanova, J.-L. (2013). The human gene connectome as a map of short cuts for morbid allele discovery. <i>Proceedings of the National Academy of Sciences</i>, <i>110</i>(14), 5558–5563.</span></li>
<li><span id="itan2014hgcs">Itan, Y., Mazel, M., Mazel, B., Abhyankar, A., Nitschke, P., Quintana-Murci, L., Boisson-Dupuis, S., Boisson, B., Abel, L., Zhang, S.-Y., &amp; others. (2014). HGCS: an online tool for prioritizing disease-causing gene variants by biological distance. <i>BMC Genomics</i>, <i>15</i>(1), 1–8.</span></li>
<li><span id="itan2015novel">Itan, Y., &amp; Casanova, J.-L. (2015). Novel primary immunodeficiency candidate genes predicted by the human gene connectome. <i>Frontiers in Immunology</i>, <i>6</i>, 142.</span></li>
<li><span id="itan2015human">Itan, Y., Shang, L., Boisson, B., Patin, E., Bolze, A., Moncada-Vélez, M., Scott, E., Ciancanelli, M. J., Lafaille, F. G., Markle, J. G., &amp; others. (2015). The human gene damage index as a gene-level approach to prioritizing exome variants. <i>Proceedings of the National Academy of Sciences</i>, <i>112</i>(44), 13615–13620.</span></li>
<li><span id="itan2016mutation">Itan, Y., Shang, L., Boisson, B., Ciancanelli, M. J., Markle, J. G., Martinez-Barricarte, R., Scott, E., Shah, I., Stenson, P. D., Gleeson, J., &amp; others. (2016). The mutation significance cutoff: gene-level thresholds for variant predictions. <i>Nature Methods</i>, <i>13</i>(2), 109–110.</span></li>
<li><span id="itan2015can">Itan, Y., &amp; Casanova, J.-L. (2015). Can the impact of human genetic variations be predicted? <i>Proceedings of the National Academy of Sciences</i>, <i>112</i>(37), 11426–11427.</span></li>
<li><span id="zhang2018popviz">Zhang, P., Bigio, B., Rapaport, F., Zhang, S.-Y., Casanova, J.-L., Abel, L., Boisson, B., &amp; Itan, Y. (2018). PopViz: a webserver for visualizing minor allele frequencies and damage prediction scores of human genetic variations. <i>Bioinformatics</i>, <i>34</i>(24), 4307–4309.</span></li>
<li><span id="maffucci2019blacklisting">Maffucci, P., Bigio, B., Rapaport, F., Cobat, A., Borghesi, A., Lopez, M., Patin, E., Bolze, A., Shang, L., Bendavid, M., &amp; others. (2019). Blacklisting variants common in private cohorts but not in public databases optimizes human exome analysis. <i>Proceedings of the National Academy of Sciences</i>, <i>116</i>(3), 950–959.</span></li>
<li><span id="zhang2019seqtailor">Zhang, P., Boisson, B., Stenson, P. D., Cooper, D. N., Casanova, J.-L., Abel, L., &amp; Itan, Y. (2019). SeqTailor: a user-friendly webserver for the extraction of DNA or protein sequences from next-generation sequencing data. <i>Nucleic Acids Research</i>, <i>47</i>(W1), W623–W631.</span></li>
<li><span id="casanova2020human">Casanova, J.-L., &amp; Abel, L. (2020). The human genetic determinism of life-threatening infectious diseases: genetic heterogeneity and physiological homogeneity? <i>Human Genetics</i>, <i>139</i>, 681–694.</span></li>
<li><span id="bastard2021herpes">Bastard, P., Manry, J., Chen, J., Rosain, J., Seeleuthner, Y., AbuZaitun, O., Lorenzo, L., Khan, T., Hasek, M., Hernandez, N., &amp; others. (2021). Herpes simplex encephalitis in a patient with a distinctive form of inherited IFNAR1 deficiency. <i>The Journal of Clinical Investigation</i>, <i>131</i>(1).</span></li>
<li><span id="gao2021tlr3">Gao, D., Ciancanelli, M. J., Zhang, P., Harschnitz, O., Bondet, V., Hasek, M., Chen, J., Mu, X., Itan, Y., Cobat, A., &amp; others. (2021). TLR3 controls constitutive IFN-βantiviral immunity in human fibroblasts and cortical neurons. <i>The Journal of Clinical Investigation</i>, <i>131</i>(1).</span></li></ol>

</div>

<!--

<div id="related">
  <h2>Related Topics</h2>
  <ul class="topics">
    
  </ul>
</div>

-->


	<div class="footer">
		<div class="foot-container">
				<a href="/" class="logo-foot">
				<img style="margin-bottom:-16px" 
				src="/images/logos/logo4.2_150ppi.png" 
				alt="Logo image" width="35.7" height="55"/></a>
					<div class="right-root">
						<p> Maintained by Dylan Lawless. Postdoc at <a href="https://people.epfl.ch/dylan.lawless">EPFL</a>.
						<a href="/resume">Resume</a>
						and <a href="https://scholar.google.com/citations?user=RPBxP1wAAAAJ&hl=en&oi=ao">Google scholar</a>.
						</p>
					</div>
		</div>
	</div>

</div>


<!-- Banner on top right corner linking to github
<a href="https://github.com/mojombo"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_red_aa0000.png" alt="Fork me on GitHub" /></a>
-->

<!-- ClickTale Bottom part -->
<div id="ClickTaleDiv" style="display: none;"></div>
<script type="text/javascript">
if(document.location.protocol!='https:')
  document.write(unescape("%3Cscript%20src='https://s.clicktale.net/WRb.js'%20type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
if(typeof ClickTale=='function') ClickTale(206,0.3,"www03");
</script>
<!-- ClickTale end of Bottom part -->

<!-- Google Analytics -->
<!--
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "https://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
var pageTracker = _gat._getTracker("UA-6016902-1");
pageTracker._trackPageview();
</script>
-->
<!-- Google Analytics end -->

</body>
</html>
