<!DOCTYPE html>
<html xmlns="https://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
	<!-- link to MathJax for LaTeX stlye eqn -->
	 <!-- {% include head.html %} -->
	 
	 <!-- https://realfavicongenerator.net -->
	 <link rel="apple-touch-icon" sizes="180x180" href="/images/logos/favicon_package_v0.16_bw/apple-touch-icon.png">
	 <link rel="icon" type="image/png" sizes="32x32" href="/images/logos/favicon_package_v0.16_bw/favicon-32x32.png">
	 <link rel="icon" type="image/png" sizes="16x16" href="/images/logos/favicon_package_v0.16_bw/favicon-16x16.png">
	 <link rel="manifest" href="/images/logos/favicon_package_v0.16_bw/site.webmanifest">
	 <meta name="msapplication-TileColor" content="#da532c">
	 <meta name="theme-color" content="#ffffff">
	
	 <meta name="viewport" content="width=device-width, initial-scale=1">
	
	<meta http-equiv="content-type" content="text/html; charset=utf-8" />
	<meta name="description"
content="A home for topics in human precision medicine, genomic analysis, and data visualisation.">
	<title>Evidence builder</title>
	<meta name="author" content="LawlessGenomics" />
	<link href='https://lawlessgenomics.com/feed.xml' rel='alternate' title="LawlessGenomics" type='application/atom+xml'>
	<!-- <meta name="readability-verification" content="QCzSs992GxmRYRKVpPeZ6LE2tS8aYKxsSSQKV8YM"/> -->
	
	<!-- include a google font -->
	<link href="https://fonts.googleapis.com/css?family=Roboto+Mono|Roboto+Slab:300|Roboto:500" rel="stylesheet">
	<link href="https://fonts.googleapis.com/css?family=Lora" rel="stylesheet">
	
	<!-- syntax highlighting CSS -->
	<link rel="stylesheet" href="/css/syntax.css" type="text/css" />
	
	<!-- Homepage CSS -->
	<link rel="stylesheet" href="/css/screen.css" type="text/css" media="screen, projection" />
	
	<!-- Google font -->
	<link rel="preconnect" href="https://fonts.googleapis.com"> 
	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin> 
	<link href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,400;0,600;0,700;1,400;1,600;1,700&display=swap" rel="stylesheet">
	
	<!-- Typekit -->
	<script type="text/javascript" src="https://use.typekit.com/jpd0pfm.js"></script>
	<script type="text/javascript">try{Typekit.load();}catch(e){}</script>





<style>
body {
	background-color: white;
	font-family: Baskerville;
	}

h1, h2, h3, h4, h5, h6 { 
	width:70%;
	margin: 0 auto;
	align:left;
	color: black;
}

p {
	width:70%;
	margin: 0 auto;
	align:center;
	color: black;
}

@media only screen and (max-width:320px)  { /* smartphones, iPhone, portrait below 320 phones */ 
	.subjects { margin-top: 2.5rem ; }
	.logo { font-size: 1rem; }
	body {
		font-family: 'Garamond';
		font-size: 0.8rem;
		line-height: 1.4rem;
		}
	h1 { 
		font: 1.2rem 'Didot';
		color: black;
	}
	h2, h3, h4, h5, h6 {
		font: 1.0rem 'Didot';
		font-style: italic;
		color: black;
		}
    }

@media only screen and (min-width:320px)  { /* smartphones, iPhone, portrait 480x320 phones */ 
	.subjects { margin-top: 2.5rem ; }
	.logo { font-size: 1.4rem; }
	body {
		font-family: 'Garamond';
		font-size: 1.1rem;
		line-height: 1.7rem;
		}
	h1 { 
		font: 1.2rem 'Didot';
		color: black;
		}
	h2, h3, h4, h5, h6 {
		font: 1.1rem 'Didot';
		font-style: italic;
		color: black;
		}
    }

@media only screen and (min-width:481px)  { /* portrait e-readers (Nook/Kindle), smaller tablets @ 600 or @ 640 wide. */ 
	.subjects { margin-top: 2.5rem ; }
	.logo { font-size: 1.5rem; }
	body {
		font-family: 'Garamond';
		font-size: 1.1rem;
		line-height: 1.7rem;
		}
	h1 { 
		font: 1.2rem 'Didot';
		color: black;
		}
	h2, h3, h4, h5, h6 {
		font: 1.1rem 'Didot';
		font-style: italic;
		color: black;
		}
    }

@media only screen and (min-width:641px)  { /* portrait tablets, portrait iPad, landscape e-readers, landscape 800x480 or 854x480 phones */ 
	.subjects { margin-top: 3rem ; }
	.logo { font-size: 1.6rem; }
	body {
		font-size: 1.2rem;
		line-height: 2rem;
		}
	h1 { 
		font: 1.6rem 'Didot';
		color: black;
		}
	h2, h3, h4, h5, h6 {
		font: 1.3rem 'Didot';
		font-style: italic;
		color: black;
		}
    }

@media only screen and (min-width:961px)  { /* tablet, landscape iPad, lo-res laptops ands desktops */
	.subjects { margin-top: 4rem ; }
	.logo { font-size: 1.7rem; }
	body {
		font-size: 1.3rem;
		line-height: 2rem;
		}
	h1 { 
		font: 1.7rem 'Didot';
		color: black;
		}
	h2 ,h3, h4, h5, h6 {
		font: 1.3rem 'Didot';
		font-style: italic;
		color: black;
		}
    }

@media only screen and (min-width:1025px) { /* big landscape tablets, laptops, and desktops */
	.subjects { margin-top: 4rem ; }
	.logo { font-size: 1.7rem; }
	body {
		font-size: 1.3rem;
		line-height: 2rem;
		}
	h1 { 
		font: 1.7rem 'Didot';
		color: black;
		}
	h2, h3, h4, h5, h6 {
		font: 1.3rem 'Didot';
		font-style: italic;
		color: black;
		}
    }

@media only screen and (min-width:1281px) { /* hi-res laptops and desktops */ 
	.subjects { margin-top: 4rem ; }
	body {
		font-size: 1.3rem;
		line-height: 2rem;
		}
	h1 { 
		font: 1.7rem 'Didot';
		color: black;
		}
	h2, h3, h4, h5, h6 {
		font: 1.3rem 'Didot';
		color: black;
	}
    }

</style>



</head>
<body>

<!-- ClickTale Top part -->
<script type="text/javascript">
	var WRInitTime=(new Date()).getTime();
</script>
<!-- ClickTale end of Top part -->

<!-- Logo and navigation bar -->



<!-- SITE START --------------------------------------------------------- -->
<div class="site">

<!-- NAV START --------------------------------------------------------- -->
<div class="sitewidthnav">
	<div class="title">
		<div class="nav-container">
			<nav>

			<div class="logoimage">
				<a href="/">
				<img style="margin-bottom:-16px" 
				src="/images/logos/logo4.2_150ppi.png" 
				alt="Logo image" width="25" height="38.5" /></a>
				<!-- alt="Logo image" width="35.7" height="55" /></a> -->
			</div>

			<div class="logo">
				<a href="/">LawlessGenomics</a>
			</div>

				<input type="checkbox" id="nav-toggle">
					<label for="nav-toggle" class="gg-menu"></label>
					<div class="right-menu">
						<a href="/resume">About</a>
						<a href="/">Home</a>
					</div>
			</nav>
		</div>
	</div>
</div>
<!-- NAV END ----------------------------------------------------------- -->

<!-- MAIN START -------------------------------------------------------- -->
<div>
	</br>
<h1>Evidence builder</h1>
<p>
This page is a work in progress. 
I publish this intermediate version to demonstrate the purpose and for discussion with both data curators and front-end devs.
</br>
</br>

<strong>Shown here:</strong> two randomly chosen coding variants in DDX58.<br>
<!-- <strong>Variant table:</strong> gnomAD data for the variants of interest. <br> -->
<strong>Annotation table:</strong> Default output from variant annotation. <br>
<strong>Top:</strong> protein-coding functional data with coding positions. Protein annotations match that of Uniprot:ENST00000379883; ENSP00000369213; ENSG00000107201 [O95786-1].<br>
<strong>Bottom:</strong> transcript sequence data with genomic coordinates, GRCh37.
Blue dotted lines indicate the variants of interest.
The canonical transcript ENST00000379883.2 is indicated by the green diamond.
</br>
</br>

<strong>Under dev and coming soon:</strong> <br>
	&#9989; Hover tip annotations for <a href="https://lawlessgenomics.com/topic/gene_illustrate_all_positions.html">all positions</a>.<br>
	&#9997;&#65039; Browsable variant evidence table.<br>
	&#9997;&#65039; Internal links to
<a href="https://lawlessgenomics.com/topic/evidence_finder_gene_all.html">gene evidence finder</a> and
<a href="https://lawlessgenomics.com/topic/evidence-finder-variant">variant evidence finder</a>.<br>
	&#9995; External links to all evidence sources that have been used for analysis; 141db prepared but unlinked e.g.
 <a href="https://gnomad.broadinstitute.org">gnomAD</a>
 <a href="http://database.liulab.science/dbNSFP">dbNSFP</a>
 <a href="https://grch37.ensembl.org/Tools/VEP">VEP</a>
 <a href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a>
 <a href="https://www.clinicalgenome.org">ClinGen</a>
 <a href="http://www.hgmd.cf.ac.uk/ac/index.php">HGMD</a>
	and many more including the top pathogenicity prediction databases.<br>
	&#9989; Data prep and AWS server.<br>
	&#9997;&#65039; Crosslinking.<br>
	&#9997;&#65039; Automated protocols for evidence scores.<br>
	&#9997;&#65039; <a href="https://lawlessgenomics.com/topic/acgm-criteria-table-main">ACMG standards and guidelines</a> for the interpretation of sequence variants (doi:<a href="https://www.nature.com/articles/gim201530">10.1038/gim.2015.30</a>).<br>
	&#9989; All-coding VEP and dbNSFP annotation pipeline.<br>
	&#9989; Account for hg18, hg19 and GRCh38.<br>
	<!-- - SNP1 <b>GRCh37</b> GnomAD v2.1.1 9-32493805-T-C <br> -->
	<!-- - SNP2 <b>GRCh37</b> GnomAD v2.1.1 9-32491328-T-C <br> -->
	<!-- - SNP1 <b>GRCh38</b> GnomAD v3.1.2 9-32493807-T-C <br> -->
	<!-- - SNP2 <b>GRCh38</b> GnomAD v3.1.2 9-32491330-T-C <br> -->
	&#9989; Annotation databases <a href="https://lawlessgenomics.com/topic/annotation-datasets-table-main">are updating live</a>.<br>
	&#9989; Annotation graph protocols <a href="https://lawlessgenomics.com/topic/variant-annotation-graph">are available (demo)</a>.<br>
</p>

<div class="evidence-graph-container">
  <div class="evidence-graph left" margin-bottom="3vh">

<!-- <iframe id="igraph" scrolling="yes" style="border:none;" seamless="seamless" src=" -->
<!-- 	./gnomad_table.html -->
<!-- 	" -->
<!-- 	height="240" --> 
<!-- 	width="80%"></iframe> -->

<iframe id="igraph" scrolling="yes" style="border:none;" seamless="seamless" src="
	./ddx58_grch38.vep.html
	"
	height="430" 
	width="80%"></iframe>
	<!-- https://lawlessgenomics.com/topic/ddx58_grch38.vep.html -->

<iframe id="igraph" scrolling="no" style="border:none;" seamless="seamless" src="
	https://lawlessgenomics.com/topic/gene_illustrate.html
	"
	height="1000" 
	width="95%"></iframe>

<iframe id="igraph" scrolling="no" style="border:none;" seamless="seamless" allowFullScreen="true" src="
https://www.ncbi.nlm.nih.gov/Structure/icn3d/full.html?mmdbid=2YKG&buidx=0&date=20220119&v=3.7.1&command=set background white; color domain
	"
	height="500" 
	width="90%"></iframe>

 </div>
</div>

<div>
	<p>
	Bookmark this page to return for future updates containing the full genome public version. 
	Protein visualisation using iCn3D by
	<a href="https://www.ncbi.nlm.nih.gov/Structure/icn3d/icn3d.html#citing">Wang J, et al. Bioinformatics 2020</a>.
	</p>
</div>


</div>
<!-- # MAIN END ---------------------------------------------------------- -->

<!-- FOOT START -------------------------------------------------------- -->
<div class="sitewidthnav">
	<div class="footer">
		<div class="foot-container">
				<a href="/" class="logo-foot">
				<img style="margin-bottom:-16px" 
				src="/images/logos/logo4.2_150ppi.png" 
				alt="Logo image" width="25" height="38.5" /></a>
					<div class="right-root">
						<p><a href="https://github.com/DylanLawless/DylanLawless.github.io/tree/source">Maintained</a> by Dylan Lawless. Postdoc at <a href="https://people.epfl.ch/dylan.lawless">EPFL</a>.
						<a href="/resume">Resume</a>
						and <a href="https://scholar.google.com/citations?user=RPBxP1wAAAAJ&hl=en&oi=ao">Google scholar</a>.
						<a href='https://lawlessgenomics.com/feed.xml'>RSS feed</a>
						</p>
					</div>
		</div>
	</div>
</div>
<!-- FOOT END ---------------------------------------------------------- -->

</div>
<!-- SITE END ----------------------------------------------------------- -->

<!-- ClickTale Bottom part -->
<div id="ClickTaleDiv" style="display: none;"></div>
<script type="text/javascript">
if(document.location.protocol!='https:')
  document.write(unescape("%3Cscript%20src='https://s.clicktale.net/WRb.js'%20type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
if(typeof ClickTale=='function') ClickTale(206,0.3,"www03");
</script>
<!-- ClickTale end of Bottom part -->

</body>
</html>
